Index: abiword-trunk/src/wp/ap/xp/ap_Toolbar_Layouts.cpp
===================================================================
--- abiword-trunk/src/wp/ap/xp/ap_Toolbar_Layouts.cpp	(revision 22159)
+++ abiword-trunk/src/wp/ap/xp/ap_Toolbar_Layouts.cpp	(working copy)
@@ -342,7 +342,7 @@
 		return pLayout;
 	} 
 	else {
-		fprintf (stderr, "%s:%s: Layout `%s' not found\n", __FILE__, __FUNCTION__, szName);
+		fprintf (stderr, "%s:%s: Layout `%s' not found\n", __FILE__, G_GNUC_PRETTY_FUNCTION, szName);
 	}
 	return NULL;
 }
