Index: abiword-trunk/src/af/xap/unix/xap_UnixFrameImpl.cpp
===================================================================
--- abiword-trunk/src/af/xap/unix/xap_UnixFrameImpl.cpp	(revision 22159)
+++ abiword-trunk/src/af/xap/unix/xap_UnixFrameImpl.cpp	(working copy)
@@ -933,7 +933,7 @@
 
 		if(!pUnixFrameImpl->m_bDoZoomUpdate && (pUnixFrameImpl->m_iZoomUpdateID == 0))
 		{
-			pUnixFrameImpl->m_iZoomUpdateID = g_idle_add(static_cast<GSourceFunc>(do_ZoomUpdate), static_cast<gpointer>(pUnixFrameImpl));
+			pUnixFrameImpl->m_iZoomUpdateID = g_idle_add(reinterpret_cast<GSourceFunc>(do_ZoomUpdate), static_cast<gpointer>(pUnixFrameImpl));
 		}
 	}
 	return 1;
Index: abiword-trunk/src/af/util/xp/ut_png.cpp
===================================================================
--- abiword-trunk/src/af/util/xp/ut_png.cpp	(revision 22159)
+++ abiword-trunk/src/af/util/xp/ut_png.cpp	(working copy)
@@ -84,7 +84,7 @@
 	myBB.pBB = pBB;
 	myBB.iCurPos = 0;
 	
-	png_set_read_fn(png_ptr, static_cast<void *>(&myBB), static_cast<png_rw_ptr>(_png_read));
+	png_set_read_fn(png_ptr, static_cast<void *>(&myBB), reinterpret_cast<png_rw_ptr>(_png_read));
 
 	/* The call to png_read_info() gives us all of the information from the
 	 * PNG file before the first IDAT (image data chunk).  REQUIRED
Index: abiword-trunk/src/wp/ap/unix/abiwidget.cpp
===================================================================
--- abiword-trunk/src/wp/ap/unix/abiwidget.cpp	(revision 22159)
+++ abiword-trunk/src/wp/ap/unix/abiwidget.cpp	(working copy)
@@ -1588,7 +1588,7 @@
 //
 // Can't load until this event has finished propagating
 //
-	  g_idle_add(static_cast<GSourceFunc>(s_abi_widget_load_file),static_cast<gpointer>(abi));
+	  g_idle_add(reinterpret_cast<GSourceFunc>(s_abi_widget_load_file),static_cast<gpointer>(abi));
   }
 }
 
Index: abiword-trunk/src/wp/impexp/unix/ie_impGraphic_GdkPixbuf.cpp
===================================================================
--- abiword-trunk/src/wp/impexp/unix/ie_impGraphic_GdkPixbuf.cpp	(revision 22159)
+++ abiword-trunk/src/wp/impexp/unix/ie_impGraphic_GdkPixbuf.cpp	(working copy)
@@ -454,7 +454,7 @@
 	m_pPngBB = new UT_ByteBuf;  /* Byte Buffer for Converted Data */
 	
 	/* Setting up the Data Writing Function */
-	png_set_write_fn(m_pPNG, static_cast<void *>(m_pPngBB), static_cast<png_rw_ptr>(_write_png), NULL);
+	png_set_write_fn(m_pPNG, static_cast<void *>(m_pPngBB), reinterpret_cast<png_rw_ptr>(_write_png), NULL);
 	
 	return UT_OK;
 }
