Index: abiword-trunk/src/af/xap/xp/xap_Prefs.cpp
===================================================================
--- abiword-trunk/src/af/xap/xp/xap_Prefs.cpp	(revision 22159)
+++ abiword-trunk/src/af/xap/xp/xap_Prefs.cpp	(working copy)
@@ -1693,7 +1693,7 @@
 		return false;
 
 	std::vector<UT_UTF8String>::const_iterator i =
-		find(m_vecFonts.begin(), m_vecFonts.end(), name);
+		std::find(m_vecFonts.begin(), m_vecFonts.end(), name);
 
 	return i != m_vecFonts.end();
 }
