Index: abiword-trunk/src/text/ptbl/xp/pt_PT_InsertObject.cpp
===================================================================
--- abiword-trunk/src/text/ptbl/xp/pt_PT_InsertObject.cpp	(revision 22159)
+++ abiword-trunk/src/text/ptbl/xp/pt_PT_InsertObject.cpp	(working copy)
@@ -279,8 +279,8 @@
 	}
 	if(sProps.size() > 0)
 	{
-	    Atts.addItem(static_cast<gchar *>("props"));
-	    Atts.addItem(static_cast<gchar *>(const_cast<char *>(sProps.utf8_str())));
+		Atts.addItem(const_cast<gchar *>("props"));
+		Atts.addItem(static_cast<gchar *>(const_cast<char *>(sProps.utf8_str())));
 	}
 	PT_AttrPropIndex indexAP;
 	if (!m_varset.storeAP(&Atts,&indexAP))
