--- alpine-0.99999.orig/pico/basic.c	Tue Nov  6 18:51:13 2007
+++ alpine-0.99999/pico/basic.c	Fri Nov  9 10:43:09 2007
@@ -939,7 +939,7 @@
 toggle_xterm_mouse(int f, int n)
 {
 #ifndef _WINDOWS
-    mouseexist() ? end_mouse() : init_mouse();
+    mouseexist() ? end_mouse() : (void)init_mouse();
     mouseexist() ? emlwrite(_("Xterm mouse tracking on!"), NULL)
 		 : emlwrite(_("Xterm mouse tracking off!"), NULL);
 #endif
diff -ru alpine-0.99999.orig/alpine/osdep/termin.unx.c alpine-0.99999/alpine/osdep/termin.unx.c
--- alpine-0.99999.orig/alpine/osdep/termin.unx.c	Tue Nov  6 18:51:13 2007
+++ alpine-0.99999/alpine/osdep/termin.unx.c	Fri Nov  9 10:50:55 2007
@@ -526,7 +526,7 @@
 #ifdef MOUSE
 	else if(ch == ctrl('\\')){
 	    dprint((9, "Read char got ^\\, toggle xterm mouse\n"));
-	    mouseexist() ? end_mouse() : init_mouse();
+	    mouseexist() ? end_mouse() : (void)init_mouse();
 	    q_status_message1(SM_ASYNC, 0, 2, "Xterm mouse tracking %s!",
 	                                       mouseexist() ? "on" : "off");
 	    return(NO_OP_COMMAND);
