--- alpine-0.9999.orig/pith/ldap.c	2007年  8月 20日 一 15:46:37
+++ alpine-0.9999/pith/ldap.c	2007年 10月  8日 一 16:51:24
@@ -436,11 +436,17 @@
     if(wp_err->mangled)
       *(wp_err->mangled) = 1;
 
+#ifdef _SOLARIS_SDK
+    if(info->tls || info->tlsmust)
+      ldapssl_client_init(NULL, NULL);
+    if((ld = ldap_init(serv, info->port)) == NULL)
+#else
 #if (LDAPAPI >= 11)
     if((ld = ldap_init(serv, info->port)) == NULL)
 #else
     if((ld = ldap_open(serv, info->port)) == NULL)
 #endif
+#endif
     {
       /* TRANSLATORS: All of the three args together are an error message */
       snprintf(ebuf, sizeof(ebuf), _("Access to LDAP server failed: %s%s(%s)"),
@@ -471,6 +477,11 @@
 
       memset(&mb, 0, sizeof(mb));
 
+#ifdef _SOLARIS_SDK
+      if(info->tls || info->tlsmust)
+	rc = ldapssl_install_routines(ld);
+#endif
+
       if(ldap_v3_is_supported(ld) &&
 	 our_ldap_set_option(ld, LDAP_OPT_PROTOCOL_VERSION, &proto) == 0){
 	dprint((5, "ldap: using version 3 protocol\n"));
@@ -509,7 +520,11 @@
 	  if(mb.tlsflag
 	     && (pwdtrial > 0 || 
 #ifndef _WINDOWS
+#ifdef _SOLARIS_SDK
+		 (rc == LDAP_SUCCESS)
+#else
 		 ((rc=ldap_start_tls_s(ld, NULL, NULL)) == LDAP_SUCCESS)
+#endif
 #else
 		 0  /* TODO: find a way to do this in Windows */
 #endif
--- alpine-0.9999.orig/configure.ac	2007年  8月 21日 二 18:13:29
+++ alpine-0.9999/configure.ac	2007年 10月  8日 一 15:58:16
@@ -822,8 +822,13 @@
 
 dnl provide LDAP support?
 if test "$alpine_with_ldap" = "yes" ; then
+  AC_CHECK_LIB(lber, ber_alloc,
+    [
+      $LIBS="$LIBS -llber"
+    ],,)
+
   alpine_has_ldap=
-  AC_SEARCH_LIBS(ldap_init,ldap_init,
+  AC_SEARCH_LIBS(ldap_init,ldap,
     [
       alpine_has_ldap=yes
     ],
@@ -831,9 +836,8 @@
       AC_SEARCH_LIBS(ldap_open,ldap,
 	[
 	  alpine_has_ldap=yes
-	],, [-llber])
-    ],
-    [-llber])
+	],,)
+    ])
 
   if test "$alpine_has_ldap" = "yes" ; then
     AC_MSG_NOTICE([Including LDAP Support])
