diff -ur alsa-lib-1.0.14a.orig/configure.in alsa-lib-1.0.14a/configure.in
--- alsa-lib-1.0.14a.orig/configure.in	2007-06-11 15:53:13.000000000 +0700
+++ alsa-lib-1.0.14a/configure.in	2007-08-11 14:53:26.474751213 +0700
@@ -278,6 +278,47 @@
   AC_DEFINE(SUPPORT_RESMGR, "1", [Support resmgr with alsa-lib])
 fi
 
+dnl Check for headers
+AC_CHECK_HEADERS([alloca.h endian.h byteswap.h sys/byteorder.h search.h err.h])
+AC_CHECK_HEADERS([sys/sockio.h features.h sys/ioccom.h])
+
+dnl Check for -lnsl
+AC_CHECK_LIB([nsl], [gethostent], [HAVE_LIBNSL="yes"])
+if test "$HAVE_LIBNSL" = "yes"; then
+  ALSA_DEPLIBS="$ALSA_DEPLIBS -lnsl"
+  AC_DEFINE([HAVE_LIBNSL], 1, [Have libnsl])
+fi
+
+dnl Check for -lsocket
+AC_CHECK_LIB([socket], [setsockopt], [HAVE_LIBSOCKET="yes"])
+if test "$HAVE_LIBSOCKET" = "yes"; then
+  ALSA_DEPLIBS="$ALSA_DEPLIBS -lsocket"
+  AC_DEFINE([HAVE_LIBSOCKET], 1, [Have libsocket])
+fi
+
+dnl not all systems have fcntl(, F_SETFL, O_ASYNC)
+AC_MSG_CHECKING([if fcntl() supports F_SETFL+O_ASYNC])
+AC_COMPILE_IFELSE(
+  [AC_LANG_PROGRAM(
+    [[
+#if HAVE_SYS_TYPES_H
+#include <sys/types.h>
+#endif
+#if HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+#if HAVE_FCNTL_H
+#include <fcntl.h>
+#endif
+    ]],[[
+  fcntl(STDIN_FILENO, F_GETFL);
+  fcntl(STDIN_FILENO, F_SETFL, O_ASYNC);
+    ]]
+  )],
+  [AC_DEFINE(HAVE_FCNTL_ASYNC, 1, [Define to 1 if fcntl() supports F_SETFL+O_ASYNC])
+   AC_MSG_RESULT(yes)],
+  [AC_MSG_RESULT(no)])
+
 dnl Check for aload* support...
 AC_MSG_CHECKING(for aload* support)
 AC_ARG_ENABLE(aload,
