diff -ur alsa-lib-1.0.14a.orig/include/asoundlib.h alsa-lib-1.0.14a/include/asoundlib.h
--- alsa-lib-1.0.14a.orig/include/asoundlib.h	2007-06-11 15:53:13.000000000 +0700
+++ alsa-lib-1.0.14a/include/asoundlib.h	2007-08-11 14:33:53.978819804 +0700
@@ -28,6 +28,18 @@
 #ifndef __ASOUNDLIB_H
 #define __ASOUNDLIB_H
 
+#if defined(sun)
+#include <sys/inttypes.h>
+typedef uint8_t u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+typedef uint64_t u_int64_t;
+#endif
+
+#ifdef HAVE_SYS_IOCCOM_H
+#include <sys/ioccom.h>
+#endif
+
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -ur alsa-lib-1.0.14a.orig/include/sound/asound.h alsa-lib-1.0.14a/include/sound/asound.h
--- alsa-lib-1.0.14a.orig/include/sound/asound.h	2007-06-11 15:53:13.000000000 +0700
+++ alsa-lib-1.0.14a/include/sound/asound.h	2007-08-11 14:37:19.175407508 +0700
@@ -23,6 +23,18 @@
 #ifndef __SOUND_ASOUND_H
 #define __SOUND_ASOUND_H
 
+#if defined(sun)
+#include <sys/inttypes.h>
+typedef uint8_t u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+typedef uint64_t u_int64_t;
+#endif
+
+#ifdef HAVE_SYS_IOCCOM_H
+#include <sys/ioccom.h>
+#endif
+
 #if defined(LINUX) || defined(__LINUX__) || defined(__linux__)
 
 #include <linux/ioctl.h>
