diff -ur alsa-lib-1.0.14a.orig/include/Makefile.am alsa-lib-1.0.14a/include/Makefile.am
--- alsa-lib-1.0.14a.orig/include/Makefile.am	2007-06-11 15:53:13.000000000 +0700
+++ alsa-lib-1.0.14a/include/Makefile.am	2007-08-11 18:30:28.178039422 +0700
@@ -10,7 +10,8 @@
 		      seq_event.h seq.h seqmid.h seq_midi_event.h \
 		      conv.h instr.h iatomic.h \
 		      alisp.h pcm_external.h pcm_ioplug.h pcm_extplug.h \
-		      pcm_rate.h control_external.h
+		      pcm_rate.h control_external.h byteswap-compat.h \
+		      endian-compat.h
 
 noinst_HEADERS = alsa sys.h search.h list.h aserver.h local.h alsa-symbols.h
 
