diff -ur alsa-plugins-1.0.14.orig/configure.in alsa-plugins-1.0.14/configure.in
--- alsa-plugins-1.0.14.orig/configure.in	2007-05-31 15:06:51.000000000 +0700
+++ alsa-plugins-1.0.14/configure.in	2007-08-12 00:10:12.158016751 +0700
@@ -31,6 +31,23 @@
         AC_DEFINE(USE_RESOURCE_MANAGER, 1,"Use dbus server as a resouce manager")
 fi
 
+AC_CHECK_HEADERS([alloca.h endian.h byteswap.h sys/byteorder.h search.h err.h])
+AC_CHECK_HEADERS([sys/sockio.h features.h sys/ioccom.h linux/soundcard.h])
+
+dnl Check for -lnsl
+AC_CHECK_LIB([nsl], [gethostent], [HAVE_LIBNSL="yes"])
+if test "$HAVE_LIBNSL" = "yes"; then
+  LIBS="$LIBS -lnsl"
+  AC_DEFINE([HAVE_LIBNSL], 1, [Have libnsl])
+fi
+
+dnl Check for -lsocket
+AC_CHECK_LIB([socket], [setsockopt], [HAVE_LIBSOCKET="yes"])
+if test "$HAVE_LIBSOCKET" = "yes"; then
+  LIBS="$LIBS -lsocket"
+  AC_DEFINE([HAVE_LIBSOCKET], 1, [Have libsocket])
+fi
+
 AC_ARG_WITH([avcodec-includedir],
 	AS_HELP_STRING([--with-avcodec-includedir=dir],
 		[AVcodec include directory]),
--- alsa-plugins-1.0.14/configure.in.orig	2007-08-12 17:27:04.984403672 +0700
+++ alsa-plugins-1.0.14/configure.in	2007-08-12 16:49:55.651001349 +0700
@@ -18,7 +18,16 @@
 PKG_CHECK_MODULES(JACK, jack >= 0.98, [HAVE_JACK=yes], [HAVE_JACK=no])
 AM_CONDITIONAL(HAVE_JACK, test x$HAVE_JACK = xyes)
 
+AC_ARG_WITH([pulseaudio],
+     AS_HELP_STRING([--with-pulseaudio={yes|no}],
+	[build with Pulse Audio [[default=yes]]]),
+     [TESTPULSE=$withval] , [TESTPULSE="yes"])
+
+if test "$TESTPULSE" = "yes"; then
 PKG_CHECK_MODULES(pulseaudio, [libpulse >= 0.9.2], [HAVE_PULSE=yes], [HAVE_PULSE=no])
+else
+HAVE_PULSE=no
+fi
 AM_CONDITIONAL(HAVE_PULSE, test x$HAVE_PULSE = xyes)
 
 PKG_CHECK_MODULES(samplerate, [samplerate], [HAVE_SAMPLERATE=yes], [HAVE_SAMPLERATE=no])
