--- amrnb-6.1.0.4/prepare_sources.sh.orig	2007-08-01 14:28:35.797179645 +0700
+++ amrnb-6.1.0.4/prepare_sources.sh	2007-08-01 14:27:14.657711599 +0700
@@ -19,9 +19,9 @@
 
 rm -r c-code
 
-patch <amrnb-intsizes.patch
-patch <amrnb-strict-aliasing.patch
-patch <amrnb-any-cflags.patch
+gpatch <amrnb-intsizes.patch
+gpatch <amrnb-strict-aliasing.patch
+gpatch <amrnb-any-cflags.patch
 
 for FILE in interf_{dec,enc}.{c,h} ; do
     echo "modifying file $FILE"
