--- amrwb-7.0.0.1/prepare_sources.sh.orig	2007-08-01 14:42:14.288391298 +0700
+++ amrwb-7.0.0.1/prepare_sources.sh	2007-08-01 14:42:29.909685933 +0700
@@ -20,9 +20,9 @@
 
 rm -r c-code
 
-patch <amrwb-intsizes.patch
-patch <amrwb-any-cflags.patch
-patch <amrwb-dhf-declaration.patch
+gpatch <amrwb-intsizes.patch
+gpatch <amrwb-any-cflags.patch
+gpatch <amrwb-dhf-declaration.patch
 
 for FILE in {dec,enc}_if.{c,h} ; do
     echo "modifying file $FILE"
