Index: trunk/libanjuta/anjuta-launcher.c
===================================================================
--- trunk/libanjuta/anjuta-launcher.c	(revision 3982)
+++ trunk/libanjuta/anjuta-launcher.c	(working copy)
@@ -67,13 +67,13 @@
 #define FILE_INPUT_BUFFER_SIZE		1048576
 #ifndef __MAX_BAUD
 #  if defined(B460800)
-#    define __MAX_BAUX B460800
+#    define __MAX_BAUD B460800
 #  elif defined(B307200)
-#    define __MAX_BAUX B307200
+#    define __MAX_BAUD B307200
 #  elif defined(B256000)
-#    define __MAX_BAUX B256000
+#    define __MAX_BAUD B256000
 #  else
-#    define __MAX_BAUX B230400
+#    define __MAX_BAUD B230400
 #  endif
 #endif
 
