Index: anjuta-head/configure.in
===================================================================
--- anjuta-head/configure.in	(revision 2875)
+++ anjuta-head/configure.in	(working copy)
@@ -97,13 +97,13 @@
 CFLAGS_WARNINGS="-Wall -Wmissing-prototypes -Wmissing-declarations -Wparentheses -Wpointer-arith"
 WCXXFLAGS_WARNINGS="-Wall -Wparentheses -Wpointer-arith"
 
-if test "x$GCC" = xyes; then
+if test "x$GCC" = x; then
     CFLAGS_WARNINGS=""
     CXXFLAGS_WARNINGS=""
 fi
 
-AM_CFLAGS="$AM_CFLAGS -O0 -g $CFLAGS_WARNINGS"
-AM_CXXFLAGS="$AM_CXXFLAGS -O0 -g $CXXFLAGS_WARNINGS"
+AM_CFLAGS="$AM_CFLAGS $CFLAGS_WARNINGS"
+AM_CXXFLAGS="$AM_CXXFLAGS $CXXFLAGS_WARNINGS"
 
 dnl Enable debugging mode
 AC_ARG_ENABLE(debug,
Index: anjuta-head/global-tags/Makefile.am
===================================================================
--- anjuta-head/global-tags/Makefile.am	(revision 2875)
+++ anjuta-head/global-tags/Makefile.am	(working copy)
@@ -7,7 +7,7 @@
 bindir = $(anjuta_data_dir)/scripts
 dist_bin_SCRIPTS = create_global_tags.sh
 
-INCLUDES = -I$(top_srcdir)/tagmanager/include -Wall $(GLIB_CFLAGS)
+INCLUDES = -I$(top_srcdir)/tagmanager/include $(GLIB_CFLAGS)
 
 bin_PROGRAMS = anjuta-tags
 
