Index: anjuta-head/plugins/class-gen/window.c
===================================================================
--- anjuta-head/plugins/class-gen/window.c	(revision 2875)
+++ anjuta-head/plugins/class-gen/window.c	(working copy)
@@ -1105,7 +1105,7 @@
 	button = GTK_CHECK_BUTTON (glade_xml_get_widget (priv->gxml,
 	                                                 "add_project"));
 
-	return gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (button), enable);
+	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (button), enable);
 }
 
 void
@@ -1119,7 +1119,7 @@
 	button = GTK_CHECK_BUTTON (glade_xml_get_widget (priv->gxml,
 	                                                 "add_repository"));
 
-	return gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (button), enable);
+	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (button), enable);
 }
 
 gboolean
