--- anjuta-trunk/libtool.orig	2007-05-14 00:44:16.204338000 +0800
+++ anjuta-trunk/libtool	2007-05-14 00:44:32.811973000 +0800
@@ -255,7 +255,7 @@
 allow_undefined_flag=""
 
 # Flag that forces no undefined symbols.
-no_undefined_flag=" -z text"
+no_undefined_flag=""
 
 # Commands used to finish a libtool library installation in a directory.
 finish_cmds=""
@@ -7730,7 +7730,7 @@
 allow_undefined_flag=""
 
 # Flag that forces no undefined symbols.
-no_undefined_flag=" -z text"
+no_undefined_flag=""
 
 # Commands used to finish a libtool library installation in a directory.
 finish_cmds=""
