Index: trunk/manuals/reference/libanjuta/libanjuta-docs.sgml
===================================================================
--- trunk/manuals/reference/libanjuta/libanjuta-docs.sgml	(revision 4064)
+++ trunk/manuals/reference/libanjuta/libanjuta-docs.sgml	(revision 4167)
@@ -3,7 +3,6 @@
     "http://www.oasis-open.org/docbook/xml/4.1.2/docbookx.dtd" [
 <!ENTITY AnjutaArchitecture SYSTEM "anjuta-architecture.sgml">
 <!ENTITY WritingPlugins SYSTEM "writing-plugins.sgml">
-<!ENTITY WritingPluginsInVala SYSTEM "writing-plugins-vala.sgml">
 <!ENTITY PluginDescriptionFile SYSTEM "plugin-description-file.sgml">
 <!ENTITY ProjectWizardFormat SYSTEM "project-wizard-format.sgml">
 ]>
@@ -44,7 +43,6 @@
 
   &AnjutaArchitecture;
   &WritingPlugins;
-  &WritingPluginsInVala;
   &PluginDescriptionFile;
   &ProjectWizardFormat;
 
