diff -ur antlr-2.7.7-orig/Makefile.in antlr-2.7.7/Makefile.in
--- antlr-2.7.7-orig/Makefile.in	2006-11-02 04:37:18.000000000 +0700
+++ antlr-2.7.7/Makefile.in	2007-05-13 17:05:01.505188044 +0700
@@ -124,16 +124,16 @@
 extradir = $(datadir)/$(versioneddir)
 
 install ::
-	$(MKDIR) -p "$(bindir)"
-	$(MKDIR) -p "$(extradir)"
-	$(MKDIR) -p "$(docdir)"
-	$(INSTALL) -m 755 scripts/run-antlr                          "$(bindir)/antlr"
-	$(INSTALL) -m 755 scripts/antlr-config                       "$(bindir)/antlr-config"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(docdir)"
+	$(MKDIR) -p "$(DESTDIR)$(bindir)"
+	$(MKDIR) -p "$(DESTDIR)$(extradir)"
+	$(MKDIR) -p "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 755 scripts/run-antlr                          "$(DESTDIR)$(bindir)/antlr"
+	$(INSTALL) -m 755 scripts/antlr-config                       "$(DESTDIR)$(bindir)/antlr-config"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(DESTDIR)$(docdir)"
 
 install ::
 	@ECHO@ "installation done"
diff -ur antlr-2.7.7-orig/antlr/Makefile.in antlr-2.7.7/antlr/Makefile.in
--- antlr-2.7.7-orig/antlr/Makefile.in	2006-11-02 04:37:17.000000000 +0700
+++ antlr-2.7.7/antlr/Makefile.in	2007-05-13 17:10:48.660761885 +0700
@@ -321,11 +321,11 @@
 
 ## install antlr.jar in libdirectory if exists
 install ::	@ANTLR_JAR@ 
-	-$(MKDIR) -p "$(libdir)"
-	-$(MKDIR) -p "$(datadir)/$(versioneddir)"
+	-$(MKDIR) -p "$(DESTDIR)$(libdir)"
+	-$(MKDIR) -p "$(DESTDIR)$(datadir)/$(versioneddir)"
 	-test -f @ANTLR_JAR@ && {\
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(datadir)/$(versioneddir)" ; \
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(libdir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(libdir)" ; \
 	}
 
 distclean :: clean
diff -ur antlr-2.7.7-orig/doc/Makefile.in antlr-2.7.7/doc/Makefile.in
--- antlr-2.7.7-orig/doc/Makefile.in	2006-11-02 04:37:17.000000000 +0700
+++ antlr-2.7.7/doc/Makefile.in	2007-05-13 17:11:22.428191167 +0700
@@ -43,12 +43,12 @@
 	$(eol)
 
 install ::
-	@$(MKDIR) -p "$(antlr_doc_DIR)"
+	@$(MKDIR) -p "$(DESTDIR)$(antlr_doc_DIR)"
 	@@ECHO@ "install doc files .. "
 	@for f in $(antlr_doc_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_doc_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_doc_DIR)" ; \
 		fi ;\
 	done
 
diff -ur antlr-2.7.7-orig/lib/cpp/antlr/Makefile.in antlr-2.7.7/lib/cpp/antlr/Makefile.in
--- antlr-2.7.7-orig/lib/cpp/antlr/Makefile.in	2006-11-02 04:37:17.000000000 +0700
+++ antlr-2.7.7/lib/cpp/antlr/Makefile.in	2007-05-13 17:11:29.956236184 +0700
@@ -75,11 +75,11 @@
 
 install:
 	@@ECHO@ "install hpp files .. "
-	@$(MKDIR) -p "$(includedir)/antlr"
+	@$(MKDIR) -p "$(DESTDIR)$(includedir)/antlr"
 	@for f in $(antlr_hpp_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(includedir)/antlr" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(includedir)/antlr" ; \
 		fi ;\
 	done
 
diff -ur antlr-2.7.7-orig/lib/cpp/src/Makefile.in antlr-2.7.7/lib/cpp/src/Makefile.in
--- antlr-2.7.7-orig/lib/cpp/src/Makefile.in	2006-11-02 04:37:17.000000000 +0700
+++ antlr-2.7.7/lib/cpp/src/Makefile.in	2007-05-13 17:08:39.854205161 +0700
@@ -114,12 +114,12 @@
   $(eol)
 
 this-install: @ANTLR_LIB@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C++ core files .. "
 	@for f in $(antlr_lib_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 
diff -ur antlr-2.7.7-orig/lib/csharp/antlr.astframe/Makefile.in antlr-2.7.7/lib/csharp/antlr.astframe/Makefile.in
--- antlr-2.7.7-orig/lib/csharp/antlr.astframe/Makefile.in	2006-11-02 04:37:17.000000000 +0700
+++ antlr-2.7.7/lib/csharp/antlr.astframe/Makefile.in	2007-05-13 17:08:09.494692930 +0700
@@ -45,13 +45,13 @@
 
 ## install our target ..
 install :: @ASTFRAME_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C# core files .. "
 	@for f in @ASTFRAME_NET@  ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
diff -ur antlr-2.7.7-orig/lib/csharp/antlr.runtime/Makefile.in antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in
--- antlr-2.7.7-orig/lib/csharp/antlr.runtime/Makefile.in	2006-11-02 04:37:17.000000000 +0700
+++ antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in	2007-05-13 17:07:23.127331202 +0700
@@ -135,13 +135,13 @@
 	@RMF@ Makefile
 
 install :: @ANTLR_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C# core files .. "
 	@for f in @ANTLR_NET@  ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
diff -ur antlr-2.7.7-orig/lib/python/Makefile.in antlr-2.7.7/lib/python/Makefile.in
--- antlr-2.7.7-orig/lib/python/Makefile.in	2006-11-02 04:37:17.000000000 +0700
+++ antlr-2.7.7/lib/python/Makefile.in	2007-05-13 17:06:23.095918397 +0700
@@ -41,26 +41,26 @@
 extradir = $(datadir)/$(versioneddir)
 
 install-this:
-	$(MKDIR) -p "$(antlr_py_DIR)"
-	$(MKDIR) -p "$(libdir)"
+	$(MKDIR) -p "$(DESTDIR)$(antlr_py_DIR)"
+	$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install python files .. "
 	@for f in $(antlr_py_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 	@for f in $(antlr_py1_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_py_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_py_DIR)" ; \
 		fi ;\
 	done
 	@f="$(objdir)/scripts/pyantlr.sh"; \
 	if test -f "$${f}" ; then \
 			@ECHO@ "install Python installer" ; \
-			$(MKDIR) -p "$(sbindir)" ; \
-			$(INSTALL) -m 555 "$${f}" "$(sbindir)" ; \
+			$(MKDIR) -p "$(DESTDIR)$(sbindir)" ; \
+			$(INSTALL) -m 555 "$${f}" "$(DESTDIR)$(sbindir)" ; \
 	fi
 
 
