--- AssaultCube/assaultcube_server.sh.orig	2007-06-05 20:23:39.886803142 +0700
+++ AssaultCube/assaultcube_server.sh	2007-06-05 20:23:54.315405687 +0700
@@ -1,8 +1,8 @@
 #!/bin/sh
 # CUBE_DIR should refer to the directory in which Cube is placed.
 #CUBE_DIR=~/cube
-#CUBE_DIR=/usr/local/cube
-CUBE_DIR=./
+CUBE_DIR=/usr/share/cube
+#CUBE_DIR=./
 
 # SYSTEM_NAME should be set to the name of your operating system.
 #SYSTEM_NAME=Linux
--- AssaultCube/assaultcube.sh.orig	2007-06-05 20:02:14.820337638 +0700
+++ AssaultCube/assaultcube.sh	2007-06-05 20:23:29.472820232 +0700
@@ -1,8 +1,8 @@
 #!/bin/sh
 # CUBE_DIR should refer to the directory in which Cube is placed.
 #CUBE_DIR=~/cube
-#CUBE_DIR=/usr/local/cube
-CUBE_DIR=./
+CUBE_DIR=/usr/share/cube
+#CUBE_DIR=./
 
 # SYSTEM_NAME should be set to the name of your operating system.
 #SYSTEM_NAME=Linux
