diff -uNr asterisk-1.4.10.1/Makefile asterisk-1.4.10.1.patched/Makefile
--- asterisk-1.4.10.1/Makefile	2007-08-10 04:40:46.000000000 +0530
+++ asterisk-1.4.10.1.patched/Makefile	2007-08-13 00:55:19.084389000 +0530
@@ -428,7 +428,8 @@
 		chmod 755 $(DESTDIR)$(ASTSBINDIR)/safe_asterisk;\
 	fi
 	$(INSTALL) -d $(DESTDIR)$(ASTHEADERDIR)
-	$(INSTALL) -m 644 include/asterisk.h $(DESTDIR)$(includedir)
+	#$(INSTALL) -m 644 include/asterisk.h $(DESTDIR)$(includedir)
+	$(INSTALL) -m 644 include/asterisk.h $(DESTDIR)$(ASTHEADERDIR)
 	$(INSTALL) -m 644 include/asterisk/*.h $(DESTDIR)$(ASTHEADERDIR)
 	if [ -n "$(OLDHEADERS)" ]; then \
 		rm -f $(addprefix $(DESTDIR)$(ASTHEADERDIR)/,$(OLDHEADERS)) ;\
