--- audacity-src-1.3.5-beta/src/Makefile.in-orig	2008-05-12 17:30:16.132711000 -0500
+++ audacity-src-1.3.5-beta/src/Makefile.in	2008-05-12 17:30:30.874247000 -0500
@@ -234,7 +234,8 @@ libaudacity.a: $(LIBAUDACITY_OBJS)
 	ar rcs libaudacity.a $(LIBAUDACITY_OBJS)
 
 ../audacity: $(OBJS) $(LOCAL_LIBS:%=$(top_builddir)/lib-src/%) $(EXTRAOBJS) libaudacity.a Makefile
-	$(CCC) -o ../audacity $(OBJS) $(LDFLAGS) $(LOCAL_LIBS:%=$(top_builddir)/lib-src/%) $(EXTRAOBJS) libaudacity.a $(LIBS)
+	$(CCC) -o ../audacity $(OBJS) $(LDFLAGS) $(LOCAL_LIBS:%=$(top_builddir)/lib-src/%) $(EXTRAOBJS) libaudacity.a $(LIBS) `pkg-config --libs gtk+-2.0`
+
 	@AFTERBUILD@
 
 $(DIRS):
