--- audacity-src-1.3.3-beta/lib-src/portaudio-v19/Makefile.in-orig	2008-03-08 17:26:26.430765000 -0600
+++ audacity-src-1.3.3-beta/lib-src/portaudio-v19/Makefile.in	2008-03-08 17:37:00.020154000 -0600
@@ -188,10 +188,17 @@ Makefile: Makefile.in config.status
 	$(SHELL) config.status
 
 all-recursive:
-	for dir in $(SUBDIRS); do make -C $$dir all; done
+	if test -n "$(SUBDIRS)"; then \
+           for dir in "$(SUBDIRS)"; do make -C $$dir all; done; \
+        fi
 
 install-recursive:
-	for dir in $(SUBDIRS); do make -C $$dir install; done
+	if test -n "$(SUBDIRS)"; then \
+           for dir in "$(SUBDIRS)"; do make -C $$dir install; done; \
+        fi
 
 uninstall-recursive:
-	for dir in $(SUBDIRS); do make -C $$dir uninstall; done
+	if test -n "$(SUBDIRS)"; then \
+           for dir in "$(SUBDIRS)"; do make -C $$dir uninstall; done; \
+        fi
+
