--- audacity-src-1.3.4-beta/lib-src/libvamp/vamp-sdk/hostext/PluginLoader.cpp-orig	2008-03-17 20:10:29.506184000 -0500
+++ audacity-src-1.3.4-beta/lib-src/libvamp/vamp-sdk/hostext/PluginLoader.cpp	2008-03-17 20:10:35.527271000 -0500
@@ -550,7 +550,9 @@ PluginLoader::Impl::listFiles(string dir
     struct dirent *e = 0;
     while ((e = readdir(d))) {
         
+/*
         if (!(e->d_type & DT_REG) || !e->d_name) continue;
+*/
         
         size_t len = strlen(e->d_name);
         if (len < extlen + 2 ||
