--- audacity-src-1.3.4-beta/configure.in-orig	2008-03-17 19:45:33.380546000 -0500
+++ audacity-src-1.3.4-beta/configure.in	2008-03-17 20:02:04.435930000 -0500
@@ -371,9 +371,6 @@ dnl Gather wx arguments
 LIBS="$LIBS `$WX_CONFIG $static_preference $unicode_preference $debug_preference $wx_preference --libs`"
 CPPFLAGS="$CPPFLAGS `$WX_CONFIG $static_preference $unicode_preference $debug_preference $wx_preference --cxxflags`"
 
-dnl API define is only needed on Windows
-CPPFLAGS="$CPPFLAGS -DAUDACITY_DLL_API=''"
-
 AC_CANONICAL_HOST
 
 dnl OS-specific configuration
