--- audacity-src-1.3.3-beta/src/Matrix.cpp-orig	2008-03-08 16:51:53.609554000 -0600
+++ audacity-src-1.3.3-beta/src/Matrix.cpp	2008-03-08 16:53:43.297642000 -0600
@@ -262,7 +262,7 @@ Matrix operator+(const Matrix &left, con
    return M;
 }
 
-Matrix operator*(const Matrix &left, double right)
+Matrix operator*(const Matrix &left, const double right)
 {
    Matrix M(left.Rows(), left.Cols());
    int i, j;
