--- audacity-src-1.3.5-beta/locale/Makefile.in-orig	2008-05-12 18:07:40.712680000 -0500
+++ audacity-src-1.3.5-beta/locale/Makefile.in	2008-05-12 18:12:27.379482000 -0500
@@ -8,7 +8,7 @@
 # This makefile may be invoked to build either audacity.pot or any lang.mo
 
 PROGNAME=audacity
-LINGUAS=af ar bn bg ca cs cy da de el es eu fi fr ga gl he hu it ja ko lt mk nb nl oc pl pt_BR ro ru sk sl sv tr uk vi zh zh_TW
+LINGUAS=af ar bn bg cy es eu fi ga gl he hu it ja ko lt mk nl ro sk sv tr uk vi zh zh_TW
 
 prefix=@prefix@
 DATADIR=@datadir@
