--- audacity-src-1.3.3-beta/acinclude.m4.orig	2007-09-22 16:21:06.666140644 +0700
+++ audacity-src-1.3.3-beta/acinclude.m4	2007-09-22 16:20:52.269784270 +0700
@@ -49,8 +49,20 @@
    fi
 
    dnl Libtwolame is so new, we should really use our patched local version for now
-   dnl FIXME: Might want to change this at some point of time in the future...
-   LIBTWOLAME_SYSTEM_AVAILABLE="no"
+   PKG_CHECK_MODULES(TWOLAME, twolame,
+        libtwolame_available_system="yes",
+        libtwolame_available_system="no")
+
+   if test "x$libtwolame_available_system" = "xyes" ; then
+	LIBTWOLAME_SYSTEM_AVAILABLE="yes"
+	LIBTWOLAME_SYSTEM_LIBS=$TWOLAME_LIBS
+	LIBTWOLAME_SYSTEM_CXXFLAGS=$TWOLAME_CFLAGS
+	LIBTWOLAME_SYSTEM_CPPSYMBOLS="USE_TWOLAME"
+	AC_MSG_NOTICE([twolame libraries are available as system libraries])
+   else
+	LIBTWOLAME_SYSTEM_AVAILABLE="no"
+	AC_MSG_NOTICE([twolame libraries are NOT available as system libraries])
+   fi
 
    dnl see if libtwolame is available locally
 
@@ -72,6 +84,7 @@
       LIBTWOLAME_LOCAL_AVAILABLE="no"
       AC_MSG_NOTICE([libtwolame library is NOT available in the local tree])
    fi
+
 ])
 
 AC_DEFUN([AUDACITY_CHECKLIB_LIBSOUNDTOUCH], [
