--- audacity-src-1.3.3-beta/lib-src/portaudio-v19/configure.in-orig	2008-03-08 17:20:51.427029000 -0600
+++ audacity-src-1.3.3-beta/lib-src/portaudio-v19/configure.in	2008-03-08 17:21:02.809670000 -0600
@@ -287,7 +287,7 @@ case "${host_os}" in
                 AC_DEFINE(PA_USE_ASIHPI)
 	fi
 	
-	THREAD_CFLAGS="-pthread"
+	THREAD_CFLAGS=""
 	DLL_LIBS="$DLL_LIBS -lm -lpthread";
 	LIBS="$LIBS -lm -lpthread";
 	PADLL="libportaudio.so";
