--- audacity-src-1.3.3-beta/locale/Makefile.in-orig	2008-03-08 17:41:03.802962000 -0600
+++ audacity-src-1.3.3-beta/locale/Makefile.in	2008-03-08 17:42:28.192237000 -0600
@@ -8,7 +8,7 @@
 # This makefile may be invoked to build either audacity.pot or any lang.mo
 
 PROGNAME=audacity
-LINGUAS=af ar bn bg ca cs cy da de el es eu fi fr ga gl hu it ja lt mk nb nl oc pl pt ro ru sk sl sv tr uk zh zh_TW
+LINGUAS=af ar bn bg ca cs cy el es eu fi ga gl hu it ja lt mk nl pt ro ru sk sl sv tr zh zh_TW
 
 prefix=@prefix@
 # this defaults to /usr/share
