--- audacity-src-1.3.4-beta/locale/Makefile.in-orig	2008-03-18 02:07:47.925150000 -0500
+++ audacity-src-1.3.4-beta/locale/Makefile.in	2008-03-18 02:08:11.991282000 -0500
@@ -8,7 +8,8 @@
 # This makefile may be invoked to build either audacity.pot or any lang.mo
 
 PROGNAME=audacity
-LINGUAS=af ar bn bg ca cs cy da de el es eu fi fr ga gl hu it ja ko lt mk nb nl oc pl pt_BR ro ru sk sl sv tr uk zh zh_TW
+LINGUAS=af ar bn bg cs cy es eu fi ga gl hu it ja lt mk nl ro ru sk sv tr zh_TW
+
 
 prefix=@prefix@
 DATADIR=@datadir@
