diff -ur CEGUI-0.5.0-orig/include/CEGUIDynamicModule.h CEGUI-0.5.0/include/CEGUIDynamicModule.h
--- CEGUI-0.5.0-orig/include/CEGUIDynamicModule.h	2006-05-13 02:42:23.000000000 +0700
+++ CEGUI-0.5.0/include/CEGUIDynamicModule.h	2007-05-07 12:49:07.566455751 +0700
@@ -50,6 +50,15 @@
 #    define DYNLIB_UNLOAD( a ) dlclose( a )
 #    define DYNLIB_ERROR( ) dlerror( )
 
+#elif defined(sun)
+#include <dlfcn.h>
+#include <link.h>
+#    define DYNLIB_HANDLE void*
+#    define DYNLIB_LOAD( a ) dlopen( a, RTLD_LAZY )
+#    define DYNLIB_GETSYM( a, b ) dlsym( a, b )
+#    define DYNLIB_UNLOAD( a ) dlclose( a )
+#    define DYNLIB_ERROR( ) dlerror( )
+
 #elif defined(__APPLE_CC__)
 #    define DYNLIB_HANDLE CFBundleRef
 #    define DYNLIB_LOAD( a ) mac_loadExeBundle( a )
