diff -ur CEGUI-0.5.0-orig/Samples/common/src/CEGuiSample.cpp CEGUI-0.5.0/Samples/common/src/CEGuiSample.cpp
--- CEGUI-0.5.0-orig/Samples/common/src/CEGuiSample.cpp	2006-10-30 16:44:57.000000000 +0700
+++ CEGUI-0.5.0/Samples/common/src/CEGuiSample.cpp	2007-05-07 12:59:04.778386191 +0700
@@ -35,7 +35,7 @@
 // includes for renderer selector classes
 #if defined( __WIN32__ ) || defined( _WIN32 )
 #   include "Win32CEGuiRendererSelector.h"
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(sun)
 #   ifdef CEGUI_SAMPLES_USE_GTK2
 #       include "GTK2CEGuiRendererSelector.h"
 #   else
@@ -153,7 +153,7 @@
     d_rendererSelector->setRendererAvailability(Direct3D9GuiRendererType);
 #endif
 
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(sun)
     // decide which method to use for renderer selection
 #   ifdef CEGUI_SAMPLES_USE_GTK2
         d_rendererSelector = new GTK2CEGuiRendererSelector();
