Index: check-trunk/src/check.h.in
===================================================================
--- check-trunk/src/check.h.in	(revision 408)
+++ check-trunk/src/check.h.in	(working copy)
@@ -182,6 +182,10 @@
 /* Start a unit test with START_TEST(unit_name), end with END_TEST
    One must use braces within a START_/END_ pair to declare new variables
 */ 
+#ifndef __attribute__
+#define __attribute__(X)
+#endif
+
 #define START_TEST(__testname)\
 static void __testname (int _i __attribute__((unused)))\
 {\
