--- ./src/cheese-webcam.c.orig	2008-09-01 11:16:01.916287000 +0800
+++ ./src/cheese-webcam.c	2008-09-01 11:17:05.996507000 +0800
@@ -620,8 +620,13 @@
     }
   }
     
+#ifdef SOLARIS    
+  *numerator = 15;
+  *denominator = 2;
+#else  
   *numerator = framerate_numerator;
   *denominator = framerate_denominator;
+#endif  
 }
 
 static gboolean 
@@ -956,7 +961,7 @@
     return FALSE;
 
   gst_bin_add_many (GST_BIN (priv->video_save_bin), priv->audio_source, audio_queue,
-                    audio_convert, audio_enc, video_save_csp, video_save_rate, video_save_scale, video_enc, 
+                    audio_convert, audio_enc, video_save_csp, video_save_scale, video_enc, 
                     mux, priv->video_file_sink, NULL);
 
   /* add ghostpad */
@@ -968,7 +973,7 @@
   ok = gst_element_link_many (priv->audio_source, audio_queue, audio_convert, 
                               audio_enc, mux, priv->video_file_sink, NULL);
 
-  ok &= gst_element_link_many (video_save_csp, video_save_rate, video_save_scale, video_enc,
+  ok &= gst_element_link_many (video_save_csp, video_save_scale, video_enc,
                                NULL);
   ok &= gst_element_link (video_enc, mux);
 
--- ./src/cheese-gconf.c.orig	2008-08-19 02:32:53.000000000 +0800
+++ ./src/cheese-gconf.c	2008-09-01 11:17:05.997071000 +0800
@@ -103,14 +103,18 @@
       g_slist_free (tmp);
       break;
     case GCONF_PROP_X_RESOLUTION:
+      gdk_threads_enter();
       g_value_set_int (value, gconf_client_get_int (priv->client,
                                                     CHEESE_GCONF_PREFIX "/x_resolution",
                                                     NULL));
+      gdk_threads_leave();                                              
       break;
     case GCONF_PROP_Y_RESOLUTION:
+      gdk_threads_enter();
       g_value_set_int (value, gconf_client_get_int (priv->client,
                                                     CHEESE_GCONF_PREFIX "/y_resolution",
                                                     NULL));
+      gdk_threads_leave();                                              
       break;
     case GCONF_PROP_VIDEO_PATH:
       g_value_set_string (value, gconf_client_get_string (priv->client,
