--- codeina-0.10.2.1/codeina/check.py-orig	2008-04-29 12:42:00.538098000 -0500
+++ codeina-0.10.2.1/codeina/check.py	2008-04-29 12:47:47.217251000 -0500
@@ -297,8 +297,17 @@ def find_system_details():
     global GST_PLUGINS_BASE_VERSION
 
     u = os.uname()
-    OS_NAME = u[0]
-    ARCH_NAME = u[4]
+#   OS_NAME = u[0]
+#   ARCH_NAME = u[4]
+# os.uname() returns "SunOS" for OS_NAME and "i386pc" for ARCH_NAME
+# on my i386 box.
+#
+# However, to work with the Fluendo store we need ARCH_NAME to be
+# intel or sparc, and OS_NAME needs to be "solaris".  So just
+# hardcode.
+#
+    OS_NAME = "solaris"
+    ARCH_NAME = "intel"
 
   # for now munge everything x86 to i386
 
@@ -306,21 +315,27 @@ def find_system_details():
         ARCH_NAME = "i386"
 
   # Find distro name
-    DISTRO_NAME = _get_distro_id_from_lsb_release()
+  # DISTRO_NAME = _get_distro_id_from_lsb_release()
 
-    if not DISTRO_NAME:
-        DISTRO_NAME = _get_distro_id_from_release_file()
+  # if not DISTRO_NAME:
+  #     DISTRO_NAME = _get_distro_id_from_release_file()
 
-    if not DISTRO_NAME:
-        DISTRO_NAME = 'generic'
+  # if not DISTRO_NAME:
+  #     DISTRO_NAME = 'generic'
 
   # Munge distro name to our liking (e.g. lsb-release will return 'fedoracore')
-    if DISTRO_NAME.startswith('fedora'):
-        DISTRO_NAME = 'fedora'
+  # if DISTRO_NAME.startswith('fedora'):
+  #     DISTRO_NAME = 'fedora'
+
+  # DISTRO_VERSION = _get_distro_version_from_lsb_release ()
+  # if not DISTRO_VERSION:
+  #     DISTRO_VERSION = 'any'
+
+  # Just hack DISTRO_NAME to "solaris" and DISTRO_VERSION to "any".
+  #
+    DISTRO_NAME = 'solaris'
+    DISTRO_VERSION = 'any'
 
-    DISTRO_VERSION = _get_distro_version_from_lsb_release ()
-    if not DISTRO_VERSION:
-        DISTRO_VERSION = 'any'
   # Find out current GStreamer core version
 
     GST_CORE_VERSION = Version(_version_to_string(gst.version()))
