--- codeina-0.10.3.1/bin/codeina-orig	2008-10-13 11:04:28.889497000 -0500
+++ codeina-0.10.3.1/bin/codeina	2008-10-13 11:04:35.571506000 -0500
@@ -3,7 +3,7 @@
 # this script sets MOZILLA_FIVE_HOME and LD_LIBRARY_PATH as needed
 # by the gtkmozembed python module.
 
-for p in /usr/lib{64,}/firefox*/libgtkembedmoz.so; do
+for p in /usr/lib{64,}/firefox*/libxul.so; do
   [ -f "$p" ] || continue
   latest_firefox="$(dirname "$p")"/
   export LD_LIBRARY_PATH="$latest_firefox":$LD_LIBRARY_PATH
