diff -urN compiz-0.6.0.orig/metadata/annotate.xml compiz-0.6.0/metadata/annotate.xml
--- compiz-0.6.0.orig/metadata/annotate.xml	2007-10-01 13:43:27.000000000 -0400
+++ compiz-0.6.0/metadata/annotate.xml	2007-10-31 00:52:10.047172630 -0400
@@ -34,7 +34,7 @@
 		<long xml:lang="pl">Rozpocznij rysowanie</long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Alt&gt;Button1</button>
+		    <button>&lt;Mod4&gt;&lt;Alt&gt;Button1</button>
 		</default>
 	    </option>
 	    <option type="action" name="draw">
@@ -58,7 +58,7 @@
 		<long xml:lang="sv">Initiera radering av anteckning</long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Alt&gt;Button3</button>
+		    <button>&lt;Mod4&gt;&lt;Alt&gt;Button3</button>
 		</default>
 	    </option>
 	    <option type="action" name="clear">
@@ -76,7 +76,7 @@
 		<long xml:lang="sv">Töm</long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <key>&lt;Super&gt;&lt;Alt&gt;k</key>
+		    <key>&lt;Mod4&gt;&lt;Alt&gt;k</key>
 		</default>
 	    </option>
 	    <option type="color" name="fill_color">
diff -urN compiz-0.6.0.orig/metadata/annotate.xml.in compiz-0.6.0/metadata/annotate.xml.in
--- compiz-0.6.0.orig/metadata/annotate.xml.in	2007-08-23 06:06:48.000000000 -0400
+++ compiz-0.6.0/metadata/annotate.xml.in	2007-10-31 00:52:10.050263204 -0400
@@ -8,7 +8,7 @@
 		<_long>Initiate annotate drawing</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Alt&gt;Button1</button>
+		    <button>&lt;Mod4&gt;&lt;Alt&gt;Button1</button>
 		</default>
 	    </option>
 	    <option name="draw" type="action">
@@ -20,7 +20,7 @@
 		<_long>Initiate annotate erasing</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Alt&gt;Button3</button>
+		    <button>&lt;Mod4&gt;&lt;Alt&gt;Button3</button>
 		</default>
 	    </option>
 	    <option name="clear" type="action">
@@ -28,7 +28,7 @@
 		<_long>Clear</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <key>&lt;Super&gt;&lt;Alt&gt;k</key>
+		    <key>&lt;Mod4&gt;&lt;Alt&gt;k</key>
 		</default>
 	    </option>
 	    <option name="fill_color" type="color">
diff -urN compiz-0.6.0.orig/metadata/clone.xml compiz-0.6.0/metadata/clone.xml
--- compiz-0.6.0.orig/metadata/clone.xml	2007-10-01 13:43:27.000000000 -0400
+++ compiz-0.6.0/metadata/clone.xml	2007-10-31 00:52:10.050053819 -0400
@@ -33,7 +33,7 @@
 		<long xml:lang="pl">Rozpocznij klonowanie zaznaczenia</long>
 		<allowed button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Shift&gt;Button1</button>
+		    <button>&lt;Mod4&gt;&lt;Shift&gt;Button1</button>
 		</default>
 	    </option>
 	</display>
diff -urN compiz-0.6.0.orig/metadata/clone.xml.in compiz-0.6.0/metadata/clone.xml.in
--- compiz-0.6.0.orig/metadata/clone.xml.in	2007-08-23 06:06:48.000000000 -0400
+++ compiz-0.6.0/metadata/clone.xml.in	2007-10-31 00:52:10.049274921 -0400
@@ -8,7 +8,7 @@
 		<_long>Initiate clone selection</_long>
 		<allowed button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Shift&gt;Button1</button>
+		    <button>&lt;Mod4&gt;&lt;Shift&gt;Button1</button>
 		</default>
 	    </option>
 	</display>
diff -urN compiz-0.6.0.orig/metadata/compiz-annotate.schemas compiz-0.6.0/metadata/compiz-annotate.schemas
--- compiz-0.6.0.orig/metadata/compiz-annotate.schemas	2007-10-01 13:43:31.000000000 -0400
+++ compiz-0.6.0/metadata/compiz-annotate.schemas	2007-10-31 00:52:10.049867257 -0400
@@ -85,7 +85,7 @@
       <applyto>/apps/compiz/plugins/annotate/allscreens/options/initiate_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;&lt;Alt&gt;Button1</default>
+      <default>&lt;Mod4&gt;&lt;Alt&gt;Button1</default>
       <locale name="C">
         <short>Initiate</short>
         <long>Initiate annotate drawing</long>
@@ -187,7 +187,7 @@
       <applyto>/apps/compiz/plugins/annotate/allscreens/options/erase_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;&lt;Alt&gt;Button3</default>
+      <default>&lt;Mod4&gt;&lt;Alt&gt;Button3</default>
       <locale name="C">
         <short>Initiate erase</short>
         <long>Initiate annotate erasing</long>
@@ -210,7 +210,7 @@
       <applyto>/apps/compiz/plugins/annotate/allscreens/options/clear_key</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;&lt;Alt&gt;k</default>
+      <default>&lt;Mod4&gt;&lt;Alt&gt;k</default>
       <locale name="C">
         <short>Clear</short>
         <long>Clear</long>
diff -urN compiz-0.6.0.orig/metadata/compiz-clone.schemas compiz-0.6.0/metadata/compiz-clone.schemas
--- compiz-0.6.0.orig/metadata/compiz-clone.schemas	2007-10-01 13:43:31.000000000 -0400
+++ compiz-0.6.0/metadata/compiz-clone.schemas	2007-10-31 00:52:10.050508599 -0400
@@ -6,7 +6,7 @@
       <applyto>/apps/compiz/plugins/clone/allscreens/options/initiate_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;&lt;Shift&gt;Button1</default>
+      <default>&lt;Mod4&gt;&lt;Shift&gt;Button1</default>
       <locale name="C">
         <short>Initiate</short>
         <long>Initiate clone selection</long>
diff -urN compiz-0.6.0.orig/metadata/compiz-screenshot.schemas compiz-0.6.0/metadata/compiz-screenshot.schemas
--- compiz-0.6.0.orig/metadata/compiz-screenshot.schemas	2007-10-01 13:43:31.000000000 -0400
+++ compiz-0.6.0/metadata/compiz-screenshot.schemas	2007-10-31 00:52:10.049115442 -0400
@@ -6,7 +6,7 @@
       <applyto>/apps/compiz/plugins/screenshot/allscreens/options/initiate_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;Button1</default>
+      <default>&lt;Mod4&gt;Button1</default>
       <locale name="C">
         <short>Initiate</short>
         <long>Initiate rectangle screenshot</long>
diff -urN compiz-0.6.0.orig/metadata/compiz-water.schemas compiz-0.6.0/metadata/compiz-water.schemas
--- compiz-0.6.0.orig/metadata/compiz-water.schemas	2007-10-01 13:43:31.000000000 -0400
+++ compiz-0.6.0/metadata/compiz-water.schemas	2007-10-31 00:52:10.048613191 -0400
@@ -6,7 +6,7 @@
       <applyto>/apps/compiz/plugins/water/allscreens/options/initiate_key</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Control&gt;&lt;Super&gt;</default>
+      <default>&lt;Control&gt;&lt;Mod4&gt;</default>
       <locale name="C">
         <short>Initiate</short>
         <long>Enable pointer water effects</long>
diff -urN compiz-0.6.0.orig/metadata/compiz-zoom.schemas compiz-0.6.0/metadata/compiz-zoom.schemas
--- compiz-0.6.0.orig/metadata/compiz-zoom.schemas	2007-10-01 13:43:31.000000000 -0400
+++ compiz-0.6.0/metadata/compiz-zoom.schemas	2007-10-31 00:52:10.046611690 -0400
@@ -85,7 +85,7 @@
       <applyto>/apps/compiz/plugins/zoom/allscreens/options/initiate_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;Button3</default>
+      <default>&lt;Mod4&gt;Button3</default>
       <locale name="C">
         <short>Initiate</short>
         <long>Zoom In</long>
@@ -243,7 +243,7 @@
       <applyto>/apps/compiz/plugins/zoom/allscreens/options/zoom_in_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;Button4</default>
+      <default>&lt;Mod4&gt;Button4</default>
       <locale name="C">
         <short>Zoom In</short>
         <long>Zoom In</long>
@@ -401,7 +401,7 @@
       <applyto>/apps/compiz/plugins/zoom/allscreens/options/zoom_out_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;Button5</default>
+      <default>&lt;Mod4&gt;Button5</default>
       <locale name="C">
         <short>Zoom Out</short>
         <long>Zoom Out</long>
@@ -491,7 +491,7 @@
       <applyto>/apps/compiz/plugins/zoom/allscreens/options/zoom_pan_button</applyto>
       <owner>compiz</owner>
       <type>string</type>
-      <default>&lt;Super&gt;Button2</default>
+      <default>&lt;Mod4&gt;Button2</default>
       <locale name="C">
         <short>Zoom Pan</short>
         <long>Zoom pan</long>
diff -urN compiz-0.6.0.orig/metadata/core.xml.in compiz-0.6.0/metadata/core.xml.in
--- compiz-0.6.0.orig/metadata/core.xml.in	2007-09-17 12:48:31.000000000 -0400
+++ compiz-0.6.0/metadata/core.xml.in	2007-10-31 00:51:19.222758679 -0400
@@ -7,6 +7,33 @@
 		<_short>Active Plugins</_short>
 		<_long>List of currently active plugins</_long>
 		<type>string</type>
+		<default>
+			<value>jpeg</value>
+			<value>svg</value>
+			<value>png</value>
+			<value>decoration</value>
+			<value>wobbly</value>
+			<value>fade</value>
+			<value>expo</value>
+			<value>minimize</value>
+			<value>cube</value>
+			<value>showdesktop</value>
+			<value>switcher</value>
+			<value>move</value>
+			<value>resize</value>
+			<value>place</value>
+			<value>rotate</value>
+			<value>zoom</value>
+			<value>scale</value>
+			<value>dbus</value>
+			<value>water</value>		
+			<value>snow</value>		
+			<value>animation</value>		
+			<value>firepaint</value>		
+			<value>thumbnail</value>		
+			<value>switcher</value>		
+			<value>cubereflex</value>		
+		</default>
 	    </option>
 	    <option name="texture_filter" type="int">
 		<_short>Texture Filter</_short>
diff -urN compiz-0.6.0.orig/metadata/cube.xml.in compiz-0.6.0/metadata/cube.xml.in
--- compiz-0.6.0.orig/metadata/cube.xml.in	2007-08-23 06:06:48.000000000 -0400
+++ compiz-0.6.0/metadata/cube.xml.in	2007-10-31 00:51:19.223248342 -0400
@@ -42,9 +42,9 @@
 		<_short>Cube Color</_short>
 		<_long>Color of top and bottom sides of the cube</_long>
 		<default>
-		    <red>0xfefe</red>
+		    <red>0xffff</red>
 		    <green>0xffff</green>
-		    <blue>0xc7c7</blue>
+		    <blue>0xffff</blue>
 		</default>
 	    </option>
 	    <option name="in" type="bool">
@@ -55,14 +55,14 @@
 	    <option name="scale_image" type="bool">
 		<_short>Scale image</_short>
 		<_long>Scale images to cover top face of cube</_long>
-		<default>false</default>
+		<default>true</default>
 	    </option>
 	    <option name="images" type="list">
 		<_short>Image files</_short>
 		<_long>List of PNG and SVG files that should be rendered on top face of cube</_long>
 		<type>string</type>
 		<default>
-		    <value>freedesktop</value>
+		    <value>/usr/share/pixmaps/backgrounds/sun/di-logo-sun-blue.png</value>
 		</default>
 	    </option>
 	    <option name="skydome" type="bool">
@@ -139,7 +139,7 @@
 	    <option name="active_opacity" type="float">
 		<_short>Opacity During Rotation</_short>
 		<_long>Opacity of desktop window during rotation.</_long>
-		<default>100.0</default>
+		<default>60.0</default>
 		<min>0.0</min>
 		<max>100.0</max>
 		<precision>1.0</precision>
diff -urN compiz-0.6.0.orig/metadata/screenshot.xml compiz-0.6.0/metadata/screenshot.xml
--- compiz-0.6.0.orig/metadata/screenshot.xml	2007-10-01 13:43:30.000000000 -0400
+++ compiz-0.6.0/metadata/screenshot.xml	2007-10-31 00:52:10.048829146 -0400
@@ -34,7 +34,7 @@
 		<long xml:lang="pl">Rozpocznik prostokątny zrzut ekranu</long>
 		<allowed button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button1</button>
+		    <button>&lt;Mod4&gt;Button1</button>
 		</default>
 	    </option>
 	    <option type="string" name="directory">
diff -urN compiz-0.6.0.orig/metadata/screenshot.xml.in compiz-0.6.0/metadata/screenshot.xml.in
--- compiz-0.6.0.orig/metadata/screenshot.xml.in	2007-08-23 06:06:48.000000000 -0400
+++ compiz-0.6.0/metadata/screenshot.xml.in	2007-10-31 00:52:10.051214817 -0400
@@ -8,7 +8,7 @@
 		<_long>Initiate rectangle screenshot</_long>
 		<allowed button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button1</button>
+		    <button>&lt;Mod4&gt;Button1</button>
 		</default>
 	    </option>
 	    <option name="directory" type="string">
diff -urN compiz-0.6.0.orig/metadata/water.xml compiz-0.6.0/metadata/water.xml
--- compiz-0.6.0.orig/metadata/water.xml	2007-10-01 13:43:30.000000000 -0400
+++ compiz-0.6.0/metadata/water.xml	2007-10-31 00:52:10.051078107 -0400
@@ -83,7 +83,7 @@
 		<long xml:lang="zh_TW">啟用指標水波特效</long>
 		<allowed key="true"/>
 		<default>
-		    <key>&lt;Control&gt;&lt;Super&gt;</key>
+		    <key>&lt;Control&gt;&lt;Mod4&gt;</key>
 		</default>
 	    </option>
 	    <option type="action" name="toggle_rain">
diff -urN compiz-0.6.0.orig/metadata/water.xml.in compiz-0.6.0/metadata/water.xml.in
--- compiz-0.6.0.orig/metadata/water.xml.in	2007-08-23 06:06:48.000000000 -0400
+++ compiz-0.6.0/metadata/water.xml.in	2007-10-31 00:52:10.047945174 -0400
@@ -14,7 +14,7 @@
 		<_long>Enable pointer water effects</_long>
 		<allowed key="true"/>
 		<default>
-		    <key>&lt;Control&gt;&lt;Super&gt;</key>
+		    <key>&lt;Control&gt;&lt;Mod4&gt;</key>
 		</default>
 	    </option>
 	    <option name="toggle_rain" type="action">
diff -urN compiz-0.6.0.orig/metadata/zoom.xml compiz-0.6.0/metadata/zoom.xml
--- compiz-0.6.0.orig/metadata/zoom.xml	2007-10-01 13:43:31.000000000 -0400
+++ compiz-0.6.0/metadata/zoom.xml	2007-10-31 00:52:10.047716734 -0400
@@ -77,7 +77,7 @@
 		<long xml:lang="zh_TW">放大</long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button3</button>
+		    <button>&lt;Mod4&gt;Button3</button>
 		</default>
 	    </option>
 	    <option type="action" name="zoom_in">
@@ -119,7 +119,7 @@
 		<long xml:lang="zh_TW">放大</long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button4</button>
+		    <button>&lt;Mod4&gt;Button4</button>
 		</default>
 	    </option>
 	    <option type="action" name="zoom_out">
@@ -161,7 +161,7 @@
 		<long xml:lang="zh_TW">縮小</long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button5</button>
+		    <button>&lt;Mod4&gt;Button5</button>
 		</default>
 	    </option>
 	    <option type="action" name="zoom_pan">
@@ -169,7 +169,7 @@
 		<long>Zoom pan</long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button2</button>
+		    <button>&lt;Mod4&gt;Button2</button>
 		</default>
 	    </option>
 	</display>
diff -urN compiz-0.6.0.orig/metadata/zoom.xml.in compiz-0.6.0/metadata/zoom.xml.in
--- compiz-0.6.0.orig/metadata/zoom.xml.in	2007-08-23 06:06:48.000000000 -0400
+++ compiz-0.6.0/metadata/zoom.xml.in	2007-10-31 00:52:10.046852442 -0400
@@ -8,7 +8,7 @@
 		<_long>Zoom In</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button3</button>
+		    <button>&lt;Mod4&gt;Button3</button>
 		</default>
 	    </option>
 	    <option name="zoom_in" type="action">
@@ -16,7 +16,7 @@
 		<_long>Zoom In</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button4</button>
+		    <button>&lt;Mod4&gt;Button4</button>
 		</default>
 	    </option>
 	    <option name="zoom_out" type="action">
@@ -24,7 +24,7 @@
 		<_long>Zoom Out</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button5</button>
+		    <button>&lt;Mod4&gt;Button5</button>
 		</default>
 	    </option>
 	    <option name="zoom_pan" type="action">
@@ -32,7 +32,7 @@
 		<_long>Zoom pan</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button2</button>
+		    <button>&lt;Mod4&gt;Button2</button>
 		</default>
 	    </option>
 	</display>
