diff -rup compiz-0.5.2/configure.ac ../compiz-0.5.2/configure.ac
--- compiz-0.5.2/configure.ac	2007-08-03 22:00:55.000000000 +0200
+++ ../compiz-0.5.2/configure.ac	2007-08-30 17:12:05.898401000 +0200
@@ -102,6 +102,16 @@ if test "x$GCC" = "xyes"; then
   esac
 fi
 
+compiling_on_solaris=no
+case "$host" in
+        *-*-solaris*)
+	  AC_MSG_CHECKING(if compiling on Solaris)
+	  compiling_on_solaris=yes
+	  AC_MSG_RESULT($compiling_on_solaris)
+	  AC_DEFINE(SOLARIS, 1, [Compiling on Solaris])
+	;;
+esac
+
 AC_C_BIGENDIAN
 
 plugindir=$libdir/compiz
diff -rup compiz-0.5.2/metadata/annotate.xml.in ../compiz-0.5.2/metadata/annotate.xml.in
--- compiz-0.5.2/metadata/annotate.xml.in	2007-05-21 16:33:36.000000000 +0200
+++ ../compiz-0.5.2/metadata/annotate.xml.in	2007-08-30 17:13:51.177995000 +0200
@@ -8,7 +8,7 @@
 		<_long>Initiate annotate drawing</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Alt&gt;Button1</button>
+		    <button>&lt;Mod4&gt;&lt;Alt&gt;Button1</button>
 		</default>
 	    </option>
 	    <option name="draw" type="action">
@@ -20,7 +20,7 @@
 		<_long>Initiate annotate erasing</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Alt&gt;Button3</button>
+		    <button>&lt;Mod4&gt;&lt;Alt&gt;Button3</button>
 		</default>
 	    </option>
 	    <option name="clear" type="action">
@@ -28,7 +28,7 @@
 		<_long>Clear</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <key>&lt;Super&gt;&lt;Alt&gt;k</key>
+		    <key>&lt;Mod4&gt;&lt;Alt&gt;k</key>
 		</default>
 	    </option>
 	    <option name="fill_color" type="color">
diff -rup compiz-0.5.2/metadata/clone.xml.in ../compiz-0.5.2/metadata/clone.xml.in
--- compiz-0.5.2/metadata/clone.xml.in	2007-04-27 20:36:57.000000000 +0200
+++ ../compiz-0.5.2/metadata/clone.xml.in	2007-08-30 17:13:51.178044000 +0200
@@ -8,7 +8,7 @@
 		<_long>Initiate clone selection</_long>
 		<allowed button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Shift&gt;Button1</button>
+		    <button>&lt;Mod4&gt;&lt;Shift&gt;Button1</button>
 		</default>
 	    </option>
 	</display>
diff -rup compiz-0.5.2/metadata/core.xml.in ../compiz-0.5.2/metadata/core.xml.in
--- compiz-0.5.2/metadata/core.xml.in	2007-05-21 21:10:20.000000000 +0200
+++ ../compiz-0.5.2/metadata/core.xml.in	2007-08-30 17:13:51.178385000 +0200
@@ -7,6 +7,33 @@
 		<_short>Active Plugins</_short>
 		<_long>List of currently active plugins</_long>
 		<type>string</type>
+		<default>
+			<value>jpeg</value>
+			<value>svg</value>
+			<value>png</value>
+			<value>decoration</value>
+			<value>wobbly</value>
+			<value>fade</value>
+			<value>expo</value>
+			<value>minimize</value>
+			<value>cube</value>
+			<value>showdesktop</value>
+			<value>switcher</value>
+			<value>move</value>
+			<value>resize</value>
+			<value>place</value>
+			<value>rotate</value>
+			<value>zoom</value>
+			<value>scale</value>
+			<value>dbus</value>
+			<value>water</value>		
+			<value>snow</value>		
+			<value>animation</value>		
+			<value>firepaint</value>		
+			<value>thumbnail</value>		
+			<value>switcher</value>		
+			<value>cubereflex</value>		
+		</default>
 	    </option>
 	    <option name="texture_filter" type="int">
 		<_short>Texture Filter</_short>
diff -rup compiz-0.5.2/metadata/cube.xml.in ../compiz-0.5.2/metadata/cube.xml.in
--- compiz-0.5.2/metadata/cube.xml.in	2007-08-02 23:33:10.000000000 +0200
+++ ../compiz-0.5.2/metadata/cube.xml.in	2007-08-30 18:08:08.071979000 +0200
@@ -42,9 +42,9 @@
 		<_short>Cube Color</_short>
 		<_long>Color of top and bottom sides of the cube</_long>
 		<default>
-		    <red>0xfefe</red>
+		    <red>0xffff</red>
 		    <green>0xffff</green>
-		    <blue>0xc7c7</blue>
+		    <blue>0xffff</blue>
 		</default>
 	    </option>
 	    <option name="in" type="bool">
@@ -55,14 +55,14 @@
 	    <option name="scale_image" type="bool">
 		<_short>Scale image</_short>
 		<_long>Scale images to cover top face of cube</_long>
-		<default>false</default>
+		<default>true</default>
 	    </option>
 	    <option name="images" type="list">
 		<_short>Image files</_short>
 		<_long>List of PNG and SVG files that should be rendered on top face of cube</_long>
 		<type>string</type>
 		<default>
-		    <value>freedesktop</value>
+		    <value>/usr/share/pixmaps/backgrounds/sun/di-logo-sun-blue.png</value>
 		</default>
 	    </option>
 	    <option name="skydome" type="bool">
@@ -139,7 +139,7 @@
 	    <option name="active_opacity" type="float">
 		<_short>Opacity During Rotation</_short>
 		<_long>Opacity of desktop window during rotation.</_long>
-		<default>100.0</default>
+		<default>60.0</default>
 		<min>0.0</min>
 		<max>100.0</max>
 		<precision>1.0</precision>
diff -rup compiz-0.5.2/metadata/screenshot.xml.in ../compiz-0.5.2/metadata/screenshot.xml.in
--- compiz-0.5.2/metadata/screenshot.xml.in	2007-05-03 17:51:54.000000000 +0200
+++ ../compiz-0.5.2/metadata/screenshot.xml.in	2007-08-30 17:13:51.179498000 +0200
@@ -8,7 +8,7 @@
 		<_long>Initiate rectangle screenshot</_long>
 		<allowed button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button1</button>
+		    <button>&lt;Mod4&gt;Button1</button>
 		</default>
 	    </option>
 	    <option name="directory" type="string">
diff -rup compiz-0.5.2/metadata/water.xml.in ../compiz-0.5.2/metadata/water.xml.in
--- compiz-0.5.2/metadata/water.xml.in	2007-07-13 01:30:57.000000000 +0200
+++ ../compiz-0.5.2/metadata/water.xml.in	2007-08-30 17:13:51.179531000 +0200
@@ -14,7 +14,7 @@
 		<_long>Enable pointer water effects</_long>
 		<allowed key="true"/>
 		<default>
-		    <key>&lt;Control&gt;&lt;Super&gt;</key>
+		    <key>&lt;Control&gt;&lt;Mod4&gt;</key>
 		</default>
 	    </option>
 	    <option name="toggle_rain" type="action">
diff -rup compiz-0.5.2/metadata/zoom.xml.in ../compiz-0.5.2/metadata/zoom.xml.in
--- compiz-0.5.2/metadata/zoom.xml.in	2007-05-30 00:00:46.000000000 +0200
+++ ../compiz-0.5.2/metadata/zoom.xml.in	2007-08-30 17:13:51.179948000 +0200
@@ -8,7 +8,7 @@
 		<_long>Zoom In</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button3</button>
+		    <button>&lt;Mod4&gt;Button3</button>
 		</default>
 	    </option>
 	    <option name="zoom_in" type="action">
@@ -16,7 +16,7 @@
 		<_long>Zoom In</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button4</button>
+		    <button>&lt;Mod4&gt;Button4</button>
 		</default>
 	    </option>
 	    <option name="zoom_out" type="action">
@@ -24,7 +24,7 @@
 		<_long>Zoom Out</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button5</button>
+		    <button>&lt;Mod4&gt;Button5</button>
 		</default>
 	    </option>
 	    <option name="zoom_pan" type="action">
@@ -32,7 +32,7 @@
 		<_long>Zoom pan</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;Button2</button>
+		    <button>&lt;Mod4&gt;Button2</button>
 		</default>
 	    </option>
 	</display>
diff -rup compiz-0.5.2/plugins/decoration.c ../compiz-0.5.2/plugins/decoration.c
--- compiz-0.5.2/plugins/decoration.c	2007-07-30 20:54:20.000000000 +0200
+++ ../compiz-0.5.2/plugins/decoration.c	2007-08-30 17:12:05.899488000 +0200
@@ -33,12 +33,12 @@
 #include <math.h>
 #include <unistd.h>
 
-#include <X11/Xatom.h>
-#include <X11/extensions/shape.h>
-
 #include <compiz.h>
 #include <decoration.h>
 
+#include <X11/Xatom.h>
+#include <X11/extensions/shape.h>
+
 static CompMetadata decorMetadata;
 
 typedef struct _Vector {
diff -rup compiz-0.5.2/plugins/video.c ../compiz-0.5.2/plugins/video.c
--- compiz-0.5.2/plugins/video.c	2007-07-13 04:08:27.000000000 +0200
+++ ../compiz-0.5.2/plugins/video.c	2007-08-30 17:12:05.900921000 +0200
@@ -29,12 +29,12 @@
 #include <math.h>
 #include <unistd.h>
 
-#include <X11/Xatom.h>
-#include <X11/extensions/shape.h>
-
 #include <compiz.h>
 #include <decoration.h>
 
+#include <X11/Xatom.h>
+#include <X11/extensions/shape.h>
+
 /*
  * compiz composited video
  *
diff -rup compiz-0.5.2/src/option.c ../compiz-0.5.2/src/option.c
--- compiz-0.5.2/src/option.c	2007-06-12 18:46:05.000000000 +0200
+++ ../compiz-0.5.2/src/option.c	2007-08-30 17:12:05.901379000 +0200
@@ -25,6 +25,10 @@
 
 #define _GNU_SOURCE
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 #include <strings.h>
@@ -32,6 +36,9 @@
 #include <math.h>
 
 #include <compiz.h>
+#ifdef SOLARIS
+#include <ctype.h>
+#endif
 
 struct _Modifier {
     char *name;
@@ -64,6 +71,26 @@ static char *edgeName[] = {
     "BottomRight"
 };
 
+#ifdef SOLARIS
+char* strcasestr (char *haystack, char *needle)
+{
+	char *p, *startn = 0, *np = 0;
+	for (p = haystack; *p; p++) {
+		if (np) {
+			if (toupper(*p) == toupper(*np)) {
+				if (!*++np)
+					return startn;
+			} else
+				np = 0;
+		} else if (toupper(*p) == toupper(*needle)) {
+			np = needle + 1;
+			startn = p;
+		}
+	}
+	return 0;
+}
+#endif
+
 static void
 finiOptionValue (CompOptionValue *v,
 		 CompOptionType  type)
