diff -urN compiz-fusion-plugins-extra-0.6.0.orig/configure.ac compiz-fusion-plugins-extra-0.6.0/configure.ac
--- compiz-fusion-plugins-extra-0.6.0.orig/configure.ac	2007-10-15 13:56:43.000000000 -0400
+++ compiz-fusion-plugins-extra-0.6.0/configure.ac	2007-10-31 16:15:32.065836169 -0400
@@ -1,6 +1,6 @@
 AC_PREREQ(2.57)
 
-AC_INIT([compiz-fusion-plugins-extra],esyscmd(. ./VERSION;echo -n $VERSION), [maniac@opencompositing.org])
+AC_INIT([compiz-fusion-plugins-extra],esyscmd(. ./VERSION;echo $VERSION | tr -d "\n"), [maniac@opencompositing.org])
 
 AC_DEFINE_UNQUOTED(VERSION,"$VERSION",[Version])
 
diff -urN compiz-fusion-plugins-extra-0.6.0.orig/metadata/addhelper.xml.in compiz-fusion-plugins-extra-0.6.0/metadata/addhelper.xml.in
--- compiz-fusion-plugins-extra-0.6.0.orig/metadata/addhelper.xml.in	2007-10-15 13:56:43.000000000 -0400
+++ compiz-fusion-plugins-extra-0.6.0/metadata/addhelper.xml.in	2007-10-31 16:16:43.533035255 -0400
@@ -17,7 +17,7 @@
           <_long>Use this to enable/disable AddHelper on the fly.</_long>
           <allowed key="true"/>
           <default>
-            <key>&lt;Super&gt;p</key>
+            <key>&lt;Mod4&gt;p</key>
           </default>
         </option>
       </group>
diff -urN compiz-fusion-plugins-extra-0.6.0.orig/metadata/bench.xml.in compiz-fusion-plugins-extra-0.6.0/metadata/bench.xml.in
--- compiz-fusion-plugins-extra-0.6.0.orig/metadata/bench.xml.in	2007-10-15 13:56:43.000000000 -0400
+++ compiz-fusion-plugins-extra-0.6.0/metadata/bench.xml.in	2007-10-31 16:16:43.534369597 -0400
@@ -12,7 +12,7 @@
           <_long>Start benchmark</_long>
           <allowed key="true"/>
           <default>
-            <key>&lt;Super&gt;F12</key>
+            <key>&lt;Mod4&gt;F12</key>
           </default>
         </option>
         <option name="disable_limiter" type="bool">
diff -urN compiz-fusion-plugins-extra-0.6.0.orig/metadata/firepaint.xml.in compiz-fusion-plugins-extra-0.6.0/metadata/firepaint.xml.in
--- compiz-fusion-plugins-extra-0.6.0.orig/metadata/firepaint.xml.in	2007-10-15 13:56:43.000000000 -0400
+++ compiz-fusion-plugins-extra-0.6.0/metadata/firepaint.xml.in	2007-10-31 16:16:43.534099360 -0400
@@ -9,7 +9,7 @@
 		<_long>Initiate fire drawing</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <button>&lt;Super&gt;&lt;Shift&gt;Button1</button>
+		    <button>&lt;Mod4&gt;&lt;Shift&gt;Button1</button>
 		</default>
 	    </option>
 	    <option type="action" name="clear">
@@ -17,7 +17,7 @@
 		<_long>Clear</_long>
 		<allowed key="true" button="true"/>
 		<default>
-		    <key>&lt;Super&gt;&lt;Shift&gt;c</key>
+		    <key>&lt;Mod4&gt;&lt;Shift&gt;c</key>
 		</default>
 	    </option>
 	</display>
diff -urN compiz-fusion-plugins-extra-0.6.0.orig/metadata/group.xml.in compiz-fusion-plugins-extra-0.6.0/metadata/group.xml.in
--- compiz-fusion-plugins-extra-0.6.0.orig/metadata/group.xml.in	2007-10-15 13:56:43.000000000 -0400
+++ compiz-fusion-plugins-extra-0.6.0/metadata/group.xml.in	2007-10-31 16:16:43.533869052 -0400
@@ -26,7 +26,7 @@
             <_long>The key for selecting the current window.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;s</key>
+              <key>&lt;Mod4&gt;s</key>
             </default>
           </option>
         </subgroup>
@@ -40,7 +40,7 @@
             <_long>The key for grouping windows.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;g</key>
+              <key>&lt;Mod4&gt;g</key>
             </default>
           </option>
           <option name="ungroup" type="action">
@@ -48,7 +48,7 @@
             <_long>The key for ungrouping the current group.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;u</key>
+              <key>&lt;Mod4&gt;u</key>
             </default>
           </option>
           <option name="remove" type="action">
@@ -56,7 +56,7 @@
             <_long>The key for removing the selected window from its group.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;r</key>
+              <key>&lt;Mod4&gt;r</key>
             </default>
           </option>
           <option name="close" type="action">
@@ -64,7 +64,7 @@
             <_long>The key for closing all windows in the group.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;c</key>
+              <key>&lt;Mod4&gt;c</key>
             </default>
           </option>
           <option name="ignore" type="action">
@@ -72,7 +72,7 @@
             <_long>The key for ignoring the group. If this key is pressed, you can do actions for a single window in the group only.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;x</key>
+              <key>&lt;Mod4&gt;x</key>
             </default>
           </option>
         </subgroup>
@@ -86,7 +86,7 @@
             <_long>The key for toggling the tab mode for the current group.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;t</key>
+              <key>&lt;Mod4&gt;t</key>
             </default>
           </option>
           <option name="change_tab_left" type="action">
@@ -94,7 +94,7 @@
             <_long>The key for switching to the tab left of the current one.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;Left</key>
+              <key>&lt;Mod4&gt;Left</key>
             </default>
           </option>
           <option name="change_tab_right" type="action">
@@ -102,7 +102,7 @@
             <_long>The key for switching to the tab right of the current one.</_long>
             <allowed key="true"/>
             <default>
-              <key>&lt;Super&gt;Right</key>
+              <key>&lt;Mod4&gt;Right</key>
             </default>
           </option>
         </subgroup>
diff -urN compiz-fusion-plugins-extra-0.6.0.orig/src/scalefilter/scalefilter.c compiz-fusion-plugins-extra-0.6.0/src/scalefilter/scalefilter.c
--- compiz-fusion-plugins-extra-0.6.0.orig/src/scalefilter/scalefilter.c	2007-10-15 13:56:44.000000000 -0400
+++ compiz-fusion-plugins-extra-0.6.0/src/scalefilter/scalefilter.c	2007-10-31 16:22:01.561972942 -0400
@@ -34,7 +34,7 @@
 #include <locale.h>
 
 #include <X11/Xlib.h>
-#include <X11/keysymdef.h>
+#include <X11/keysym.h>
 
 #include <compiz.h>
 #include <scale.h>
@@ -464,6 +464,7 @@
     memset (buffer, 0, sizeof (buffer));
     memset (wbuffer, 0, sizeof (wbuffer));
 
+#ifdef X_HAVE_UTF8_STRING
     if (fd->xic)
     {
 	Status status;
@@ -473,6 +474,7 @@
 	XUnsetICFocus (fd->xic);
     }
     else
+#endif
     {
 	count = XLookupString (event, buffer, 9, &ks, NULL);
     }
