--- ConsoleKit-0.2.9/src/ck-vt-monitor.c.orig	2008-02-25 17:31:11.481823000 +0800
+++ ConsoleKit-0.2.9/src/ck-vt-monitor.c	2008-02-25 17:31:26.827987000 +0800
@@ -40,7 +40,7 @@
 #include "ck-sysdeps.h"
 #include "ck-marshal.h"
 
-#ifdef __sun
+#if defined (__sun) && defined (HAVE_SYS_VT_H)
 #include <sys/vt.h>
 #include <signal.h>
 #include <stropts.h>
@@ -163,7 +163,7 @@
         return TRUE;
 }
 
-#ifdef __sun
+#if defined (__sun) && defined (HAVE_SYS_VT_H)
 static void
 handle_vt_active (void)
 {
@@ -381,7 +381,9 @@
         int    i;
         gint32 current_num;
 
-#ifdef __sun
+#if defined (__sun) && !defined (HAVE_SYS_VT_H)
+        /* Best to do nothing if VT is not supported */
+#elif defined (__sun) && defined (HAVE_SYS_VT_H)
         /*
          * Solaris supports synchronous event notification in STREAMS.
          * Applications that open the virtual console device  can
