--- ConsoleKit-0.2.9/src/ck-vt-monitor.c.orig	2008-02-18 14:27:20.845403000 +0800
+++ ConsoleKit-0.2.9/src/ck-vt-monitor.c	2008-02-18 14:04:11.449622000 +0800
@@ -40,6 +40,12 @@
 #include "ck-sysdeps.h"
 #include "ck-marshal.h"
 
+#ifdef __sun
+#include <sys/vt.h>
+#include <signal.h>
+#include <stropts.h>
+#endif
+
 #ifndef ERROR
 #define ERROR -1
 #endif
@@ -157,6 +163,43 @@
         return TRUE;
 }
 
+#ifdef __sun
+static void
+handle_vt_active (void)
+{
+	struct vt_stat state;
+        guint num;
+        CkVtMonitor *vt_monitor = CK_VT_MONITOR (vt_object);
+
+        g_return_if_fail (CK_IS_VT_MONITOR (vt_monitor));
+  
+      /*
+	 * state.v_active value: [1 .. N]
+	 *
+	 * VT device file	 VT #
+	 * 
+	 * /dev/console		--- VT #1
+	 * /dev/vt/2		--- VT #2
+	 * /dev/vt/3		--- VT #3
+	 * /dev/vt/N		--- VT #4
+	 */
+	if (ioctl(vt_monitor->priv->vfd, VT_GETSTATE, &state) != -1)
+                num = state.v_active;
+        else 
+                g_debug ("Fails to ioctl VT_GETSTATE");
+        
+        if (vt_monitor->priv->active_num != num) {
+                g_debug ("Changing active VT: %d", num);
+
+                vt_monitor->priv->active_num = num;
+
+                g_signal_emit (vt_monitor, signals[ACTIVE_CHANGED], 0, num);
+        } else {
+                g_debug ("VT activated but already active: %d", num);
+        }
+}
+#endif
+
 static void
 change_active_num (CkVtMonitor *vt_monitor,
                    guint        num)
@@ -338,6 +381,20 @@
         int    i;
         gint32 current_num;
 
+#ifdef __sun
+        /*
+         * Solaris supports synchronous event notification in STREAMS.
+         * Applications that open the virtual console device  can
+         * get a asynchronous notification of VT switching by setting 
+         * the S_MSG flag in an I_SETSIG STREAMS ioctl. Such processes
+         * receive a SIGPOLL signal when a VT switching succeeds.
+         */
+	struct sigaction act;
+	act.sa_handler = handle_vt_active;
+	sigaction (SIGPOLL, &act, NULL);
+
+	ioctl (vt_monitor->priv->vfd, I_SETSIG, S_MSG);
+#else
         G_LOCK (hash_lock);
 
         current_num = vt_monitor->priv->active_num;
@@ -365,6 +422,7 @@
         }
 
         G_UNLOCK (hash_lock);
+#endif
 }
 
 static void
--- ConsoleKit-0.2.9/src/ck-sysdeps-solaris.c.orig	2008-02-18 10:16:37.917264000 +0800
+++ ConsoleKit-0.2.9/src/ck-sysdeps-solaris.c	2008-02-18 10:17:01.915080000 +0800
@@ -364,7 +364,7 @@
         error = NULL;
         svcprop_stdout = NULL;
         status = 0;
-        res = g_spawn_command_line_sync ("/usr/bin/svcprop -p options/vtnodecount vtdaemon",
+        res = g_spawn_command_line_sync ("/usr/bin/svcprop -p options/nodecount vtdaemon",
                                          &svcprop_stdout,
                                          NULL,
                                          &status,
