--- ck-old/src/ck-session.c-orig	2008-01-25 23:22:09.006898000 -0600
+++ ck-old/src/ck-session.c	2008-01-25 23:22:47.083388000 -0600
@@ -1230,7 +1230,8 @@ ck_session_run_programs (CkSession  *ses
         if (session->priv->x11_display_device != NULL && strlen (session->priv->x11_display_device) > 0) {
                 extra_env[n++] = g_strdup_printf ("CK_SESSION_X11_DISPLAY_DEVICE=%s", session->priv->x11_display_device);
         }
-        extra_env[n++] = g_strdup_printf ("CK_SESSION_X11_DISPLAY=%s", session->priv->x11_display);
+        extra_env[n++] = g_strdup_printf ("CK_SESSION_X11_DISPLAY=%s",
+                session->priv->x11_display ? session->priv->x11_display : "");
         if (session->priv->remote_host_name != NULL && strlen (session->priv->remote_host_name) > 0) {
                 extra_env[n++] = g_strdup_printf ("CK_SESSION_REMOTE_HOST_NAME=%s", session->priv->remote_host_name);
         }
--- ck-old/src/ck-sysdeps-solaris.c-orig	2008-01-26 01:55:56.364088000 -0600
+++ ck-old/src/ck-sysdeps-solaris.c	2008-01-26 01:57:49.064138000 -0600
@@ -260,7 +260,7 @@ ck_unix_pid_get_env_hash (pid_t pid)
 
                         if (skip_prefix != NULL) {
                                 char **vals;
-                                vals = g_strsplit (buf, "=", 2);
+                                vals = g_strsplit (skip_prefix + 1, "=", 2);
                                 if (vals != NULL) {
                                         g_hash_table_insert (hash,
                                                              g_strdup (vals[0]),
