--- ConsoleKit-0.3.0/src/ck-log-event.h-orig	2008-08-07 02:40:30.585808000 -0500
+++ ConsoleKit-0.3.0/src/ck-log-event.h	2008-08-07 02:40:34.180115000 -0500
@@ -45,18 +45,22 @@ typedef enum
 
 typedef struct
 {
+   gboolean foo;
 } CkLogNoneEvent;
 
 typedef struct
 {
+   gboolean foo;
 } CkLogSystemStopEvent;
 
 typedef struct
 {
+   gboolean foo;
 } CkLogSystemRestartEvent;
 
 typedef struct
 {
+   gboolean foo;
 } CkLogSystemStartEvent;
 
 typedef struct
