--- ConsoleKit-0.2.10/configure.ac.orig	2008-03-04 15:23:40.479007000 +0800
+++ ConsoleKit-0.2.10/configure.ac	2008-03-04 15:24:26.757612000 +0800
@@ -212,7 +212,7 @@
 AC_SUBST(HAVE_PAM)
 AC_SUBST(PAM_LIBS)
 
-AC_CHECK_HEADERS([security/pam_modutil.h security/pam_ext.h])
+AC_CHECK_HEADERS([security/pam_modutil.h security/pam_ext.h security/pam_macro.h])
 AC_CHECK_LIB(pam, pam_syslog, [AC_DEFINE(HAVE_PAM_SYSLOG, [], [Define to 1 if you have the pam_syslog function])])
 
 # Check if we should build the PAM module
--- ConsoleKit-0.2.10/pam-ck-connector/Makefile.am.orig	2008-03-04 14:35:03.648181000 +0800
+++ ConsoleKit-0.2.10/pam-ck-connector/Makefile.am	2008-03-04 14:42:51.367637000 +0800
@@ -22,6 +22,7 @@
 
 man_MANS = pam_ck_connector.8
 
+if CK_COMPILE_LINUX
 noinst_PROGRAMS = 				\
 	test-pam				\
 	$(NULL)
@@ -34,6 +35,7 @@
 	$(PAM_LIBS)				\
 	-lpam_misc				\
 	$(NULL)
+endif
 
 endif
 
--- ConsoleKit-0.2.10/pam-ck-connector/pam-ck-connector.c.orig	2008-03-04 17:51:04.718537000 +0800
+++ ConsoleKit-0.2.10/pam-ck-connector/pam-ck-connector.c	2008-03-04 17:46:23.996292000 +0800
@@ -51,8 +51,18 @@
 
 #define PAM_SM_SESSION
 
-#include <security/pam_modules.h>
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+#ifndef PAM_EXTERN
+#define PAM_EXTERN 
+#endif
+
+#ifdef HAVE_SECURITY_PAM_MACROS_H
 #include <security/_pam_macros.h>
+#endif
+#include <security/pam_appl.h>
+#include <security/pam_modules.h>
 #ifdef HAVE_SECURITY_PAM_MODUTIL_H
 #include <security/pam_modutil.h>
 #endif
@@ -151,7 +161,7 @@
         }
 }
 
-
+#ifndef __sun
 PAM_EXTERN int
 pam_sm_authenticate (pam_handle_t *pamh,
                      int           flags,
@@ -169,6 +179,7 @@
 {
         return PAM_IGNORE;
 }
+#endif
 
 static uid_t
 _util_name_to_uid (const char *username,
@@ -185,10 +196,16 @@
 
         bufsize = sysconf (_SC_GETPW_R_SIZE_MAX);
         buf = calloc (sizeof (char), bufsize);
+#ifdef __sun
+        pwdp = getpwnam_r (username, &pwd, buf, bufsize);
+	if (pwdp == NULL)
+		goto out;
+#else
         rc = getpwnam_r (username, &pwd, buf, bufsize, &pwdp);
         if (rc != 0 || pwdp == NULL) {
                 goto out;
         }
+#endif
 
         res = pwdp->pw_uid;
         if (default_gid != NULL) {
