--- ConsoleKit-0.2.9/src/ck-vt-monitor.c-orig	2008-02-22 12:33:54.462905000 -0600
+++ ConsoleKit-0.2.9/src/ck-vt-monitor.c	2008-02-22 12:37:19.076843000 -0600
@@ -40,7 +40,7 @@
 #include "ck-sysdeps.h"
 #include "ck-marshal.h"
 
-#ifdef __sun
+#if defined (__sun) && (HAVE_SYS_VT_H)
 #include <sys/vt.h>
 #include <signal.h>
 #include <stropts.h>
@@ -163,7 +163,7 @@ ck_vt_monitor_get_active (CkVtMonitor   
         return TRUE;
 }
 
-#ifdef __sun
+#if defined (__sun) && (HAVE_SYS_VT_H)
 static void
 handle_vt_active (void)
 {
@@ -381,7 +381,9 @@ vt_add_watches (CkVtMonitor *vt_monitor)
         int    i;
         gint32 current_num;
 
-#ifdef __sun
+#ifdef (__sun) 
+        /* Best to do nothing if VT is not supported */
+#ifdef (HAVE_SYS_VT_H)
         /*
          * Solaris supports synchronous event notification in STREAMS.
          * Applications that open the virtual console device  can
@@ -394,6 +396,7 @@ vt_add_watches (CkVtMonitor *vt_monitor)
 	sigaction (SIGPOLL, &act, NULL);
 
 	ioctl (vt_monitor->priv->vfd, I_SETSIG, S_MSG);
+#endif
 #else
         G_LOCK (hash_lock);
 
