--- ConsoleKit-0.2.9/src/ck-seat.c-orig	2008-02-22 16:40:16.161110000 -0600
+++ ConsoleKit-0.2.9/src/ck-seat.c	2008-02-22 16:41:16.503412000 -0600
@@ -500,7 +500,7 @@ change_active_session (CkSeat    *seat,
                 ck_session_set_active (session, TRUE, NULL);
         }
 
-        g_debug ("Active session changed: %s", ssid);
+        g_debug ("Active session changed: %s", ssid ? ssid : "(null)");
 
         g_signal_emit (seat, signals [ACTIVE_SESSION_CHANGED], 0, ssid);
 
