--- dbus-0.35.2/glib/Makefile.am.orig	Fri Oct 21 15:30:12 2005
+++ dbus-0.35.2/glib/Makefile.am	Fri Oct 21 15:31:15 2005
@@ -1,6 +1,6 @@
 SUBDIRS = . examples
 
-INCLUDES=-I$(top_srcdir) $(DBUS_CLIENT_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_GLIB_TOOL_CFLAGS) -DDBUS_COMPILATION=1 -DDBUS_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\"
+INCLUDES=-I$(top_srcdir) $(DBUS_CLIENT_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_GLIB_TOOL_CFLAGS) -DDBUS_COMPILATION=1 -DDBUS_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\" $(CFLAGS)
 
 lib_LTLIBRARIES=libdbus-glib-1.la
 
--- dbus-0.35.2/glib/Makefile.in.orig	Fri Oct 21 15:30:19 2005
+++ dbus-0.35.2/glib/Makefile.in	Fri Oct 21 15:31:57 2005
@@ -361,7 +361,7 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 SUBDIRS = . examples
-INCLUDES = -I$(top_srcdir) $(DBUS_CLIENT_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_GLIB_TOOL_CFLAGS) -DDBUS_COMPILATION=1 -DDBUS_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\"
+INCLUDES = -I$(top_srcdir) $(DBUS_CLIENT_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_GLIB_TOOL_CFLAGS) -DDBUS_COMPILATION=1 -DDBUS_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\" $(CFLAGS)
 lib_LTLIBRARIES = libdbus-glib-1.la
 BUILT_SOURCES = dbus-glib-error-switch.h
 CLEANFILES = $(BUILT_SOURCES)
