--- dbus-0.62/configure.in-orig	2006-07-20 16:20:26.716244000 -0500
+++ dbus-0.62/configure.in	2006-07-20 16:23:28.194175000 -0500
@@ -1422,6 +1422,21 @@ fi
 
 AM_CONDITIONAL(HAVE_PYTHON, test x$have_python = xyes)
 
+if test ${exec_prefix} != "NONE"; then
+   real_dbus_prefix=$exec_prefix
+elif test ${prefix} != "NONE"; then
+   real_dbus_prefix=$prefix
+else
+   real_dbus_prefix=$ac_default_prefix
+fi
+
+if test ${libexecdir} = "\${exec_prefix}/libexec"; then
+   DBUS_DAEMON_LOCATION="$real_dbus_prefix/libexec"
+else
+   DBUS_DAEMON_LOCATION="${libexecdir}"
+fi
+AC_SUBST(DBUS_DAEMON_LOCATION)
+
 AC_OUTPUT([
 Doxyfile
 dbus/dbus-arch-deps.h
--- dbus-0.62/tools/Makefile.am-orig	2006-07-20 16:22:03.668619000 -0500
+++ dbus-0.62/tools/Makefile.am	2006-07-20 16:28:54.688750000 -0500
@@ -1,4 +1,4 @@
-INCLUDES=-I$(top_srcdir) $(DBUS_CLIENT_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_X_CFLAGS) $(DBUS_GTK_THREADS_CFLAGS) -DDBUS_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\" -DDBUS_COMPILATION
+INCLUDES=-I$(top_srcdir) $(DBUS_CLIENT_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_X_CFLAGS) $(DBUS_GTK_THREADS_CFLAGS) -DDBUS_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\" -DDBUS_COMPILATION -DDBUS_DAEMON_LOCATION=\"@DBUS_DAEMON_LOCATION@\"
 
 if HAVE_GLIB
 GLIB_TOOLS=dbus-monitor
--- dbus-0.62/bus/Makefile.am-orig	2006-07-20 16:04:12.651077000 -0500
+++ dbus-0.62/bus/Makefile.am	2006-07-20 16:04:29.441199000 -0500
@@ -16,7 +16,7 @@ config_DATA=					\
 	session.conf				\
 	system.conf
 
-bin_PROGRAMS=dbus-daemon
+libexec_PROGRAMS=dbus-daemon
 
 if DBUS_USE_LIBXML
 XML_SOURCES=config-loader-libxml.c
--- /tmp/dbus-launch.c	2006-07-20 16:55:40.804739310 -0500
+++ dbus-0.62/tools/dbus-launch.c	2006-07-20 16:53:48.243775000 -0500
@@ -777,6 +777,25 @@ main (int argc, char **argv)
 
       verbose ("Calling exec()\n");
       
+      execlp (DBUS_DAEMON_LOCATION"/dbus-daemon",
+              DBUS_DAEMON_LOCATION"/dbus-daemon",
+              "--fork",
+              "--print-pid", write_pid_fd_as_string,
+              "--print-address", write_address_fd_as_string,
+              config_file ? "--config-file" : "--session",
+              config_file, /* has to be last in this varargs list */
+              NULL);
+
+      fprintf (stderr,
+               "Failed to execute message bus daemon %s: %s.  Will try again without full path.\n",
+               DBUS_DAEMON_LOCATION"/dbus-daemon", strerror (errno));
+
+      /*
+       * If it failed, try running without full PATH.  Note this is needed
+       * because the build process builds the run-with-tmp-session-bus.conf
+       * file and the dbus-daemon will not be in the install location during
+       * build time.
+       */
       execlp ("dbus-daemon",
               "dbus-daemon",
               "--fork",
