--- deluge-torrent-0.5.8.5.orig/libtorrent/src/enum_net.cpp	Wed Mar 26 01:29:40 2008
+++ deluge-torrent-0.5.8.5/libtorrent/src/enum_net.cpp	Wed Mar 26 01:50:02 2008
@@ -37,6 +37,7 @@
 #include <sys/ioctl.h>
 #include <netinet/in.h>
 #include <net/if.h>
+#include <string.h>
 #elif defined TORRENT_WINDOWS
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN
@@ -155,9 +156,9 @@
 				ret.push_back(iface);
 			}
 
-#if defined TORRENT_BSD
+#if defined TORRENT_BSD && defined BSD4_4
 			int current_size = item.ifr_addr.sa_len + IFNAMSIZ;
-#elif defined TORRENT_LINUX
+#elif defined TORRENT_LINUX || defined TORRENT_BSD
 			int current_size = sizeof(ifreq);
 #endif
 			ifr += current_size;
