--- dosbox-0.72/configure.in.orig	Thu Aug 23 06:24:49 2007
+++ dosbox-0.72/configure.in	Mon Feb 25 01:19:36 2008
@@ -371,6 +371,11 @@
   AC_MSG_WARN([Can't find libSDL_sound, libSDL_sound support disabled])
 fi
 
+dnl Check for -lsocket -lnsl
+
+AC_CHECK_FUNC(gethostent, , AC_CHECK_LIB(nsl, gethostent))
+AC_CHECK_FUNC(setsockopt, , AC_CHECK_LIB(socket, setsockopt))
+
 dnl Check for mprotect. Needed for 64 bits linux 
 AH_TEMPLATE(C_HAVE_MPROTECT,[Define to 1 if you have the mprotect function])
 AC_CHECK_HEADER([sys/mman.h], [
@@ -407,6 +412,9 @@
        LIBS="$LIBS -framework AudioUnit"
        AC_DEFINE(C_DIRECTSERIAL, 1, [ Define to 1 if you want serial passthrough support (Win32, Posix and OS/2).])
        ;;
+    *-*-solaris*)
+       AC_DEFINE(C_DIRECTSERIAL, 1, [ Define to 1 if you want serial passthrough support (Win32, Posix and OS/2).])
+       ;;
     *-*-linux*)
        AC_DEFINE(LINUX, 1, [Compiling on GNU/Linux])
        AC_DEFINE(C_DIRECTSERIAL, 1, [ Define to 1 if you want serial passthrough support (Win32, Posix and OS/2).])
--- dosbox-0.72/src/hardware/serialport/directserial_posix.cpp.orig	Sun Aug 26 13:19:46 2007
+++ dosbox-0.72/src/hardware/serialport/directserial_posix.cpp	Mon Feb 25 01:16:56 2008
@@ -23,7 +23,7 @@
 #if C_DIRECTSERIAL
 
 // Posix version
-#if defined (LINUX) || defined (MACOSX)
+#if defined (LINUX) || defined (MACOSX) || defined (_sun)
 
 #include "serialport.h"
 #include "directserial_posix.h"
