--- doxygen-1.5.4/tmake/lib/solaris-g++/tmake.conf.orig	2008-01-18 13:36:59.690049000 +0530
+++ doxygen-1.5.4/tmake/lib/solaris-g++/tmake.conf	2008-01-18 13:38:16.873357000 +0530
@@ -43,7 +43,7 @@
 TMAKE_LFLAGS_SHLIB	= -shared -h $(TARGET1)
 TMAKE_LFLAGS_SONAME	=
 
-TMAKE_LIBS		= -liconv
+TMAKE_LIBS		= -L/usr/gnu/lib -R/usr/gnu/lib -liconv
 TMAKE_LIBS_X11		= -lXext -lX11 -lm
 TMAKE_LIBS_QT		= -lqt
 TMAKE_LIBS_QT_MT	= -lqt-mt
