Index: trunk/tmake/lib/solaris-cc/tmake.conf
===================================================================
--- trunk/tmake/lib/solaris-cc/tmake.conf	(revision 644)
+++ trunk/tmake/lib/solaris-cc/tmake.conf	(working copy)
@@ -11,7 +11,7 @@
 TMAKE_CFLAGS		=
 TMAKE_CFLAGS_WARN_ON	=
 TMAKE_CFLAGS_WARN_OFF	= -w
-TMAKE_CFLAGS_RELEASE	= -O
+TMAKE_CFLAGS_RELEASE	= -xO4 -xspace -xstrconst -mr -xregs=no%frameptr
 TMAKE_CFLAGS_DEBUG	= -g
 TMAKE_CFLAGS_SHLIB	= -KPIC
 TMAKE_CFLAGS_YACC	=
@@ -20,7 +20,7 @@
 TMAKE_CXXFLAGS		= $$TMAKE_CFLAGS
 TMAKE_CXXFLAGS_WARN_ON	= $$TMAKE_CFLAGS_WARN_ON
 TMAKE_CXXFLAGS_WARN_OFF	= $$TMAKE_CFLAGS_WARN_OFF
-TMAKE_CXXFLAGS_RELEASE	= -O2
+TMAKE_CXXFLAGS_RELEASE	= -xO4 -xspace -mr -xregs=no%frameptr
 TMAKE_CXXFLAGS_DEBUG	= $$TMAKE_CFLAGS_DEBUG
 TMAKE_CXXFLAGS_SHLIB	= -PIC
 TMAKE_CXXFLAGS_YACC	= $$TMAKE_CFLAGS_YACC
@@ -38,7 +38,7 @@
 # Jan Wortelboer <janw@wins.uva.nl> suggests avoiding $LD_LIBRARY_PATH:
 TMAKE_LINK_SHLIB	= CC -R$(QTDIR)/lib:/usr/openwin/lib
 TMAKE_LFLAGS		=
-TMAKE_LFLAGS_RELEASE	=
+TMAKE_LFLAGS_RELEASE	= -Wl,-zignore -Wl,-zcombreloc -Wl,-Bdirect
 TMAKE_LFLAGS_DEBUG	=
 TMAKE_LFLAGS_SHLIB	= -G -h $(TARGET1)
 TMAKE_LFLAGS_SONAME	=
