diff -ur doxygen-1.5.2-orig/src/portable.cpp doxygen-1.5.2/src/portable.cpp
--- doxygen-1.5.2-orig/src/portable.cpp	2007-03-13 23:58:17.000000000 +0700
+++ doxygen-1.5.2/src/portable.cpp	2007-04-21 19:07:44.101394160 +0700
@@ -376,13 +376,13 @@
 #if (defined(_LIBICONV_VERSION) && (_LIBICONV_VERSION==0x0109))
 #define CASTNEEDED(x) (x)
 #else
-#define CASTNEEDED(x) (char **)(x)
+#define CASTNEEDED(x) (const char **)(x)
 #endif
-  return iconv(cd,CASTNEEDED(inbuf),inbytesleft,outbuf,outbytesleft);
+  return iconv((iconv_t) cd,CASTNEEDED(inbuf),inbytesleft,outbuf,outbytesleft);
 }
 
 int portable_iconv_close (void *cd)
 {
-  return iconv_close(cd);
+  return iconv_close((iconv_t)cd);
 }
 
