diff -ur dvdauthor-0.6.11-orig/src/compat.h dvdauthor-0.6.11/src/compat.h
--- dvdauthor-0.6.11-orig/src/compat.h	2005-02-01 09:26:18.000000000 +0700
+++ dvdauthor-0.6.11/src/compat.h	2006-11-23 12:13:58.809155911 +0700
@@ -106,3 +106,10 @@
 #define FT_FREETYPE_H <freetype/freetype.h>
 #define FT_GLYPH_H <freetype/ftglyph.h>
 #endif
+
+#ifdef sun
+typedef uint8_t u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+typedef uint64_t u_int64_t;
+#endif sun
