--- easytag-1.99.12/configure.in.orig	2006-06-23 21:15:32.203052000 -0400
+++ easytag-1.99.12/configure.in	2006-06-23 21:16:16.249998000 -0400
@@ -183,10 +183,9 @@
 dnl ################################################
 if test "x$enable_mp3" = "xyes"; then
     dnl Check which libs are required by id3lib, libid3.la is fucked up
-    AC_SEARCH_LIBS(ID3Tag_Link,"id3" "id3 -lstdc++" "id3 -lz" "id3 -lz -lstdc++",,AC_MSG_WARN(id3lib not found); id3lib_available=no)
+    AC_SEARCH_LIBS(ID3Tag_Link,"id3" "id3 -lCstd -lCrun -lz" "id3 -lCstd -lCrun -lz" "id3 -lCstd -lCrun -lz",,AC_MSG_WARN(id3lib not found); id3lib_available=no)
 
-    dnl Needed to link correctly with id3lib 3.8.0
-    LIBS="-lz -lstdc++ $LIBS"
+    LIBS="-lz -lCstd -lCrun -lz $LIBS"
     dnl AC_CHECK_LIB(z,uncompress,AC_DEFINE_UNQUOTED(HAVE_ZLIB))
 
     dnl expected version for cross compiling
