--- easytag-2.1/configure.in.orig	Sun Feb 24 06:04:33 2008
+++ easytag-2.1/configure.in	Sun Feb 24 06:22:51 2008
@@ -71,7 +71,8 @@
 CFLAGS="`echo $CFLAGS|sed -e 's%\-I/usr/include%%g' -e 's%\-L/usr/lib%%g'`"
 
 dnl Patch from Ben Taylor to fix a Solaris bug (for CDDB)
-AC_CHECK_LIB(socket, recv)
+AC_CHECK_FUNC(gethostbyname, , AC_CHECK_LIB(nsl, gethostbyname))
+AC_CHECK_FUNC(connect, , AC_CHECK_LIB(socket, connect))
 
 
 dnl ################################################
@@ -189,11 +190,19 @@
 dnl # Id3lib libraries
 dnl ################################################
 if test "x$enable_mp3" = "xyes"; then
+    if test "x$GCC" != "xyes"; then
+        dnl Check if we should use non-GNU C++ libraries
+        AC_CHECK_LIB([Crun],[main])
+        AC_CHECK_LIB([Cstd],[main])
+    else
+        AC_CHECK_LIB([stdc++],[main])
+    fi
+
     dnl Check which libs are required by id3lib, libid3.la is fucked up
-    AC_SEARCH_LIBS(ID3Tag_Link,"id3" "id3 -lstdc++" "id3 -lz" "id3 -lz -lstdc++", , AC_MSG_WARN(id3lib not found); id3lib_available=no)
+    AC_SEARCH_LIBS(ID3Tag_Link,"id3" "id3 -lz", , AC_MSG_WARN(id3lib not found); id3lib_available=no)
 
     dnl Needed to link correctly with id3lib 3.8.0
-    LIBS="-lz -lstdc++ $LIBS"
+    LIBS="-lz $LIBS"
     dnl AC_CHECK_LIB(z,uncompress,AC_DEFINE_UNQUOTED(HAVE_ZLIB))
 
     dnl expected version for cross compiling
