--- elisa/elisa/plugins/base/plugin.conf-orig	2007-07-18 05:15:14.033713000 -0500
+++ elisa/elisa/plugins/base/plugin.conf	2007-07-18 05:20:31.038600000 -0500
@@ -15,15 +15,15 @@ description="To access file:// media"
 name="audiocd"
 
 [media_providers.ipod_media:IpodMedia]
-platforms=["linux",]
+platforms=["linux", "posix"]
 external_dependencies=["gpod",]
 
 [media_providers.daap_media:DaapMedia]
-platforms=["linux",]
+platforms=["linux", "posix"]
 external_dependencies=["daap",]
 
 [media_providers.gnomevfs_media:GnomevfsMedia]
-platforms=["linux",]
+platforms=["linux", "posix"]
 external_dependencies=["gnomevfs",]
 description="Any media location supported by GNOME vfs: smb://, http:// ..."
 
@@ -32,12 +32,12 @@ name="upnp_media"
 component_dependencies = ["base:coherence_service"]
 
 [metadata_providers.taglib_metadata:TaglibMetadata]
-platforms=["linux",]
+platforms=["linux", "posix"]
 external_dependencies=["tagpy",]
 description="Local audio files generaldata retrieval support"
 
 [metadata_providers.amazon_metadata:AmazonCovers]
-platforms=["linux",]
+platforms=["linux", "posix"]
 description="search and download Covers using amazon"
 
 [metadata_providers.coverindir_metadata:CoverInDir]
@@ -48,13 +48,13 @@ description="Looks in cache dirs for cov
 
 [input_providers.lirc_input:LircInput]
 external_dependencies=["pylirc",]
-platforms=["linux",]
+platforms=["linux", "posix"]
 
 [input_providers.raw_input:RawInput]
 
 [input_providers.bluetooth_input:BluetoothInput]
 external_dependencies=["bluetooth",]
-platforms=["linux",]
+platforms=["linux", "posix"]
 
 [input_providers.webcam_input:WebcamInput]
 external_dependencies=["gst",]
@@ -62,7 +62,7 @@ external_dependencies=["gst",]
 [metadata_providers.gst_metadata:GstMetadata]
 
 [services.hal_service:HalService]
-platforms=["linux",]
+platforms=["linux"]
 
 
 [services.coherence_service:CoherenceService]
