--- elisa-0.3.2/elisa/plugins/good/pigment/pigment_context.py-orig	2007-10-08 13:54:54.378577000 -0500
+++ elisa-0.3.2/elisa/plugins/good/pigment/pigment_context.py	2007-10-08 13:55:10.125739000 -0500
@@ -159,7 +159,7 @@ class PigmentContext(Context):
 
     def _compiz_check(self):
         # FIXME: this is a work-a-round against the flickering in compiz
-        ps_aux = os.popen('ps aux')
+        ps_aux = os.popen('/usr/ucb/ps aux')
 
         compiz = False
 
