--- elisa-0.5.2/elisa/core/utils/i18n.py-orig	2008-07-25 08:01:46.362926000 -0500
+++ elisa-0.5.2/elisa/core/utils/i18n.py	2008-07-25 08:01:57.755537000 -0500
@@ -80,14 +80,14 @@ def install_translation(plugin_name):
 
     if platform.system().lower() == 'windows':
         # windows hack for locale setting
-        default_lang, default_enc = locale.getdefaultlocale()
+        default_lang, default_enc = locale.nl_langinfo(locale.CODESET)
         lang = os.getenv('LANG')
         if lang is None:
             lang = default_lang
             os.environ['LANG'] = lang
         current_locale = lang
     else:
-        current_locale = locale.getdefaultlocale()[0]
+        current_locale = locale.nl_langinfo(locale.CODESET)
     try:
         t = gettext.translation(domain, i18n_dir, languages=[current_locale,])
         translator = t.ugettext
