--- elisa-plugins-bad-0.5.18/setup.py-orig	2008-11-17 16:00:06.954905000 -0600
+++ elisa-plugins-bad-0.5.18/setup.py	2008-11-17 16:06:38.136921000 -0600
@@ -4,27 +4,15 @@ from setup_bundle import setup
 # List of plugins to be packaged in the 'bad' bundle
 bad_plugins = ['amazon',
                'amp',
-               'avahi',
-               'coherence',
-               'daap',
                'database',
-               'discogs',
-               'dvd',
                'elisa_updater',
                'favorites',
-               'filtered_shares',
                'gstreamer',
                'http_client',
-               'ipod',
-               'osso',
                'pigment',
                'poblesec',
                'search',
                'shelf',
-               'smbwin32',
-               'winremote',
-               'wmd',
-               'yesfm',
               ]
 
 if __name__ == '__main__':
