--- elisa-plugins-bad-0.5.15/setup.py.orig	2008-10-24 18:07:47.051906000 +0800
+++ elisa-plugins-bad-0.5.15/setup.py	2008-10-24 18:09:47.443280000 +0800
@@ -2,29 +2,18 @@
 from setup_bundle import setup
 
 # List of plugins to be packaged in the 'bad' bundle
-bad_plugins = ['amazon',
-               'amp',
+bad_plugins = ['amp',
                'avahi',
-               'coherence',
-               'daap',
                'database',
                'discogs',
                'dvd',
-               'elisa_updater',
                'favorites',
-               'filtered_shares',
                'gstreamer',
                'http_client',
-               'ipod',
-               'osso',
                'pigment',
                'poblesec',
                'search',
                'shelf',
-               'smbwin32',
-               'winremote',
-               'wmd',
-               'yesfm',
               ]
 
 if __name__ == '__main__':
