--- elisa-plugins-bad-0.5.12/elisa/plugins/poblesec/music_library.py-orig	2008-09-30 23:24:14.828078000 -0500
+++ elisa-plugins-bad-0.5.12/elisa/plugins/poblesec/music_library.py	2008-09-30 23:25:26.955173000 -0500
@@ -42,14 +42,14 @@ except ImportError:
 
 from elisa.core.utils import defer
 
-import os, hashlib
+import os, md5
 
 
 # TODO: factorize this code in the core of Elisa in a caching utilities module
 def _thumbnail_file(thumbnail_uri):
     if not os.path.exists(PICTURES_CACHE):
         os.makedirs(PICTURES_CACHE, 0755)
-    thumbnail = hashlib.md5(str(thumbnail_uri)).hexdigest() + '.jpg'
+    thumbnail = md5.md5(str(thumbnail_uri)).hexdigest() + '.jpg'
     return os.path.join(PICTURES_CACHE, thumbnail)
 
 
--- elisa-plugins-bad-0.5.12/elisa/plugins/database/dbus_service.py-orig	2008-09-30 23:24:46.842282000 -0500
+++ elisa-plugins-bad-0.5.12/elisa/plugins/database/dbus_service.py	2008-09-30 23:25:05.351048000 -0500
@@ -36,7 +36,7 @@ from elisa.plugins.database.models impor
 
 from storm.expr import Desc, Asc
 
-import hashlib, os
+import md5, os
 
 
 def result_set_order_offset_and_limit(result_set, field, offset, limit):
@@ -86,7 +86,7 @@ def build_last_track_played_tuple(music_
 
 
 def _thumbnail_file(uri):
-    thumbnail = hashlib.md5(str(uri)).hexdigest() + '.jpg'
+    thumbnail = md5.md5(str(uri)).hexdigest() + '.jpg'
     filename = os.path.join(PICTURES_CACHE, thumbnail)
     return filename
 
