diff -urN emerald-0.5.2.orig/configure.ac emerald-0.5.2/configure.ac
--- emerald-0.5.2.orig/configure.ac	2007-08-12 17:00:01.000000000 -0400
+++ emerald-0.5.2/configure.ac	2007-11-01 20:58:11.397594288 -0400
@@ -1,6 +1,6 @@
 AC_PREREQ(2.57)
 
-AC_INIT([emerald],esyscmd(. ./VERSION;echo -n $VERSION), [livinglatexkali@gmail.com])
+AC_INIT([emerald],esyscmd(. ./VERSION;echo $VERSION | tr -d "\n"), [livinglatexkali@gmail.com])
 
 AC_DEFINE_UNQUOTED(VERSION,"$VERSION",[Version])
 
@@ -19,6 +19,7 @@
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package.])
 AC_SUBST(GETTEXT_PACKAGE)
 AC_CHECK_HEADERS([stdlib.h sys/time.h unistd.h])
+AC_CHECK_FUNCS(strverscmp)
 
 enginesdir=$libdir/emerald/engines
 AC_SUBST(enginesdir)
diff -urN emerald-0.5.2.orig/engines/line.c emerald-0.5.2/engines/line.c
--- emerald-0.5.2.orig/engines/line.c	2007-08-12 17:00:01.000000000 -0400
+++ emerald-0.5.2/engines/line.c	2007-11-01 20:44:46.895838033 -0400
@@ -45,6 +45,7 @@
  */
 typedef struct _private_ws
 {
+  int dummy;
 } private_ws;
 
 void get_meta_info (EngineMetaInfo * emi)
diff -urN emerald-0.5.2.orig/themer/compat.c emerald-0.5.2/themer/compat.c
--- emerald-0.5.2.orig/themer/compat.c	1969-12-31 19:00:00.000000000 -0500
+++ emerald-0.5.2/themer/compat.c	2007-11-01 20:56:51.991921457 -0400
@@ -0,0 +1,74 @@
+/* Copyright (c) 2007 Albert Lee <trisk@acm.jhu.edu>.
+ * Copyright (c) 2006 Ulrich Spoerlein <q@galgenberg.net>.
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "config.h"
+#include "compat.h"
+
+#ifndef HAVE_STRVERSCMP
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int strverscmp(const char *s1, const char *s2)
+{
+	static const char *digits = "0123456789";
+	char *t1, *t2;
+	int ret;
+	long n1, n2;
+	size_t p1, p2;
+
+	p1 = strcspn(s1, digits);
+	p2 = strcspn(s2, digits);
+	while (p1 == p2 && s1[p1] != '\0' && s2[p2] != '\0') {
+		/* Different prefix */
+		if ((ret = strncmp(s1, s2, p1)) != 0)
+			return ret;
+
+		s1 += p1;
+		s2 += p2;
+		n1 = strtol(s1, &t1, 10);
+		n2 = strtol(s2, &t2, 10);
+
+		if (n1 < n2)
+			return -1;
+		else if (n1 > n2)
+			return 1;
+
+		/* One number is "shorter", e.g., "07" vs "007" */
+		if (t1-s1 < t2-s2)
+			return 1;
+		else if (t1-s1 > t2-s2)
+			return -1;
+
+		/* Numbers are equal or not present, try with next ones. */
+		s1 = t1;
+		s2 = t2;
+		p1 = strcspn(s1, digits);
+		p2 = strcspn(s2, digits);
+	}
+
+	return strcmp(s1, s2);
+}
+#endif
diff -urN emerald-0.5.2.orig/themer/compat.h emerald-0.5.2/themer/compat.h
--- emerald-0.5.2.orig/themer/compat.h	1969-12-31 19:00:00.000000000 -0500
+++ emerald-0.5.2/themer/compat.h	2007-11-01 20:57:01.548400361 -0400
@@ -0,0 +1,33 @@
+/* Copyright (c) 2007 Albert Lee <trisk@acm.jhu.edu>.
+ * Copyright (c) 2006 Ulrich Spoerlein <q@galgenberg.net>.
+ * 
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef _COMPAT_H
+#include "config.h"
+
+#ifndef HAVE_STRVERSCMP
+int strverscmp(const char *s1, const char *s2);
+#endif
+
+#endif /* _COMPAT_H */
diff -urN emerald-0.5.2.orig/themer/main.c emerald-0.5.2/themer/main.c
--- emerald-0.5.2.orig/themer/main.c	2007-08-12 17:00:01.000000000 -0400
+++ emerald-0.5.2/themer/main.c	2007-11-01 20:44:46.897672231 -0400
@@ -38,6 +38,7 @@
 GtkWidget * QuitButton;
 gchar * svnpath;
 gchar * themecache;
+#include "compat.h"
 
 static void theme_list_append(gchar * value,gchar * dir, gchar * fil)
 {
diff -urN emerald-0.5.2.orig/themer/Makefile.am emerald-0.5.2/themer/Makefile.am
--- emerald-0.5.2.orig/themer/Makefile.am	2007-08-12 17:00:01.000000000 -0400
+++ emerald-0.5.2/themer/Makefile.am	2007-11-01 20:44:46.896398910 -0400
@@ -1,5 +1,5 @@
 ## Process this file with automake to produce Makefile.in
-emerald_theme_manager_SOURCES = main.c
+emerald_theme_manager_SOURCES = main.c compat.c
 emerald_theme_manager_LDADD = $(EMERALD_LIBS) ../libengine/libemeraldengine.la
 
 bin_PROGRAMS = emerald-theme-manager
