--- espeak-1.31-source.org/src/speech.h	2008-01-19 16:48:52.000000000 -0500
+++ espeak-1.31-source/src/speech.h	2008-01-29 04:04:34.403265000 -0500
@@ -28,7 +28,7 @@
 
 #define PLATFORM_POSIX
 #define PATHSEP  '/'
-#define USE_PORTAUDIO
+//#define USE_PORTAUDIO
 //#define USE_PULSEAUDIO
 #define USE_NANOSLEEP
 #define __cdecl 
--- espeak-1.31-source/src/Makefile-orig	2008-03-21 19:15:59.868061000 -0500
+++ espeak-1.31-source/src/Makefile	2008-03-21 19:16:55.392041000 -0500
@@ -15,6 +15,8 @@ INSTALL = install
 LN_SF = /bin/ln -sf
 MKDIR = mkdir -p
 
+AUDIO = pulseaudio
+
 ifeq ($(AUDIO),pulseaudio)
 WAVE=wave_pulse.cpp
 LIB_AUDIO=pulse
