--- f-spot-0.1.11/src/main.cs.prctl	2006-07-24 20:42:12.561579000 -0400
+++ f-spot-0.1.11/src/main.cs	2006-07-24 20:42:26.455362000 -0400
@@ -128,21 +128,8 @@
 		}
 	}
 
-	[DllImport("libc")]
-	private static extern int prctl(int option, string name, ulong arg3,
-					ulong arg4, ulong arg5);
-	
 	public static void SetProcessName(string name)
 	{
-		try {
-			if(prctl(15 /* PR_SET_NAME */, name,
-				 0, 0, 0) != 0) {
-				throw new ApplicationException("Error setting process name: " +
-							       Mono.Unix.Native.Stdlib.GetLastError());
-			}
-		} catch (DllNotFoundException de) {
-			/* noop */
-		}
 	}
 }
 
