--- faad2-orig/plugins/xmms/src/Makefile.am	2006-09-25 19:52:53.000000000 +0000
+++ faad2/plugins/xmms/src/Makefile.am	2007-11-03 04:50:08.654764130 +0000
@@ -1,9 +1,9 @@
-local_CFLAGS=`$(XMMS_CONFIG) --cflags` -Wall
+local_CFLAGS=`$(XMMS_CONFIG) --cflags`
 local_LDFLAGS=`$(XMMS_CONFIG) --libs`
 libdir = `$(XMMS_CONFIG) --input-plugin-dir`
 lib_LTLIBRARIES = libmp4.la
 
-libmp4_la_CFLAGS = $(local_CFLAGS) -Wall \
+libmp4_la_CFLAGS = $(local_CFLAGS) \
        -I$(top_srcdir)/include -I$(top_srcdir)/common/mp4ff
 
 libmp4_la_LIBADD = $(top_builddir)/libfaad/libfaad.la \
