--- faad2-orig/libfaad/common.h	2007-11-01 12:33:30.000000000 +0000
+++ faad2/libfaad/common.h	2007-11-03 04:45:18.044003192 +0000
@@ -39,11 +39,15 @@
 #  include "../config.h"
 #endif
 
-#if 1
+#if (defined(sun) && defined(__SVR4))
+#define INLINE
+#else
+#if 0
 #define INLINE __inline
 #else
 #define INLINE inline
 #endif
+#endif
 
 #if 0 //defined(_WIN32) && !defined(_WIN32_WCE)
 #define ALIGN __declspec(align(16))
