--- ffmpeg-export-2007-07-31.orig/libavcodec/mlib/dsputil_mlib.c	Tue Jul 31 02:16:05 2007
+++ ffmpeg-export-2007-07-31/libavcodec/mlib/dsputil_mlib.c	Tue Mar 18 23:55:41 2008
@@ -416,7 +416,10 @@
 
 static void ff_fdct_mlib(DCTELEM *data)
 {
+    const mlib_s16 c[1] = {8};
+
     mlib_VideoDCT8x8_S16_S16 (data, data);
+    mlib_VectorMulS_S16_Sat(data, (mlib_s16 *)&c, 64); /* scaling data by 8 */
 }
 
 void dsputil_init_mlib(DSPContext* c, AVCodecContext *avctx)
