Index: trunk/src/interface/prefix.cpp
===================================================================
--- trunk/src/interface/prefix.cpp	(revision 1538)
+++ trunk/src/interface/prefix.cpp	(working copy)
@@ -372,7 +372,8 @@
 char *
 br_extract_dir (const char *path)
 {
-	char *end, *result;
+	const char *end;
+	char *result;
 
 	br_return_val_if_fail (path != (char *) NULL, (char *) NULL);
 
@@ -407,7 +408,8 @@
 char *
 br_extract_prefix (const char *path)
 {
-	char *end, *tmp, *result;
+	const char *end;
+	char *tmp, *result;
 
 	br_return_val_if_fail (path != (char *) NULL, (char *) NULL);
 
