Index: trunk/src/interface/QueueView.h
===================================================================
--- trunk/src/interface/QueueView.h	(revision 2163)
+++ trunk/src/interface/QueueView.h	(working copy)
@@ -75,7 +75,7 @@
 
 	bool IsEmpty() const;
 	int IsActive() const { return m_activeMode; }
-	bool SetActive(const bool active = true);
+	bool SetActive(bool active = true);
 	bool Quit();
 
 	// If the settings are changed, this function will recalculate some
