Index: trunk/src/interface/QueueView.cpp
===================================================================
--- trunk/src/interface/QueueView.cpp	(revision 2163)
+++ trunk/src/interface/QueueView.cpp	(working copy)
@@ -2791,7 +2791,8 @@
 
 	t_EngineData* pEngineData = *ppEngineData;
 
-	for (std::vector<t_EngineData*>::iterator iter = ++m_engineData.begin(); iter != m_engineData.end(); iter++)
+	std::vector<t_EngineData*>::iterator tmp_iter = m_engineData.begin();
+	for (std::vector<t_EngineData*>::iterator iter = ++tmp_iter; iter != m_engineData.end(); iter++)
 	{
 		t_EngineData* pNewEngineData = *iter;
 		if (pNewEngineData == pEngineData)
