diff -ur fltk-2.0.x-r5783-orig/glut/Makefile fltk-2.0.x-r5783/glut/Makefile
--- fltk-2.0.x-r5783-orig/glut/Makefile	2007-01-07 14:23:21.000000000 +0700
+++ fltk-2.0.x-r5783/glut/Makefile	2007-04-30 10:47:21.249181376 +0700
@@ -97,21 +97,21 @@
 
 install_static: static
 	@echo "Installing static libraries..."
-	@mkdir -p $(libdir)
-	@rm -f $(libdir)/$(LIBNAME)_glut$(LIBEXT)
-	@cp $(LIBNAME) $(libdir)
-	@-chmod 644 $(libdir)/$(LIBNAME)_glut$(LIBEXT) 
+	@mkdir -p $(DESTDIR)$(libdir)
+	@rm -f $(DESTDIR)$(libdir)/$(LIBNAME)_glut$(LIBEXT)
+	@cp $(DESTDIR)$(LIBNAME) $(libdir)
+	@-chmod 644 $(DESTDIR)$(libdir)/$(LIBNAME)_glut$(LIBEXT) 
 
 install_shared: shared
 	@echo "Installing shared libraries..."
-	@rm -f $(libdir)/$(DLLIMPORTNAME)
-	@if [ -f "$(libdir)/$(DSOLIBglutNAME)$(DSOEXT)" ]; then rm -f $(libdir)/$(DSOLIBglutNAME)$(DSOEXT); fi
-	@if [ "$(DSOSHORTEXT)" ]; then rm -f $(libdir)/$(DSOLIBglutNAME)$(DSOSHORTEXT); fi
-	@if [ "$(DSO)$(DSOEXT)" ]; then cp $(DSO)$(DSOEXT) $(libdir); fi
-	@chmod 755 $(libdir)/$(DSOLIBglutNAME)$(DSOEXT)
+	@rm -f $(DESTDIR)$(libdir)/$(DLLIMPORTNAME)
+	@if [ -f "$(DESTDIR)$(libdir)/$(DSOLIBglutNAME)$(DSOEXT)" ]; then rm -f $(DESTDIR)$(libdir)/$(DSOLIBglutNAME)$(DSOEXT); fi
+	@if [ "$(DSOSHORTEXT)" ]; then rm -f $(DESTDIR)$(libdir)/$(DSOLIBglutNAME)$(DSOSHORTEXT); fi
+	@if [ "$(DSO)$(DSOEXT)" ]; then cp $(DSO)$(DSOEXT) $(DESTDIR)$(libdir); fi
+	@chmod 755 $(DESTDIR)$(libdir)/$(DSOLIBglutNAME)$(DSOEXT)
 	@if [ "$(DSOSHORTEXT)" ]; then \
-		ln -s $(DSOLIBglutNAME)$(DSOEXT) $(libdir)/$(DSOLIBglutNAME)$(DSOSHORTEXT); fi
-	@if [ -f "../lib/$(DLLIMPORTNAME)" ]; then cp ../lib/$(DLLIMPORTNAME) $(libdir); fi
+		ln -s $(DSOLIBglutNAME)$(DSOEXT) $(DESTDIR)$(libdir)/$(DSOLIBglutNAME)$(DSOSHORTEXT); fi
+	@if [ -f "../lib/$(DLLIMPORTNAME)" ]; then cp ../lib/$(DLLIMPORTNAME) $(DESTDIR)$(libdir); fi
 
 install_programs:
 
