diff -ur fltk-2.0.x-r5783-orig/documentation/Makefile fltk-2.0.x-r5783/documentation/Makefile
--- fltk-2.0.x-r5783-orig/documentation/Makefile	2006-06-25 13:11:31.000000000 +0700
+++ fltk-2.0.x-r5783/documentation/Makefile	2007-04-30 10:54:13.265954278 +0700
@@ -42,11 +42,11 @@
 # 	$(CP) $(HTMLFILES) *.gif *.jpg index.html $(docdir)
 # 	$(CHMOD) 644 $(docdir)/*
 	echo "Installing man pages in $(mandir)"
-	-$(MKDIR) -p $(DESTDIR)$(mandir)/man1
+	-$(MKDIR) -p $(DESTDIR)$(mandir)/man1 ; \
 	for file in $(MAN1FILES); do \
 		$(CP) $$file $(DESTDIR)$(mandir)/man1/`basename $$file .man`.1; \
 	done
-	-$(MKDIR) -p $(DESTDIR)$(mandir)/man3
+	-$(MKDIR) -p $(DESTDIR)$(mandir)/man3 ; \
 	for file in $(MAN3FILES); do \
 		$(CP) $$file $(DESTDIR)$(mandir)/man3/`basename $$file .man`.3; \
 	done
