diff -ur fltk-2.0.x-r5783-orig/configure fltk-2.0.x-r5783/configure
--- fltk-2.0.x-r5783-orig/configure	2007-04-27 11:00:55.000000000 +0700
+++ fltk-2.0.x-r5783/configure	2007-04-30 10:19:04.250772711 +0700
@@ -10141,7 +10141,7 @@
 echo "$as_me: WARNING: Ignoring libraries \"$X_PRE_LIBS\" requested by configure." >&2;}
 	fi
 
-	LIBS="$LIBS -lXext $X_EXTRA_LIBS"
+	LIBS="$LIBS -lX11 -lXext $X_EXTRA_LIBS"
 	CFLAGS="$CFLAGS$X_CFLAGS"
 	CXXFLAGS="$CXXFLAGS$X_CFLAGS"
 	LDFLAGS="$X_LIBS $LDFLAGS"
diff -ur fltk-2.0.x-r5783-orig/configure.in fltk-2.0.x-r5783/configure.in
--- fltk-2.0.x-r5783-orig/configure.in	2007-01-03 11:06:26.000000000 +0700
+++ fltk-2.0.x-r5783/configure.in	2007-04-30 10:19:22.143645754 +0700
@@ -605,7 +605,7 @@
 	    AC_MSG_WARN(Ignoring libraries \"$X_PRE_LIBS\" requested by configure.)
 	fi
 
-	LIBS="$LIBS -lXext $X_EXTRA_LIBS"
+	LIBS="$LIBS -lX11 -lXext $X_EXTRA_LIBS"
 	CFLAGS="$CFLAGS$X_CFLAGS"
 	CXXFLAGS="$CXXFLAGS$X_CFLAGS"
 	LDFLAGS="$X_LIBS $LDFLAGS"
