diff -ur freeciv-2.1.0-beta4-orig/acinclude.m4 freeciv-2.1.0-beta4/acinclude.m4
--- freeciv-2.1.0-beta4-orig/acinclude.m4	2007-04-01 04:33:43.000000000 +0700
+++ freeciv-2.1.0-beta4/acinclude.m4	2007-04-22 00:07:41.469710830 +0700
@@ -1636,8 +1636,8 @@
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
 	[[void signedness(void){char c;if(c==-1)c=0;}]])],
 	[],
-	[save_cflags="$save_cflags -fsigned-char"
-	 save_cxxflags="$save_cxxflags -fsigned-char"])
+	[save_cflags="$save_cflags -xchar=s"
+	 save_cxxflags="$save_cxxflags -xchar=s"])
 CFLAGS=$save_cflags
 CXXFLAGS=$save_cxxflags
 ])
diff -ur freeciv-2.1.0-beta4-orig/configure freeciv-2.1.0-beta4/configure
--- freeciv-2.1.0-beta4-orig/configure	2007-04-01 04:34:26.000000000 +0700
+++ freeciv-2.1.0-beta4/configure	2007-04-22 00:07:57.027255894 +0700
@@ -16642,8 +16642,8 @@
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-	save_cflags="$save_cflags -fsigned-char"
-	 save_cxxflags="$save_cxxflags -fsigned-char"
+	save_cflags="$save_cflags -xchar=s"
+	 save_cxxflags="$save_cxxflags -xchar=s"
 fi
 
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
