diff -ur freeciv-2.1.0-beta1-orig/utility/support.h freeciv-2.1.0-beta1/utility/support.h
--- freeciv-2.1.0-beta1-orig/utility/support.h	2006-03-02 13:33:24.000000000 +0700
+++ freeciv-2.1.0-beta1/utility/support.h	2006-05-18 16:19:41.157909385 +0700
@@ -83,7 +83,11 @@
 size_t mystrlcat(char *dest, const char *src, size_t n);
 
 /* convenience macros for use when dest is a char ARRAY: */
-#define sz_strlcpy(dest,src) ((void)mystrlcpy((dest),(src),sizeof(dest)))
+#ifdef __SUNPRO_C
+#define sz_strlcpy(dest,src) ((size_t)mystrlcpy((dest),(src),sizeof(dest)))
+#else
+#define sz_strlcpy(dest,src) ((size_t)mystrlcpy((dest),(src),sizeof(dest)))
+#endif
 #define sz_strlcat(dest,src) ((void)mystrlcat((dest),(src),sizeof(dest)))
 
 int my_snprintf(char *str, size_t n, const char *format, ...)
