diff -ur gamin-0.1.8-orig/server/gam_server.c gamin-0.1.8/server/gam_server.c
--- gamin-0.1.8-orig/server/gam_server.c	2006-10-25 23:13:49.000000000 +0700
+++ gamin-0.1.8/server/gam_server.c	2007-04-15 15:26:09.807568834 +0700
@@ -164,7 +164,7 @@
 #endif	
 	}
 
-	if (gam_poll_generic_init()) {
+	if (gam_poll_basic_init()) {
 		GAM_DEBUG(DEBUG_INFO, "Using poll as backend\n");
 		return(TRUE);
 	}
