--- gargoyle-2008.12.25/terps/advsys/advint.c-orig	2008-12-30 18:25:58.144289000 -0600
+++ gargoyle-2008.12.25/terps/advsys/advint.c	2008-12-30 18:28:41.680399000 -0600
@@ -17,7 +17,7 @@ jmp_buf restart;
 /* CHANGED TO WORK WITH GLK */
 /* Modernize it */
 void play(void);
-int single(void);
+int single_action(void);
 
 /* GLK Specifics */
 winid_t window;
@@ -95,8 +95,8 @@ void play()
 
 	/* parse the next input command */
 	if (parse()) {
-	    if (single())
-		while (next() && single())
+	    if (single_action())
+		while (next() && single_action())
 		    ;
 	}
 
@@ -107,7 +107,7 @@ void play()
 }
 
 /* single - handle a single action */
-int single()
+int single_action()
 {
     /* execute the before code */
     switch (execute(h_before)) {
