Index: gdl-trunk/gdl/gdl-tools.h
===================================================================
--- gdl-trunk/gdl/gdl-tools.h	(revision 372)
+++ gdl-trunk/gdl/gdl-tools.h	(working copy)
@@ -137,7 +137,7 @@
  * the _WITH_DEFAULT version for that */
 #define GDL_CALL_PARENT(parent_class_cast, name, args)          \
 	((parent_class_cast(parent_class)->name != NULL) ?      \
-	 parent_class_cast(parent_class)->name args : (void *)0)
+	 parent_class_cast(parent_class)->name args : (void)0)
 
 #define GDL_CALL_PARENT_GBOOLEAN(parent_class_cast, name, args)          \
        ((parent_class_cast(parent_class)->name != NULL) ?      \
