--- gdm-2.21.9/gui/simple-greeter/gdm-greeter-session.c.orig	2008-05-02 01:00:56.185966000 +0800
+++ gdm-2.21.9/gui/simple-greeter/gdm-greeter-session.c	2008-05-02 01:12:57.405620000 +0800
@@ -637,13 +637,23 @@
 {
         GError  *error;
         gboolean ret;
+        char *argv[] = {NULL, NULL, NULL};
 
         g_debug ("GdmGreeterSession: Launching settings daemon");
 
         ret = FALSE;
 
         error = NULL;
-        g_spawn_command_line_async (LIBEXECDIR "/gnome-settings-daemon --gconf-prefix=/apps/gdm/simple-greeter/settings-manager-plugins", &error);
+        argv[0] = g_strdup_printf ("%s%s", LIBEXECDIR, "/gnome-settings-daemon"); 
+        argv[1] = "--gconf-prefix=/apps/gdm/simple-greeter/settings-manager-plugins";
+        /* 
+         * Below enviroment variable will be inherited by child process
+         * and used to override user home directory in child process.
+         */
+        g_setenv ("GNOME22_USER_DIR", "/tmp/.gdm_gnome2", TRUE);
+        g_spawn_async (NULL, argv, NULL, G_SPAWN_SEARCH_PATH,
+		       NULL, NULL, NULL, &error);
+	g_free (argv[0]);
         if (error != NULL) {
                 g_warning ("Error starting settings daemon: %s", error->message);
                 g_error_free (error);
@@ -662,7 +672,7 @@
                    GdmGreeterSession *session)
 {
         if (enabled) {
-                activate_settings_daemon (session);
+                start_settings_daemon (session);
                 start_window_manager (session);
         } else {
         }
