--- gdm-2.25.1/gui/simple-greeter/gdm-option-widget.c-orig	2008-12-15 23:14:36.580573116 -0600
+++ gdm-2.25.1/gui/simple-greeter/gdm-option-widget.c	2008-12-15 23:14:53.554337419 -0600
@@ -618,9 +618,10 @@ path_is_separator (GdmOptionWidget *widg
 static gboolean
 gdm_option_widget_check_visibility (GdmOptionWidget *widget)
 {
-        if (widget->priv->number_of_top_rows > 0 ||
-            widget->priv->number_of_middle_rows > 1 || 
-            widget->priv->number_of_bottom_rows > 0) {
+        if ((widget->priv->number_of_middle_rows != 0) &&
+            (widget->priv->number_of_top_rows > 0 ||
+             widget->priv->number_of_middle_rows > 1 || 
+             widget->priv->number_of_bottom_rows > 0)) {
                 gtk_widget_show (widget->priv->items_combo_box);
                 gtk_widget_show (widget->priv->label);
 
