--- gdm-2.24.0/configure.ac-orig	2008-11-04 11:10:11.527216000 -0600
+++ gdm-2.24.0/configure.ac	2008-11-04 11:10:33.081167000 -0600
@@ -1410,6 +1410,15 @@ if test "$enable_profiling" = "yes"; the
     AC_DEFINE(ENABLE_PROFILING,1,[enable profiling])
 fi
 
+#
+# Set SHELL to use in scripts.
+#
+if test x$os_solaris = xyes ; then
+   XSESSION_SHELL=/bin/ksh
+else
+   XSESSION_SHELL=/bin/sh
+fi
+
 # Set configuration choices.
 #
 AC_SUBST(XSESSION_SHELL)
