--- gdm-2.22.0/daemon/gdm-display-access-file.c.orig	2008-05-21 23:00:26.383800000 +0800
+++ gdm-2.22.0/daemon/gdm-display-access-file.c	2008-05-21 22:50:30.444254000 +0800
@@ -227,6 +227,23 @@
 
         fp = NULL;
 
+        /* Create directory if not exist then set permission 01775 and ownership root:gdm */ 
+        if (g_file_test (GDM_XAUTH_DIR, G_FILE_TEST_IS_DIR) == FALSE) {
+                g_unlink (GDM_XAUTH_DIR);
+                if (g_mkdir (GDM_XAUTH_DIR, S_ISVTX|S_IRWXU|S_IRWXG|S_IROTH|S_IXOTH) != 0) {
+                        g_set_error (error,
+                                     G_FILE_ERROR,
+                                     g_file_error_from_errno (errno),
+                                     "%s", g_strerror (errno));
+                        fp = -1;
+                        goto out;
+                }
+
+                g_chmod (GDM_XAUTH_DIR, S_ISVTX|S_IRWXU|S_IRWXG|S_IROTH|S_IXOTH);
+                _get_uid_and_gid_for_user ("gdm", &uid, &gid);
+                chown (GDM_XAUTH_DIR, 0, gid);
+        }
+ 
         template = g_strdup_printf (GDM_XAUTH_DIR
                                     "/auth-cookie-XXXXXXXX-for-%s",
                                     username);
