Index: daemon/gdm-simple-slave.c
===================================================================
--- gdm-2.22.0/daemon/gdm-simple-slave.c	(revision 6253)
+++ gdm-2.22.0/daemon/gdm-simple-slave.c	(working copy)
@@ -99,8 +99,14 @@
                     int               pid,
                     GdmSimpleSlave   *slave)
 {
+        char *username;
+
         g_debug ("GdmSimpleSlave: session started %d", pid);
 
+        /* Run the PreSession script. gdmslave suspends until script has terminated */
+        username = gdm_session_direct_get_username (slave->priv->session);
+        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/PreSession", username);
+
         /* FIXME: should we do something here? */
 }
 
@@ -109,8 +115,14 @@
                    int             exit_code,
                    GdmSimpleSlave *slave)
 {
+        char *username;
+
         g_debug ("GdmSimpleSlave: session exited with code %d\n", exit_code);
 
+        /* Run the PostSession script. gdmslave suspends until script has terminated */
+        username = gdm_session_direct_get_username (slave->priv->session);
+        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/PostSession", username);
+
         gdm_slave_stopped (GDM_SLAVE (slave));
 }
 
@@ -353,6 +365,9 @@
                        GdmSimpleSlave *slave)
 {
         queue_start_session (slave);
+
+        /* Run the PostLogin script. gdmslave suspends until script has terminated */
+        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/PostLogin", "gdm");
 }
 
 static void
@@ -840,7 +855,7 @@
         }
 
         /* Run the init script. gdmslave suspends until script has terminated */
-        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/Init/Default", "gdm");
+        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/Init", "gdm");
 
         create_new_session (slave);
 
--- gdm-2.22.0/daemon/gdm-slave.c-orig	2008-06-03 01:29:06.653359000 -0500
+++ gdm-2.22.0/daemon/gdm-slave.c	2008-06-03 01:29:27.383615000 -0500
@@ -265,7 +265,8 @@ gdm_slave_run_script (GdmSlave   *slave,
         }
 
         if (script == NULL &&
-            slave->priv->display_hostname != NULL) {
+            slave->priv->display_hostname != NULL &&
+            slave->priv->display_hostname[0] != NULL) {
                 script = g_build_filename (dir, slave->priv->display_hostname, NULL);
                 if (g_access (script, R_OK | X_OK) != 0) {
                         g_debug ("GdmSlave: script %s not found; skipping", script);
Index: gdm-factory-slave.c
===================================================================
--- gdm-factory-slave.c	(revision 6253)
+++ gdm-factory-slave.c	(working copy)
@@ -489,7 +489,7 @@
         /* FIXME: send a signal back to the master */
 
         /* Run the init script. gdmslave suspends until script has terminated */
-        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/Init/Default", "gdm");
+        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/Init", "gdm");
 
         slave->priv->greeter_server = gdm_greeter_server_new (display_id);
         g_signal_connect (slave->priv->greeter_server,
Index: gdm-xdmcp-chooser-slave.c
===================================================================
--- gdm-xdmcp-chooser-slave.c	(revision 6253)
+++ gdm-xdmcp-chooser-slave.c	(working copy)
@@ -190,7 +190,7 @@
         }
 
         /* Run the init script. gdmslave suspends until script has terminated */
-        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/Init/Default", "gdm");
+        gdm_slave_run_script (GDM_SLAVE (slave), GDMCONFDIR "/Init", "gdm");
 
         slave->priv->chooser_server = gdm_chooser_server_new (display_id);
         g_signal_connect (slave->priv->chooser_server,
