--- gdm-2.21.8/gui/simple-greeter/gdm-greeter-login-window.c.orig	2008-02-29 10:36:34.465962000 +0800
+++ gdm-2.21.8/gui/simple-greeter/gdm-greeter-login-window.c	2008-02-29 10:37:49.331362000 +0800
@@ -1163,7 +1163,10 @@
                         g_error_free (error);
                 }
                 g_object_unref (client);
-                g_debug ("GdmGreeterLoginWindow: Got greeter logo '%s'", icon_name);
+                if (icon_name) 
+			g_debug ("GdmGreeterLoginWindow: Got greeter logo '%s'", icon_name);
+		else
+			g_debug ("GdmGreeterLoginWindow: Got greeter logo: icon_name is NULL");
                 if (icon_name != NULL) {
                         gtk_image_set_from_icon_name (GTK_IMAGE (image),
                                                       icon_name,
