--- gdm-2.25.1/daemon/gdm-display.c-orig	2008-12-12 15:18:43.780058408 -0600
+++ gdm-2.25.1/daemon/gdm-display.c	2008-12-12 15:29:28.523914783 -0600
@@ -380,7 +380,7 @@ gdm_display_get_timed_login_details (Gdm
         g_debug ("GdmSlave: Got timed login details for display %s: %d '%s' %d",
                  display->priv->x11_display_name,
                  *enabled,
-                 *username,
+                 *username ? *username : "(null)",
                  *delay);
 
         return TRUE;
