--- daemon/gdm-welcome-session.c.orig	2008-09-06 18:40:04.753453000 +0800
+++ daemon/gdm-welcome-session.c	2008-09-06 18:42:20.633895000 +0800
@@ -278,6 +278,8 @@
 
         g_hash_table_insert (hash, g_strdup ("XAUTHORITY"), g_strdup (welcome_session->priv->x11_authority_file));
         g_hash_table_insert (hash, g_strdup ("DISPLAY"), g_strdup (welcome_session->priv->x11_display_name));
+	/* Due to HOME dirctory of gdm user, use tmp dir instead */
+        g_hash_table_insert (hash, g_strdup ("ICEAUTHORITY"), g_strdup ("/tmp/.ICEauthority"));
 
 #if 0
         /* hackish ain't it */
--- gui/simple-greeter/gdm-option-widget.c.orig	2008-09-08 16:25:23.929558000 +0800
+++ gui/simple-greeter/gdm-option-widget.c	2008-09-08 16:30:11.180315000 +0800
@@ -281,9 +281,10 @@
                                     const char      *item)
 {
         g_return_if_fail (GDM_IS_OPTION_WIDGET (widget));
-        g_return_if_fail (item == NULL ||
-                          gdm_option_widget_lookup_item (widget, item,
-                                                         NULL, NULL, NULL));
+        if ((item != NULL &&
+                          ! gdm_option_widget_lookup_item (widget, item,
+                                                         NULL, NULL, NULL)))
+	return;
 
         if (widget->priv->default_item_id == NULL ||
             item == NULL ||
