diff -uNr geany-0.11.orig/src/Makefile geany-0.11/src/Makefile
--- geany-0.11.orig/src/Makefile	2007-08-04 20:34:33.245681000 +0530
+++ geany-0.11/src/Makefile	2007-08-04 20:42:47.817032000 +0530
@@ -246,14 +246,14 @@
 #AM_CFLAGS = -DGEANY_DEBUG -Wall -pipe -g -O0
 geany_SOURCES = $(SRCS) vte.c vte.h
 #geany_SOURCES = $(SRCS) win32.c win32.h
-geany_LDADD = ../scintilla/libscintilla.a ../tagmanager/libtagmanager.a -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgdk_pixbuf-2.0 -lm -lmlib -lpangocairo-1.0 -lpango-1.0 -lcairo -lgobject-2.0 -lgmodule-2.0 -lglib-2.0   -lstdc++ $(INTLLIBS)
+geany_LDADD = ../scintilla/libscintilla.a ../tagmanager/libtagmanager.a -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgdk_pixbuf-2.0 -lm -lmlib -lpangocairo-1.0 -lpango-1.0 -lcairo -lgobject-2.0 -lgmodule-2.0 -lglib-2.0   -lCrun $(INTLLIBS)
 #geany_LDADD = ../scintilla/libscintilla.a ../tagmanager/libtagmanager.a -lstdc++ -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgdk_pixbuf-2.0 -lm -lmlib -lpangocairo-1.0 -lpango-1.0 -lcairo -lgobject-2.0 -lgmodule-2.0 -lglib-2.0   \
 #			$(INTLLIBS) -lgdi32 -limm32 -lshell32 -lole32 -luuid -liberty -lcomdlg32 -lcomctl32 \
 #			geany_private.res
 
 
 # build Geany for all other platforms
-AM_CFLAGS = -Wall -pipe
+AM_CFLAGS = 
 #AM_CFLAGS = -Wall -pipe -mms-bitfields
 #AM_CFLAGS = -DGEANY_DEBUG -Wall -pipe -mms-bitfields -g -O0
 #geany_LDFLAGS = -mwindows
diff -uNr geany-0.11.orig/tagmanager/Makefile geany-0.11/tagmanager/Makefile
--- geany-0.11.orig/tagmanager/Makefile	2007-08-04 20:34:32.771101000 +0530
+++ geany-0.11/tagmanager/Makefile	2007-08-04 20:42:17.205746000 +0530
@@ -208,7 +208,7 @@
 #	-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED \
 #	-DGDK_PIXBUF_DEPRECATED \
 #	-DGTK_DISABLE_DEPRECATED -DGNOME_DISABLE_DEPRECATED
-AM_CFLAGS = -Wall
+AM_CFLAGS = 
 noinst_LIBRARIES = libtagmanager.a
 libtagmanager_a_SOURCES = \
 	general.h\
