diff -Naur GeoIP-1.4.5.old/Makefile.am GeoIP-1.4.5/Makefile.am
--- GeoIP-1.4.5.old/Makefile.am	2008-10-27 20:05:14.015985938 +0200
+++ GeoIP-1.4.5/Makefile.am	2008-10-27 20:31:58.385580374 +0200
@@ -1,6 +1,6 @@
 NULL =
 
-INCLUDES = -Wall -ansi
+INCLUDES =
 
 SUBDIRS =		\
 	libGeoIP	\
diff -Naur GeoIP-1.4.5.old/Makefile.in GeoIP-1.4.5/Makefile.in
--- GeoIP-1.4.5.old/Makefile.in	2008-10-27 20:05:13.952071131 +0200
+++ GeoIP-1.4.5/Makefile.in	2008-10-27 20:31:45.725524309 +0200
@@ -89,7 +89,7 @@
 
 NULL = 
 
-INCLUDES = -Wall
+INCLUDES =
 
 SUBDIRS =  	libGeoIP		apps			conf			data			test			man			$(NULL)
 
diff -Naur GeoIP-1.4.5.old/apps/Makefile.am GeoIP-1.4.5/apps/Makefile.am
--- GeoIP-1.4.5.old/apps/Makefile.am	2008-10-27 20:05:14.027048801 +0200
+++ GeoIP-1.4.5/apps/Makefile.am	2008-10-27 20:34:25.411562926 +0200
@@ -1,8 +1,8 @@
 INCLUDES = 	\
-	-I$(top_srcdir)/libGeoIP		\
-	-Wall
+	-I$(top_srcdir)/libGeoIP		
 
-AM_CPPFLAGS = -DSYSCONFDIR=\"$(sysconfdir)\" -Wall
+
+AM_CPPFLAGS = -DSYSCONFDIR=\"$(sysconfdir)\"
 
 DEPS = $(top_builddir)/libGeoIP/libGeoIP.la
 LDADDS = $(top_builddir)/libGeoIP/libGeoIP.la
diff -Naur GeoIP-1.4.5.old/apps/Makefile.in GeoIP-1.4.5/apps/Makefile.in
--- GeoIP-1.4.5.old/apps/Makefile.in	2008-10-27 20:05:14.026242730 +0200
+++ GeoIP-1.4.5/apps/Makefile.in	2008-10-27 20:34:39.595585673 +0200
@@ -87,10 +87,10 @@
 VERSION = @VERSION@
 WITH_CITY = @WITH_CITY@
 
-INCLUDES =  	-I$(top_srcdir)/libGeoIP			-Wall
+INCLUDES =  	-I$(top_srcdir)/libGeoIP
 
 
-AM_CPPFLAGS = -DSYSCONFDIR=\"$(sysconfdir)\" -Wall
+AM_CPPFLAGS = -DSYSCONFDIR=\"$(sysconfdir)\"
 
 DEPS = $(top_builddir)/libGeoIP/libGeoIP.la
 LDADDS = $(top_builddir)/libGeoIP/libGeoIP.la
diff -Naur GeoIP-1.4.5.old/libGeoIP/GeoIPCity.h GeoIP-1.4.5/libGeoIP/GeoIPCity.h
--- GeoIP-1.4.5.old/libGeoIP/GeoIPCity.h	2008-10-27 20:05:14.020398096 +0200
+++ GeoIP-1.4.5/libGeoIP/GeoIPCity.h	2008-10-27 22:51:49.822806357 +0200
@@ -36,10 +36,7 @@
 	char *postal_code;
 	float latitude;
 	float longitude;
-	union {
-	  int metro_code; /* metro_code is a alias for dma_code */
-	  int dma_code;
-        };
+	int metro_code;
 	int area_code;
 	int charset;
 	char *continent_code;
diff -Naur GeoIP-1.4.5.old/libGeoIP/GeoIPUpdate.c GeoIP-1.4.5/libGeoIP/GeoIPUpdate.c
--- GeoIP-1.4.5.old/libGeoIP/GeoIPUpdate.c	2008-10-27 20:05:14.025348346 +0200
+++ GeoIP-1.4.5/libGeoIP/GeoIPUpdate.c	2008-10-27 21:55:02.806511890 +0200
@@ -22,6 +22,7 @@
 #include "GeoIP.h"
 #include "GeoIPUpdate.h"
 #include "GeoIP_internal.h"
+#include "vasprintf.h"
 
 #include "global.h"
 #include "md5.h"
diff -Naur GeoIP-1.4.5.old/libGeoIP/Makefile.am GeoIP-1.4.5/libGeoIP/Makefile.am
--- GeoIP-1.4.5.old/libGeoIP/Makefile.am	2008-10-27 20:05:14.023969014 +0200
+++ GeoIP-1.4.5/libGeoIP/Makefile.am	2008-10-27 20:33:37.454787884 +0200
@@ -2,7 +2,7 @@
 
 EXTRA_DIST = md5.h global.h types.h GeoIP_internal.h
 
-AM_CPPFLAGS = -DGEOIPDATADIR=\"$(pkgdatadir)\" -Wall
+AM_CPPFLAGS = -DGEOIPDATADIR=\"$(pkgdatadir)\"
 
 if WITH_CITY
 libGeoIP_la_SOURCES = GeoIP.c GeoIPCity.c regionName.c timeZone.c
diff -Naur GeoIP-1.4.5.old/libGeoIP/Makefile.in GeoIP-1.4.5/libGeoIP/Makefile.in
--- GeoIP-1.4.5.old/libGeoIP/Makefile.in	2008-10-27 20:05:14.025065840 +0200
+++ GeoIP-1.4.5/libGeoIP/Makefile.in	2008-10-27 20:34:00.748334148 +0200
@@ -91,7 +91,7 @@
 
 EXTRA_DIST = md5.h global.h types.h GeoIP_internal.h
 
-AM_CPPFLAGS = -DGEOIPDATADIR=\"$(pkgdatadir)\" -Wall
+AM_CPPFLAGS = -DGEOIPDATADIR=\"$(pkgdatadir)\"
 @WITH_CITY_TRUE@libGeoIP_la_SOURCES = GeoIP.c GeoIPCity.c regionName.c timeZone.c
 @WITH_CITY_FALSE@libGeoIP_la_SOURCES = GeoIP.c regionName.c timeZone.c
 @WITH_CITY_TRUE@include_HEADERS = GeoIP.h GeoIPCity.h GeoIPUpdate.h
diff -Naur GeoIP-1.4.5.old/libGeoIP/vasprintf.h GeoIP-1.4.5/libGeoIP/vasprintf.h
--- GeoIP-1.4.5.old/libGeoIP/vasprintf.h	1970-01-01 02:00:00.000000000 +0200
+++ GeoIP-1.4.5/libGeoIP/vasprintf.h	2008-10-27 21:53:59.233652558 +0200
@@ -0,0 +1,79 @@
+#ifndef VASPRINTF_H
+#define VASPRINTF_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef HAVE_VASPRINTF
+#include <errno.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#ifndef VA_COPY
+# ifdef HAVE_VA_COPY
+#  define VA_COPY(dest, src) va_copy(dest, src)
+# else
+#  ifdef HAVE___VA_COPY
+#   define VA_COPY(dest, src) __va_copy(dest, src)
+#  else
+#   define VA_COPY(dest, src) (dest) = (src)
+#  endif
+# endif
+#endif
+
+#define INIT_SZ	128
+
+static int vasprintf(char **str, const char *fmt, va_list ap);
+
+static int vasprintf(char **str, const char *fmt, va_list ap)
+{
+	int ret = -1;
+	va_list ap2;
+	char *string, *newstr;
+	size_t len;
+
+	VA_COPY(ap2, ap);
+	if ((string = malloc(INIT_SZ)) == NULL)
+		goto fail;
+
+	ret = vsnprintf(string, INIT_SZ, fmt, ap2);
+	if (ret >= 0 && ret < INIT_SZ) { /* succeeded with initial alloc */
+		*str = string;
+	} else if (ret == INT_MAX) { /* shouldn't happen */
+		goto fail;
+	} else {	/* bigger than initial, realloc allowing for nul */
+		len = (size_t)ret + 1;
+		if ((newstr = realloc(string, len)) == NULL) {
+			free(string);
+			goto fail;
+		} else {
+			va_end(ap2);
+			VA_COPY(ap2, ap);
+			ret = vsnprintf(newstr, len, fmt, ap2);
+			if (ret >= 0 && (size_t)ret < len) {
+				*str = newstr;
+			} else { /* failed with realloc'ed string, give up */
+				free(newstr);
+				goto fail;
+			}
+		}
+	}
+	va_end(ap2);
+	return (ret);
+
+fail:
+	*str = NULL;
+	errno = ENOMEM;
+	va_end(ap2);
+	return (-1);
+}
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* VASPRINTF_H */
diff -Naur GeoIP-1.4.5.old/test/Makefile.am GeoIP-1.4.5/test/Makefile.am
--- GeoIP-1.4.5.old/test/Makefile.am	2008-10-27 20:05:13.947914268 +0200
+++ GeoIP-1.4.5/test/Makefile.am	2008-10-27 20:35:08.668756354 +0200
@@ -2,7 +2,6 @@
 
 INCLUDES =	\
 	-I$(top_srcdir)/libGeoIP		\
-	-Wall					\
 	$(NULL)
 
 check_PROGRAMS = test-geoip
diff -Naur GeoIP-1.4.5.old/test/Makefile.in GeoIP-1.4.5/test/Makefile.in
--- GeoIP-1.4.5.old/test/Makefile.in	2008-10-27 20:05:13.949233894 +0200
+++ GeoIP-1.4.5/test/Makefile.in	2008-10-27 20:35:21.873243969 +0200
@@ -89,7 +89,7 @@
 
 NULL = 
 
-INCLUDES =  	-I$(top_srcdir)/libGeoIP			-Wall					$(NULL)
+INCLUDES =  	-I$(top_srcdir)/libGeoIP $(NULL)
 
 
 check_PROGRAMS = test-geoip
