--- gettext-0.17/gettext-tools/src/Makefile.am-orig	2007-12-10 21:26:48.898968000 -0600
+++ gettext-0.17/gettext-tools/src/Makefile.am	2007-12-10 21:41:17.118388000 -0600
@@ -235,22 +235,24 @@ msgmerge_CXXFLAGS = $(AM_CXXFLAGS) $(OPE
 # INTL_MACOSX_LIBS is needed because the programs depend on libintl.la
 # but libtool doesn't put -Wl,-framework options into .la files.
 # For msginit, it is also needed because of localename.c.
-msgcmp_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgfmt_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgmerge_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ @MSGMERGE_LIBM@ $(WOE32_LDADD) $(OPENMP_CFLAGS)
-msgunfmt_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-xgettext_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ @LTLIBEXPAT@ $(WOE32_LDADD)
-msgattrib_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgcat_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgcomm_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgconv_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgen_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgexec_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msgfilter_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msggrep_LDADD = $(LIBGREP) libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msginit_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-msguniq_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
-recode_sr_latin_LDADD = ../gnulib-lib/libgettextlib.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD)
+msgcmp_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgfmt_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgmerge_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ @MSGMERGE_LIBM@ $(WOE32_LDADD) $(OPENMP_CFLAGS) -lsec
+msgunfmt_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+xgettext_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ @LTLIBEXPAT@ $(WOE32_LDADD) -lsec
+msgattrib_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgcat_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgcomm_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgconv_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgen_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgexec_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msgfilter_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msggrep_LDADD = $(LIBGREP) libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msginit_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+msguniq_LDADD = libgettextsrc.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+recode_sr_latin_LDADD = ../gnulib-lib/libgettextlib.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+hostname_LDADD = ../gnulib-lib/libgettextlib.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
+urlget_LDADD = ../gnulib-lib/libgettextlib.la @INTL_MACOSX_LIBS@ $(WOE32_LDADD) -lsec
 
 # Specify installation directory, for --enable-relocatable.
 msgcmp_CPPFLAGS = $(AM_CPPFLAGS) -DINSTALLDIR=\"$(bindir)\"
--- gettext-0.17/gettext-tools/tests/Makefile.am-orig	2007-12-10 21:49:16.767562000 -0600
+++ gettext-0.17/gettext-tools/tests/Makefile.am	2007-12-10 21:52:31.148824000 -0600
@@ -202,17 +202,17 @@ LDADD_no = ../intl/libgnuintl.la @LTLIBT
 noinst_PROGRAMS = tstgettext tstngettext testlocale gettext-3-prg gettext-4-prg gettext-5-prg gettext-6-prg gettext-7-prg cake fc3 fc4 fc5
 tstgettext_SOURCES = tstgettext.c setlocale.c
 tstgettext_CFLAGS = -DINSTALLDIR=\".\"
-tstgettext_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+tstgettext_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 tstngettext_SOURCES = tstngettext.c setlocale.c
 tstngettext_CFLAGS = -DINSTALLDIR=\".\"
-tstngettext_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+tstngettext_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 testlocale_SOURCES = testlocale.c
 gettext_3_prg_SOURCES = gettext-3-prg.c
-gettext_3_prg_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+gettext_3_prg_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 gettext_4_prg_SOURCES = gettext-4-prg.c
-gettext_4_prg_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+gettext_4_prg_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 gettext_5_prg_SOURCES = gettext-5-prg.c
-gettext_5_prg_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+gettext_5_prg_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 gettext_6_prg_SOURCES = gettext-6-prg.c
 gettext_6_prg_CFLAGS = $(gettext_6_prg_CFLAGS_@GLIBC2@)
 gettext_6_prg_CFLAGS_yes = -DUSE_POSIX_THREADS
@@ -224,13 +224,13 @@ gettext_7_prg_CFLAGS_yes = -DUSE_POSIX_T
 gettext_7_prg_LDADD = $(gettext_7_prg_LDADD_@GLIBC2@)
 gettext_7_prg_LDADD_yes = ../gnulib-lib/libgettextlib.la $(LDADD) -lpthread
 cake_SOURCES = plural-1-prg.c setlocale.c
-cake_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+cake_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 fc3_SOURCES = format-c-3-prg.c setlocale.c
-fc3_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+fc3_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 fc4_SOURCES = format-c-4-prg.c setlocale.c
-fc4_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+fc4_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 fc5_SOURCES = format-c-5-prg.c
-fc5_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD)
+fc5_LDADD = ../gnulib-lib/libgettextlib.la $(LDADD) -lsec
 
 # Clean up after Solaris cc.
 clean-local:
