Index: trunk/gio/gdrive.c
===================================================================
--- trunk/gio/gdrive.c	(revision 722)
+++ trunk/gio/gdrive.c	(working copy)
@@ -252,7 +252,7 @@
       return;
     }
   
-  return (* iface->mount) (drive, mount_operation, callback, user_data);
+  (* iface->mount) (drive, mount_operation, callback, user_data);
 }
 
 /**
@@ -311,7 +311,7 @@
       return;
     }
   
-  return (* iface->eject) (drive, callback, user_data);
+  (* iface->eject) (drive, callback, user_data);
 }
 
 /**
Index: trunk/gio/gfile.c
===================================================================
--- trunk/gio/gfile.c	(revision 722)
+++ trunk/gio/gfile.c	(working copy)
@@ -2218,7 +2218,7 @@
   g_return_if_fail (G_IS_FILE_INFO (info));
 
   iface = G_FILE_GET_IFACE (file);
-  return (* iface->set_attributes_async) (file, info, flags, io_priority, cancellable, callback, user_data);
+  (* iface->set_attributes_async) (file, info, flags, io_priority, cancellable, callback, user_data);
   
 }
 
@@ -3378,7 +3378,7 @@
       return;
     }
   
-  return (* iface->mount_for_location) (location, mount_operation, cancellable, callback, user_data);
+  (* iface->mount_for_location) (location, mount_operation, cancellable, callback, user_data);
 
 }
 
Index: trunk/gio/gvolume.c
===================================================================
--- trunk/gio/gvolume.c	(revision 722)
+++ trunk/gio/gvolume.c	(working copy)
@@ -228,7 +228,7 @@
       return;
     }
   
-  return (* iface->unmount) (volume, callback, user_data);
+  (* iface->unmount) (volume, callback, user_data);
 }
 
 /**
@@ -289,7 +289,7 @@
       return;
     }
   
-  return (* iface->eject) (volume, callback, user_data);
+  (* iface->eject) (volume, callback, user_data);
 }
 
 /**
