--- gmediaserver-0.12.0/src/interface.c.orig	2007-07-15 17:07:20.758352141 +0700
+++ gmediaserver-0.12.0/src/interface.c	2007-07-15 17:40:24.506612594 +0700
@@ -23,6 +23,9 @@
 #include <stdio.h>		/* C89 */
 #include <string.h>		/* C89 */
 #include <sys/socket.h>		/* POSIX */
+#ifdef sun
+#include <sys/sockio.h>
+#endif
 #include <sys/ioctl.h>		/* ? */
 #include <sys/param.h>		/* ? */
 #include <net/if.h>		/* ? */
@@ -30,6 +33,9 @@
 #include <arpa/inet.h>		/* ? */
 #include "strnlen.h"		/* Gnulib */
 #include "gmediaserver.h"
+#ifdef sun
+#include "minmax.h"
+#endif
 
 char *
 get_ip_by_spec(char *spec)
--- gmediaserver-0.12.0/src/webclient.c.orig	2007-07-15 17:17:14.983125084 +0700
+++ gmediaserver-0.12.0/src/webclient.c	2007-07-15 17:17:34.303327717 +0700
@@ -371,7 +371,9 @@
     if (res == NULL || *res != '\0')
         return false;
 
+#ifndef sun
     tm.tm_zone = "GMT";
+#endif
     t = mktime(&tm);
     if (t == (time_t) -1)
         return false;
