diff -r -u gmime-2.2.2//gmime/gmime-gpg-context.c ../gmime-2.2.2//gmime/gmime-gpg-context.c
--- gmime-2.2.2//gmime/gmime-gpg-context.c	Mon May  1 21:46:02 2006
+++ ../gmime-2.2.2//gmime/gmime-gpg-context.c	Mon Jun 26 20:34:43 2006
@@ -30,7 +30,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/time.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
@@ -41,6 +40,8 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <ctype.h>
+#include <sys/time.h>
+#include <sys/select.h>
 
 #include "gmime-gpg-context.h"
 #include "gmime-filter-charset.h"
@@ -1104,6 +1105,11 @@
 	gpg->statusleft -= len;                                           \
 } G_STMT_END
 
+struct timeval {
+        time_t          tv_sec;         /* seconds */
+        suseconds_t     tv_usec;        /* and microseconds */
+};
+
 static int
 gpg_ctx_op_step (struct _GpgCtx *gpg, GError **err)
 {
