--- src/gpm-engine.c.org	2007-12-20 16:30:40.305566000 +0800
+++ src/gpm-engine.c	2007-12-20 16:37:22.087976000 +0800
@@ -548,7 +548,7 @@
 			return FALSE;
 		}
 		/* icon before, now none */
-		gpm_debug ("** EMIT: icon-changed: %s", icon);
+		gpm_debug ("** EMIT: icon-changed: none");
 
 		/* we let the icon stick around for a couple of seconds */
 		g_timeout_add (1000*2, (GSourceFunc) gpm_engine_icon_clear_delay, engine);
