--- src/Makefile.am.org	2007-12-17 18:58:48.299754000 +0800
+++ src/Makefile.am	2007-12-17 18:56:36.079825000 +0800
@@ -105,6 +105,8 @@
 	gpm-common.c					\
 	gpm-screensaver.h				\
 	gpm-screensaver.c				\
+	gpm-brightness-lcd.h				\
+	gpm-brightness-lcd.c				\
 	$(NULL)
 
 gnome_power_preferences_LDADD =				\
--- src/gpm-prefs-core.c.org	2007-12-20 16:54:37.514707000 +0800
+++ src/gpm-prefs-core.c	2007-12-20 16:17:57.171458000 +0800
@@ -44,6 +44,7 @@
 #include "gpm-stock-icons.h"
 #include "gpm-screensaver.h"
 #include "gpm-prefs-server.h"
+#include "gpm-brightness-lcd.h"
 
 static void     gpm_prefs_class_init (GpmPrefsClass *klass);
 static void     gpm_prefs_init       (GpmPrefs      *prefs);
@@ -917,6 +918,21 @@
 				 ACTION_SUSPEND,
 				 ACTION_HIBERNATE,
 				 NULL};
+/* Disable AC tab on Solaris Sparc in Phase 1 development */
+#ifdef SOLARIS
+        GtkWidget *notebook;
+        gint page;
+
+        if ((prefs->priv->has_button_lid == FALSE)
+	    && (prefs->priv->hal_cpufreq == FALSE)
+	    && (gpm_brightness_lcd_has_hw () == FALSE)) {
+                notebook = glade_xml_get_widget (prefs->priv->glade_xml, "notebook_preferences");
+                widget = glade_xml_get_widget (prefs->priv->glade_xml, "vbox_ac");
+                page = gtk_notebook_page_num (GTK_NOTEBOOK (notebook), GTK_WIDGET (widget));
+                gtk_notebook_remove_page (GTK_NOTEBOOK (notebook), page);
+                return;
+        }
+#endif
 
 	gpm_prefs_setup_action_combo (prefs, "combobox_ac_lid",
 				      GPM_CONF_BUTTON_LID_AC,
@@ -957,6 +973,22 @@
 		widget = glade_xml_get_widget (prefs->priv->glade_xml, "hbox_ac_cpu");
 		gtk_widget_hide_all (widget);
 	}
+
+/* Disable sleep configuration in Phase 1 development */
+#ifdef SOLARIS
+	widget = glade_xml_get_widget (prefs->priv->glade_xml, "hbox_ac_computer");
+	gtk_widget_hide_all (widget);
+
+	if (gpm_brightness_lcd_has_hw () == FALSE) {
+		widget = glade_xml_get_widget (prefs->priv->glade_xml, "vbox_ac_display");
+		gtk_widget_hide_all (widget);
+	} else {
+		widget = glade_xml_get_widget (prefs->priv->glade_xml, "hbox_ac_display");
+		gtk_widget_hide_all (widget);
+		widget = glade_xml_get_widget (prefs->priv->glade_xml, "checkbutton_ac_display_dim");
+		gtk_widget_hide_all (widget);
+	}
+#endif
 }
 
 static void
@@ -1037,6 +1069,22 @@
 		widget = glade_xml_get_widget (prefs->priv->glade_xml, "hbox_battery_cpu");
 		gtk_widget_hide_all (widget);
 	}
+
+/* Disable sleep configuration in Phase 1 development */
+#ifdef SOLARIS
+        widget = glade_xml_get_widget (prefs->priv->glade_xml, "hbox_battery_computer");
+        gtk_widget_hide_all (widget);
+
+        if (gpm_brightness_lcd_has_hw () == FALSE) {
+                widget = glade_xml_get_widget (prefs->priv->glade_xml, "vbox_battery_display");
+                gtk_widget_hide_all (widget);
+        } else {
+                widget = glade_xml_get_widget (prefs->priv->glade_xml, "hbox_battery_display");
+                gtk_widget_hide_all (widget);
+                widget = glade_xml_get_widget (prefs->priv->glade_xml, "checkbutton_battery_display_dim");
+                gtk_widget_hide_all (widget);
+        }
+#endif
 }
 
 static void
