diff -Nur gnome-device-manager-0.2-org/configure.ac gnome-device-manager-0.2/configure.ac
--- gnome-device-manager-0.2-org/configure.ac	2007-12-18 14:55:30.644301000 +0800
+++ gnome-device-manager-0.2/configure.ac	2007-12-18 14:58:21.675613000 +0800
@@ -26,7 +26,7 @@
 # Check for required packages
 # ***************************
 
-LIBHAL_REQUIRED=0.5.10
+LIBHAL_REQUIRED=0.5.8
 LIBGTK_REQUIRED=2.6.0
 LIBGNOME_REQUIRED=2.14.0
 LIBGNOMEUI_REQUIRED=2.14.0
diff -Nur gnome-device-manager-0.2-org/gnome-device-manager/gdm-pool.c gnome-device-manager-0.2/gnome-device-manager/gdm-pool.c
--- gnome-device-manager-0.2-org/gnome-device-manager/gdm-pool.c	2007-12-18 14:55:30.643917000 +0800
+++ gnome-device-manager-0.2/gnome-device-manager/gdm-pool.c	2007-12-18 14:44:54.974983000 +0800
@@ -210,7 +210,7 @@
         pool->priv->hal_ctx = hal_ctx;
         pool->priv->devices = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_object_unref);
 
-        libhal_device_property_watch_all (hal_ctx, NULL);
+        libhal_device_property_watch_all (hal_ctx, &error);
         libhal_ctx_set_device_added (hal_ctx, _hal_device_added);
         libhal_ctx_set_device_removed (hal_ctx, _hal_device_removed);
         libhal_ctx_set_device_property_modified (hal_ctx, _hal_property_modified);
