--- gnome-screensaver-2.20.0/configure.ac.orig	2008-01-11 12:13:10.327621309 +0000
+++ gnome-screensaver-2.20.0/configure.ac	2008-01-11 12:16:44.936903969 +0000
@@ -58,6 +58,7 @@
         gtk+-2.0 >= $GTK_REQUIRED_VERSION
         dbus-glib-1 >= $DBUS_REQUIRED_VERSION
         gconf-2.0 >= $GCONF_REQUIRED_VERSION
+        gthread-2.0
         libgnome-menu >= $LIBGNOME_MENU_REQUIRED_VERSION)
 AC_SUBST(GNOME_SCREENSAVER_CFLAGS)
 AC_SUBST(GNOME_SCREENSAVER_LIBS)
