--- gnome-commander-1.2.4/src/libgviewer/viewer-widget.c.orig	2007-06-11 14:18:48.051936526 +0100
+++ gnome-commander-1.2.4/src/libgviewer/viewer-widget.c	2007-06-11 14:18:59.250584656 +0100
@@ -608,7 +608,7 @@
     g_return_if_fail (IS_GVIEWER (obj));
     g_return_if_fail (obj->priv->imgr);
 
-    return image_render_operation(obj->priv->imgr, op);
+    image_render_operation(obj->priv->imgr, op);
 }
 
 
