--- gnomescan-0.3/flegita/flegitaactionwidget.c.orig	2006-11-08 17:13:30.264984000 -0500
+++ gnomescan-0.3/flegita/flegitaactionwidget.c	2006-11-08 17:13:38.045080000 -0500
@@ -151,6 +151,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
--- gnomescan-0.3/libgnomescan/gnomescancontext.c.orig	2006-11-08 17:06:41.545221000 -0500
+++ gnomescan-0.3/libgnomescan/gnomescancontext.c	2006-11-08 17:06:32.234729000 -0500
@@ -947,7 +947,7 @@
       result = g_boxed_copy (GNOME_TYPE_SCAN_RESULT, result);
       image = result->image;
       result->image = gdk_pixbuf_rotate_simple (image, priv->rotation);
-      g_debug ("%s: Unref image %p", __FUNCTION__, image);
+      g_debug ("%s: Unref image %p", G_GNUC_FUNCTION, image);
       g_object_unref (image);
     }
     g_signal_emit_by_name (context, "image-received", result);
--- gnomescan-0.3/libgnomescan/gnomescanner.c.orig	2006-11-08 17:07:47.917684000 -0500
+++ gnomescan-0.3/libgnomescan/gnomescanner.c	2006-11-08 17:08:32.867622000 -0500
@@ -52,6 +52,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
--- gnomescan-0.3/libgnomescanui/gnomescanpreview.c.orig	2006-11-08 17:10:13.571447000 -0500
+++ gnomescan-0.3/libgnomescanui/gnomescanpreview.c	2006-11-08 17:10:22.389769000 -0500
@@ -41,6 +41,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
--- gnomescan-0.3/libgnomescanui/gnomescanadvancedpreview.c.orig	2006-11-08 17:08:59.889778000 -0500
+++ gnomescan-0.3/libgnomescanui/gnomescanadvancedpreview.c	2006-11-08 17:09:17.432529000 -0500
@@ -85,6 +85,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
--- gnomescan-0.3/libgnomescanui/gnomescanoptionwidget.c.orig	2006-11-08 17:09:42.469238000 -0500
+++ gnomescan-0.3/libgnomescanui/gnomescanoptionwidget.c	2006-11-08 17:09:51.727540000 -0500
@@ -66,6 +66,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
--- gnomescan-0.3/libgnomescanui/gnomescanpreviewarea.c.orig	2006-11-08 17:10:42.599585000 -0500
+++ gnomescan-0.3/libgnomescanui/gnomescanpreviewarea.c	2006-11-08 17:10:52.015489000 -0500
@@ -63,6 +63,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
--- gnomescan-0.3/libgnomescanui/gnomescanresolutionselector.c.orig	2006-11-08 17:11:14.020401000 -0500
+++ gnomescan-0.3/libgnomescanui/gnomescanresolutionselector.c	2006-11-08 17:11:21.034307000 -0500
@@ -79,6 +79,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
--- gnomescan-0.3/libgnomescanui/gnomescansourceselector.c.orig	2006-11-08 17:12:18.347172000 -0500
+++ gnomescan-0.3/libgnomescanui/gnomescansourceselector.c	2006-11-08 17:12:25.573334000 -0500
@@ -76,6 +76,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
