--- src/gpm-backlight.c.org	2007-11-17 11:05:23.741370000 +0800
+++ src/gpm-backlight.c	2007-11-17 14:57:54.114228000 +0800
@@ -32,6 +32,7 @@
 #include <string.h>
 #include <sys/time.h>
 #include <sys/types.h>
+#include <math.h>
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif /* HAVE_UNISTD_H */
@@ -422,7 +423,7 @@
 	gpm_debug ("4. ambient scale %f, brightness %f", scale, brightness);
 
 	/* convert to percentage */
-	value = brightness * 100.0f;
+	value = roundf (brightness * 100.0f);
 
 	/* only do stuff if the brightness is different */
 	gpm_brightness_lcd_get (backlight->priv->brightness, &old_value);
