--- gnome-commander-1.2.4/tests/gviewer/search_dlg_test.c.orig	2007-06-12 12:19:25.657304404 +0100
+++ gnome-commander-1.2.4/tests/gviewer/search_dlg_test.c	2007-06-12 12:19:39.735544131 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <err.h>
 
 #include <glib.h>
 #include <gtk/gtk.h>
--- gnome-commander-1.2.4/tests/gviewer/imagerenderer.c.orig	2007-06-12 12:17:31.884860379 +0100
+++ gnome-commander-1.2.4/tests/gviewer/imagerenderer.c	2007-06-12 12:17:47.347215289 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <err.h>
 
 #include <glib.h>
 #include <gtk/gtk.h>
@@ -61,7 +60,7 @@
             best_fit = FALSE;
             scale_factor = atof(optarg);
             if (scale_factor<0.1 || scale_factor>3.0) {
-                warnx("Invalid scale factor \"%f\".\n", scale_factor);
+                g_warning("Invalid scale factor \"%f\".\n", scale_factor);
                 usage();
             }
             break;
@@ -73,7 +72,7 @@
     }
 
     if (optind == argc) {
-        warnx("Need file name to work with...\n");
+        g_warning("Need file name to work with...\n");
         usage();
     }
     filename = g_strdup(argv[optind++]);
--- gnome-commander-1.2.4/tests/gviewer/viewerwindow.c.orig	2007-06-12 12:19:02.270560442 +0100
+++ gnome-commander-1.2.4/tests/gviewer/viewerwindow.c	2007-06-12 12:19:11.996222845 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <err.h>
 
 #include <glib.h>
 #include <gtk/gtk.h>
--- gnome-commander-1.2.4/tests/gviewer/dualrenderer.c.orig	2007-06-12 12:17:55.855405530 +0100
+++ gnome-commander-1.2.4/tests/gviewer/dualrenderer.c	2007-06-12 12:18:05.021337366 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <err.h>
 
 #include <glib.h>
 #include <gtk/gtk.h>
--- gnome-commander-1.2.4/tests/gviewer/textrenderer.c.orig	2007-06-12 12:08:35.611104637 +0100
+++ gnome-commander-1.2.4/tests/gviewer/textrenderer.c	2007-06-12 12:17:20.864556060 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <err.h>
 
 #include <glib.h>
 #include <gtk/gtk.h>
@@ -86,7 +85,7 @@
                 dispmode = TR_DISP_MODE_HEXDUMP;
             else
             {
-                warnx("Invalid display mode \"%s\".\n", optarg);
+                g_warning("Invalid display mode \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -95,7 +94,7 @@
             tab_size = atoi(optarg);
             if (tab_size <=0)
             {
-                warnx("Invalid tab size \"%s\".\n", optarg);
+                g_warning("Invalid tab size \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -103,7 +102,7 @@
         case 'f':
             fixed_limit = atoi(optarg);
             if (fixed_limit<=0) {
-                warnx("Invalid fixed limit \"%s\".\n", optarg);
+                g_warning("Invalid fixed limit \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -116,7 +115,7 @@
 
     if (optind == argc)
     {
-        warnx("Need file name to work with...\n");
+        g_warning("Need file name to work with...\n");
         usage();
     }
     filename = g_strdup(argv[optind++]);
--- gnome-commander-1.2.4/tests/gviewer/datapresentation.c.orig	2007-06-12 11:45:15.019894887 +0100
+++ gnome-commander-1.2.4/tests/gviewer/datapresentation.c	2007-06-12 12:16:51.677632193 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <err.h>
 
 #include <libgviewer/libgviewer.h>
 #include <libgviewer/gvtypes.h>
@@ -98,7 +97,7 @@
             else if (g_ascii_strcasecmp(optarg,"FIXED")==0)
                 presentation = PRSNT_BIN_FIXED;
             else {
-                warnx("Invalid presentation mode \"%s\".\n", optarg);
+                g_warning("Invalid presentation mode \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -106,7 +105,7 @@
         case 'w':
             wrap_limit = atoi(optarg);
             if (wrap_limit<=0) {
-                warnx("Invalid wrap limit \"%s\".\n", optarg);
+                g_warning("Invalid wrap limit \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -114,7 +113,7 @@
         case 't':
             tab_size = atoi(optarg);
             if (tab_size <=0) {
-                warnx("Invalid tab size \"%s\".\n", optarg);
+                g_warning("Invalid tab size \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -122,7 +121,7 @@
         case 'f':
             fixed_limit = atoi(optarg);
             if (fixed_limit<=0) {
-                warnx("Invalid fixed limit \"%s\".\n", optarg);
+                g_warning("Invalid fixed limit \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -130,7 +129,7 @@
     }
 
     if (hexdump && (g_ascii_strcasecmp(encoding, "UTF8")==0)) {
-        warnx("Can't use HexDump mode with UTF8 encoding. (Hexdump requires each character to be one byte exactly)\n");
+        g_warning("Can't use HexDump mode with UTF8 encoding. (Hexdump requires each character to be one byte exactly)\n");
         exit(0);
     }
 
@@ -140,7 +139,7 @@
     }
 
     if (optind == argc) {
-        warnx("Need file name to work with...\n");
+        g_warning("Need file name to work with...\n");
         usage();
     }
     filename = g_strdup(argv[optind++]);
--- gnome-commander-1.2.4/tests/gviewer/viewerwidget.c.orig	2007-06-12 12:18:16.346462976 +0100
+++ gnome-commander-1.2.4/tests/gviewer/viewerwidget.c	2007-06-12 12:18:47.273649967 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <err.h>
 
 #include <glib.h>
 #include <gtk/gtk.h>
@@ -78,7 +77,7 @@
             best_fit = FALSE;
             scale_factor = atof(optarg);
             if (scale_factor<0.1 || scale_factor>3.0) {
-                warnx("Invalid scale factor \"%f\".\n", scale_factor);
+                g_warning("Invalid scale factor \"%f\".\n", scale_factor);
                 usage();
             }
             break;
@@ -105,7 +104,7 @@
             else if (g_ascii_strcasecmp(optarg,"auto")==0)
                 auto_detect_display_mode = TRUE;
             else {
-                warnx("Invalid display mode \"%s\".\n", optarg);
+                g_warning("Invalid display mode \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -113,7 +112,7 @@
         case 't':
             tab_size = atoi(optarg);
             if (tab_size <=0) {
-                warnx("Invalid tab size \"%s\".\n", optarg);
+                g_warning("Invalid tab size \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -121,7 +120,7 @@
         case 'f':
             fixed_limit = atoi(optarg);
             if (fixed_limit<=0) {
-                warnx("Invalid fixed limit \"%s\".\n", optarg);
+                g_warning("Invalid fixed limit \"%s\".\n", optarg);
                 usage();
             }
             break;
@@ -133,7 +132,7 @@
     }
 
     if (optind == argc) {
-        warnx("Need file name to work with...\n");
+        g_warning("Need file name to work with...\n");
         usage();
     }
     filename = g_strdup(argv[optind++]);
