diff -Nur gnome-power-manager-2.21.1-org/help/Makefile.am gnome-power-manager-2.21.1/help/Makefile.am
--- gnome-power-manager-2.21.1-org/help/Makefile.am	2007-12-17 17:12:56.218240000 +0800
+++ gnome-power-manager-2.21.1/help/Makefile.am	2007-12-17 17:08:33.198369000 +0800
@@ -19,4 +19,4 @@
 	figures/gpm-unplugged.png \
 	figures/gs-prefs.png
 
-DOC_LINGUAS = ca es fr hu it oc pa ru sv
+DOC_LINGUAS = ca es fr it oc pa ru sv
diff -Nur gnome-power-manager-2.21.1-org/src/gpm-array-float.c gnome-power-manager-2.21.1/src/gpm-array-float.c
--- gnome-power-manager-2.21.1-org/src/gpm-array-float.c	2007-12-17 17:12:56.275650000 +0800
+++ gnome-power-manager-2.21.1/src/gpm-array-float.c	2007-12-17 17:02:07.642824000 +0800
@@ -75,7 +75,7 @@
  *
  * @array: input array
  **/
-inline gfloat
+gfloat
 gpm_array_float_get (GArray *array, guint i)
 {
 	if (i >= array->len) {
@@ -89,7 +89,7 @@
  *
  * @array: input array
  **/
-inline void
+void
 gpm_array_float_set (GArray *array, guint i, gfloat value)
 {
 	g_array_index (array, gfloat, i) = value;
diff -Nur gnome-power-manager-2.21.1-org/src/gpm-array-float.h gnome-power-manager-2.21.1/src/gpm-array-float.h
--- gnome-power-manager-2.21.1-org/src/gpm-array-float.h	2007-12-17 17:12:56.277090000 +0800
+++ gnome-power-manager-2.21.1/src/gpm-array-float.h	2007-12-17 17:02:07.643637000 +0800
@@ -49,9 +49,9 @@
 gboolean	 gpm_array_float_print			(GArray		*array);
 GArray		*gpm_array_float_convolve		(GArray		*data,
 							 GArray		*kernel);
-inline gfloat	 gpm_array_float_get			(GArray		*array,
+gfloat		 gpm_array_float_get			(GArray		*array,
 							 guint		 i);
-inline void	 gpm_array_float_set			(GArray		*array,
+void		 gpm_array_float_set			(GArray		*array,
 							 guint		 i,
 							 gfloat		 value);
 
diff -Nur gnome-power-manager-2.21.1-org/src/gpm-debug.h gnome-power-manager-2.21.1/src/gpm-debug.h
--- gnome-power-manager-2.21.1-org/src/gpm-debug.h	2007-12-17 17:12:56.284090000 +0800
+++ gnome-power-manager-2.21.1/src/gpm-debug.h	2007-12-17 17:02:07.650493000 +0800
@@ -28,7 +28,7 @@
 
 G_BEGIN_DECLS
 
-#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199409L
 #define gpm_debug(...) gpm_debug_real (__func__, __FILE__, __LINE__, __VA_ARGS__)
 #define gpm_warning(...) gpm_warning_real (__func__, __FILE__, __LINE__, __VA_ARGS__)
 #define gpm_error(...) gpm_error_real (__func__, __FILE__, __LINE__, __VA_ARGS__)
--- gnome-power-manager-2.21.1/configure.in	2007-11-18 23:36:20.000000000 +0800
+++ gnome-power-manager-2.21.1-new/configure.in	2007-12-17 18:15:26.399221000 +0800
@@ -136,6 +136,18 @@
 AC_SUBST(SBINDIR, $sbindir)
 
 dnl ---------------------------------------------------------------------------
+dnl - Is it compiling on Solais?
+dnl ---------------------------------------------------------------------------
+AC_MSG_CHECKING(if we compile on Solaris)
+case "$host" in
+       *-solaris*)
+         AC_MSG_RESULT(yes)
+         AC_DEFINE(SOLARIS, 1, [Define if we compile on Solaris])
+       ;;
+esac
+
+
+dnl ---------------------------------------------------------------------------
 dnl - Where should we put documentation ?
 dnl ---------------------------------------------------------------------------
 AC_ARG_WITH(doc-dir,
