--- applets/brightness/GNOME_BrightnessApplet.server.in.in.org	2007-12-17 19:21:42.319597000 +0800
+++ applets/brightness/GNOME_BrightnessApplet.server.in.in	2007-12-17 19:22:22.897177000 +0800
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 
 <oaf_info>
-  <oaf_server iid="OAFIID:GNOME_BrightnessApplet_Factory" type="exe" location="@BINDIR@/gnome-brightness-applet">
+  <oaf_server iid="OAFIID:GNOME_BrightnessApplet_Factory" type="exe" location="@LIBDIR@/gnome-brightness-applet">
     <oaf_attribute name="repo_ids" type="stringv">
       <item value="IDL:Bonobo/GenericFactory:1.0"/>
       <item value="IDL:Bonobo/Unknown:1.0"/>
