diff -Nur gnomescan-0.4.0.4/flegita/flegitaactionwidget.c gnomescan-0.4.0.4-new/flegita/flegitaactionwidget.c
--- gnomescan-0.4.0.4/flegita/flegitaactionwidget.c	2006-12-19 01:53:20.000000000 +0800
+++ gnomescan-0.4.0.4-new/flegita/flegitaactionwidget.c	2007-02-28 18:12:47.129518000 +0800
@@ -151,6 +151,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
diff -Nur gnomescan-0.4.0.4/libgnomescan/gnomescancontext.c gnomescan-0.4.0.4-new/libgnomescan/gnomescancontext.c
--- gnomescan-0.4.0.4/libgnomescan/gnomescancontext.c	2006-12-19 01:51:51.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescan/gnomescancontext.c	2007-02-28 18:12:47.131625000 +0800
@@ -1056,7 +1056,7 @@
       result = g_boxed_copy (GNOME_TYPE_SCAN_RESULT, result);
       image = result->image;
       result->image = gdk_pixbuf_rotate_simple (image, priv->rotation);
-      g_debug ("%s: Unref image %p", __FUNCTION__, image);
+      g_debug ("%s: Unref image %p", G_GNUC_FUNCTION, image);
       g_object_unref (image);
     }
     g_signal_emit_by_name (context, "image-received", result);
diff -Nur gnomescan-0.4.0.4/libgnomescan/gnomescanner.c gnomescan-0.4.0.4-new/libgnomescan/gnomescanner.c
--- gnomescan-0.4.0.4/libgnomescan/gnomescanner.c	2006-12-19 02:41:50.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescan/gnomescanner.c	2007-02-28 18:12:47.133290000 +0800
@@ -55,6 +55,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
diff -Nur gnomescan-0.4.0.4/libgnomescanui/gnomescanareaselector.c gnomescan-0.4.0.4-new/libgnomescanui/gnomescanareaselector.c
--- gnomescan-0.4.0.4/libgnomescanui/gnomescanareaselector.c	2006-12-19 00:30:08.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescanui/gnomescanareaselector.c	2007-02-28 18:06:37.538795000 +0800
@@ -108,6 +108,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
diff -Nur gnomescan-0.4.0.4/libgnomescanui/gnomescanliststore.c gnomescan-0.4.0.4-new/libgnomescanui/gnomescanliststore.c
--- gnomescan-0.4.0.4/libgnomescanui/gnomescanliststore.c	2006-12-19 20:27:01.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescanui/gnomescanliststore.c	2007-02-28 18:08:05.025994000 +0800
@@ -92,6 +92,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
diff -Nur gnomescan-0.4.0.4/libgnomescanui/gnomescanoptionwidget.c gnomescan-0.4.0.4-new/libgnomescanui/gnomescanoptionwidget.c
--- gnomescan-0.4.0.4/libgnomescanui/gnomescanoptionwidget.c	2006-12-18 00:59:06.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescanui/gnomescanoptionwidget.c	2007-02-28 18:12:47.135156000 +0800
@@ -66,6 +66,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
diff -Nur gnomescan-0.4.0.4/libgnomescanui/gnomescanpreviewarea.c gnomescan-0.4.0.4-new/libgnomescanui/gnomescanpreviewarea.c
--- gnomescan-0.4.0.4/libgnomescanui/gnomescanpreviewarea.c	2006-12-19 21:07:39.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescanui/gnomescanpreviewarea.c	2007-02-28 18:12:47.137911000 +0800
@@ -66,6 +66,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
diff -Nur gnomescan-0.4.0.4/libgnomescanui/gnomescanresolutionselector.c gnomescan-0.4.0.4-new/libgnomescanui/gnomescanresolutionselector.c
--- gnomescan-0.4.0.4/libgnomescanui/gnomescanresolutionselector.c	2006-12-19 19:49:58.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescanui/gnomescanresolutionselector.c	2007-02-28 18:12:47.139560000 +0800
@@ -89,6 +89,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
diff -Nur gnomescan-0.4.0.4/libgnomescanui/gnomescansourceselector.c gnomescan-0.4.0.4-new/libgnomescanui/gnomescansourceselector.c
--- gnomescan-0.4.0.4/libgnomescanui/gnomescansourceselector.c	2006-12-18 22:24:36.000000000 +0800
+++ gnomescan-0.4.0.4-new/libgnomescanui/gnomescansourceselector.c	2007-02-28 18:12:47.140686000 +0800
@@ -80,6 +80,7 @@
 };
 
 enum {
+  dummy,
   N_SIGNALS
 };
 
