--- tools/gnome-power-bugreport.sh.org	2007-12-18 13:55:22.379051000 +0800
+++ tools/gnome-power-bugreport.sh	2007-12-18 13:55:31.036043000 +0800
@@ -36,7 +36,7 @@
 }
 
 echo -n "Distro version:       "
-cat /etc/*-release | uniq
+cat /etc/release | uniq
 
 echo -n "Kernel version:       "
 uname -r
@@ -70,8 +70,8 @@
 lshal | grep "battery\."
 
 echo "GNOME Power Manager Process Information:"
-ps aux --forest | grep gnome-power | grep -v grep
+ptree -a `pgrep power` | grep -v grep
 
 echo "HAL Process Information:"
-ps aux --forest | grep hald | grep -v grep
+ptree -a `pgrep hald `| grep -v grep
 
