--- src/gpm-prefs-core.c.org	2007-12-19 09:59:39.777337000 +0800
+++ src/gpm-prefs-core.c	2007-12-19 09:57:19.955853000 +0800
@@ -1171,7 +1171,11 @@
 	prefs->priv->has_lcd = ((caps & GPM_PREFS_SERVER_BACKLIGHT) > 0);
 	prefs->priv->has_ambient = ((caps & GPM_PREFS_SERVER_AMBIENT) > 0);
 	prefs->priv->has_button_lid = ((caps & GPM_PREFS_SERVER_LID) > 0);
+#ifdef SOLARIS
+	prefs->priv->has_button_suspend = FALSE;
+#else
 	prefs->priv->has_button_suspend = TRUE;
+#endif
 	prefs->priv->can_suspend = gpm_dbus_method_bool ("CanSuspend");
 	prefs->priv->can_hibernate = gpm_dbus_method_bool ("CanHibernate");
 	gpm_debug ("caps=%i", caps);
