--- gnucash-2.2.5/configure.in.orig	2008-06-27 16:55:40.868976000 +0800
+++ gnucash-2.2.5/configure.in	2008-06-27 16:56:17.042870000 +0800
@@ -411,6 +411,7 @@
 AC_MSG_CHECKING(if guile long_long is at least as big as gint64)
 GNC_OLDCFLAGS="$CFLAGS"
 CFLAGS="${GNOME_CFLAGS} ${GUILE_INCS} ${CFLAGS} ${GLIB_CFLAGS}"
+LDFLAGS="${LDFLAGS} ${GLIB_LIBS}"
 AC_TRY_RUN([
   #include <glib.h>
   #include <libguile/__scm.h>
