Index: trunk/src/gnome-utils/dialog-transfer.c
===================================================================
--- trunk/src/gnome-utils/dialog-transfer.c	(revision 17246)
+++ trunk/src/gnome-utils/dialog-transfer.c	(working copy)
@@ -257,7 +257,10 @@
 
   if (!gnc_is_euro_currency (from_currency) ||
       !gnc_is_euro_currency (to_currency))
-    return gnc_xfer_dialog_update_price (xferData);
+  {
+    gnc_xfer_dialog_update_price (xferData);
+    return;
+  }
 
   from_rate = gnc_euro_currency_get_rate (from_currency);
   to_rate = gnc_euro_currency_get_rate (to_currency);
Index: trunk/src/gnome-utils/gnc-dense-cal.c
===================================================================
--- trunk/src/gnome-utils/gnc-dense-cal.c	(revision 17246)
+++ trunk/src/gnome-utils/gnc-dense-cal.c	(working copy)
@@ -539,7 +539,7 @@
 void
 gnc_dense_cal_set_year(GncDenseCal *dcal, guint year)
 {
-    return _gnc_dense_cal_set_year(dcal, year, TRUE);
+    _gnc_dense_cal_set_year(dcal, year, TRUE);
 }
 
 static void
Index: trunk/src/gnome-utils/gnc-dense-cal-model.c
===================================================================
--- trunk/src/gnome-utils/gnc-dense-cal-model.c	(revision 17246)
+++ trunk/src/gnome-utils/gnc-dense-cal-model.c	(working copy)
@@ -138,5 +138,5 @@
 void
 gnc_dense_cal_model_get_instance(GncDenseCalModel *model, guint tag, gint instance_index, GDate *date)
 {
-    return (*GNC_DENSE_CAL_MODEL_GET_INTERFACE(model)->get_instance)(model, tag, instance_index, date);
+    (*GNC_DENSE_CAL_MODEL_GET_INTERFACE(model)->get_instance)(model, tag, instance_index, date);
 }
Index: trunk/src/gnome-utils/gnc-plugin-page.c
===================================================================
--- trunk/src/gnome-utils/gnc-plugin-page.c	(revision 17246)
+++ trunk/src/gnome-utils/gnc-plugin-page.c	(working copy)
@@ -177,7 +177,7 @@
 	g_return_if_fail (klass != NULL);
 	g_return_if_fail (klass->destroy_widget != NULL);
 
-	return klass->destroy_widget (plugin_page);
+	klass->destroy_widget (plugin_page);
 }
 
 
