--- gnucash-2.2.6/configure.in.orig	2008-08-18 15:06:44.024407000 +0800
+++ gnucash-2.2.6/configure.in	2008-08-18 15:07:14.163758000 +0800
@@ -1124,9 +1124,10 @@
     goffice=0
     goffice_with_cairo=0
     have_goffice_0_5=0
-    PKG_CHECK_MODULES(GOFFICE, libgoffice-0.6 >= 0.6.0, [goffice=1], [
-      PKG_CHECK_MODULES(GOFFICE, libgoffice-0.5 >= 0.5.1, [goffice=1], [goffice=0])
-    ])
+    PKG_CHECK_MODULES(GOFFICE, libgoffice-0.8 >= 0.7.0, [goffice=1], [
+      PKG_CHECK_MODULES(GOFFICE, libgoffice-0.6 >= 0.6.0, [goffice=1], [
+        PKG_CHECK_MODULES(GOFFICE, libgoffice-0.5 >= 0.5.1, [goffice=1], [goffice=0])
+    ])])
     if test x$goffice = x1
     then
       AC_DEFINE(HAVE_GOFFICE_0_5,1,[System has goffice 0.5.1 or better])
