Index: trunk/src/engine/gnc-pricedb.c
===================================================================
--- trunk/src/engine/gnc-pricedb.c	(revision 17246)
+++ trunk/src/engine/gnc-pricedb.c	(working copy)
@@ -2439,32 +2439,32 @@
 
 static QofObject price_object_def =
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_PRICE,
-  type_label:        "Price",
-  create:            price_create,
-  book_begin:        NULL,
-  book_end:          NULL,
-  is_dirty:          qof_collection_is_dirty,
-  mark_clean:        qof_collection_mark_clean,
-  foreach:           price_foreach,
-  printable:         price_printable,
-  version_cmp:       NULL,
+  QOF_OBJECT_VERSION,
+  GNC_ID_PRICE,
+  "Price",
+  price_create,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  price_foreach,
+  price_printable,
+  NULL,
 };
 
 static QofObject pricedb_object_def =
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_PRICEDB,
-  type_label:        "PriceDB",
-  create:            NULL,
-  book_begin:        pricedb_book_begin,
-  book_end:          pricedb_book_end,
-  is_dirty:          qof_collection_is_dirty,
-  mark_clean:        qof_collection_mark_clean,
-  foreach:           NULL,
-  printable:         NULL,
-  version_cmp:       NULL,
+  QOF_OBJECT_VERSION,
+  GNC_ID_PRICEDB,
+  "PriceDB",
+  NULL,
+  pricedb_book_begin,
+  pricedb_book_end,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  NULL,
+  NULL,
+  NULL,
 };
 
 gboolean
Index: trunk/src/engine/gnc-budget.c
===================================================================
--- trunk/src/engine/gnc-budget.c	(revision 17246)
+++ trunk/src/engine/gnc-budget.c	(working copy)
@@ -425,17 +425,17 @@
 /* Define the QofObject. */
 static QofObject budget_object_def =
 {
-    interface_version: QOF_OBJECT_VERSION,
-    e_type:            GNC_ID_BUDGET,
-    type_label:        "Budget",
-    create:            (gpointer)gnc_budget_new,
-    book_begin:        NULL,
-    book_end:          NULL,
-    is_dirty:          qof_collection_is_dirty,
-    mark_clean:        qof_collection_mark_clean,
-    foreach:           qof_collection_foreach,
-    printable:         (const char* (*)(gpointer)) gnc_budget_get_name,
-    version_cmp:       (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+    QOF_OBJECT_VERSION,
+    GNC_ID_BUDGET,
+    "Budget",
+    (gpointer)gnc_budget_new,
+    NULL,
+    NULL,
+    qof_collection_is_dirty,
+    qof_collection_mark_clean,
+    qof_collection_foreach,
+    (const char* (*)(gpointer)) gnc_budget_get_name,
+    (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 
Index: trunk/src/engine/Account.c
===================================================================
--- trunk/src/engine/Account.c	(revision 17246)
+++ trunk/src/engine/Account.c	(working copy)
@@ -4473,17 +4473,17 @@
 /* QofObject function implementation and registration */
 
 static QofObject account_object_def = {
-  interface_version:     QOF_OBJECT_VERSION,
-  e_type:                GNC_ID_ACCOUNT,
-  type_label:            "Account",
-  create:                (gpointer)xaccMallocAccount,
-  book_begin:            NULL,
-  book_end:              NULL,
-  is_dirty:              qof_collection_is_dirty,
-  mark_clean:            qof_collection_mark_clean,
-  foreach:               qof_collection_foreach,
-  printable:             (const char* (*)(gpointer)) xaccAccountGetName,
-  version_cmp:           (int (*)(gpointer,gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  GNC_ID_ACCOUNT,
+  "Account",
+  (gpointer)xaccMallocAccount,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  (const char* (*)(gpointer)) xaccAccountGetName,
+  (int (*)(gpointer,gpointer)) qof_instance_version_cmp,
 };
 
 gboolean xaccAccountRegister (void)
Index: trunk/src/engine/SX-book.c
===================================================================
--- trunk/src/engine/SX-book.c	(revision 17246)
+++ trunk/src/engine/SX-book.c	(working copy)
@@ -162,15 +162,17 @@
 
 static QofObject sxtg_object_def = 
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_SXTG,
-  type_label:        "Scheduled Transaction Templates",
-  book_begin:        sxtg_book_begin,
-  book_end:          sxtg_book_end,
-  is_dirty:          sxtg_is_dirty,
-  mark_clean:        sxtg_mark_clean,
-  foreach:           NULL,
-  printable:         NULL,
+  QOF_OBJECT_VERSION,
+  GNC_ID_SXTG,
+  "Scheduled Transaction Templates",
+  NULL,
+  sxtg_book_begin,
+  sxtg_book_end,
+  sxtg_is_dirty,
+  sxtg_mark_clean,
+  NULL,
+  NULL,
+  NULL,
 };
 
 /* ====================================================================== */
@@ -294,32 +296,32 @@
 
 static QofObject sxes_object_def =
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_SXES,
-  type_label:        "Scheduled Transactions List",
-  create:            NULL,
-  book_begin:        book_sxes_setup,
-  book_end:          NULL,
-  is_dirty:          book_sxlist_notsaved,
-  mark_clean:        book_sxns_mark_saved,
-  foreach:           NULL,
-  printable:         NULL,
-  version_cmp:       NULL
+  QOF_OBJECT_VERSION,
+  GNC_ID_SXES,
+  "Scheduled Transactions List",
+  NULL,
+  book_sxes_setup,
+  NULL,
+  book_sxlist_notsaved,
+  book_sxns_mark_saved,
+  NULL,
+  NULL,
+  NULL
 };
   
 static QofObject sxtt_object_def = 
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_SXTT,
-  type_label:        "Scheduled Transaction Templates",
-  create:            NULL,
-  book_begin:        NULL,
-  book_end:          NULL,
-  is_dirty:          NULL,
-  mark_clean:        NULL,
-  foreach:           NULL,
-  printable:         NULL,
-  version_cmp:       NULL,
+  QOF_OBJECT_VERSION,
+  GNC_ID_SXTT,
+  "Scheduled Transaction Templates",
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
 };
 
 gboolean 
Index: trunk/src/engine/gnc-lot.c
===================================================================
--- trunk/src/engine/gnc-lot.c	(revision 17246)
+++ trunk/src/engine/gnc-lot.c	(working copy)
@@ -420,17 +420,17 @@
 
 static QofObject gncLotDesc =
 {
-    interface_version:  QOF_OBJECT_VERSION,
-    e_type:             GNC_ID_LOT,
-    type_label:         "Lot",
-    create:             (gpointer)gnc_lot_new,
-    book_begin:         NULL,
-    book_end:           NULL,
-    is_dirty:           qof_collection_is_dirty,
-    mark_clean:         qof_collection_mark_clean,
-    foreach:            qof_collection_foreach,
-    printable:          NULL,
-    version_cmp:        (int (*)(gpointer,gpointer))qof_instance_version_cmp,
+    QOF_OBJECT_VERSION,
+    GNC_ID_LOT,
+    "Lot",
+    (gpointer)gnc_lot_new,
+    NULL,
+    NULL,
+    qof_collection_is_dirty,
+    qof_collection_mark_clean,
+    qof_collection_foreach,
+    NULL,
+    (int (*)(gpointer,gpointer))qof_instance_version_cmp,
 };
 
 
Index: trunk/src/engine/Transaction.c
===================================================================
--- trunk/src/engine/Transaction.c	(revision 17246)
+++ trunk/src/engine/Transaction.c	(working copy)
@@ -1911,17 +1911,17 @@
 
 /* Hook into the QofObject registry */
 static QofObject trans_object_def = {
-  interface_version:   QOF_OBJECT_VERSION,
-  e_type:              GNC_ID_TRANS,
-  type_label:          "Transaction",
-  create:              (gpointer)xaccMallocTransaction,
-  book_begin:          NULL,
-  book_end:            NULL,
-  is_dirty:            qof_collection_is_dirty,
-  mark_clean:          qof_collection_mark_clean,
-  foreach:             qof_collection_foreach,
-  printable:           (const char* (*)(gpointer)) xaccTransGetDescription,
-  version_cmp:         (int (*)(gpointer,gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  GNC_ID_TRANS,
+  "Transaction",
+  (gpointer)xaccMallocTransaction,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  (const char* (*)(gpointer)) xaccTransGetDescription,
+  (int (*)(gpointer,gpointer)) qof_instance_version_cmp,
 };
 
 static gboolean
Index: trunk/src/engine/gnc-commodity.c
===================================================================
--- trunk/src/engine/gnc-commodity.c	(revision 17246)
+++ trunk/src/engine/gnc-commodity.c	(working copy)
@@ -1991,28 +1991,32 @@
 
 static QofObject commodity_object_def = 
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_COMMODITY,
-  type_label:        "Commodity",
-  book_begin:        NULL,
-  book_end:          NULL,
-  is_dirty:          qof_collection_is_dirty,
-  mark_clean:        qof_collection_mark_clean,
-  foreach:           qof_collection_foreach,
-  printable:         (const char* (*)(gpointer)) gnc_commodity_get_fullname,
+  QOF_OBJECT_VERSION,
+  GNC_ID_COMMODITY,
+  "Commodity",
+  NULL,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  (const char* (*)(gpointer)) gnc_commodity_get_fullname,
+  NULL,
 };
 
 static QofObject namespace_object_def = 
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_COMMODITY_NAMESPACE,
-  type_label:        "Namespace",
-  book_begin:        NULL,
-  book_end:          NULL,
-  is_dirty:          NULL,
-  mark_clean:        NULL,
-  foreach:           NULL,
-  printable:         NULL,
+  QOF_OBJECT_VERSION,
+  GNC_ID_COMMODITY_NAMESPACE,
+  "Namespace",
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
+  NULL,
 };
 
 static void 
@@ -2047,17 +2051,17 @@
 
 static QofObject commodity_table_object_def = 
 {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_COMMODITY_TABLE,
-  type_label:        "CommodityTable",
-  create:            NULL,
-  book_begin:        commodity_table_book_begin,
-  book_end:          commodity_table_book_end,
-  is_dirty:          qof_collection_is_dirty,
-  mark_clean:        qof_collection_mark_clean,
-  foreach:           NULL,
-  printable:         NULL,
-  version_cmp:       NULL,
+  QOF_OBJECT_VERSION,
+  GNC_ID_COMMODITY_TABLE,
+  "CommodityTable",
+  NULL,
+  commodity_table_book_begin,
+  commodity_table_book_end,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  NULL,
+  NULL,
+  NULL,
 };
 
 gboolean 
Index: trunk/src/engine/SchedXaction.c
===================================================================
--- trunk/src/engine/SchedXaction.c	(revision 17246)
+++ trunk/src/engine/SchedXaction.c	(working copy)
@@ -849,17 +849,17 @@
 
 static QofObject SXDesc = 
 {
-	interface_version : QOF_OBJECT_VERSION,
-	e_type            : GNC_SX_ID,
-	type_label        : "Scheduled Transaction",
-	create            : (gpointer)xaccSchedXactionMalloc,
-	book_begin        : NULL,
-	book_end          : NULL,
-	is_dirty          : qof_collection_is_dirty,
-	mark_clean        : qof_collection_mark_clean,
-	foreach           : qof_collection_foreach,
-	printable         : NULL,
-	version_cmp       : (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+	QOF_OBJECT_VERSION,
+	GNC_SX_ID,
+	"Scheduled Transaction",
+	(gpointer)xaccSchedXactionMalloc,
+	NULL,
+	NULL,
+	qof_collection_is_dirty,
+	qof_collection_mark_clean,
+	qof_collection_foreach,
+	NULL,
+	(int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean
Index: trunk/src/engine/Split.c
===================================================================
--- trunk/src/engine/Split.c	(revision 17246)
+++ trunk/src/engine/Split.c	(working copy)
@@ -1743,17 +1743,17 @@
 /* Hook into the QofObject registry */
 
 static QofObject split_object_def = {
-  interface_version: QOF_OBJECT_VERSION,
-  e_type:            GNC_ID_SPLIT,
-  type_label:        "Split",
-  create:            (gpointer)xaccMallocSplit,
-  book_begin:        NULL,
-  book_end:          NULL,
-  is_dirty:          qof_collection_is_dirty,
-  mark_clean:        qof_collection_mark_clean,
-  foreach:           qof_collection_foreach,
-  printable:         (const char* (*)(gpointer)) xaccSplitGetMemo,
-  version_cmp:       (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  GNC_ID_SPLIT,
+  "Split",
+  (gpointer)xaccMallocSplit,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  (const char* (*)(gpointer)) xaccSplitGetMemo,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 static gpointer 
Index: trunk/src/business/business-core/gncAddress.c
===================================================================
--- trunk/src/business/business-core/gncAddress.c	(revision 17246)
+++ trunk/src/business/business-core/gncAddress.c	(working copy)
@@ -372,17 +372,17 @@
 
 static QofObject GncAddressDesc =
 {
-	interface_version:  QOF_OBJECT_VERSION,
-	e_type:             GNC_ID_ADDRESS,
-	type_label:         "Address",
-	create:             (gpointer)qofAddressCreate,
-	book_begin:         NULL,
-	book_end:           NULL,
-	is_dirty:           qof_collection_is_dirty,
-	mark_clean:         qof_collection_mark_clean,
-	foreach:            qof_collection_foreach,
-	printable:          NULL,
-	version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+	QOF_OBJECT_VERSION,
+	GNC_ID_ADDRESS,
+	"Address",
+	(gpointer)qofAddressCreate,
+	NULL,
+	NULL,
+	qof_collection_is_dirty,
+	qof_collection_mark_clean,
+	qof_collection_foreach,
+	NULL,
+	(int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncAddressRegister (void)
Index: trunk/src/business/business-core/gncBillTerm.c
===================================================================
--- trunk/src/business/business-core/gncBillTerm.c	(revision 17246)
+++ trunk/src/business/business-core/gncBillTerm.c	(working copy)
@@ -702,17 +702,17 @@
 
 static QofObject gncBillTermDesc = 
 {
-  interface_version:   QOF_OBJECT_VERSION,
-  e_type:              _GNC_MOD_NAME,
-  type_label:          "Billing Term",
-  create:              (gpointer)gncBillTermCreate,
-  book_begin:          _gncBillTermCreate,
-  book_end:            _gncBillTermDestroy,
-  is_dirty:            qof_collection_is_dirty,
-  mark_clean:          qof_collection_mark_clean,
-  foreach:             qof_collection_foreach,
-  printable:           NULL,
-  version_cmp:         (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Billing Term",
+  (gpointer)gncBillTermCreate,
+  _gncBillTermCreate,
+  _gncBillTermDestroy,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  NULL,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncBillTermRegister (void)
Index: trunk/src/business/business-core/gncCustomer.c
===================================================================
--- trunk/src/business/business-core/gncCustomer.c	(revision 17246)
+++ trunk/src/business/business-core/gncCustomer.c	(working copy)
@@ -577,17 +577,17 @@
 
 static QofObject gncCustomerDesc =
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Customer",
-  create:             (gpointer)gncCustomerCreate,
-  book_begin:         NULL,
-  book_end:           NULL,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          (const char* (*)(gpointer))gncCustomerGetName,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Customer",
+  (gpointer)gncCustomerCreate,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  (const char* (*)(gpointer))gncCustomerGetName,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncCustomerRegister (void)
Index: trunk/src/business/business-core/gncEmployee.c
===================================================================
--- trunk/src/business/business-core/gncEmployee.c	(revision 17246)
+++ trunk/src/business/business-core/gncEmployee.c	(working copy)
@@ -419,17 +419,17 @@
 
 static QofObject gncEmployeeDesc = 
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Employee",
-  create:             (gpointer)gncEmployeeCreate,
-  book_begin:         NULL,
-  book_end:           NULL,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          _gncEmployeePrintable,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Employee",
+  (gpointer)gncEmployeeCreate,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  _gncEmployeePrintable,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncEmployeeRegister (void)
Index: trunk/src/business/business-core/gncEntry.c
===================================================================
--- trunk/src/business/business-core/gncEntry.c	(revision 17246)
+++ trunk/src/business/business-core/gncEntry.c	(working copy)
@@ -1239,17 +1239,17 @@
 
 static QofObject gncEntryDesc = 
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Order/Invoice/Bill Entry",
-  create:             (gpointer)gncEntryCreate,
-  book_begin:         NULL,
-  book_end:           NULL,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          NULL,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Order/Invoice/Bill Entry",
+  (gpointer)gncEntryCreate,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  NULL,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncEntryRegister (void)
Index: trunk/src/business/business-core/gncInvoice.c
===================================================================
--- trunk/src/business/business-core/gncInvoice.c	(revision 17246)
+++ trunk/src/business/business-core/gncInvoice.c	(working copy)
@@ -1543,17 +1543,17 @@
 
 static QofObject gncInvoiceDesc = 
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Invoice",
-  create:             (gpointer)gncInvoiceCreate,
-  book_begin:         NULL,
-  book_end:           NULL,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          _gncInvoicePrintable,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Invoice",
+  (gpointer)gncInvoiceCreate,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  _gncInvoicePrintable,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 static void
Index: trunk/src/business/business-core/gncJob.c
===================================================================
--- trunk/src/business/business-core/gncJob.c	(revision 17246)
+++ trunk/src/business/business-core/gncJob.c	(working copy)
@@ -367,17 +367,17 @@
 
 static QofObject gncJobDesc = 
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Job",
-  create:             (gpointer)gncJobCreate,
-  book_begin:         NULL,
-  book_end:           NULL,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          _gncJobPrintable,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Job",
+  (gpointer)gncJobCreate,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  _gncJobPrintable,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncJobRegister (void)
Index: trunk/src/business/business-core/gncOrder.c
===================================================================
--- trunk/src/business/business-core/gncOrder.c	(revision 17246)
+++ trunk/src/business/business-core/gncOrder.c	(working copy)
@@ -421,17 +421,17 @@
 
 static QofObject gncOrderDesc =
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Order",
-  create:             (gpointer)gncOrderCreate,
-  book_begin:         NULL,
-  book_end:           NULL,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          _gncOrderPrintable,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Order",
+  (gpointer)gncOrderCreate,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  _gncOrderPrintable,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncOrderRegister (void)
Index: trunk/src/business/business-core/gncTaxTable.c
===================================================================
--- trunk/src/business/business-core/gncTaxTable.c	(revision 17246)
+++ trunk/src/business/business-core/gncTaxTable.c	(working copy)
@@ -795,17 +795,17 @@
 
 static QofObject gncTaxTableDesc = 
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Tax Table",
-  create:             (gpointer)gncTaxTableCreate,
-  book_begin:         _gncTaxTableCreate,
-  book_end:           _gncTaxTableDestroy,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          NULL,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Tax Table",
+  (gpointer)gncTaxTableCreate,
+  _gncTaxTableCreate,
+  _gncTaxTableDestroy,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  NULL,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncTaxTableRegister (void)
Index: trunk/src/business/business-core/gncVendor.c
===================================================================
--- trunk/src/business/business-core/gncVendor.c	(revision 17246)
+++ trunk/src/business/business-core/gncVendor.c	(working copy)
@@ -520,17 +520,17 @@
 
 static QofObject gncVendorDesc = 
 {
-  interface_version:  QOF_OBJECT_VERSION,
-  e_type:             _GNC_MOD_NAME,
-  type_label:         "Vendor",
-  create:             (gpointer)gncVendorCreate,
-  book_begin:         NULL,
-  book_end:           NULL,
-  is_dirty:           qof_collection_is_dirty,
-  mark_clean:         qof_collection_mark_clean,
-  foreach:            qof_collection_foreach,
-  printable:          _gncVendorPrintable,
-  version_cmp:        (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
+  QOF_OBJECT_VERSION,
+  _GNC_MOD_NAME,
+  "Vendor",
+  (gpointer)gncVendorCreate,
+  NULL,
+  NULL,
+  qof_collection_is_dirty,
+  qof_collection_mark_clean,
+  qof_collection_foreach,
+  _gncVendorPrintable,
+  (int (*)(gpointer, gpointer)) qof_instance_version_cmp,
 };
 
 gboolean gncVendorRegister (void)
