--- gnumeric-1.8.2.orig/src/dialogs/dialog-preferences.c	Mon Jan 14 18:54:32 2008
+++ gnumeric-1.8.2/src/dialogs/dialog-preferences.c	Tue Apr 15 09:45:51 2008
@@ -765,7 +765,7 @@
 	{N_("Header/Footer"), GTK_STOCK_ITALIC,		 "0",  &pref_font_hf_initializer,	NULL},
 	{N_("Sorting"),       GTK_STOCK_SORT_ASCENDING,  "3", &pref_sort_page_initializer,	NULL},
 	{N_("Screen"),        GTK_STOCK_PREFERENCES,     "5", &pref_screen_page_initializer,	NULL},
-	{NULL, NULL, NULL, NULL, NULL, NULL},
+	{NULL, NULL, NULL, NULL, NULL},
 };
 
 static void
