--- gobby-0.4.4.orig/src/window.cpp	2007年  6月 15日 五 17:59:10
+++ gobby-0.4.4/src/window.cpp	2007年  8月 17日 五 22:05:34
@@ -1330,7 +1330,7 @@
 		m_statusbar.update_connection(str.str() );
 
 		// Start session
-		m_buffer = buffer;
+		m_buffer.reset(buffer.release());
 		obby_start();
 
 		// Session is open, no need to reshow join dialog
@@ -1373,7 +1373,7 @@
 		str << port;
 		m_statusbar.update_connection(str.str() );
 
-		m_buffer = buffer;
+		m_buffer.reset(buffer.release());
 
 		// Start session
 		obby_start();
