--- gpac/applications/osmo4_wx/Makefile.orig	2007-07-29 20:07:12.121839809 +0700
+++ gpac/applications/osmo4_wx/Makefile	2007-07-29 20:10:35.909799781 +0700
@@ -2,7 +2,7 @@
 
 vpath %.cpp $(SRC_PATH)/applications/osmo4_wx
 
-CFLAGS= $(CPPFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(CPPFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
@@ -41,7 +41,7 @@
 all: $(PROG)
 
 Osmo4$(EXE): $(OBJS)
-	$(CC) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac $(WX_LFLAGS) -lz
+	$(CXX) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac $(WX_LFLAGS) -lz -lgtk-x11-2.0 -lgdk-x11-2.0 -lX11
 
 %.o: %.cpp
 	$(CXX) $(CFLAGS) $(INSTALL_FLAGS) $(WX_CFLAGS) -c -o $@ $< 
