--- gpac/src/Makefile.orig	2007-07-29 20:44:56.908019022 +0700
+++ gpac/src/Makefile	2007-07-29 20:44:56.936540030 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/src
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/isom_in/Makefile.orig	2007-07-29 20:44:57.291784781 +0700
+++ gpac/modules/isom_in/Makefile	2007-07-29 20:44:57.318807351 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/isom_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/ffmpeg_in/Makefile.orig	2007-07-29 20:44:57.337699375 +0700
+++ gpac/modules/ffmpeg_in/Makefile	2007-07-29 20:44:57.364941383 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ffmpeg_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/timedtext/Makefile.orig	2007-07-29 20:44:57.384050300 +0700
+++ gpac/modules/timedtext/Makefile	2007-07-29 20:44:57.412841988 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/timedtext
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/render2d/Makefile.orig	2007-07-29 20:44:57.441486966 +0700
+++ gpac/modules/render2d/Makefile	2007-07-29 20:44:57.468561960 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/render2d
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/ogg/Makefile.orig	2007-07-29 20:44:57.486127299 +0700
+++ gpac/modules/ogg/Makefile	2007-07-29 20:44:57.513230348 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ogg
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/xvid_dec/Makefile.orig	2007-07-29 20:44:57.537718567 +0700
+++ gpac/modules/xvid_dec/Makefile	2007-07-29 20:44:57.565969279 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/xvid_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/laser_dec/Makefile.orig	2007-07-29 20:44:57.585490556 +0700
+++ gpac/modules/laser_dec/Makefile	2007-07-29 20:44:57.612326866 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/laser_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/svg_in/Makefile.orig	2007-07-29 20:44:57.632118020 +0700
+++ gpac/modules/svg_in/Makefile	2007-07-29 20:44:57.660054846 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/svg_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/svg_loader/Makefile.orig	2007-07-29 20:44:57.675559584 +0700
+++ gpac/modules/svg_loader/Makefile	2007-07-29 20:44:57.703941164 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/svg_loader
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include -I$(VPATH)/svg_parser
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include -I$(VPATH)/svg_parser
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/render3d/Makefile.orig	2007-07-29 20:44:57.721505574 +0700
+++ gpac/modules/render3d/Makefile	2007-07-29 20:44:57.750076907 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/render3d
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/oss_audio/Makefile.orig	2007-07-29 20:44:57.765342796 +0700
+++ gpac/modules/oss_audio/Makefile	2007-07-29 20:44:57.792750416 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/oss_audio
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(OSS_CFLAGS)
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include $(OSS_CFLAGS)
 LDFLAGS+=$(OSS_LDFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
--- gpac/modules/ctx_load/Makefile.orig	2007-07-29 20:44:57.811635520 +0700
+++ gpac/modules/ctx_load/Makefile	2007-07-29 20:44:57.838237545 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ctx_load
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/mp3_in/Makefile.orig	2007-07-29 20:44:57.855803847 +0700
+++ gpac/modules/mp3_in/Makefile	2007-07-29 20:44:57.882971294 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/mp3_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/wav_out/Makefile.orig	2007-07-29 20:44:57.918341272 +0700
+++ gpac/modules/wav_out/Makefile	2007-07-29 20:44:57.950880095 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/wav_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include -DDISABLE_WAVE_EX
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include -DDISABLE_WAVE_EX
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/aac_in/Makefile.orig	2007-07-29 20:44:58.297147864 +0700
+++ gpac/modules/aac_in/Makefile	2007-07-29 20:44:58.323809704 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/aac_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/x11_out/Makefile.orig	2007-07-29 20:44:58.345389220 +0700
+++ gpac/modules/x11_out/Makefile	2007-07-29 20:44:58.372621777 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/x11_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 # -I/usr/local/arm/3.3.2/include
 
 
--- gpac/modules/raw_out/Makefile.orig	2007-07-29 20:44:58.398516222 +0700
+++ gpac/modules/raw_out/Makefile	2007-07-29 20:44:58.427998497 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/raw_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/img_in/Makefile.orig	2007-07-29 20:44:58.445977778 +0700
+++ gpac/modules/img_in/Makefile	2007-07-29 20:44:58.472700968 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/img_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/soft_raster/Makefile.orig	2007-07-29 20:44:58.491115672 +0700
+++ gpac/modules/soft_raster/Makefile	2007-07-29 20:44:58.517966156 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/soft_raster
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/ismacryp/Makefile.orig	2007-07-29 20:44:58.542351734 +0700
+++ gpac/modules/ismacryp/Makefile	2007-07-29 20:44:58.570153559 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ismacryp
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/bifs_dec/Makefile.orig	2007-07-29 20:44:58.587497965 +0700
+++ gpac/modules/bifs_dec/Makefile	2007-07-29 20:44:58.614677164 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/bifs_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/sdl_out/Makefile.orig	2007-07-29 20:44:58.632412398 +0700
+++ gpac/modules/sdl_out/Makefile	2007-07-29 20:44:58.659186163 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/sdl_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(SDL_CFLAGS)
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include $(SDL_CFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/alsa/Makefile.orig	2007-07-29 20:44:58.677504865 +0700
+++ gpac/modules/alsa/Makefile	2007-07-29 20:44:58.705861242 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/alsa
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(OSS_CFLAGS)
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include $(OSS_CFLAGS)
 LDFLAGS+=$(OSS_LDFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
--- gpac/modules/ft_font/Makefile.orig	2007-07-29 20:44:58.723235880 +0700
+++ gpac/modules/ft_font/Makefile	2007-07-29 20:44:58.752080782 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ft_font
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(FT_CFLAGS)
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include $(FT_CFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/dummy_in/Makefile.orig	2007-07-29 20:44:58.767341466 +0700
+++ gpac/modules/dummy_in/Makefile	2007-07-29 20:44:58.794874526 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/dummy_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/odf_dec/Makefile.orig	2007-07-29 20:44:58.812868525 +0700
+++ gpac/modules/odf_dec/Makefile	2007-07-29 20:44:58.839479536 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/odf_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/modules/rtp_in/Makefile.orig	2007-07-29 20:44:58.859261976 +0700
+++ gpac/modules/rtp_in/Makefile	2007-07-29 20:44:58.887997818 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/rtp_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/osmozilla/Makefile.orig	2007-07-29 20:44:58.913462667 +0700
+++ gpac/applications/osmozilla/Makefile	2007-07-29 20:44:58.940452768 +0700
@@ -14,7 +14,7 @@
 endif
 endif
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include \
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include \
   -I$(GECKO_PATH) \
   -I$(GECKO_PATH)/plugin/include  \
   -I$(GECKO_PATH)/nspr/include \
--- gpac/applications/generators/MPEG4/Makefile.orig	2007-07-29 20:44:58.958582796 +0700
+++ gpac/applications/generators/MPEG4/Makefile	2007-07-29 20:44:58.985831609 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/generators/MPEG4
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/generators/X3D/Makefile.orig	2007-07-29 20:44:59.004172912 +0700
+++ gpac/applications/generators/X3D/Makefile	2007-07-29 20:44:59.031731411 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/generators/X3D
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/generators/SVG/Makefile.orig	2007-07-29 20:44:59.049883874 +0700
+++ gpac/applications/generators/SVG/Makefile	2007-07-29 20:44:59.078356904 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/generators/SVG
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/mp4client/Makefile.orig	2007-07-29 20:44:59.094712917 +0700
+++ gpac/applications/mp4client/Makefile	2007-07-29 20:44:59.121512801 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/mp4client
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/mp4box/Makefile.orig	2007-07-29 20:44:59.141145998 +0700
+++ gpac/applications/mp4box/Makefile	2007-07-29 20:44:59.167945853 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/mp4box
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/mp42avi/Makefile.orig	2007-07-29 20:44:59.186549738 +0700
+++ gpac/applications/mp42avi/Makefile	2007-07-29 20:44:59.215105611 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/mp42avi
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/testapps/loadcompare/Makefile.orig	2007-07-29 20:44:59.240363590 +0700
+++ gpac/applications/testapps/loadcompare/Makefile	2007-07-29 20:44:59.267234861 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/testapps/loadcompare
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/testapps/mpedemux/Makefile.orig	2007-07-29 20:44:59.286225935 +0700
+++ gpac/applications/testapps/mpedemux/Makefile	2007-07-29 20:44:59.313833492 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/test_apps/mpedemux
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/testapps/mp4_streamer/Makefile.orig	2007-07-29 20:44:59.331881343 +0700
+++ gpac/applications/testapps/mp4_streamer/Makefile	2007-07-29 20:44:59.360334079 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/test_apps/mp4_streamer
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/testapps/mp42ts/Makefile.orig	2007-07-29 20:44:59.383777627 +0700
+++ gpac/applications/testapps/mp42ts/Makefile	2007-07-29 20:44:59.413619632 +0700
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/test_apps/mp42ts
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS)  -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
--- gpac/applications/testapps/broadcaster/Makefile.orig	2007-07-29 20:44:59.428606727 +0700
+++ gpac/applications/testapps/broadcaster/Makefile	2007-07-29 20:44:59.455412831 +0700
@@ -1,5 +1,5 @@
 CC=gcc
-CFLAGS=-Wall -DUSE_TCP_STANDARD
+CFLAGS= -DUSE_TCP_STANDARD
 # -pedantic -ansi
 SOURCES=
 INCLUDES=-Iinclude -I.
