--- gpac/applications/osmo4_wx/Makefile.orig	2007-07-29 21:22:14.829346423 +0700
+++ gpac/applications/osmo4_wx/Makefile	2007-07-29 21:27:02.240127373 +0700
@@ -59,17 +59,18 @@
 	   $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/Resources/English.lproj/InfoPlist.strings
 	cp ./Darwin.Osmo.icns \
 	   $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/Resources/Osmo.icns
-	install -c -s -m 755 ../../bin/gcc/Osmo4 \
+	install -c -m 755 ../../bin/gcc/Osmo4 \
 	      $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/MacOS
 	echo -n 'APPLOsm4' > $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/PkgInfo
 else
 	rm -f wxOsmo4.o
 	make -override BUILD_INSTALL=yes all
-	install -c -s -m 755 ../../bin/gcc/Osmo4 "$(prefix)/bin"
+	mkdir -p $(DESTDIR)$(prefix)/bin
+	install -c -m 755 ../../bin/gcc/Osmo4 "$(DESTDIR)$(prefix)/bin"
 endif
 
-uninstall:
-	rm -rf $(prefix)/bin/Osmo4
+
+	rm -rf $(DESTDIR)$(prefix)/bin/Osmo4
 
 dep:
 
