diff -rbu gpac.orig/configure gpac/configure
--- gpac.orig/configure	Wed May 30 18:45:28 2007
+++ gpac/configure	Fri Nov 21 13:17:00 2008
@@ -345,9 +345,9 @@
 #include <stdio.h>
 int main( void ) { return 0; }
 EOF
-if $cc -o $TMPO $TMPC -fno-strict-aliasing 2> /dev/null  ; then
-CFLAGS="$CFLAGS -fno-strict-aliasing"
-fi
+#if $cc -o $TMPO $TMPC -fno-strict-aliasing 2> /dev/null  ; then
+#CFLAGS="$CFLAGS -fno-strict-aliasing"
+#fi
 CPPFLAGS="$CFLAGS"
 
 if $cc -o $TMPO $TMPC -lz -Wno-pointer-sign 2> /dev/null  ; then
diff -rbu gpac.orig/applications/generators/MPEG4/Makefile gpac/applications/generators/MPEG4/Makefile
--- gpac.orig/applications/generators/MPEG4/Makefile	Wed May 24 11:52:52 2006
+++ gpac/applications/generators/MPEG4/Makefile	Fri Nov 21 12:20:19 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/generators/MPEG4
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/generators/SVG/Makefile gpac/applications/generators/SVG/Makefile
--- gpac.orig/applications/generators/SVG/Makefile	Tue Jan  9 13:43:57 2007
+++ gpac/applications/generators/SVG/Makefile	Fri Nov 21 12:20:24 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/generators/SVG
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/generators/X3D/Makefile gpac/applications/generators/X3D/Makefile
--- gpac.orig/applications/generators/X3D/Makefile	Wed May 24 11:52:52 2006
+++ gpac/applications/generators/X3D/Makefile	Fri Nov 21 12:20:30 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/generators/X3D
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/mp42avi/Makefile gpac/applications/mp42avi/Makefile
--- gpac.orig/applications/mp42avi/Makefile	Wed May 24 11:52:52 2006
+++ gpac/applications/mp42avi/Makefile	Fri Nov 21 12:19:45 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/mp42avi
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/mp4box/Makefile gpac/applications/mp4box/Makefile
--- gpac.orig/applications/mp4box/Makefile	Wed May 24 11:52:53 2006
+++ gpac/applications/mp4box/Makefile	Fri Nov 21 12:19:55 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/mp4box
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/mp4client/Makefile gpac/applications/mp4client/Makefile
--- gpac.orig/applications/mp4client/Makefile	Sun Jan 14 17:11:40 2007
+++ gpac/applications/mp4client/Makefile	Fri Nov 21 12:20:03 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/mp4client
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/osmo4_wx/Makefile gpac/applications/osmo4_wx/Makefile
--- gpac.orig/applications/osmo4_wx/Makefile	Sun Jun 11 22:48:02 2006
+++ gpac/applications/osmo4_wx/Makefile	Fri Nov 21 15:11:05 2008
@@ -2,7 +2,7 @@
 
 vpath %.cpp $(SRC_PATH)/applications/osmo4_wx
 
-CFLAGS= $(CPPFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(CPPFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
@@ -41,7 +41,7 @@
 all: $(PROG)
 
 Osmo4$(EXE): $(OBJS)
-	$(CC) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac $(WX_LFLAGS) -lz
+	$(CXX) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac $(WX_LFLAGS) -lz -lgdk-x11-2.0 -lgtk-x11-2.0 -lX11
 
 %.o: %.cpp
 	$(CXX) $(CFLAGS) $(INSTALL_FLAGS) $(WX_CFLAGS) -c -o $@ $< 
@@ -59,17 +59,18 @@
 	   $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/Resources/English.lproj/InfoPlist.strings
 	cp ./Darwin.Osmo.icns \
 	   $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/Resources/Osmo.icns
-	install -c -s -m 755 ../../bin/gcc/Osmo4 \
+	install -c -m 755 ../../bin/gcc/Osmo4 \
 	      $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/MacOS
 	echo -n 'APPLOsm4' > $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/PkgInfo
 else
 	rm -f wxOsmo4.o
 	make -override BUILD_INSTALL=yes all
-	install -c -s -m 755 ../../bin/gcc/Osmo4 "$(prefix)/bin"
+	mkdir -p $(DESTDIR)$(prefix)/bin
+	install -c -m 755 ../../bin/gcc/Osmo4 "$(DESTDIR)$(prefix)/bin"
 endif
 
 uninstall:
-	rm -rf $(prefix)/bin/Osmo4
+	rm -rf $(DESTDIR)$(prefix)/bin/Osmo4
 
 dep:
 
diff -rbu gpac.orig/applications/osmo4_wx/Playlist.cpp gpac/applications/osmo4_wx/Playlist.cpp
--- gpac.orig/applications/osmo4_wx/Playlist.cpp	Tue Apr 10 12:56:34 2007
+++ gpac/applications/osmo4_wx/Playlist.cpp	Fri Nov 21 14:47:31 2008
@@ -38,8 +38,8 @@
 	if (_url) {_url += 3; is_remote = 1; }
 	else _url = (const char *) the_url;
 
-	char *str = strrchr(_url, '\\');
-	if (!str) str = strrchr(_url, '/');
+	char *str = (char*)strrchr(_url, '\\');
+	if (!str) str = (char*)strrchr(_url, '/');
 	if (str && strlen(str+1)) {
 		m_disp_name = strdup(str+1);
 		str = strrchr(m_disp_name, '.');
@@ -763,7 +763,7 @@
 
 void wxPlaylist::QueueURL(wxString filename)
 {
-	char *ext = strrchr(filename.mb_str(wxConvUTF8), '.');
+	char *ext = (char*)strrchr(filename.mb_str(wxConvUTF8), '.');
 	if (ext && (!stricmp(ext, ".m3u") || !stricmp(ext, ".pls")) ) {
 		OpenPlaylist(filename);
 	} else {
diff -rbu gpac.orig/applications/osmo4_wx/wxOsmo4.cpp gpac/applications/osmo4_wx/wxOsmo4.cpp
--- gpac.orig/applications/osmo4_wx/wxOsmo4.cpp	Wed May 30 19:50:37 2007
+++ gpac/applications/osmo4_wx/wxOsmo4.cpp	Fri Nov 21 14:45:28 2008
@@ -390,7 +390,7 @@
 	u32 count = filenames.GetCount();
 
 	if (count==1) {
-		char *ext = strrchr(filenames.Item(0).mb_str(wxConvUTF8) , '.');
+		const char *ext = strrchr(filenames.Item(0).mb_str(wxConvUTF8) , '.');
 		/*if playing and sub d&d, open sub in current presentation*/
 		if (m_pMain->m_connected && ext && ( !stricmp(ext, ".srt") || !stricmp(ext, ".sub") || !stricmp(ext, ".ttxt") || !stricmp(ext, ".xml") ) ) {
 			m_pMain->AddSubtitle(filenames.Item(0).mb_str(wxConvUTF8) , 1);
@@ -2255,13 +2255,13 @@
 		if (gf_term_get_object_info(m_term, odm, &info) != GF_OK) break;
 
 		if (info.owns_service) {
-			char *szName = strrchr(info.service_url, '\\');
-			if (!szName) szName = strrchr(info.service_url, '/');
+			char *szName = (char*)strrchr(info.service_url, '\\');
+			if (!szName) szName = (char*)strrchr(info.service_url, '/');
 			if (!szName) szName = (char *) info.service_url;
 			else szName += 1;
 			strcpy(szLabel, szName);
 			szName = strrchr(szLabel, '.');
-			if (szName) szName[0] = 0;
+			if (szName) szName[0] = '\0';
 		}
 		switch (info.od_type) {
 		case GF_STREAM_AUDIO:
diff -rbu gpac.orig/applications/osmozilla/Makefile gpac/applications/osmozilla/Makefile
--- gpac.orig/applications/osmozilla/Makefile	Fri May  4 18:16:42 2007
+++ gpac/applications/osmozilla/Makefile	Fri Nov 21 12:20:13 2008
@@ -14,7 +14,7 @@
 endif
 endif
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include \
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include \
   -I$(GECKO_PATH) \
   -I$(GECKO_PATH)/plugin/include  \
   -I$(GECKO_PATH)/nspr/include \
diff -rbu gpac.orig/applications/testapps/broadcaster/Makefile gpac/applications/testapps/broadcaster/Makefile
--- gpac.orig/applications/testapps/broadcaster/Makefile	Fri Feb 16 14:07:20 2007
+++ gpac/applications/testapps/broadcaster/Makefile	Fri Nov 21 12:21:01 2008
@@ -1,5 +1,5 @@
 CC=gcc
-CFLAGS=-Wall -DUSE_TCP_STANDARD
+CFLAGS=-DUSE_TCP_STANDARD
 # -pedantic -ansi
 SOURCES=
 INCLUDES=-Iinclude -I.
diff -rbu gpac.orig/applications/testapps/loadcompare/Makefile gpac/applications/testapps/loadcompare/Makefile
--- gpac.orig/applications/testapps/loadcompare/Makefile	Tue Jul 11 10:26:29 2006
+++ gpac/applications/testapps/loadcompare/Makefile	Fri Nov 21 12:21:18 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/testapps/loadcompare
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/testapps/mp42ts/Makefile gpac/applications/testapps/mp42ts/Makefile
--- gpac.orig/applications/testapps/mp42ts/Makefile	Wed May 30 18:08:40 2007
+++ gpac/applications/testapps/mp42ts/Makefile	Fri Nov 21 12:21:35 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/test_apps/mp42ts
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/testapps/mp4_streamer/Makefile gpac/applications/testapps/mp4_streamer/Makefile
--- gpac.orig/applications/testapps/mp4_streamer/Makefile	Thu Feb 22 21:12:49 2007
+++ gpac/applications/testapps/mp4_streamer/Makefile	Fri Nov 21 12:21:28 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/test_apps/mp4_streamer
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/applications/testapps/mpedemux/Makefile gpac/applications/testapps/mpedemux/Makefile
--- gpac.orig/applications/testapps/mpedemux/Makefile	Wed Mar 14 09:36:29 2007
+++ gpac/applications/testapps/mpedemux/Makefile	Fri Nov 21 12:21:51 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/applications/test_apps/mpedemux
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/include/gpac/esi.h gpac/include/gpac/esi.h
--- gpac.orig/include/gpac/esi.h	Fri Mar 30 14:37:56 2007
+++ gpac/include/gpac/esi.h	Fri Nov 21 14:16:06 2008
@@ -131,10 +131,8 @@
 	/*average bit rate in bit/sec - 0 if unknown*/
 	u32 bit_rate;
 
-	union {
 		struct __esi_video_info info_video;
 		struct __esi_audio_info info_audio;
-	};
 
 	/*input ES control from caller*/
 	GF_Err (*input_ctrl)(struct __elementary_stream_ifce *_self, u32 ctrl_type, void *param);
diff -rbu gpac.orig/include/gpac/media_tools.h gpac/include/gpac/media_tools.h
--- gpac.orig/include/gpac/media_tools.h	Wed Dec 20 18:34:55 2006
+++ gpac/include/gpac/media_tools.h	Fri Nov 21 14:25:03 2008
@@ -99,10 +99,8 @@
 	/*possible import flags*/
 	u32 flags;
 	/*media format info*/
-	union {
 		struct __track_video_info video_info;
 		struct __track_audio_info audio_info;
-	};
 	u32 lang;
 	/*for MPEG2 TS: program number*/
 	u16 prog_num;
diff -rbu gpac.orig/include/gpac/scenegraph.h gpac/include/gpac/scenegraph.h
--- gpac.orig/include/gpac/scenegraph.h	Wed May 16 09:40:57 2007
+++ gpac/include/gpac/scenegraph.h	Fri Nov 21 14:17:57 2008
@@ -215,7 +215,8 @@
 void gf_node_allow_cyclic_render(GF_Node *node);
 
 /*blindly calls RenderNode on all nodes in the "children" list*/
-void gf_node_render_children(GF_Node *node, void *renderStack);;
+void gf_node_render_children(GF_Node *node, void *renderStack);
+
 /*returns number of parent for this node (parent are kept regardless of DEF state)*/
 u32 gf_node_get_parent_count(GF_Node *node);
 /*returns desired parent for this node (parent are kept regardless of DEF state)
diff -rbu gpac.orig/modules/aac_in/Makefile gpac/modules/aac_in/Makefile
--- gpac.orig/modules/aac_in/Makefile	Wed Dec 13 16:12:04 2006
+++ gpac/modules/aac_in/Makefile	Fri Nov 21 12:14:55 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/aac_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/alsa/Makefile gpac/modules/alsa/Makefile
--- gpac.orig/modules/alsa/Makefile	Sun Apr 22 19:15:53 2007
+++ gpac/modules/alsa/Makefile	Fri Nov 21 12:15:04 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/alsa
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(OSS_CFLAGS)
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include $(OSS_CFLAGS)
 LDFLAGS+=$(OSS_LDFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
diff -rbu gpac.orig/modules/bifs_dec/Makefile gpac/modules/bifs_dec/Makefile
--- gpac.orig/modules/bifs_dec/Makefile	Wed May 24 11:52:55 2006
+++ gpac/modules/bifs_dec/Makefile	Fri Nov 21 12:15:49 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/bifs_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/ctx_load/Makefile gpac/modules/ctx_load/Makefile
--- gpac.orig/modules/ctx_load/Makefile	Wed May 24 11:52:55 2006
+++ gpac/modules/ctx_load/Makefile	Fri Nov 21 12:15:58 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ctx_load
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/dummy_in/Makefile gpac/modules/dummy_in/Makefile
--- gpac.orig/modules/dummy_in/Makefile	Wed May 24 11:52:55 2006
+++ gpac/modules/dummy_in/Makefile	Fri Nov 21 12:16:08 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/dummy_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/ffmpeg_in/Makefile gpac/modules/ffmpeg_in/Makefile
--- gpac.orig/modules/ffmpeg_in/Makefile	Wed Dec 13 16:12:07 2006
+++ gpac/modules/ffmpeg_in/Makefile	Fri Nov 21 12:16:23 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ffmpeg_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/ft_font/Makefile gpac/modules/ft_font/Makefile
--- gpac.orig/modules/ft_font/Makefile	Wed May 24 11:52:57 2006
+++ gpac/modules/ft_font/Makefile	Fri Nov 21 12:16:31 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ft_font
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(FT_CFLAGS)
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include $(FT_CFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/img_in/Makefile gpac/modules/img_in/Makefile
--- gpac.orig/modules/img_in/Makefile	Thu Feb 22 19:31:30 2007
+++ gpac/modules/img_in/Makefile	Fri Nov 21 12:16:40 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/img_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/ismacryp/Makefile gpac/modules/ismacryp/Makefile
--- gpac.orig/modules/ismacryp/Makefile	Fri Apr 20 18:11:16 2007
+++ gpac/modules/ismacryp/Makefile	Fri Nov 21 12:16:50 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ismacryp
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/isom_in/Makefile gpac/modules/isom_in/Makefile
--- gpac.orig/modules/isom_in/Makefile	Wed May 24 11:52:57 2006
+++ gpac/modules/isom_in/Makefile	Fri Nov 21 12:08:52 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/isom_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/laser_dec/Makefile gpac/modules/laser_dec/Makefile
--- gpac.orig/modules/laser_dec/Makefile	Wed May 24 11:52:58 2006
+++ gpac/modules/laser_dec/Makefile	Fri Nov 21 12:16:59 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/laser_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/mp3_in/Makefile gpac/modules/mp3_in/Makefile
--- gpac.orig/modules/mp3_in/Makefile	Wed Dec 13 16:12:11 2006
+++ gpac/modules/mp3_in/Makefile	Fri Nov 21 12:17:05 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/mp3_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/odf_dec/Makefile gpac/modules/odf_dec/Makefile
--- gpac.orig/modules/odf_dec/Makefile	Wed May 24 11:52:58 2006
+++ gpac/modules/odf_dec/Makefile	Fri Nov 21 12:17:55 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/odf_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/ogg/Makefile gpac/modules/ogg/Makefile
--- gpac.orig/modules/ogg/Makefile	Wed Dec 13 16:12:12 2006
+++ gpac/modules/ogg/Makefile	Fri Nov 21 12:18:02 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/ogg
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/oss_audio/Makefile gpac/modules/oss_audio/Makefile
--- gpac.orig/modules/oss_audio/Makefile	Fri Mar  9 14:51:52 2007
+++ gpac/modules/oss_audio/Makefile	Fri Nov 21 12:18:10 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/oss_audio
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(OSS_CFLAGS)
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include $(OSS_CFLAGS)
 LDFLAGS+=$(OSS_LDFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
diff -rbu gpac.orig/modules/raw_out/Makefile gpac/modules/raw_out/Makefile
--- gpac.orig/modules/raw_out/Makefile	Wed May 24 11:52:59 2006
+++ gpac/modules/raw_out/Makefile	Fri Nov 21 12:18:17 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/raw_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/render2d/Makefile gpac/modules/render2d/Makefile
--- gpac.orig/modules/render2d/Makefile	Mon Apr 23 16:08:38 2007
+++ gpac/modules/render2d/Makefile	Fri Nov 21 12:18:23 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/render2d
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/render3d/Makefile gpac/modules/render3d/Makefile
--- gpac.orig/modules/render3d/Makefile	Tue Feb 27 12:53:16 2007
+++ gpac/modules/render3d/Makefile	Fri Nov 21 12:18:29 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/render3d
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/rtp_in/Makefile gpac/modules/rtp_in/Makefile
--- gpac.orig/modules/rtp_in/Makefile	Sun Mar 11 22:59:08 2007
+++ gpac/modules/rtp_in/Makefile	Fri Nov 21 12:18:35 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/rtp_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/sdl_out/Makefile gpac/modules/sdl_out/Makefile
--- gpac.orig/modules/sdl_out/Makefile	Wed May 24 11:53:03 2006
+++ gpac/modules/sdl_out/Makefile	Fri Nov 21 12:18:48 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/sdl_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include $(SDL_CFLAGS)
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include $(SDL_CFLAGS)
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/soft_raster/Makefile gpac/modules/soft_raster/Makefile
--- gpac.orig/modules/soft_raster/Makefile	Wed May 24 11:53:03 2006
+++ gpac/modules/soft_raster/Makefile	Fri Nov 21 12:18:55 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/soft_raster
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/svg_in/Makefile gpac/modules/svg_in/Makefile
--- gpac.orig/modules/svg_in/Makefile	Mon Feb 26 13:31:31 2007
+++ gpac/modules/svg_in/Makefile	Fri Nov 21 12:19:01 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/svg_in
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/svg_loader/Makefile gpac/modules/svg_loader/Makefile
--- gpac.orig/modules/svg_loader/Makefile	Wed May 24 11:53:04 2006
+++ gpac/modules/svg_loader/Makefile	Fri Nov 21 12:19:07 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/svg_loader
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include -I$(VPATH)/svg_parser
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include -I$(VPATH)/svg_parser
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/timedtext/Makefile gpac/modules/timedtext/Makefile
--- gpac.orig/modules/timedtext/Makefile	Wed May 24 11:53:04 2006
+++ gpac/modules/timedtext/Makefile	Fri Nov 21 12:19:13 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/timedtext
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/wav_out/Makefile gpac/modules/wav_out/Makefile
--- gpac.orig/modules/wav_out/Makefile	Wed May 24 11:53:04 2006
+++ gpac/modules/wav_out/Makefile	Fri Nov 21 12:19:19 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/wav_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include -DDISABLE_WAVE_EX
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include -DDISABLE_WAVE_EX
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/modules/x11_out/Makefile gpac/modules/x11_out/Makefile
--- gpac.orig/modules/x11_out/Makefile	Wed May 24 11:53:05 2006
+++ gpac/modules/x11_out/Makefile	Fri Nov 21 12:19:25 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/x11_out
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 # -I/usr/local/arm/3.3.2/include
 
 
diff -rbu gpac.orig/modules/xvid_dec/Makefile gpac/modules/xvid_dec/Makefile
--- gpac.orig/modules/xvid_dec/Makefile	Wed Dec 13 16:12:25 2006
+++ gpac/modules/xvid_dec/Makefile	Fri Nov 21 12:19:32 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/modules/xvid_dec
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/src/Makefile gpac/src/Makefile
--- gpac.orig/src/Makefile	Tue May  8 17:15:25 2007
+++ gpac/src/Makefile	Fri Nov 21 12:08:15 2008
@@ -2,7 +2,7 @@
 
 vpath %.c $(SRC_PATH)/src
 
-CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include
+CFLAGS= $(OPTFLAGS) -I$(SRC_PATH)/include
 
 ifeq ($(DEBUGBUILD), yes)
 CFLAGS+=-g
diff -rbu gpac.orig/src/odf/ipmpx_code.c gpac/src/odf/ipmpx_code.c
--- gpac.orig/src/odf/ipmpx_code.c	Mon Nov 13 19:07:10 2006
+++ gpac/src/odf/ipmpx_code.c	Fri Nov 21 14:27:54 2008
@@ -34,7 +34,7 @@
 
 #define GF_IPMPX_DELETE_ARRAY(__ar) if (__ar) { if (__ar->data) free(__ar->data); free(__ar); }
 
-u32 GFINLINE get_field_size(u32 size_desc)
+u32 get_field_size(u32 size_desc)
 {
 	if (size_desc < 0x00000080) return 1;
 	else if (size_desc < 0x00004000) return 2;
diff -rbu gpac.orig/src/scenegraph/base_scenegraph.c gpac/src/scenegraph/base_scenegraph.c
--- gpac.orig/src/scenegraph/base_scenegraph.c	Sun May 20 20:25:17 2007
+++ gpac/src/scenegraph/base_scenegraph.c	Fri Nov 21 14:27:02 2008
@@ -435,7 +435,7 @@
 	if (sg) sg->RootNode = node;
 }
 
-GFINLINE void remove_node_id(GF_SceneGraph *sg, GF_Node *node)
+void remove_node_id(GF_SceneGraph *sg, GF_Node *node)
 {
 	NodeIDedItem *reg_node = sg->id_node;
 	if (reg_node && (reg_node->node==node)) {
diff -rbu gpac.orig/src/utils/os_divers.c gpac/src/utils/os_divers.c
--- gpac.orig/src/utils/os_divers.c	Wed May 23 17:51:18 2007
+++ gpac/src/utils/os_divers.c	Fri Nov 21 14:08:30 2008
@@ -274,7 +274,7 @@
 /*enumerate directories*/
 GF_Err gf_enum_directory(const char *dir, Bool enum_directory, gf_enum_dir_item enum_dir_fct, void *cbck, const char *filter)
 {
-	unsigned char item_path[GF_MAX_PATH];
+	char item_path[GF_MAX_PATH];
 
 #if defined(_WIN32_WCE)
 	unsigned char _path[GF_MAX_PATH];
@@ -282,7 +282,7 @@
 	unsigned short w_filter[GF_MAX_PATH];
 	unsigned char file[GF_MAX_PATH];
 #else
-	unsigned char path[GF_MAX_PATH], *file;
+	char path[GF_MAX_PATH], *file;
 #endif
 
 #ifdef WIN32
diff -rbu gpac.orig/src/utils/os_net.c gpac/src/utils/os_net.c
--- gpac.orig/src/utils/os_net.c	Wed May 30 17:53:28 2007
+++ gpac/src/utils/os_net.c	Fri Nov 21 13:15:40 2008
@@ -57,7 +57,12 @@
 #define ENETDOWN			WSAENETDOWN
 
 #define LASTSOCKERROR WSAGetLastError()
+#if (defined(sun) && defined(__SVR4)) && !defined(INADDR_NONE)
+#define INADDR_NONE -1
+#endif
 
+
+
 /*the number of sockets used. This because the WinSock lib needs init*/
 static int wsa_init = 0;
 
