--- graphviz-2.14.1/lib/common/Makefile.am.orig	2007-09-11 15:46:15.832059000 +0800
+++ graphviz-2.14.1/lib/common/Makefile.am	2007-09-11 15:46:47.611811000 +0800
@@ -14,10 +14,10 @@
 AM_CFLAGS = -D_BLD_common=1
 endif
 
-pkginclude_HEADERS = geom.h color.h types.h textpara.h usershape.h
+pkginclude_HEADERS = arith.h geom.h color.h types.h textpara.h usershape.h
 noinst_HEADERS = render.h utils.h memory.h \
 	geomprocs.h colorprocs.h colortbl.h entities.h \
-	logic.h arith.h globals.h \
+	logic.h globals.h \
 	const.h macros.h htmllex.h htmltable.h pointset.h
 noinst_LTLIBRARIES = libcommon_C.la
 
