diff -r 5b220aad28b5 configure.ac
--- a/configure.ac	Sat Oct 13 14:55:39 2007 +0530
+++ b/configure.ac	Sat Oct 13 14:56:04 2007 +0530
@@ -137,9 +137,18 @@ AC_CHECK_DECLS(ldexp,,,[#include <math.h
 AC_CHECK_DECLS(ldexp,,,[#include <math.h>])
 AC_CHECK_DECLS(frexp,,,[#include <math.h>])
 AC_CHECK_DECLS(isinf,,,[#include <math.h>])
-AC_CHECK_DECLS(finite,,,[#include <math.h>])
-AC_CHECK_DECLS(isfinite,,,[#include <math.h>])
 AC_CHECK_DECLS(isnan,,,[#include <math.h>])
+
+dnl Solaris does not have an implementation of finite
+case "$host" in
+    *-*-*solaris*)
+       AC_MSG_RESULT([using builtin gsl_finite implementation for solaris])
+       ;;
+    *)
+       AC_CHECK_DECLS(finite,,,[#include <math.h>])
+       AC_CHECK_DECLS(isfinite,,,[#include <math.h>])
+       ;;
+esac
 
 dnl OpenBSD has a broken implementation of log1p.
 case "$host" in
