--- gst-ffmpeg.orig/gst-libs/ext/ffmpeg/libavformat/rtp.c	2007-07-31 13:16:43.000000000 +0700
+++ gst-ffmpeg/gst-libs/ext/ffmpeg/libavformat/rtp.c	2007-07-31 13:27:32.036401201 +0700
@@ -30,6 +30,8 @@
 
 //#define DEBUG
 
+#define BE_16(x)  ((((uint8_t*)(x))[0] << 8) | ((uint8_t*)(x))[1])
+
 /* from http://www.iana.org/assignments/rtp-parameters last updated 05 January 2005 */
 /* payload types >= 96 are dynamic;
  * payload types between 72 and 76 are reserved for RTCP conflict avoidance;
--- gst-ffmpeg.orig/gst-libs/ext/ffmpeg/libavformat/rtp_h264.c	2007-07-31 13:16:43.000000000 +0700
+++ gst-ffmpeg/gst-libs/ext/ffmpeg/libavformat/rtp_h264.c	2007-07-31 13:27:32.036904445 +0700
@@ -49,6 +49,8 @@
 #include "base64.h"
 #include "avstring.h"
 
+#define BE_16(x)  ((((uint8_t*)(x))[0] << 8) | ((uint8_t*)(x))[1])
+
 /**
     RTP/H264 specific private data.
 */
