--- gst-ffmpeg-0.10.4.orig/ext/libpostproc/gstpostproc.c	2008-05-19 15:58:10.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/libpostproc/gstpostproc.c	2008-07-22 21:01:35.708344947 -0400
@@ -32,8 +32,8 @@
 #include <avcodec.h>
 #include <postprocess.h>
 #else
-#include <ffmpeg/avcodec.h>
-#include <postproc/postprocess.h>
+#include <libavcodec/avcodec.h>
+#include <libpostproc/postprocess.h>
 #endif
 
 
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpeg.c	2008-05-19 15:58:09.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpeg.c	2008-07-22 20:52:08.986360151 -0400
@@ -31,8 +31,8 @@
 #include <avcodec.h>
 #include <avformat.h>
 #else
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include "gstffmpeg.h"
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegdeinterlace.c	2007-07-12 07:13:13.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegdeinterlace.c	2008-07-22 20:59:00.369052374 -0400
@@ -27,7 +27,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #  include <avcodec.h>
 #else
-#  include <ffmpeg/avcodec.h>
+#  include <libavcodec/avcodec.h>
 #endif
 
 #include <gst/gst.h>
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegcodecmap.h	2007-01-09 09:59:34.000000000 -0500
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegcodecmap.h	2008-07-22 20:53:26.578520860 -0400
@@ -23,7 +23,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avcodec.h>
 #else
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 #include <gst/gst.h>
 
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegdec.c	2008-05-19 15:58:09.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegdec.c	2008-07-22 20:59:00.342528770 -0400
@@ -27,7 +27,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avcodec.h>
 #else
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 
 #include <gst/gst.h>
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegaudioresample.c	2008-05-08 07:15:52.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegaudioresample.c	2008-07-22 20:59:00.407604824 -0400
@@ -27,7 +27,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avcodec.h>
 #else
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 
 #include <gst/gst.h>
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpeg.h	2008-05-19 15:58:09.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpeg.h	2008-07-22 20:52:51.706376501 -0400
@@ -28,8 +28,8 @@
 #include <avcodec.h>
 #include <avformat.h>
 #else
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include <gst/gst.h>
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegmux.c	2008-05-19 15:58:10.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegmux.c	2008-07-22 20:59:00.400123429 -0400
@@ -25,7 +25,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avformat.h>
 #else
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include <gst/gst.h>
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegcodecmap.c	2008-05-19 15:58:09.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegcodecmap.c	2008-07-22 21:00:07.188741741 -0400
@@ -27,8 +27,8 @@
 #include <avcodec.h>
 #include <libswscale/swscale.h>
 #else
-#include <ffmpeg/swscale.h>
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
+#include <libswscale/swscale.h>
 #endif
 #include <string.h>
 
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegdemux.c	2008-05-19 15:58:09.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegdemux.c	2008-07-22 20:59:00.361503380 -0400
@@ -30,7 +30,7 @@
 #include <avi.h>
 #endif
 #else
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 #ifdef HAVE_AVI_H
 #include <ffmpeg/avi.h>
 #endif
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegprotocol.c	2007-11-03 12:14:53.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegprotocol.c	2008-07-22 20:59:00.350655096 -0400
@@ -26,7 +26,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avformat.h>
 #else
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include <gst/gst.h>
--- gst-ffmpeg-0.10.4.orig/ext/ffmpeg/gstffmpegenc.c	2008-05-19 15:58:09.000000000 -0400
+++ gst-ffmpeg-0.10.4/ext/ffmpeg/gstffmpegenc.c	2008-07-22 20:59:00.390588663 -0400
@@ -31,7 +31,7 @@
 #ifdef HAVE_FFMPEG_UNINSTALLED
 #include <avcodec.h>
 #else
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 
 #include <gst/gst.h>
