--- gst-plugins-bad-0.10.8/ext/gsm/gstgsmdec.c.orig	2008-09-02 18:32:47.981202000 +0800
+++ gst-plugins-bad-0.10.8/ext/gsm/gstgsmdec.c	2008-09-02 18:35:25.428134000 +0800
@@ -206,7 +206,7 @@
 
   /* Setting up src caps based on the input sample rate. */
   srccaps = gst_caps_new_simple ("audio/x-raw-int",
-      "endianness", G_TYPE_INT, BYTE_ORDER,
+      "endianness", G_TYPE_INT, G_BIG_ENDIAN,
       "signed", G_TYPE_BOOLEAN, TRUE,
       "width", G_TYPE_INT, 16,
       "depth", G_TYPE_INT, 16,
