--- gst-plugins-bad-0.10.5.orig/gst/vmnc/vmncdec.c	Wed Oct 17 23:48:03 2007
+++ gst-plugins-bad-0.10.5/gst/vmnc/vmncdec.c	Wed Oct 17 23:48:20 2007
@@ -993,7 +993,7 @@
 
   if (res < 0) {
     ret = GST_FLOW_ERROR;
-    GST_ELEMENT_ERROR (dec, STREAM, DECODE, NULL, ("Couldn't decode packet"));
+    GST_ELEMENT_ERROR (dec, STREAM, DECODE, ("Couldn't decode packet"), (NULL));
   } else {
     GST_DEBUG_OBJECT (dec, "read %d bytes of %d", res, len);
     /* inbuf may be NULL; that's ok */
