--- gst-plugins-bad-0.10.9/ext/ladspa/search.c-orig	2009-01-08 01:34:23.604052000 -0600
+++ gst-plugins-bad-0.10.9/ext/ladspa/search.c	2009-01-08 01:35:04.515439000 -0600
@@ -98,13 +98,15 @@ LADSPAPluginSearch (LADSPAPluginSearchCa
   const char *pcEnd;
   const char *pcLADSPAPath;
   const char *pcStart;
+  char *search_path;
 
   /* thomasvs: I'm sorry, but I'm going to add glib stuff here.
    * I'm appending logical values for LADSPA_PATH here
    */
 
+  search_path = getenv ("LADSPA_PATH");
   pcLADSPAPath = g_strdup_printf ("%s:/usr/lib/ladspa:/usr/local/lib/ladspa",
-      getenv ("LADSPA_PATH"));
+      search_path ? search_path : "(null)");
 
   if (!pcLADSPAPath) {
 /*    fprintf(stderr, */
