--- gthumb-2.7.1/src/Makefile.am.orig	2006-01-03 17:11:02.459035000 +0000
+++ gthumb-2.7.1/src/Makefile.am	2006-01-03 17:12:10.930257000 +0000
@@ -143,7 +143,7 @@
 	$(RENDER_LIBS)				\
 	$(LIBGPHOTO_LIBS)			\
 	$(XTEST_LIBS)				\
-	$(top_builddir)/libgthumb/libgthumb.la	\
+	$(top_builddir)/libgthumb/libgthumb.la	-lX11 \
 	$(top_builddir)/src/jpegutils/libgthumb-jpegutils.la
 
 moduledir = $(libdir)/gthumb/modules
