--- gthumb-2.9.2/src/Makefile.am.orig	2006-01-03 17:11:02.459035000 +0000
+++ gthumb-2.9.2/src/Makefile.am	2006-01-03 17:12:10.930257000 +0000
@@ -138,7 +138,7 @@
 
 gthumb_LDADD = 					\
 	$(top_builddir)/libgthumb/jpegutils/libgthumb-jpegutils.la	\
-	$(top_builddir)/libgthumb/libgthumb.la	\
+	$(top_builddir)/libgthumb/libgthumb.la	-lX11 \
 	$(EXIF_LIBS)				\
 	$(IPTCDATA_LIBS)			\
 	$(GTHUMB_LIBS)				\
