--- gtk-vnc-0.2.0/src/coroutine.c.orig	2007-09-14 05:11:29.000000000 +0800
+++ gtk-vnc-0.2.0/src/coroutine.c	2007-10-25 23:40:24.470121000 +0800
@@ -96,7 +96,7 @@
 	return coroutine_swap(coroutine_self(), to, arg);
 }
 
-void *yield(void *arg)
+void *gvnc_yield(void *arg)
 {
 	return yieldto(coroutine_self()->caller, arg);
 }
--- gtk-vnc-0.2.0/src/coroutine.h.orig	2007-10-25 23:39:15.282475000 +0800
+++ gtk-vnc-0.2.0/src/coroutine.h	2007-10-25 23:40:24.472431000 +0800
@@ -39,7 +39,7 @@
 
 void *yieldto(struct coroutine *to, void *arg);
 
-void *yield(void *arg);
+void *gvnc_yield(void *arg);
 
 #endif
 /*
--- gtk-vnc-0.2.0/src/gvnc.c.orig	2007-10-25 23:39:30.280365000 +0800
+++ gtk-vnc-0.2.0/src/gvnc.c	2007-10-25 23:40:24.477208000 +0800
@@ -144,7 +144,7 @@
 	GIOCondition *ret;
 
 	g_io_add_watch(channel, cond | G_IO_HUP | G_IO_ERR | G_IO_NVAL, g_io_wait_helper, coroutine_self());
-	ret = yield(NULL);
+	ret = gvnc_yield(NULL);
 
 	return *ret;
 }
@@ -161,7 +161,7 @@
 	id = g_io_add_watch(channel, cond | G_IO_HUP | G_IO_ERR | G_IO_NVAL, g_io_wait_helper, wait->context);
 
 	wait->waiting = TRUE;
-	ret = yield(NULL);
+	ret = gvnc_yield(NULL);
 	wait->waiting = FALSE;
 
 	if (ret == NULL) {
@@ -240,7 +240,7 @@
 
 	g_source_attach(src, NULL);
 	g_source_set_callback(src, g_condition_wait_helper, coroutine_self(), NULL);
-	yield(NULL);
+	gvnc_yield(NULL);
 	return TRUE;
 }
 
--- gtk-vnc-0.2.0/src/libgtk-vnc_sym.version.orig	2007-09-14 05:11:29.000000000 +0800
+++ gtk-vnc-0.2.0/src/libgtk-vnc_sym.version	2007-10-25 23:40:24.484391000 +0800
@@ -57,7 +57,7 @@
     coroutine_swap;
     coroutine_self;
     yieldto;
-    yield;
+    gvnc_yield;
 
   local:
       *;
