--- gtk-vnc-0.3.2/src/gvnc.c.orig	2007-12-31 12:48:48.000000000 +0800
+++ gtk-vnc-0.3.2/src/gvnc.c	2008-01-10 19:30:20.146338000 +0800
@@ -1152,21 +1152,15 @@
 	if (gvnc->zrle_pi_bits == 0) {
 		gvnc->zrle_pi = gvnc_read_u8(gvnc);
 		gvnc->zrle_pi_bits = 8;
-	}
-
-	switch (palette_size) {
-	case 2:
+	} else if ( palette_size == 2) {
 		pi = (gvnc->zrle_pi >> (gvnc->zrle_pi_bits - 1)) & 1;
 		gvnc->zrle_pi_bits -= 1;
-		break;
-	case 3 ... 4:
+	} else if ((palette_size == 3) || (palette_size == 4)) {
 		pi = (gvnc->zrle_pi >> (gvnc->zrle_pi_bits - 2)) & 3;
 		gvnc->zrle_pi_bits -= 2;
-		break;
-	case 5 ... 16:
+	} else if ((palette_size >=5) && (palette_size <=16)){
 		pi = (gvnc->zrle_pi >> (gvnc->zrle_pi_bits - 4)) & 15;
 		gvnc->zrle_pi_bits -= 4;
-		break;
 	}
 
 	return pi;
@@ -1263,30 +1257,28 @@
 	uint8_t subencoding = gvnc_read_u8(gvnc);
 	uint8_t pixel[4];
 
-	switch (subencoding) {
-	case 0: /* Raw pixel data */
+	if (subencoding == 0 ) {
+		/* Raw pixel data */
 		gvnc_zrle_update_tile_blit(gvnc, x, y, width, height);
-		break;
-	case 1: /* Solid tile of a single color */
+	} else if (subencoding == 1) {
+		/* Solid tile of a single color */
 		gvnc_read_cpixel(gvnc, pixel);
 		gvnc_fill(gvnc, pixel, x, y, width, height);
-		break;
-	case 2 ... 16: /* Packed palette types */
+	} else if ((subencoding >= 2) && (subencoding <= 16)) {
+		/* Packed palette types */
 		gvnc_zrle_update_tile_palette(gvnc, subencoding,
 					      x, y, width, height);
-		break;
-	case 128: /* Plain RLE */
+	} else if ((subencoding >= 17) && (subencoding <= 127)) {
+		/* FIXME raise error? */
+	} else if (subencoding == 128) {
+		/* Plain RLE */
 		gvnc_zrle_update_tile_rle(gvnc, x, y, width, height);
-		break;
-	case 130 ... 255: /* Palette RLE */
+	} else if (subencoding == 129) { 
+	
+	} else if ((subencoding >= 130) && (subencoding <= 255)) {
+		/* Palette RLE */
 		gvnc_zrle_update_tile_prle(gvnc, subencoding - 128,
 					   x, y, width, height);
-		break;
-	case 129:
-	case 17 ... 127:
-	default:
-		/* FIXME raise error? */
-		break;
 	}
 }
 
