--- gtk-vnc-0.3.5/src/Makefile.am.orig	2008-04-02 22:21:57.000000000 +0800
+++ gtk-vnc-0.3.5/src/Makefile.am	2008-04-22 14:28:06.018282000 +0800
@@ -4,12 +4,12 @@
 lib_LTLIBRARIES = libgtk-vnc-1.0.la
 
 libgtk_vnc_1_0_la_LIBADD = @GTK_LIBS@ @GTKGLEXT_LIBS@ @GNUTLS_LIBS@ \
-			   @GTHREAD_LIBS@
+			   @GTHREAD_LIBS@ @AST_LIBS@
 libgtk_vnc_1_0_la_CFLAGS = @GTK_CFLAGS@ @GTKGLEXT_CFLAGS@ @GNUTLS_CFLAGS@ \
-			   @GTHREAD_CFLAGS@ @WARNING_CFLAGS@ \
+			   @GTHREAD_CFLAGS@ @WARNING_CFLAGS@ @AST_CFLAGS@ \
 			   -DSYSCONFDIR=\""$(sysconfdir)"\" \
                            -DG_LOG_DOMAIN=\"gtk-vnc\"
-libgtk_vnc_1_0_la_LDFLAGS = -Wl,--version-script=$(srcdir)/libgtk-vnc_sym.version \
+libgtk_vnc_1_0_la_LDFLAGS = -Wl, -export-symbols-regex 'vnc_display_' \
                             -version-info 0:1:0
 
 gtk_vnc_includedir = $(includedir)/gtk-vnc-1.0/
--- gtk-vnc-0.3.5/src/vncdisplay.c.orig	2008-04-07 06:35:22.000000000 +0800
+++ gtk-vnc-0.3.5/src/vncdisplay.c	2008-04-22 14:26:57.487518000 +0800
@@ -25,6 +25,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <pwd.h>
+#include <endian.h>
 
 #if WITH_GTKGLEXT
 #include <gtk/gtkgl.h>
--- gtk-vnc-0.3.5/configure.ac.orig	2008-04-22 14:29:37.134794000 +0800
+++ gtk-vnc-0.3.5/configure.ac	2008-04-22 14:26:57.485668000 +0800
@@ -168,6 +168,24 @@
 AC_DEFINE_UNQUOTED(WITH_UCONTEXT,[$WITH_UCONTEXT], [Whether to use ucontext coroutine impl])
 AM_CONDITIONAL(WITH_UCONTEXT, [test "$WITH_UCONTEXT" != "0"])
 
+case $host in
+  *-*-solaris* | *-*-sunos*)
+       AC_CHECK_LIB(ast, _ast_printf,
+      [AST_LIBS="-last"],
+      [
+        echo "Error! You need to have libast around."
+        exit -1
+      ])
+    AC_CHECK_HEADER(ast/endian.h,,
+      [AST_CFLAGS="-I/usr/include/ast"],
+      [
+        echo "Error! You need to have ast/endian.h around."
+        exit -1
+      ])
+       AC_SUBST(AST_CFLAGS)
+       AC_SUBST(AST_LIBS) ;;
+esac
+
 if test "$WITH_PYTHON" = "yes"; then
   PKG_CHECK_MODULES(PYGTK, pygtk-2.0 >= $PYGTK_REQUIRED)
   AC_SUBST(PYGTK_CFLAGS)
