--- trunk/gtk/src/toolbar.ccg	2007/07/09 09:57:32	868
+++ trunk/gtk/src/toolbar.ccg	2007/07/12 12:41:04	869
@@ -92,5 +92,13 @@
   gtk_toolbar_set_drop_highlight_item(gobj(), 0, 0);
 }
 
+#ifndef GTKMM_DISABLE_DEPRECATED
+Tooltips* Toolbar::get_tooltips_object() const
+{
+  //Note that the _tooltips field is deprecated and broken since GTK+ 2.12:
+  return Glib::wrap((GtkTooltips*)gobj()->_tooltips);
+}
+#endif //GTKMM_DISABLE_DEPRECATED
+
 } // namespace Gtk
 
--- trunk/gtk/src/toolbar.hg	2007/07/09 09:57:32	868
+++ trunk/gtk/src/toolbar.hg	2007/07/12 12:41:04	869
@@ -118,9 +118,11 @@
   _IGNORE_SIGNAL("move_focus")
   _IGNORE_SIGNAL("focus_home_or_end")
 
-  //This is called get_tooltips_object() to avoid a clash with get_tooltips(), which just says whether they are enabled.
-  
-  _MEMBER_GET(tooltips_object, tooltips, Tooltips*, GtkTooltips*)
+  #ifndef GTKMM_DISABLE_DEPRECATED
+  //This was called get_tooltips_object() to avoid a clash with get_tooltips(), which just says whether they are enabled.
+  ///@deprecated Use the Gtk::Tooltip API instead. 
+  Tooltips* get_tooltips_object() const;
+  #endif //GTKMM_DISABLE_DEPRECATED
   
   //This is probably the same as Container::children().size(), which is deprecated anyway?
   //_MEMBER_GET(num_children, num_children, int, gint)
--- gtkmm-2.10.11/gtk/gtkmm/toolbar.cc	2007/07/09 09:57:32	868
+++ gtkmm-2.10.11/gtk/gtkmm/toolbar.cc	2007/07/12 12:41:04	869
@@ -98,6 +98,14 @@
   gtk_toolbar_set_drop_highlight_item(gobj(), 0, 0);
 }
 
+#ifndef GTKMM_DISABLE_DEPRECATED
+Tooltips* Toolbar::get_tooltips_object() const
+{
+  //Note that the _tooltips field is deprecated and broken since GTK+ 2.12:
+  return Glib::wrap((GtkTooltips*)gobj()->_tooltips);
+}
+#endif //GTKMM_DISABLE_DEPRECATED
+
 } // namespace Gtk
 
 
@@ -564,11 +572,6 @@
 gtk_toolbar_set_drop_highlight_item(gobj(), (tool_item).gobj(), index); 
 }
 
- Tooltips* Toolbar::get_tooltips_object() const
-{
-  return Glib::wrap(gobj()->tooltips);
-}
- 
 
 Glib::SignalProxy1< void,Orientation > Toolbar::signal_orientation_changed()
 {
--- gtkmm-2.10.11/gtk/gtkmm/toolbar.h	2007/07/09 09:57:32	868
+++ gtkmm-2.10.11/gtk/gtkmm/toolbar.h	2007/07/12 12:41:04	869
@@ -324,9 +324,11 @@
   Glib::SignalProxy3< bool,int,int,int > signal_popup_context_menu();
 
   
-  //This is called get_tooltips_object() to avoid a clash with get_tooltips(), which just says whether they are enabled.
-  
-   Tooltips* get_tooltips_object() const;
+  #ifndef GTKMM_DISABLE_DEPRECATED
+  //This was called get_tooltips_object() to avoid a clash with get_tooltips(), which just says whether they are enabled.
+  ///@deprecated Use the Gtk::Tooltip API instead.
+  Tooltips* get_tooltips_object() const;
+  #endif //GTKMM_DISABLE_DEPRECATED
    
   //This is probably the same as Container::children().size(), which is deprecated anyway?
   //_MEMBER_GET(num_children, num_children, int, gint)
