--- gvfs-0.0.1/test/benchmark-common.c.orig	2007-11-02 14:29:52.000000000 +0000
+++ gvfs-0.0.1/test/benchmark-common.c	2007-11-08 14:28:10.167303573 +0000
@@ -184,7 +184,7 @@
 benchmark_start_wallclock_timer (gint n_seconds)
 {
   benchmark_is_running = TRUE;
-  signal (SIGALRM, (sig_t) benchmark_timeout);
+  signal (SIGALRM, benchmark_timeout);
   alarm (n_seconds);
 }
 
@@ -198,7 +198,7 @@
   memset (&itv, 0, sizeof (itv));
   itv.it_value.tv_sec = n_seconds;
 
-  signal (SIGPROF, (sig_t) benchmark_timeout);
+  signal (SIGPROF, benchmark_timeout);
   setitimer (ITIMER_PROF, &itv, NULL);
 }
 
--- gvfs-0.0.1/daemon/gvfsdaemon.c.orig	2007-11-08 14:09:23.410434175 +0000
+++ gvfs-0.0.1/daemon/gvfsdaemon.c	2007-11-08 14:11:43.277321372 +0000
@@ -25,6 +25,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/errno.h>
+#include <errno.h>
 #include <sys/un.h>
 #include <unistd.h>
 #include <string.h>
--- gvfs-0.0.1/daemon/gvfsjobmove.c.orig	2007-11-02 14:29:52.000000000 +0000
+++ gvfs-0.0.1/daemon/gvfsjobmove.c	2007-11-08 14:21:35.048016093 +0000
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <strings.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
