--- gvfs-0.0.1/daemon/gvfswritechannel.c.orig	2007-11-09 11:32:59.269688917 +0000
+++ gvfs-0.0.1/daemon/gvfswritechannel.c	2007-11-09 11:33:13.513829032 +0000
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <strings.h>
 #include <unistd.h>
 #include <errno.h>
 #include <sys/types.h>
--- gvfs-0.0.1/daemon/gvfsjobcopy.c.orig	2007-11-09 11:34:32.066438862 +0000
+++ gvfs-0.0.1/daemon/gvfsjobcopy.c	2007-11-09 11:34:45.325267669 +0000
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <strings.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
--- gvfs-0.0.1/daemon/gvfschannel.c.orig	2007-11-09 11:35:48.488607217 +0000
+++ gvfs-0.0.1/daemon/gvfschannel.c	2007-11-09 11:36:02.602036501 +0000
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <string.h>
 #include <unistd.h>
 #include <errno.h>
 #include <sys/types.h>
--- gvfs-0.0.1/daemon/gvfsmonitor.c.orig	2007-11-09 11:33:29.396863882 +0000
+++ gvfs-0.0.1/daemon/gvfsmonitor.c	2007-11-09 11:33:40.621846027 +0000
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <strings.h>
 #include <unistd.h>
 #include <errno.h>
 #include <sys/types.h>
--- gvfs-0.0.1/daemon/gvfsdaemonutils.c.orig	2007-11-09 11:33:55.141785675 +0000
+++ gvfs-0.0.1/daemon/gvfsdaemonutils.c	2007-11-09 11:36:21.079881359 +0000
@@ -22,6 +22,8 @@
 
 #include <config.h>
 
+#include <string.h>
+#include <strings.h>
 #include <unistd.h>
 #include <errno.h>
 #include <stdlib.h>
--- gvfs-0.0.1/common/gsysutils.c.orig	2007-11-09 11:32:03.287440400 +0000
+++ gvfs-0.0.1/common/gsysutils.c	2007-11-09 11:35:19.181628127 +0000
@@ -25,6 +25,8 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <stdio.h>
+#include <string.h>
+#include <strings.h>
 #include <errno.h>
 #include <unistd.h>
 #include <gio/gioerror.h>
