diff -ur hal-0.5.8.1-orig/configure.in hal-0.5.8.1/configure.in
--- hal-0.5.8.1-orig/configure.in	2006-09-20 03:24:05.000000000 +0700
+++ hal-0.5.8.1/configure.in	2007-04-21 15:34:26.381651514 +0700
@@ -243,9 +243,12 @@
 dnl Check for libpci
 AC_CHECK_HEADERS(pci/pci.h, [
 	AC_CHECK_LIB(pci, pci_init, [
-		USE_LIBPCI=yes AM_CONDITIONAL(HAVE_LIBPCI,true)], [
-		USE_LIBPCI=no AM_CONDITIONAL(HAVE_LIBPCI,false)])], [
-	USE_LIBPCI=no AM_CONDITIONAL(HAVE_LIBPCI,false)])
+		USE_LIBPCI=yes
+		AM_CONDITIONAL(HAVE_LIBPCI,true)], [
+		USE_LIBPCI=no
+		AM_CONDITIONAL(HAVE_LIBPCI,false)])], [
+		USE_LIBPCI=no
+		AM_CONDITIONAL(HAVE_LIBPCI,false)])
 
 dnl see if we should build macbookpro utils
 AC_ARG_WITH(macbookpro,     [  --with-macbookpro       Whether to build Macbook Pro utils (auto)])
