--- jack-audio-connection-kit-0.103.0/configure.ac.orig	2007-08-12 20:11:42.807628876 +0700
+++ jack-audio-connection-kit-0.103.0/configure.ac	2007-08-12 20:12:24.971267036 +0700
@@ -209,9 +209,8 @@
 AM_CONDITIONAL(USE_POSIX_SHM, $USE_POSIX_SHM)
 
 JACK_CORE_CFLAGS="-I\$(top_srcdir)/config -I\$(top_srcdir) \
--I\$(top_srcdir) -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -Wall -g"
-JACK_OPT_CFLAGS="$JACK_CORE_CFLAGS -march=pentium2 -mcpu=pentium4 -O3 \
--ffast-math -funroll-loops -fprefetch-loop-arrays"
+-I\$(top_srcdir) -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS"
+JACK_OPT_CFLAGS="$JACK_CORE_CFLAGS"
 JACK_CFLAGS="$JACK_CORE_CFLAGS $CFLAGS"
 
 dnl
