--- jokosher-0.9/bin/jokosher.desktop-orig	2008-02-07 20:02:47.652728000 -0600
+++ jokosher-0.9/bin/jokosher.desktop	2008-02-07 20:02:57.298473000 -0600
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Version=0.9
 Encoding=UTF-8
 Name=Jokosher Audio Editor
 GenericName=Jokosher audio editor
@@ -9,5 +8,5 @@ Icon=/usr/share/pixmaps/jokosher-icon.pn
 Terminal=false
 Type=Application
 MimeType=application/x-jokosher;
-Categories=AudioVideoEditing;GTK;Recorder
+Categories=AudioVideoEditing;GTK;Recorder;
 GenericName[en_CA]=Jokosher audio editor
