--- kdebase-3.5.8/startkde.orig	2008-02-01 00:28:01.909106210 +0530
+++ kdebase-3.5.8/startkde	2008-02-01 00:28:48.643270864 +0530
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/bash
 #
 #  DEFAULT KDE STARTUP SCRIPT ( KDE-3.5.8 )
 #
@@ -7,6 +7,7 @@
 # because we still need to do some cleanup.
 trap 'echo GOT SIGHUP' HUP
 
+PATH=${PATH}:/usr/sfw/bin:/usr/openwin/bin:/usr/X11/bin
 # Check if a KDE session already is running
 if kcheckrunning >/dev/null 2>&1; then
 	echo "KDE seems to be already running on this display."
@@ -377,7 +378,7 @@
 fi
 
 # wait if there's any crashhandler shown
-while dcop | grep -q ^drkonqi- ; do
+while dcop | ggrep -q ^drkonqi- ; do
     sleep 5
 done
 
