--- kdelibs-3.5.8/Makefile.in.orig	2008-01-12 12:21:31.246363000 +0530
+++ kdelibs-3.5.8/Makefile.in	2008-01-12 12:21:42.236583000 +0530
@@ -1050,11 +1050,11 @@
 uninstall-apidox:
 	@if test "$(subdir)" != "."; then \
 		if test -d $(DESTDIR)$(kde_htmldir)/en/$(PACKAGE)-apidocs/$(subdir); then \
-			rm -rfv $(DESTDIR)$(kde_htmldir)/en/$(PACKAGE)-apidocs/$(subdir); \
+			rm -rf $(DESTDIR)$(kde_htmldir)/en/$(PACKAGE)-apidocs/$(subdir); \
 		fi \
 	else \
 		if test -d $(DESTDIR)$(kde_htmldir)/en/$(PACKAGE)-apidocs; then \
-			rm -rfv $(DESTDIR)$(kde_htmldir)/en/$(PACKAGE)-apidocs; \
+			rm -rf $(DESTDIR)$(kde_htmldir)/en/$(PACKAGE)-apidocs; \
 		fi \
 	fi
 
