--- kdenetwork-3.5.8/kopete/libkopete/avdevice/videocontrol.h.orig	2007-10-08 15:19:07.000000000 +0530
+++ kdenetwork-3.5.8/kopete/libkopete/avdevice/videocontrol.h	2008-01-25 11:28:04.996209420 +0530
@@ -20,6 +20,17 @@
 #ifndef KOPETE_AVVIDEOCONTROL_H
 #define KOPETE_AVVIDEOCONTROL_H
 
+#include <qstring.h>
+#if defined(Q_OS_SOLARIS)
+
+#include <sys/types.h>
+#define __u64 uint64_t
+#define __s64 int64_t
+#define __u32 uint32_t
+#define __s32 int32_t
+
+#else
+
 #include <asm/types.h>
 #undef __STRICT_ANSI__
 #ifndef __u64 //required by videodev.h
@@ -30,7 +41,8 @@
 #define __s64 long long
 #endif // __s64
 
-#include <qstring.h>
+#endif // sun
+
 #include <kdebug.h>
 #include <qvaluevector.h>
 #include "kopete_export.h"
