--- kino-1.1.1/src/endian_types.h.orig	2007-09-04 15:25:29.641356774 +0700
+++ kino-1.1.1/src/endian_types.h	2007-09-04 15:25:56.583900680 +0700
@@ -39,6 +39,16 @@
 #include <sys/types.h>
 #include <byteswap.h>
 
+#ifdef HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif
+#ifndef HAVE_U_INT_TYPES
+typedef uint8_t u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+typedef uint64_t u_int64_t;
+#endif
+
 static inline int8_t bswap(const int8_t& x)
 {
 	return x;
