--- koffice-1.4.90/lib/kross/python/cxx/cxxsupport.cxx.orig	Sat Feb 18 21:42:02 2006
+++ koffice-1.4.90/lib/kross/python/cxx/cxxsupport.cxx	Sat Feb 18 21:42:14 2006
@@ -3,6 +3,7 @@
 // All rights reserved. See Legal.htm for full text and disclaimer.
 //---------------------------------------------------------------------------//
 
+#include <wchar.h>
 #include "Objects.hxx"
 namespace Py {
 
--- koffice-1.4.90/lib/kross/python/pythonextension.cpp.orig	Sat Feb 18 21:47:50 2006
+++ koffice-1.4.90/lib/kross/python/pythonextension.cpp	Sat Feb 18 21:47:11 2006
@@ -17,6 +17,7 @@
  * Boston, MA 02110-1301, USA.
  ***************************************************************************/
 
+#include <wchar.h>
 #include "pythonextension.h"
 #include "pythonobject.h"
 
