diff -ur lame-3.97-orig/libmp3lame/gain_analysis.c lame-3.97/libmp3lame/gain_analysis.c
--- lame-3.97-orig/libmp3lame/gain_analysis.c	2005-11-29 04:45:11.000000000 +0700
+++ lame-3.97/libmp3lame/gain_analysis.c	2006-05-08 21:24:48.534418306 +0700
@@ -248,7 +248,7 @@
 
 /* returns GAIN_ANALYSIS_OK if successful, GAIN_ANALYSIS_ERROR if not */
 
-static __inline double fsqr(const double d)
+static double fsqr(const double d)
 {  return d*d;
 }
 
