--- lame-3.97/frontend/brhist.c.orig	2007-12-02 14:15:47.477902000 +1300
+++ lame-3.97/frontend/brhist.c	2007-12-02 14:15:09.771698000 +1300
@@ -429,7 +429,7 @@
 void
 brhist_disp(const lame_global_flags * gf)
 {
-    int     i, lines = 0;
+    int     i, lame_lines = 0;
     int     br_hist[BRHIST_WIDTH]; /* how often a frame size was used */
     int     br_sm_hist[BRHIST_WIDTH][4]; /* how often a special frame size/stereo mode commbination was used */
     int     st_mode[4];
@@ -456,7 +456,7 @@
         if (most_often < br_hist[i])
             most_often = br_hist[i];
         if (br_hist[i])
-            ++lines;
+            ++lame_lines;
     }
 
     for (i = 0; i < BRHIST_WIDTH; i++) {
