diff -ur lame-3.97-orig/frontend/brhist.c lame-3.97/frontend/brhist.c
--- lame-3.97-orig/frontend/brhist.c	2005-11-29 04:45:08.000000000 +0700
+++ lame-3.97/frontend/brhist.c	2006-05-08 21:22:15.470447384 +0700
@@ -422,7 +422,7 @@
 void
 brhist_disp(const lame_global_flags * gf)
 {
-    int     i, lines = 0;
+    int     i, xlines = 0;
     int     br_hist[BRHIST_WIDTH]; /* how often a frame size was used */
     int     br_sm_hist[BRHIST_WIDTH][4]; /* how often a special frame size/stereo mode commbination was used */
     int     st_mode[4];
@@ -449,13 +449,13 @@
         if (most_often < br_hist[i])
             most_often = br_hist[i];
         if (br_hist[i])
-            ++lines;
+            ++xlines;
     }
 
     for (i = 0; i < BRHIST_WIDTH; i++) {
         int     show = br_hist[i];
 #ifdef RH_HIST
-        show = show && (lines > 1);
+        show = show && (xlines > 1);
 #endif
         if (show || (i >= brhist.vbr_bitrate_min_index && i <= brhist.vbr_bitrate_max_index))
             brhist_disp_line(gf, i, br_hist[i], br_sm_hist[i][LR], most_often, frames);
