--- lame-3.97/libmp3lame/gain_analysis.h.orig	2007-11-16 22:38:38.275758000 +0000
+++ lame-3.97/libmp3lame/gain_analysis.h	2007-11-16 22:40:52.737188000 +0000
@@ -63,6 +63,7 @@
 #define RMS_WINDOW_TIME_DENOMINATOR 20L /* numerator / denominator = time slice size [s] */
 #define STEPS_per_dB      100.          /* Table entries per dB */
 #define MAX_dB            120.          /* Table entries for 0...MAX_dB (normal max. values are 70...80 dB) */
+#define STEPS_MAX         (100*120)
 
 #define MAX_ORDER               (BUTTER_ORDER > YULE_ORDER ? BUTTER_ORDER : YULE_ORDER)
 #define MAX_SAMPLES_PER_WINDOW  (size_t) ((MAX_SAMP_FREQ * RMS_WINDOW_TIME_NUMERATOR) / RMS_WINDOW_TIME_DENOMINATOR + 1)      /* max. Samples per Time slice */
@@ -90,8 +91,8 @@
     double           rsum;
     int              freqindex;
     int              first;
-    uint32_t  A [(size_t)(STEPS_per_dB * MAX_dB)];
-    uint32_t  B [(size_t)(STEPS_per_dB * MAX_dB)];
+    uint32_t  A [(size_t)(STEPS_MAX)];
+    uint32_t  B [(size_t)(STEPS_MAX)];
 
 } replaygain_t;
 
--- lame-3.97/libmp3lame/gain_analysis.c.orig	2007-11-16 22:41:17.784462000 +0000
+++ lame-3.97/libmp3lame/gain_analysis.c	2007-11-16 22:41:28.801606000 +0000
@@ -248,7 +248,7 @@
 
 /* returns GAIN_ANALYSIS_OK if successful, GAIN_ANALYSIS_ERROR if not */
 
-static __inline double fsqr(const double d)
+static double fsqr(const double d)
 {  return d*d;
 }
 
