--- libcanberra-0.8/configure.ac-orig	2008-08-28 13:10:31.032317000 -0500
+++ libcanberra-0.8/configure.ac	2008-08-28 13:14:24.156236000 -0500
@@ -19,7 +19,7 @@
 # License along with libcanberra. If not, see
 # <http://www.gnu.org/licenses/>.
 
-AC_PREREQ(2.62)
+AC_PREREQ(2.61)
 
 AC_INIT([libcanberra], 0.8, [mzyvopnaoreen (at) 0pointer (dot) de])
 AC_CONFIG_SRCDIR([src/common.c])
@@ -54,12 +54,6 @@ AC_GNU_SOURCE
 
 AC_PROG_CXX
 
-DESIRED_FLAGS="-Wall -W -Wextra -pedantic -pipe -Wno-long-long -Wvla -Wno-overlength-strings -Wconversion -Wundef -Wformat -Wlogical-op -Wpacked -Wformat-security -Wmissing-include-dirs -Wformat-nonliteral -Wold-style-definition -Wdeclaration-after-statement -Wfloat-equal -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wshadow -Wendif-labels -Wpointer-arith -Wcast-align -Wwrite-strings -Winline -Wno-unused-parameter -ffast-math"
-
-for flag in $DESIRED_FLAGS ; do
-  CC_CHECK_CFLAGS([$flag], [CFLAGS="$CFLAGS $flag"])
-done
-
 #### libtool stuff ####
 
 AC_LTDL_ENABLE_INSTALL
--- libcanberra-0.6/src/common.h-orig	2008-08-21 00:21:50.594788000 -0500
+++ libcanberra-0.6/src/common.h	2008-08-21 00:22:25.218467000 -0500
@@ -21,6 +21,10 @@
   <http://www.gnu.org/licenses/>.
 ***/
 
+#ifdef __sun
+#include <sys/varargs.h>
+#endif
+
 #include "canberra.h"
 #include "macro.h"
 #include "mutex.h"
