--- libcanberra-0.6/src/Makefile.am-orig	2008-08-14 13:06:45.705560000 -0500
+++ libcanberra-0.6/src/Makefile.am	2008-08-14 13:08:01.465327000 -0500
@@ -59,9 +59,7 @@ libcanberra_la_CFLAGS = \
 libcanberra_la_LIBADD = \
 	$(VORBIS_LIBS)
 libcanberra_la_LDFLAGS = \
-	-export-dynamic \
-	-version-info $(LIBCANBERRA_VERSION_INFO) \
-	-Wl,-version-script=$(srcdir)/map-file
+	-export-dynamic 
 
 plugin_LTLIBRARIES =
 
@@ -215,7 +213,7 @@ libcanberra_gtk_la_LIBADD = \
 	$(GTK_LIBS) \
 	libcanberra.la
 libcanberra_gtk_la_LDFLAGS = \
-	-export-dynamic -version-info $(LIBCANBERRA_GTK_VERSION_INFO)
+	-export-dynamic $(LIBCANBERRA_GTK_VERSION_INFO)
 
 libcanberra_gtk_module_la_SOURCES = \
 	canberra-gtk-module.c
--- libcanberra-0.6/src/common.h-orig	2008-08-14 13:00:23.334879000 -0500
+++ libcanberra-0.6/src/common.h	2008-08-14 13:06:24.572488000 -0500
@@ -21,6 +21,8 @@
   <http://www.gnu.org/licenses/>.
 ***/
 
+#include <varargs.h>
+
 #include "canberra.h"
 #include "macro.h"
 #include "mutex.h"
--- libcanberra-0.6/src/sound-theme-spec.c-orig	2008-08-14 13:18:10.752437000 -0500
+++ libcanberra-0.6/src/sound-theme-spec.c	2008-08-14 13:15:04.375888000 -0500
@@ -36,6 +36,28 @@
 #define DEFAULT_OUTPUT_PROFILE "stereo"
 #define N_THEME_DIR_MAX 8
 
+#ifndef strndup
+static char *strndup (const char *s, size_t n)
+{
+    size_t length;
+    char *ret;
+
+    if (!s)
+        return 0;
+
+    if (strlen(s)+1 < n+1)
+       length = strlen(s)+1;
+    else
+       length = n+1;
+
+    ret = malloc (length * sizeof (char));
+
+    memcpy (ret, s, length);
+    ret[length-1] = '\0';
+    return ret;
+}
+#endif
+
 typedef struct ca_data_dir ca_data_dir;
 
 struct ca_data_dir {
