--- libcanberra-0.8/src/gstreamer.c-orig	2008-08-30 00:54:34.477944000 -0500
+++ libcanberra-0.8/src/gstreamer.c	2008-08-30 01:03:11.619656000 -0500
@@ -230,7 +230,7 @@ static GstBusSyncReply bus_cb(GstBus *bu
             /* g_debug (gst_element_state_get_name (pending)); */
 
             if (pending == GST_STATE_NULL || pending == GST_STATE_VOID_PENDING)
-                err = CA_SUCCESS;
+                return GST_BUS_PASS;
             else
                 return GST_BUS_DROP;
             break;
