diff -ruN libcompizconfig-0.6.0.orig/configure.ac libcompizconfig-0.6.0/configure.ac
--- libcompizconfig-0.6.0.orig/configure.ac	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/configure.ac	2007-10-30 22:32:48.015034828 -0400
@@ -1,6 +1,6 @@
 AC_PREREQ(2.57)
 
-AC_INIT([libcompizconfig],esyscmd(. ./VERSION;echo -n $VERSION), [dev@opencompositing.org])
+AC_INIT([libcompizconfig],esyscmd(. ./VERSION;echo $VERSION | tr -d "\n"), [dev@opencompositing.org])
 
 AC_DEFINE_UNQUOTED(VERSION,"$VERSION",[Version])
 
@@ -17,6 +17,7 @@
 AC_PROG_LIBTOOL
 AC_HEADER_STDC
 AC_CHECK_HEADERS([stdlib.h sys/time.h unistd.h])
+AC_CHECK_FUNCS(va_copy __va_copy vasprintf asprintf strcasestr strsep)
 IT_PROG_INTLTOOL([0.35.0])
 AC_SUBST(ALL_LINGUAS)
 AM_GLIB_GNU_GETTEXT
diff -ruN libcompizconfig-0.6.0.orig/metadata/config libcompizconfig-0.6.0/metadata/config
--- libcompizconfig-0.6.0.orig/metadata/config	1969-12-31 19:00:00.000000000 -0500
+++ libcompizconfig-0.6.0/metadata/config	2007-10-30 21:46:31.236665156 -0400
@@ -0,0 +1,6 @@
+[gnome_session]
+backend = gconf
+profile = 
+integration = true
+plugin_list_autosort = true
+
diff -ruN libcompizconfig-0.6.0.orig/metadata/global.xml libcompizconfig-0.6.0/metadata/global.xml
--- libcompizconfig-0.6.0.orig/metadata/global.xml	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/metadata/global.xml	2007-10-30 21:46:31.237063304 -0400
@@ -4,22 +4,32 @@
 			<option name="active_plugins" type="list">
 				<type>string</type>
 				<default>
-					<value>dbus</value>
-					<value>place</value>
-					<value>move</value>
-					<value>resize</value>
-					<value>decoration</value>
-					<value>png</value>
-					<value>wobbly</value>
-					<value>cube</value>
-					<value>fade</value>
-					<value>minimize</value>
-					<value>rotate</value>
-					<value>scale</value>
-					<value>switcher</value>
-					<value>regex</value>
-					<value>workarounds</value>
-					<value>zoom</value>
+	                                <value>imgjpeg</value>
+                                        <value>svg</value>
+                                        <value>png</value>
+                                        <value>decoration</value>
+                                        <value>wobbly</value>
+                                        <value>fade</value>
+                                        <value>expo</value>
+                                        <value>minimize</value>
+                                        <value>cube</value>
+                                        <value>showdesktop</value>
+                                        <value>switcher</value>
+                                        <value>move</value>
+                                        <value>resize</value>
+                                        <value>place</value>
+                                        <value>rotate</value>
+                                        <value>zoom</value>
+                                        <value>scale</value>
+                                        <value>shift</value>
+                                        <value>dbus</value>
+                                        <value>water</value>
+                                        <value>snow</value>
+                                        <value>animation</value>
+                                        <value>firepaint</value>
+                                        <value>thumbnail</value>
+                                        <value>cubereflex</value>
+                                        <value>switcher</value>      
 				</default>
 			</option>
 			<group>
diff -ruN libcompizconfig-0.6.0.orig/metadata/Makefile.am libcompizconfig-0.6.0/metadata/Makefile.am
--- libcompizconfig-0.6.0.orig/metadata/Makefile.am	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/metadata/Makefile.am	2007-10-30 21:46:31.236369035 -0400
@@ -2,5 +2,10 @@
 meta_DATA =	       \
 	global.xml
 
+configdir = $(sysconfdir)/compizconfig
+config_DATA =	      \
+	config
+
 EXTRA_DIST =	       \
-	global.xml
\ No newline at end of file
+	global.xml     \
+	config
diff -ruN libcompizconfig-0.6.0.orig/src/bindings.c libcompizconfig-0.6.0/src/bindings.c
--- libcompizconfig-0.6.0.orig/src/bindings.c	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/src/bindings.c	2007-10-30 23:14:28.426135089 -0400
@@ -38,6 +38,7 @@
 #include <X11/Xlib.h>
 
 #include <ccs.h>
+#include "compat.h"
 
 #define CompAltMask        (1 << 16)
 #define CompMetaMask       (1 << 17)
diff -ruN libcompizconfig-0.6.0.orig/src/compat.c libcompizconfig-0.6.0/src/compat.c
--- libcompizconfig-0.6.0.orig/src/compat.c	1969-12-31 19:00:00.000000000 -0500
+++ libcompizconfig-0.6.0/src/compat.c	2007-10-30 23:10:05.872599481 -0400
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2007 Albert Lee <trisk@acm.jhu.edu>.
+ * 
+ * Copyright (c) 2004 Darren Tucker.
+ *
+ * Based originally on asprintf.c from OpenBSD:
+ * Copyright (c) 1997 Todd C. Miller <Todd.Miller@courtesan.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "config.h"
+#include "compat.h"
+
+#ifndef HAVE_VASPRINTF
+#include <errno.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#ifndef VA_COPY
+# ifdef HAVE_VA_COPY
+#  define VA_COPY(dest, src) va_copy(dest, src)
+# else
+#  ifdef HAVE___VA_COPY
+#   define VA_COPY(dest, src) __va_copy(dest, src)
+#  else
+#   define VA_COPY(dest, src) (dest) = (src)
+#  endif
+# endif
+#endif
+
+#define INIT_SZ	128
+
+static int vasprintf(char **str, const char *fmt, va_list ap);
+
+static int vasprintf(char **str, const char *fmt, va_list ap)
+{
+	int ret = -1;
+	va_list ap2;
+	char *string, *newstr;
+	size_t len;
+
+	VA_COPY(ap2, ap);
+	if ((string = malloc(INIT_SZ)) == NULL)
+		goto fail;
+
+	ret = vsnprintf(string, INIT_SZ, fmt, ap2);
+	if (ret >= 0 && ret < INIT_SZ) { /* succeeded with initial alloc */
+		*str = string;
+	} else if (ret == INT_MAX) { /* shouldn't happen */
+		goto fail;
+	} else {	/* bigger than initial, realloc allowing for nul */
+		len = (size_t)ret + 1;
+		if ((newstr = realloc(string, len)) == NULL) {
+			free(string);
+			goto fail;
+		} else {
+			va_end(ap2);
+			VA_COPY(ap2, ap);
+			ret = vsnprintf(newstr, len, fmt, ap2);
+			if (ret >= 0 && (size_t)ret < len) {
+				*str = newstr;
+			} else { /* failed with realloc'ed string, give up */
+				free(newstr);
+				goto fail;
+			}
+		}
+	}
+	va_end(ap2);
+	return (ret);
+
+fail:
+	*str = NULL;
+	errno = ENOMEM;
+	va_end(ap2);
+	return (-1);
+}
+#endif
+
+#ifndef HAVE_ASPRINTF
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+
+int asprintf(char **str, const char *fmt, ...)
+{
+	va_list ap;
+	int ret;
+	
+	*str = NULL;
+	va_start(ap, fmt);
+	ret = vasprintf(str, fmt, ap);
+	va_end(ap);
+
+	return ret;
+}
+#endif
+
+#ifndef HAVE_STRCASESTR
+#include <ctype.h>
+
+char *strcasestr (char *h, char *n)
+{
+	char *hp, *np = n, *match = 0;
+
+	if(!*np) {
+		return hp;
+	}
+
+	for (hp = h; *hp; hp++) {
+		if (toupper(*hp) == toupper(*np)) {
+			if (!match) {
+				match = hp;
+			}
+			if(!*++np) {
+				return match;
+			}
+		} else {
+			if (match) { 
+				match = 0;
+				np = n;
+			}
+		}
+	}
+
+	return NULL; 
+}
+#endif
+
+#ifndef HAVE_STRSEP
+#include <string.h>
+    	
+char *strsep(char **stringp, const char *delim)
+{
+	char *s = *stringp;
+	char *e;
+
+	if (!s)
+		return NULL;
+
+	e = strpbrk(s, delim);
+	if (e)
+		*e++ = '\0';
+
+	*stringp = e;
+	return s;
+}
+#endif
+
diff -ruN libcompizconfig-0.6.0.orig/src/compat.h libcompizconfig-0.6.0/src/compat.h
--- libcompizconfig-0.6.0.orig/src/compat.h	1969-12-31 19:00:00.000000000 -0500
+++ libcompizconfig-0.6.0/src/compat.h	2007-10-30 23:12:55.926225079 -0400
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2007 Albert Lee <trisk@acm.jhu.edu>.
+ * 
+ * Copyright (c) 2004 Darren Tucker.
+ *
+ * Based originally on asprintf.c from OpenBSD:
+ * Copyright (c) 1997 Todd C. Miller <Todd.Miller@courtesan.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef _COMPAT_H
+#include "config.h"
+
+#ifndef HAVE_ASPRINTF
+int asprintf(char **str, const char *fmt, ...);
+#endif
+
+#ifndef HAVE_STRCASESTR
+char *strcasestr (char *h, char *n);
+#endif
+
+#ifndef HAVE_STRSEP
+char *strsep(char **stringp, const char *delim);
+#endif
+#endif /* _COMPAT_H */
diff -ruN libcompizconfig-0.6.0.orig/src/compiz.c libcompizconfig-0.6.0/src/compiz.c
--- libcompizconfig-0.6.0.orig/src/compiz.c	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/src/compiz.c	2007-10-30 23:15:03.372232998 -0400
@@ -42,6 +42,7 @@
 #include <compiz.h>
 #include <ccs.h>
 #include "ccs-private.h"
+#include "compat.h"
 
 static xmlDoc * globalMetadata = NULL;
 
diff -ruN libcompizconfig-0.6.0.orig/src/config.c libcompizconfig-0.6.0/src/config.c
--- libcompizconfig-0.6.0.orig/src/config.c	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/src/config.c	2007-10-30 23:14:38.939817952 -0400
@@ -26,6 +26,7 @@
 #include <string.h>
 
 #include "ccs-private.h"
+#include "compat.h"
 
 #define SETTINGPATH "compiz/compizconfig"
 
diff -ruN libcompizconfig-0.6.0.orig/src/ini.c libcompizconfig-0.6.0/src/ini.c
--- libcompizconfig-0.6.0.orig/src/ini.c	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/src/ini.c	2007-10-30 23:14:31.239326381 -0400
@@ -26,6 +26,7 @@
 
 #include <ccs.h>
 #include "iniparser.h"
+#include "compat.h"
 
 /** 
  * Creates the parent directory for @fileName, recursively creating a directory
diff -ruN libcompizconfig-0.6.0.orig/src/main.c libcompizconfig-0.6.0/src/main.c
--- libcompizconfig-0.6.0.orig/src/main.c	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/src/main.c	2007-10-30 23:14:16.075089911 -0400
@@ -37,6 +37,7 @@
 
 #include "ccs-private.h"
 #include "iniparser.h"
+#include "compat.h"
 
 Bool basicMetadata = FALSE;
 
diff -ruN libcompizconfig-0.6.0.orig/src/Makefile.am libcompizconfig-0.6.0/src/Makefile.am
--- libcompizconfig-0.6.0.orig/src/Makefile.am	2007-10-15 12:51:01.000000000 -0400
+++ libcompizconfig-0.6.0/src/Makefile.am	2007-10-30 22:14:29.617182958 -0400
@@ -19,6 +19,8 @@
 	ini.c 		\
 	bindings.c 	\
 	filewatch.c 	\
+	compat.c	\
+	compat.h	\
 	ccs-private.h	\
 	iniparser.h
 
