diff -Nrup libcompizconfig-0.5.2/config.h.in ../libcompizconfig-0.5.2/config.h.in
--- libcompizconfig-0.5.2/config.h.in	2007-08-12 22:06:16.000000000 +0200
+++ ../libcompizconfig-0.5.2/config.h.in	2007-08-30 18:10:45.064810000 +0200
@@ -78,6 +78,9 @@
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
+/* Compiling on Solaris */
+#undef SOLARIS
+
 /* Define to 1 if you have the ANSI C header files. */
 #undef STDC_HEADERS
 
diff -Nrup libcompizconfig-0.5.2/configure.ac ../libcompizconfig-0.5.2/configure.ac
--- libcompizconfig-0.5.2/configure.ac	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/configure.ac	2007-08-30 18:10:45.065438000 +0200
@@ -1,6 +1,6 @@
 AC_PREREQ(2.57)
 
-AC_INIT([libcompizconfig],esyscmd(. ./VERSION;echo -n $VERSION), [dev@opencompositing.org])
+AC_INIT([libcompizconfig],0.5.2, [dev@opencompositing.org])
 
 AC_DEFINE_UNQUOTED(VERSION,"$VERSION",[Version])
 
@@ -73,6 +73,16 @@ if test "x$GCC" = "xyes"; then
   fi
 fi
 
+compiling_on_solaris=no
+case "$host" in
+        *-*-solaris*)
+	  AC_MSG_CHECKING(if compiling on Solaris)
+	  compiling_on_solaris=yes
+	  AC_MSG_RESULT($compiling_on_solaris)
+	  AC_DEFINE(SOLARIS, 1, [Compiling on Solaris])
+	;;
+esac
+
 AC_C_BIGENDIAN
 
 PKG_CHECK_MODULES(COMPIZ, "compiz")
diff -Nrup libcompizconfig-0.5.2/metadata/Makefile.am ../libcompizconfig-0.5.2/metadata/Makefile.am
--- libcompizconfig-0.5.2/metadata/Makefile.am	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/metadata/Makefile.am	2007-08-31 09:59:12.739798000 +0200
@@ -2,5 +2,10 @@ metadir = $(datadir)/compizconfig
 meta_DATA =	       \
 	global.xml
 
+configdir = $(sysconfdir)/compizconfig
+config_DATA =	      \
+	config
+
 EXTRA_DIST =	       \
-	global.xml
\ No newline at end of file
+	global.xml     \
+	config
diff -Nrup libcompizconfig-0.5.2/metadata/config ../libcompizconfig-0.5.2/metadata/config
--- libcompizconfig-0.5.2/metadata/config	1970-01-01 01:00:00.000000000 +0100
+++ ../libcompizconfig-0.5.2/metadata/config	2007-08-31 09:59:29.926422000 +0200
@@ -0,0 +1,6 @@
+[gnome_session]
+backend = gconf
+profile = 
+integration = true
+plugin_list_autosort = true
+
diff -Nrup libcompizconfig-0.5.2/metadata/global.xml ../libcompizconfig-0.5.2/metadata/global.xml
--- libcompizconfig-0.5.2/metadata/global.xml	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/metadata/global.xml	2007-08-30 18:37:16.532477000 +0200
@@ -4,22 +4,32 @@
 			<option name="active_plugins" type="list">
 				<type>string</type>
 				<default>
-					<value>dbus</value>
-					<value>place</value>
-					<value>move</value>
-					<value>resize</value>
-					<value>decoration</value>
-					<value>png</value>
-					<value>wobbly</value>
-					<value>cube</value>
-					<value>fade</value>
-					<value>minimize</value>
-					<value>rotate</value>
-					<value>scale</value>
-					<value>switcher</value>
-					<value>regex</value>
-					<value>workarounds</value>
-					<value>zoom</value>
+	                                <value>imgjpeg</value>
+                                        <value>svg</value>
+                                        <value>png</value>
+                                        <value>decoration</value>
+                                        <value>wobbly</value>
+                                        <value>fade</value>
+                                        <value>expo</value>
+                                        <value>minimize</value>
+                                        <value>cube</value>
+                                        <value>showdesktop</value>
+                                        <value>switcher</value>
+                                        <value>move</value>
+                                        <value>resize</value>
+                                        <value>place</value>
+                                        <value>rotate</value>
+                                        <value>zoom</value>
+                                        <value>scale</value>
+                                        <value>shift</value>
+                                        <value>dbus</value>
+                                        <value>water</value>
+                                        <value>snow</value>
+                                        <value>animation</value>
+                                        <value>firepaint</value>
+                                        <value>thumbnail</value>
+                                        <value>cubereflex</value>
+                                        <value>switcher</value>      
 				</default>
 			</option>
 			<group>
diff -Nrup libcompizconfig-0.5.2/src/Makefile.am ../libcompizconfig-0.5.2/src/Makefile.am
--- libcompizconfig-0.5.2/src/Makefile.am	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/src/Makefile.am	2007-08-30 18:10:45.065601000 +0200
@@ -19,6 +19,8 @@ libcompizconfig_la_SOURCES = \
 	ini.c 		\
 	bindings.c 	\
 	filewatch.c 	\
+	solaris.c	\
+	solaris.h	\
 	ccs-private.h	\
 	iniparser.h
 
diff -Nrup libcompizconfig-0.5.2/src/bindings.c ../libcompizconfig-0.5.2/src/bindings.c
--- libcompizconfig-0.5.2/src/bindings.c	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/src/bindings.c	2007-08-30 18:10:45.065847000 +0200
@@ -38,6 +38,7 @@
 #include <X11/Xlib.h>
 
 #include <ccs.h>
+#include "solaris.h"
 
 #define CompAltMask        (1 << 16)
 #define CompMetaMask       (1 << 17)
diff -Nrup libcompizconfig-0.5.2/src/compiz.c ../libcompizconfig-0.5.2/src/compiz.c
--- libcompizconfig-0.5.2/src/compiz.c	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/src/compiz.c	2007-08-30 18:10:45.066921000 +0200
@@ -42,6 +42,7 @@
 #include <compiz.h>
 #include <ccs.h>
 #include "ccs-private.h"
+#include "solaris.h"
 
 static xmlDoc * globalMetadata = NULL;
 
diff -Nrup libcompizconfig-0.5.2/src/config.c ../libcompizconfig-0.5.2/src/config.c
--- libcompizconfig-0.5.2/src/config.c	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/src/config.c	2007-08-30 18:10:45.067196000 +0200
@@ -26,6 +26,7 @@
 #include <string.h>
 
 #include "ccs-private.h"
+#include "solaris.h"
 
 #define SETTINGPATH "compiz/compizconfig"
 
diff -Nrup libcompizconfig-0.5.2/src/ini.c ../libcompizconfig-0.5.2/src/ini.c
--- libcompizconfig-0.5.2/src/ini.c	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/src/ini.c	2007-08-30 18:10:45.067717000 +0200
@@ -26,6 +26,7 @@
 
 #include <ccs.h>
 #include "iniparser.h"
+#include "solaris.h"
 
 /** 
  * Creates the parent directory for @fileName, recursively creating a directory
diff -Nrup libcompizconfig-0.5.2/src/main.c ../libcompizconfig-0.5.2/src/main.c
--- libcompizconfig-0.5.2/src/main.c	2007-08-12 21:59:57.000000000 +0200
+++ ../libcompizconfig-0.5.2/src/main.c	2007-08-30 18:10:45.068620000 +0200
@@ -36,6 +36,7 @@
 #include <ccs.h>
 
 #include "ccs-private.h"
+#include "solaris.h"
 
 Bool basicMetadata = FALSE;
 
diff -Nrup libcompizconfig-0.5.2/src/solaris.c ../libcompizconfig-0.5.2/src/solaris.c
--- libcompizconfig-0.5.2/src/solaris.c	1970-01-01 01:00:00.000000000 +0100
+++ ../libcompizconfig-0.5.2/src/solaris.c	2007-08-30 18:10:45.069005000 +0200
@@ -0,0 +1,113 @@
+#include <config.h>
+#ifdef SOLARIS
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <varargs.h>
+char* strcasestr (char *haystack, char *needle)
+{
+	char *p, *startn = 0, *np = 0;
+	for (p = haystack; *p; p++) {
+		if (np) {
+			if (toupper(*p) == toupper(*np)) {
+				if (!*++np)
+					return startn;
+			} else
+				np = 0;
+		} else if (toupper(*p) == toupper(*needle)) {
+			np = needle + 1;
+			startn = p;
+		}
+	}
+	return 0;
+}
+
+int asprintf(char **out, const char *fmt, ...)
+{
+    va_list ap;
+    int ret_status = EOF;
+    char dir_name[2001];
+    char file_name[2000];
+    FILE *fp = NULL;
+    char *work = NULL;
+
+    va_start(ap, fmt);
+
+    /* Warning: tmpfile() does not work well on Windows (MinGW)
+     *          if user does not have write access on the drive where 
+     *          working dir is? */
+#ifdef __MINGW32__
+    /* file_name = G_tempfile(); */
+    GetTempPath ( 2000, dir_name );
+    GetTempFileName ( dir_name, "asprintf", 0, file_name );
+    fp = fopen ( file_name, "w+" );
+#else
+    fp = tmpfile(); 
+#endif /* __MINGW32__ */
+
+    if ( fp ) {
+        int count;
+
+        count = vfprintf(fp, fmt, ap);
+        if (count >= 0) {
+            work = calloc(count + 1, sizeof(char));
+            if (work != NULL) {
+                rewind(fp);
+                ret_status = fread(work, sizeof(char), count, fp);
+                if (ret_status != count) {
+                    ret_status = EOF;
+                    free(work);
+                    work = NULL;
+                }
+            }
+        }
+        fclose(fp);
+#ifdef __MINGW32__
+        unlink ( file_name );
+#endif /* __MINGW32__ */
+    }
+    va_end(ap);
+    *out = work;
+
+    return ret_status;
+}
+/*
+ * Get next token from string *stringp, where tokens are possibly-empty
+ * strings separated by characters from delim.
+ *
+ * Writes NULs into the string at *stringp to end tokens.
+ * delim need not remain constant from call to call.
+ * On return, *stringp points past the last NUL written (if there might
+ * be further tokens), or is NULL (if there are definitely no more tokens).
+ *
+ * If *stringp is NULL, strsep returns NULL.
+ */
+char *
+strsep(char ** stringp, const char *delim)
+{
+        char *s;
+        const char *spanp;
+        int c, sc;
+        char *tok;
+
+        if ((s = *stringp) == NULL)
+                return (NULL);
+        for (tok = s;;) {
+                c = *s++;
+                spanp = delim;
+                do {
+                        if ((sc = *spanp++) == c) {
+                                if (c == 0)
+                                        s = NULL;
+                                else
+                                        s[-1] = 0;
+                                *stringp = s;
+                                return (tok);
+                        }
+                } while (sc != 0);
+        }
+        /* NOTREACHED */
+}
+
+
+#endif
diff -Nrup libcompizconfig-0.5.2/src/solaris.h ../libcompizconfig-0.5.2/src/solaris.h
--- libcompizconfig-0.5.2/src/solaris.h	1970-01-01 01:00:00.000000000 +0100
+++ ../libcompizconfig-0.5.2/src/solaris.h	2007-08-30 18:10:45.069176000 +0200
@@ -0,0 +1,6 @@
+#include <config.h>
+#ifdef SOLARIS
+char* strcasestr (char *haystack, char *needle);
+int asprintf(char **out, const char *fmt, ...);
+char *strsep(char ** stringp, const char *delim);
+#endif
