diff -ur DevIL-1.6.8-orig/configure.ac DevIL-1.6.8/configure.ac
--- DevIL-1.6.8-orig/configure.ac	2006-08-17 15:32:49.000000000 +0700
+++ DevIL-1.6.8/configure.ac	2007-05-07 11:12:15.603194939 +0700
@@ -97,9 +97,9 @@
 ]) 
 
 TEST_EXT([Altivec],[-faltivec -maltivec],[vector int vec;vec=vec_splat_u32(0);],[ALTIVEC_GCC],[],[altivec])
-TEST_EXT([SSE],[-msse],[__m128 v;v=_mm_sub_ss(v,v);],[SSE],[#include <xmmintrin.h>],[sse])
-TEST_EXT([SSE2],[-msse2],[__m128i v;v=_mm_add_epi32(v,v);],[SSE2],[#include <emmintrin.h>],[sse2])
-TEST_EXT([SSE3],[-msse3],[__m128d v;v=_mm_addsub_pd(v,v);],[SSE3],[#include <emmintrin.h>],[sse3])
+TEST_EXT([SSE],[-xarch=sse],[__m128 v;v=_mm_sub_ss(v,v);],[SSE],[#include <xmmintrin.h>],[sse])
+TEST_EXT([SSE2],[-xarch=sse2],[__m128i v;v=_mm_add_epi32(v,v);],[SSE2],[#include <emmintrin.h>],[sse2])
+TEST_EXT([SSE3],[-xarch=sse3],[__m128d v;v=_mm_addsub_pd(v,v);],[SSE3],[#include <emmintrin.h>],[sse3])
 
 dnl
 dnl Check CPU Assembly
@@ -126,9 +126,7 @@
 [build_release="$enable_release"],[build_release="no"])
 
 if test "$build_release" = "no" ; then
-	ADD_CFLAGS([-Wall -g -O0])
-else
-	ADD_CFLAGS([-O3])
+	ADD_CFLAGS([-g])
 fi
 
 dnl
