--- libgpod-0.4.0/configure.ac.patch1	2006-09-24 09:55:50.000000000 -0400
+++ libgpod-0.4.0/configure.ac	2006-09-26 21:37:09.458338000 -0400
@@ -52,7 +52,7 @@
 AC_CHECK_FUNCS(mremap)
 
 PKG_CHECK_MODULES(LIBGPOD, glib-2.0 >= 2.4.0 gobject-2.0)
-LIBGPOD_CFLAGS="$LIBGPOD_CFLAGS -Wall"
+LIBGPOD_CFLAGS="$LIBGPOD_CFLAGS"
 
 dnl **************************************************
 dnl * TagLib is only used by test-rebuild-db 
