--- /dev/null	Mon Apr 12 15:39:47 2004
+++ libgsf-1.8.2/libgsf-1-uninstalled.pc	Mon Apr 12 15:25:34 2004
@@ -0,0 +1,12 @@
+# the standard variables don't make sense for an uninstalled copy
+prefix=
+exec_prefix=
+libdir=${pcfiledir}
+includedir=${pcfiledir}
+
+Name: libgsf-1
+Description: A library for reading and writing structured files (eg MS OLE and Zip)
+Version: 1.8.2
+Requires: glib-2.0 gobject-2.0 libxml-2.0
+Libs: ${pc_top_builddir}/${pcfiledir}/gsf/libgsf-1.la ${pc_top_builddir}/${pcfiledir}/gsf-gnome/libgsf-gnome-1.la
+Cflags: -I${pc_top_builddir}/${pcfiledir} 
