--- libmpcdec-1.2.5-orig/configure	Sun Feb 11 21:03:53 2007
+++ libmpcdec-1.2.5/configure	Wed Apr  4 23:24:25 2007
@@ -19626,7 +19626,7 @@
 
 
 
-CFLAGS="$CFLAGS -O3 -fomit-frame-pointer -fPIC"
+CFLAGS="$CFLAGS -KPIC"
 
 { echo "$as_me:$LINENO: checking whether byte ordering is bigendian" >&5
 echo $ECHO_N "checking whether byte ordering is bigendian... $ECHO_C" >&6; }
--- libmpcdec-1.2.5-orig/configure.ac	Sun Feb 11 21:02:49 2007
+++ libmpcdec-1.2.5/configure.ac	Wed Apr  4 23:25:01 2007
@@ -6,7 +6,7 @@
 
 AM_PROG_LIBTOOL
 
-CFLAGS="$CFLAGS -O3 -fomit-frame-pointer -fPIC"
+CFLAGS="$CFLAGS -KPIC"
 
 AC_C_BIGENDIAN(,CFLAGS="$CFLAGS -DMPC_LITTLE_ENDIAN",)
 
