--- libmtp-0.2.1/configure.ac.orig	2007-09-18 15:57:27.571218159 +0700
+++ libmtp-0.2.1/configure.ac	2007-09-18 15:57:32.866226254 +0700
@@ -87,6 +87,26 @@
 # (No need to use AC_SUBST on this default substituted environment variable.)
 # CFLAGS="$CFLAGS -Wall -Wmissing-prototypes"
 
+AC_MSG_CHECKING([if inttypes has u_int32_t etc])
+AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+        [[
+#if HAVE_SYS_TYPES_H
+#include <sys/types.h>
+#endif
+#if HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif
+#if HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+        ]],[[
+    u_int32_t  s;
+    sleep((unsigned int)s);
+        ]]
+    )],
+    [AC_DEFINE(HAVE_U_INT_TYPES, 1, [Define to 1 if have u_int32_t types etc])
+    AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)])
 # Output files
 
 # Create a stdint.h-like file containing size-specific integer definitions
--- libmtp-0.2.1/examples/common.h.orig	2007-09-18 16:00:32.671373913 +0700
+++ libmtp-0.2.1/examples/common.h	2007-09-18 16:01:02.820289580 +0700
@@ -32,3 +32,12 @@
 // Only if using MSVC...
 #include "..\windows\getopt.h"
 #endif
+#ifdef HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif
+#ifndef HAVE_U_INT_TYPES
+typedef uint8_t u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+typedef uint64_t u_int64_t;
+#endif
