--- libmusicbrainz-3.0.0/test/main.cpp.orig	2006-05-27 20:54:06.000000000 +0200
+++ libmusicbrainz-3.0.0/test/main.cpp	2007-05-28 21:13:09.985711321 +0200
@@ -1,9 +1,10 @@
+//#include <string>
+#include <cppunit/CompilerOutputter.h> 
 #include <cppunit/TestResult.h>
 #include <cppunit/TestResultCollector.h> 
 #include <cppunit/TestRunner.h>
 #include <cppunit/extensions/TestFactoryRegistry.h>
 #include <cppunit/BriefTestProgressListener.h> 
-#include <cppunit/CompilerOutputter.h> 
 
 int 
 main(int argc, char **argv)
