--- libopensync-0.20/opensync/opensync_group.c.orig     2006-11-14 13:29:18.671089000 +0800
+++ libopensync-0.20/opensync/opensync_group.c  2006-11-14 15:09:44.666840000 +0800
@@ -25,6 +25,40 @@

 extern int errno;

+#ifndef HAVE_FLOCK
+#define LOCK_SH 1
+#define LOCK_EX 2
+#define LOCK_NB 4
+#define LOCK_UN 8
+
+static int
+flock(int fd, int operation)
+{
+       struct flock flock;
+
+       switch (operation & ~LOCK_NB) {
+       case LOCK_SH:
+               flock.l_type = F_RDLCK;
+               break;
+       case LOCK_EX:
+               flock.l_type = F_WRLCK;
+               break;
+       case LOCK_UN:
+               flock.l_type = F_UNLCK;
+               break;
+       default:
+               errno = EINVAL;
+               return -1;
+       }
+
+       flock.l_whence = 0;
+       flock.l_start = 0;
+       flock.l_len = 0;
+
+       return fcntl(fd, (operation & LOCK_NB) ? F_SETLK : F_SETLKW, &flock);
+}
+#endif
+
 /**
  * @defgroup OSyncGroupPrivateAPI OpenSync Group Internals
  * @ingroup OSyncPrivate
--- libopensync-0.20/configure.in.orig  2006-11-14 15:13:00.851363000 +0800
+++ libopensync-0.20/configure.in       2006-11-14 13:40:40.901830000 +0800
@@ -155,6 +155,9 @@
 AC_SUBST(PACKAGE_CFLAGS)
 AC_SUBST(PACKAGE_LIBS)

+### Check for flock function ###
+AC_CHECK_FUNCS(flock)
+
 # Workaround for autoconf 2.58, in which abs_builddir/abs_srcdir are not
 # absolute at all.
 # These are used in *-uninstalled.pc.in
