--- libopensync-0.22/osengine/Makefile.am.orig  2006-11-14 16:09:06.600092000 +0800
+++ libopensync-0.22/osengine/Makefile.am       2006-11-14 16:09:17.443973000 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = -Werror -Wall @GCOV_CFLAGS@ @XML_CFLAGS@
+AM_CFLAGS = @GCOV_CFLAGS@ @XML_CFLAGS@
 
 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ 
 
--- libopensync-0.20/tools/Makefile.am.orig     2006-11-14 16:23:38.862343000 +0800
+++ libopensync-0.20/tools/Makefile.am  2006-11-14 16:23:49.803734000 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in

-AM_CFLAGS = -Wall -Werror @GCOV_CFLAGS@
+AM_CFLAGS = @GCOV_CFLAGS@

 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ @XML_CFLAGS@ -I$(top_srcdir)/osengine

--- libopensync-0.20/osplugin/Makefile.am.orig  2006-11-14 16:24:55.670245000 +0800
+++ libopensync-0.20/osplugin/Makefile.am       2006-11-14 16:25:14.321365000 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in

-AM_CFLAGS = -Wall @XML_CFLAGS@ -Werror @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"
+AM_CFLAGS = @XML_CFLAGS@ @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"

 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ @XML_CFLAGS@ -I$(top_srcdir)/osengine -I$(top_srcdir)/opensync

--- libopensync-0.20/formats/vformats-xml/Makefile.am.orig      2006-11-14 16:10:03.030607000 +0800
+++ libopensync-0.20/formats/vformats-xml/Makefile.am   2006-11-14 16:10:23.172303000 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in

-AM_CFLAGS = -Werror -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\" -Wall @XML_CFLAGS@  @GCOV_CFLAGS@
+AM_CFLAGS = -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\" @XML_CFLAGS@  @GCOV_CFLAGS@

 formatsdir=@OPENSYNC_FORMATSDIR@
 opensyncheaderdir=@OPENSYNC_HEADERDIR@
--- libopensync-0.20/formats/Makefile.am.orig   2006-11-14 16:10:52.649806000 +0800
+++ libopensync-0.20/formats/Makefile.am        2006-11-14 16:11:08.860505000 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in

-AM_CFLAGS = -Werror -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\" -Wall @XML_CFLAGS@ @GCOV_CFLAGS@
+AM_CFLAGS = -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\" @XML_CFLAGS@ @GCOV_CFLAGS@

 formatsdir=@OPENSYNC_FORMATSDIR@
 opensyncheaderdir=@OPENSYNC_HEADERDIR@
--- libopensync-0.20/opensync/Makefile.am.orig  2006-11-14 11:57:42.456541000 +0800
+++ libopensync-0.20/opensync/Makefile.am       2006-11-14 11:58:03.936735000 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in

-AM_CFLAGS = -Wall @XML_CFLAGS@ -Werror @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"
+AM_CFLAGS = @XML_CFLAGS@ @GCOV_CFLAGS@ -DOPENSYNC_PLUGINDIR=\"@OPENSYNC_PLUGINDIR@\" -DOPENSYNC_CONFIGDIR=\"@OPENSYNC_CONFIGDIR@\" -DOPENSYNC_FORMATSDIR=\"@OPENSYNC_FORMATSDIR@\"

 INCLUDES = @PACKAGE_CFLAGS@

--- libopensync-0.20/wrapper/Makefile.am.orig   2006-11-14 16:24:11.212795000 +0800
+++ libopensync-0.20/wrapper/Makefile.am        2006-11-14 16:24:23.845826000 +0800
@@ -1,5 +1,5 @@

-AM_CFLAGS = -Werror
+AM_CFLAGS =

 INCLUDES= @PACKAGE_CFLAGS@ $(PYTHON_INCLUDES) -I$(top_srcdir) -I$(top_srcdir)/opensync @XML_CFLAGS@
