--- libquicktime-1.0.0/configure.ac.orig	2007-09-04 20:24:08.783630622 +0700
+++ libquicktime-1.0.0/configure.ac	2007-09-04 20:30:29.863912454 +0700
@@ -687,17 +687,9 @@
 dnl Build CFLAGS
 dnl
 
-COMMON_CFLAGS=""
-lqt_test_cflags="-finline-functions -Wall -Winline"
-for i in $lqt_test_cflags; do
-  LQT_TRY_CFLAGS($i, COMMON_CFLAGS="$COMMON_CFLAGS $i") ;
-done
+COMMON_CFLAGS="$CFLAGS"
 
 
-dnl Optimizing flags
-
-LQT_OPT_CFLAGS($host_cpu, ["-O3 -funroll-all-loops -fomit-frame-pointer"])
-
 AH_TEMPLATE([NDEBUG],
             [Causes debug code to e removed])
 
