diff -ur libraw1394-1.2.1.orig/configure.ac libraw1394-1.2.1/configure.ac
--- libraw1394-1.2.1.orig/configure.ac	2005-06-23 03:12:08.000000000 +0700
+++ libraw1394-1.2.1/configure.ac	2007-09-04 08:30:14.441286216 +0700
@@ -25,6 +25,30 @@
 AC_SUBST(lt_age)
 
 
+AC_CHECK_HEADERS(endian.h byteswap.h sys/byteorder.h sys/ioccom.h sys/types.h)
+AC_CHECK_HEADERS(unistd.h inttypes.h linux/ioctl.h linux/types.h stropts.h sys/ioccom.h)
+
+AC_MSG_CHECKING([if inttypes has u_int32_t etc])
+AC_COMPILE_IFELSE(
+    [AC_LANG_PROGRAM(
+        [[
+#if HAVE_SYS_TYPES_H
+#include <sys/types.h>
+#endif
+#if HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif
++#if HAVE_UNISTD_H
++#include <unistd.h>
++#endif
+        ]],[[
+    u_int32_t	s;
+    sleep((unsigned int)s);
+        ]]
+    )],
+    [AC_DEFINE(HAVE_U_INT_TYPES, 1, [Define to 1 if have u_int32_t types etc])
+    AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)])
+
 #CFLAGS=${CFLAGS:-"-Wall"}
 AC_OUTPUT([
 Makefile
