diff -ur libraw1394-1.2.1.orig/src/raw1394.h libraw1394-1.2.1/src/raw1394.h
--- libraw1394-1.2.1.orig/src/raw1394.h	2006-03-04 08:25:13.000000000 +0700
+++ libraw1394-1.2.1/src/raw1394.h	2007-09-04 08:35:57.944626320 +0700
@@ -13,6 +13,15 @@
 #define _LIBRAW1394_RAW1394_H
 
 #include <sys/types.h>
+#ifdef HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif
+#ifndef HAVE_U_INT_TYPES
+typedef uint8_t u_int8_t;
+typedef uint16_t u_int16_t;
+typedef uint32_t u_int32_t;
+typedef uint64_t u_int64_t;
+#endif
 
 #define RAW1394_ARM_READ  1
 #define RAW1394_ARM_WRITE 2
