--- libstroke-0.4/libstroke/Makefile.am~	Wed Mar 15 17:15:10 2000
+++ libstroke-0.4/libstroke/Makefile.am	Sat Mar 23 13:12:05 2002
@@ -5,11 +5,6 @@
 # level Makefile.
 AUTOMAKE_OPTIONS = foreign
 
-EXTRA_DIST = 
-
-#tclpath::
-	#echo ${ENABLE_TCL}
-
 #if ENABLE_TCL
 #SEL_BIB = libstroke.la libstroke_tcl.la
 #else
@@ -35,8 +30,6 @@
 
 INCLUDES = -I$(top_srcdir)/libstroke @X_CFLAGS@  
 
-EXTRA_DIST = 
-
 MOSTLYCLEANFILES = core *~ 
 CLEANFILES = core *~ 
 DISTCLEANFILES = core *~ 
