--- libsyncml-0.4.6/libsyncml/sml_support.c.orig	2008-03-07 11:57:29.466249000 +0800
+++ libsyncml-0.4.6/libsyncml/sml_support.c	2008-03-07 11:57:56.584426000 +0800
@@ -150,7 +150,7 @@
 		{
 			// FIXME: according to the manpage strcasestr returns char *
 			// FIXME: but it gives a warning if I do not cast the return value
-			if (NULL != (char *) strcasestr(buffer, filters[i]))
+			if (strstr(buffer, filters[i]))
 				matched = TRUE;
 		}
 		g_strfreev(filters);
