--- libsyncml-0.4.4/tests/check_devinf.c.orig	2007-03-30 17:04:24.976197500 +0100
+++ libsyncml-0.4.4/tests/check_devinf.c	2007-03-30 17:04:38.331407163 +0100
@@ -51,7 +51,7 @@
 			break;
 		case SML_MANAGER_CONNECT_DONE:
 		case SML_MANAGER_DISCONNECT_DONE:
-			fail();
+			fail(NULL);
 			break;
 		case SML_MANAGER_TRANSPORT_ERROR:
 			transport_errors++;
--- libsyncml-0.4.4/tests/check_ds.c.orig	2007-03-30 17:04:39.055508844 +0100
+++ libsyncml-0.4.4/tests/check_ds.c	2007-03-30 17:04:48.332277949 +0100
@@ -448,7 +448,7 @@
 			g_atomic_int_add(&session_ends, 1);
 			break;
 		default:
-			fail();
+			fail(NULL);
 	}
 	
 	smlTrace(TRACE_EXIT, "%s", __func__);
--- libsyncml-0.4.4/tests/check_manager.c.orig	2007-03-30 17:04:49.005902512 +0100
+++ libsyncml-0.4.4/tests/check_manager.c	2007-03-30 17:04:57.327173233 +0100
@@ -95,7 +95,7 @@
 			break;
 		case SML_MANAGER_CONNECT_DONE:
 		case SML_MANAGER_DISCONNECT_DONE:
-			fail();
+			fail(NULL);
 			break;
 		case SML_MANAGER_TRANSPORT_ERROR:
 			transport_errors++;
--- libsyncml-0.4.4/tests/check_xml_assembler.c.orig	2007-03-30 17:04:59.214201270 +0100
+++ libsyncml-0.4.4/tests/check_xml_assembler.c	2007-03-30 17:05:12.738964709 +0100
@@ -1392,7 +1392,7 @@
 			fail_unless(smlAssemblerAddStatus(assm, status, &error) == SML_ASSEMBLER_RESULT_OK);
 			fail_unless(error == NULL, NULL);
 		} else if (res != SML_ASSEMBLER_RESULT_OK)
-			fail();
+			fail(NULL);
 		
 		smlStatusUnref(status);
 	}
@@ -1421,7 +1421,7 @@
 			fail_unless(smlAssemblerStartCommand(assm, NULL, cmd1, &error), NULL);
 			fail_unless(error == NULL, NULL);
 		} else if (res != SML_ASSEMBLER_RESULT_OK)
-			fail();
+			fail(NULL);
 	
 		smlCommandUnref(cmd1);
 		
--- libsyncml-0.4.4/tests/check_http.c.orig	2007-03-30 16:59:38.900220229 +0100
+++ libsyncml-0.4.4/tests/check_http.c	2007-03-30 16:59:53.827244200 +0100
@@ -246,7 +246,7 @@
 				smlTransportSetError(tsp, link, &newerror);
 				smlErrorDeref(&newerror);
 			} else
-				fail();
+				fail(NULL);
 			break;
 		default:
 			fail(NULL);
--- libsyncml-0.4.4/tests/check_obex.c.orig	2007-03-30 17:00:00.680420189 +0100
+++ libsyncml-0.4.4/tests/check_obex.c	2007-03-30 17:00:20.277770389 +0100
@@ -176,7 +176,7 @@
 				smlTransportSetError(tsp, link, &newerror);
 				smlErrorDeref(&newerror);
 			} else
-				fail();
+				fail(NULL);
 			break;
 		case SML_TRANSPORT_EVENT_CONNECT_DONE:
 			g_atomic_int_inc(&client_connect_done);
