--- libsyncml-0.4.2/tools/Makefile.am.orig	2006-12-04 14:49:15.349965000 +0800
+++ libsyncml-0.4.2/tools/Makefile.am	2006-12-04 14:49:22.331822000 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = -Wall -Werror
+#AM_CFLAGS = -Wall -Werror
 
 INCLUDES = -I$(top_srcdir) @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ @LIBOPENOBEX_CFLAGS@
 
--- libsyncml-0.4.2/libsyncml/Makefile.am.orig	2006-12-04 14:46:57.363281000 +0800
+++ libsyncml-0.4.2/libsyncml/Makefile.am	2006-12-04 14:47:05.809368000 +0800
@@ -2,7 +2,7 @@
 
 SUBDIRS = parser transports objects
 
-AM_CFLAGS = -Wall -Werror
+#AM_CFLAGS = -Wall -Werror
 
 INCLUDES = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir)
 
--- libsyncml-0.4.2/libsyncml/objects/Makefile.am.orig	2006-12-04 15:01:06.571727000 +0800
+++ libsyncml-0.4.2/libsyncml/objects/Makefile.am	2006-12-04 15:01:13.585437000 +0800
@@ -1,6 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-AM_CFLAGS = -Wall -Werror
 
 INCLUDES = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir)
 
--- libsyncml-0.4.2/libsyncml/parser/Makefile.am.orig	2006-12-04 14:58:50.441016000 +0800
+++ libsyncml-0.4.2/libsyncml/parser/Makefile.am	2006-12-04 14:59:07.191117000 +0800
@@ -10,10 +10,10 @@
 	
 noinst_LTLIBRARIES = libsyncml-xml.la libsyncml-wbxml.la
 
-libsyncml_xml_la_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) -Wall -Werror
+libsyncml_xml_la_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) 
 libsyncml_xml_la_SOURCES = sml_xml_parse.c sml_xml_assm.c
 libsyncml_xml_la_LDFLAGS = @PACKAGE_LIBS@ @LIBXML_LIBS@
 
-libsyncml_wbxml_la_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@  @WBXML_CFLAGS@ -I$(top_srcdir) -Wall -Werror
+libsyncml_wbxml_la_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@  @WBXML_CFLAGS@ -I$(top_srcdir)
 libsyncml_wbxml_la_SOURCES = sml_wbxml.c
 libsyncml_wbxml_la_LDFLAGS = @PACKAGE_LIBS@ @LIBXML_LIBS@ @WBXML_LIBS@
--- libsyncml-0.4.2/libsyncml/transports/Makefile.am.orig	2006-12-04 15:01:27.364192000 +0800
+++ libsyncml-0.4.2/libsyncml/transports/Makefile.am	2006-12-04 15:01:49.602289000 +0800
@@ -16,22 +16,22 @@
 
 noinst_LTLIBRARIES = libobex-client.la libobex-server.la libhttp-client.la libhttp-server.la
 
-libhttp_client_la_CFLAGS = @LIBSOUP_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror
+libhttp_client_la_CFLAGS = @LIBSOUP_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) 
 libhttp_client_la_SOURCES = http_client.c
 libhttp_client_la_LDFLAGS = @PACKAGE_LIBS@
 libhttp_client_la_LIBADD = @LIBSOUP_LIBS@
 
-libhttp_server_la_CFLAGS = @LIBSOUP_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror
+libhttp_server_la_CFLAGS = @LIBSOUP_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) 
 libhttp_server_la_SOURCES = http_server.c
 libhttp_server_la_LDFLAGS = @PACKAGE_LIBS@
 libhttp_server_la_LIBADD = @LIBSOUP_LIBS@
 
-libobex_client_la_CFLAGS = @LIBOPENOBEX_CFLAGS@ @LIBBLUETOOTH_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror
+libobex_client_la_CFLAGS = @LIBOPENOBEX_CFLAGS@ @LIBBLUETOOTH_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) 
 libobex_client_la_SOURCES = obex_client.c
 libobex_client_la_LDFLAGS = @PACKAGE_LIBS@
 libobex_client_la_LIBADD = @LIBOPENOBEX_LIBS@ @LIBBLUETOOTH_LIBS@
 
-libobex_server_la_CFLAGS = @LIBOPENOBEX_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror
+libobex_server_la_CFLAGS = @LIBOPENOBEX_CFLAGS@ @PACKAGE_CFLAGS@ -I$(top_srcdir) 
 libobex_server_la_SOURCES = obex_server.c
 libobex_server_la_LDFLAGS = @PACKAGE_LIBS@
 libobex_server_la_LIBADD = @LIBOPENOBEX_LIBS@
--- libsyncml-0.4.2/tests/Makefile.am.orig	2006-12-04 14:47:39.756134000 +0800
+++ libsyncml-0.4.2/tests/Makefile.am	2006-12-04 14:48:53.635519000 +0800
@@ -10,82 +10,82 @@
 noinst_PROGRAMS = 
 endif
 
-error_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+error_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 error_SOURCES = check_error.c support.c
 error_LIBS = @CHECK_LIBS@
 error_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-elements_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+elements_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 elements_SOURCES = check_elements.c support.c
 elements_LIBS = @CHECK_LIBS@
 elements_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-http_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+http_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 http_SOURCES = check_http.c support.c
 http_LIBS = @CHECK_LIBS@
 http_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-obex_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+obex_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 obex_SOURCES = check_obex.c support.c
 obex_LIBS = @CHECK_LIBS@
 obex_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-xmlparser_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+xmlparser_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 xmlparser_SOURCES = check_xml_parser.c support.c
 xmlparser_LIBS = @CHECK_LIBS@
 xmlparser_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-xmlassembler_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+xmlassembler_CFLAGS = @PACKAGE_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 xmlassembler_SOURCES = check_xml_assembler.c support.c
 xmlassembler_LIBS = @CHECK_LIBS@
 xmlassembler_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-wbxmlparser_CFLAGS = @PACKAGE_CFLAGS@ @WBXML_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+wbxmlparser_CFLAGS = @PACKAGE_CFLAGS@ @WBXML_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 wbxmlparser_SOURCES = check_wbxml_parser.c support.c
 wbxmlparser_LIBS = @CHECK_LIBS@ @WBXML_LIBS@
 wbxmlparser_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-wbxmlassembler_CFLAGS = @PACKAGE_CFLAGS@ @WBXML_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+wbxmlassembler_CFLAGS = @PACKAGE_CFLAGS@ @WBXML_CFLAGS@ @LIBXML_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 wbxmlassembler_SOURCES = check_wbxml_assembler.c support.c
 wbxmlassembler_LIBS = @CHECK_LIBS@
 wbxmlassembler_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-session_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @LIBXML_CFLAGS@ -Wall -Werror @CHECK_CFLAGS@
+session_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @LIBXML_CFLAGS@ @CHECK_CFLAGS@
 session_SOURCES = check_session.c support.c
 session_LIBS = @CHECK_LIBS@
 session_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-base64_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+base64_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 base64_SOURCES = check_base64.c support.c
 base64_LIBS = @CHECK_LIBS@
 base64_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-md5_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+md5_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 md5_SOURCES = check_md5.c support.c
 md5_LIBS = @CHECK_LIBS@
 md5_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-datasync_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @LIBXML_CFLAGS@ -Wall -Werror @CHECK_CFLAGS@
+datasync_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @LIBXML_CFLAGS@ @CHECK_CFLAGS@
 datasync_SOURCES = check_ds.c support.c
 datasync_LIBS = @CHECK_LIBS@
 datasync_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-san_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+san_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 san_SOURCES = check_san.c support.c
 san_LIBS = @CHECK_LIBS@
 san_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-devinf_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+devinf_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 devinf_SOURCES = check_devinf.c support.c
 devinf_LIBS = @CHECK_LIBS@
 devinf_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-manager_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+manager_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) @CHECK_CFLAGS@
 manager_SOURCES = check_manager.c support.c
 manager_LIBS = @CHECK_LIBS@
 manager_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
 
-sync_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir) -Wall -Werror @CHECK_CFLAGS@
+sync_CFLAGS = @PACKAGE_CFLAGS@ -I$(top_srcdir)  @CHECK_CFLAGS@
 sync_SOURCES = check_sync.c support.c
 sync_LIBS = @CHECK_LIBS@
 sync_LDFLAGS = @PACKAGE_LIBS@ $(top_builddir)/libsyncml/libsyncml.la -lcheck -R $(libdir)
