Index: libwbxml-trunk/tools/getopt.h
===================================================================
--- libwbxml-trunk/tools/getopt.h	(revision 148)
+++ libwbxml-trunk/tools/getopt.h	(working copy)
@@ -5,13 +5,13 @@
  *
  * @author Kannel Team (http://www.kannel.org/)
  *
- * @brief getopt() implementation
+ * @brief wbxml_getopt() implementation
  */
 
 #ifndef WBXML_GETOPT_H
 #define WBXML_GETOPT_H
 
-int getopt(int argc, char **argv, char *opts);
+int wbxml_getopt(int argc, char **argv, char *opts);
 extern int opterr;
 extern int optind;
 extern int optopt;
Index: libwbxml-trunk/tools/wbxml2xml_tool.c
===================================================================
--- libwbxml-trunk/tools/wbxml2xml_tool.c	(revision 148)
+++ libwbxml-trunk/tools/wbxml2xml_tool.c	(working copy)
@@ -236,7 +236,7 @@
     params.indent = 1;
     params.keep_ignorable_ws = FALSE;
 
-    while ((opt = (WB_TINY) getopt(argc, argv, "kh?o:m:i:l:")) != EOF)
+    while ((opt = (WB_TINY) wbxml_getopt(argc, argv, "kh?o:m:i:l:")) != EOF)
     {
         switch (opt) {
         case 'k':
Index: libwbxml-trunk/tools/xml2wbxml_tool.c
===================================================================
--- libwbxml-trunk/tools/xml2wbxml_tool.c	(revision 148)
+++ libwbxml-trunk/tools/xml2wbxml_tool.c	(working copy)
@@ -105,7 +105,7 @@
     params.use_strtbl = TRUE;
     params.keep_ignorable_ws = FALSE;
 
-    while ((opt = (WB_TINY) getopt(argc, argv, "nkh?o:v:")) != EOF)
+    while ((opt = (WB_TINY) wbxml_getopt(argc, argv, "nkh?o:v:")) != EOF)
     {
         switch (opt) {
         case 'v':
Index: libwbxml-trunk/tools/attgetopt.c
===================================================================
--- libwbxml-trunk/tools/attgetopt.c	(revision 148)
+++ libwbxml-trunk/tools/attgetopt.c	(working copy)
@@ -22,7 +22,7 @@
 int     optopt;
 char    *optarg;
 
-int getopt(int argc, char **argv, char *opts)
+int wbxml_getopt(int argc, char **argv, char *opts)
 {
     static int sp = 1;
     register int c;
