--- mjpegtools-1.8.0/lavtools/lav_io.c.orig	2007-09-04 23:57:01.844438586 +0700
+++ mjpegtools-1.8.0/lavtools/lav_io.c	2007-09-05 00:25:56.264969684 +0700
@@ -1367,7 +1367,7 @@
          break;
 #ifdef HAVE_LIBQUICKTIME
       case 'q':
-         res = fileno(((quicktime_t *)lav_file->qt_fd)->stream);
+         res = lqt_fileno((quicktime_t *)lav_file->qt_fd);
          break;
 #endif
       default:
