diff -urN monkey-bubble-0.4.0/src/audio/Makefile.am monkey-bubble-0.4.0-hacked/src/audio/Makefile.am
--- monkey-bubble-0.4.0/src/audio/Makefile.am	2004-12-03 20:05:59.000000000 +1300
+++ monkey-bubble-0.4.0-hacked/src/audio/Makefile.am	2008-08-01 03:33:21.027976716 +1200
@@ -3,7 +3,7 @@
 INCLUDES = \
 	$(LIBMONKEYAUDIO_A_CFLAGS) \
 	-I$(top_srcdir)/src/monkey \
-	-g -O2 -Wall -D_REENTRANT -DDATADIR=\""$(datadir)"\"
+	-O2 -D_REENTRANT -DDATADIR=\""$(datadir)"\"
 
 noinst_LIBRARIES = libmonkeyaudio.a
 
diff -urN monkey-bubble-0.4.0/src/input/Makefile.am monkey-bubble-0.4.0-hacked/src/input/Makefile.am
--- monkey-bubble-0.4.0/src/input/Makefile.am	2004-06-10 01:38:58.000000000 +1200
+++ monkey-bubble-0.4.0-hacked/src/input/Makefile.am	2008-08-01 03:33:21.026548817 +1200
@@ -2,7 +2,7 @@
 
 INCLUDES = \
 	$(LIBMBINPUT_A_CFLAGS) \
-	-g -O2 -Wall -Werror -D_REENTRANT -DDATADIR=\""$(datadir)"\"
+	-O2 -D_REENTRANT -DDATADIR=\""$(datadir)"\"
 
 noinst_LIBRARIES = libmbinput.a
 
diff -urN monkey-bubble-0.4.0/src/monkey/Makefile.am monkey-bubble-0.4.0-hacked/src/monkey/Makefile.am
--- monkey-bubble-0.4.0/src/monkey/Makefile.am	2004-01-31 03:57:48.000000000 +1300
+++ monkey-bubble-0.4.0-hacked/src/monkey/Makefile.am	2008-08-01 03:33:21.027051573 +1200
@@ -3,8 +3,7 @@
 
 
 INCLUDES = \
-	$(LIBMONKEY_A_CFLAGS) \
-	-Wall -Werror -g
+	$(LIBMONKEY_A_CFLAGS)
 
 
 noinst_LIBRARIES = libmonkey.a
diff -urN monkey-bubble-0.4.0/src/net/Makefile.am monkey-bubble-0.4.0-hacked/src/net/Makefile.am
--- monkey-bubble-0.4.0/src/net/Makefile.am	2004-06-15 19:13:36.000000000 +1200
+++ monkey-bubble-0.4.0-hacked/src/net/Makefile.am	2008-08-01 03:42:37.145207235 +1200
@@ -7,7 +7,7 @@
 	-I$(top_srcdir)/src/view \
 	-I$(top_srcdir)/src/ui \
 	-I$(top_srcdir)/src/util \
-	-g -O2 -Wall -Werror -D_REENTRANT  -D_DEBUG_ -D_TIME_DEBUG_ \
+	-D_REENTRANT  -D_DEBUG_ -D_TIME_DEBUG_ \
 	$(NULL)
 # -g -O2 -Wall -Wstrict-prototypes -Wmissing-prototypes -Werror -D_REENTRANT  -D_TIME_DEBUG_ -D_DEBUG_
 
@@ -29,6 +29,7 @@
 
 monkey_srv_LDADD = \
 	$(MONKEY_SRV_LIBS) \
+	-lsocket -lnsl \
 	libmonkeynet.a \
 	../monkey/libmonkey.a \
 	../util/libmonkeyutil.a 
diff -urN monkey-bubble-0.4.0/src/ui/game-sound.c monkey-bubble-0.4.0-hacked/src/ui/game-sound.c
--- monkey-bubble-0.4.0/src/ui/game-sound.c	2004-12-03 20:05:59.000000000 +1300
+++ monkey-bubble-0.4.0-hacked/src/ui/game-sound.c	2008-08-01 03:33:21.029434972 +1200
@@ -29,7 +29,7 @@
 static GObjectClass* parent_class = NULL;
 
 struct MbGameSoundPrivate {
-  
+  	char dummy;
 };
 
 MbGameSound *
diff -urN monkey-bubble-0.4.0/src/ui/Makefile.am monkey-bubble-0.4.0-hacked/src/ui/Makefile.am
--- monkey-bubble-0.4.0/src/ui/Makefile.am	2006-08-02 01:47:04.000000000 +1200
+++ monkey-bubble-0.4.0-hacked/src/ui/Makefile.am	2008-08-01 03:42:52.249773978 +1200
@@ -12,8 +12,7 @@
 	-I$(top_srcdir)/src/audio			\
 	-I$(top_srcdir)/src/util			\
 	-I$(top_srcdir)/src/net 			\
-	-I$(top_srcdir)/src/input                       \
-	-Wall -Werror
+	-I$(top_srcdir)/src/input
 
 
 bin_PROGRAMS = monkey-bubble
@@ -41,6 +40,7 @@
 
 monkey_bubble_LDADD =  \
 	$(UI_LIBS) \
+	-lsocket -lnsl \
 	../monkey/libmonkey.a \
 	../view/libmonkeyview.a \
 	../util/libmonkeyutil.a \
diff -urN monkey-bubble-0.4.0/src/util/Makefile.am monkey-bubble-0.4.0-hacked/src/util/Makefile.am
--- monkey-bubble-0.4.0/src/util/Makefile.am	2004-01-31 03:58:01.000000000 +1300
+++ monkey-bubble-0.4.0-hacked/src/util/Makefile.am	2008-08-01 03:33:21.026050433 +1200
@@ -3,8 +3,7 @@
 
 
 INCLUDES = \
-	$(LIBMONKEYUTIL_A_CFLAGS) \
-	-Wall -Werror -g
+	$(LIBMONKEYUTIL_A_CFLAGS)
 
 
 noinst_LIBRARIES = libmonkeyutil.a
diff -urN monkey-bubble-0.4.0/src/view/Makefile.am monkey-bubble-0.4.0-hacked/src/view/Makefile.am
--- monkey-bubble-0.4.0/src/view/Makefile.am	2004-07-02 03:33:34.000000000 +1200
+++ monkey-bubble-0.4.0-hacked/src/view/Makefile.am	2008-08-01 03:33:21.027510018 +1200
@@ -5,7 +5,7 @@
 	$(LIBMONKEYVIEW_A_CFLAGS) \
 	$(AM_CFLAGS) \
 	-I$(top_srcdir)/src/monkey \
-	-g -O2 -Wall -Werror -D_REENTRANT -DDATADIR=\""$(datadir)"\"
+	-D_REENTRANT -DDATADIR=\""$(datadir)"\"
 
 noinst_LIBRARIES = libmonkeyview.a
 
