--- MPlayer-1.0rc1/Makefile.orig	2007-01-03 13:29:54.739249000 -0500
+++ MPlayer-1.0rc1/Makefile	2007-01-03 13:30:15.197044000 -0500
@@ -245,7 +245,7 @@
 COMMON_DEPS += libdha/libdha.so vidix/libvidix.a
 endif
 ifeq ($(FAME),yes)
-COMMON_DEPS += libfame/libfame.a
+#COMMON_DEPS += libfame/libfame.a
 endif
 ifeq ($(DVDKIT2),yes)
 COMMON_DEPS += libmpdvdkit2/libmpdvdkit.a
--- MPlayer-1.0rc1/configure.orig	2007-01-03 13:36:32.693020000 -0500
+++ MPlayer-1.0rc1/configure	2007-01-03 13:36:58.903012000 -0500
@@ -4762,7 +4762,7 @@
 _def_libfame='#undef USE_LIBFAME'
 if test "$_libfame" = yes ; then
   _def_libfame='#define USE_LIBFAME 1'
-  _ld_libfame='libfame/libfame.a'
+  _ld_libfame='-lfame -lmp3lame'
 fi
 
 
