--- MPlayer-1.0rc2-orig/configure	2007-10-07 19:49:33.000000000 +0000
+++ MPlayer-1.0rc2/configure	2007-11-03 08:05:21.986253976 +0000
@@ -7327,6 +7327,7 @@
 if test "$_real" = yes || test "$_xanim" = yes && not win32 && not qnx && not darwin ; then
   _ld_dl_dynamic='-rdynamic'
 fi
+sunos && _ld_dl_dynamic='-dynamic'
 
 _ld_extra="$_ld_extra $_ld_pthread $_ld_dl $_ld_dl_dynamic"
 bsdos && _ld_extra="$_ld_extra -ldvd"
