--- mplayerplug-in/Makefile.in.patch1	2006-07-20 14:20:42.000000000 -0400
+++ mplayerplug-in/Makefile.in	2006-09-21 21:21:45.295455000 -0400
@@ -12,7 +12,7 @@
 
 CC= @CC@
 CXX= @CXX@
-CFLAGS= -Wall @CFLAGS@ @CPPFLAGS@ @CXXFLAGS@ @CXXFLAGS64@ -DXP_UNIX -DMOZ_X11 @MOZPLUG_CFLAGS@ -Iinclude -fPIC @GECKO_DEFINES@ @GTK_CFLAGS@ @GTHREAD_CFLAGS@ @GTK_DEFINES@
+CFLAGS= @CFLAGS@ @CPPFLAGS@ @CXXFLAGS@ @CXXFLAGS64@ -DXP_UNIX -DMOZ_X11 @MOZPLUG_CFLAGS@ -Iinclude -fPIC @GECKO_DEFINES@ @GTK_CFLAGS@ @GTHREAD_CFLAGS@ @GTK_DEFINES@
 LDFLAGS= @LDFLAGS@ @LDFLAGS64@ @LIBS@ @X_PRE_LIBS@ @XEXT_LIBS@ @XPM_LIBS@ @GTK_LIBS@ @GTHREAD_LIBS@
 LIBOBJS= @LIBOBJS@
 OBJ= plugin.o nsScriptablePeer.o npp_gate.o np_entry.o npn_gate.o \
@@ -57,26 +57,26 @@
 default all: @CREATE@ languages
 
 $(SHAREDTARGET): $(OBJ)
-	$(CXX) -shared -o $(SHAREDTARGET).so $(OBJ) $(CFLAGS) $(LDFLAGS)
+	$(CXX) -G -o $(SHAREDTARGET).so $(OBJ) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in.xpt
 $(SHAREDTARGET)-wmp: $(OBJ_WMP)
-	$(CXX) -shared -o $(SHAREDTARGET)-wmp.so $(OBJ_WMP) $(CFLAGS) $(LDFLAGS)
+	$(CXX) -G -o $(SHAREDTARGET)-wmp.so $(OBJ_WMP) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-wmp.xpt
 
 $(SHAREDTARGET)-qt: $(OBJ_QT)
-	$(CXX) -shared -o $(SHAREDTARGET)-qt.so $(OBJ_QT) $(CFLAGS) $(LDFLAGS)
+	$(CXX) -G -o $(SHAREDTARGET)-qt.so $(OBJ_QT) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-qt.xpt
 
 $(SHAREDTARGET)-rm: $(OBJ_RM)
-	$(CXX) -shared -o $(SHAREDTARGET)-rm.so $(OBJ_RM) $(CFLAGS) $(LDFLAGS)
+	$(CXX) -G -o $(SHAREDTARGET)-rm.so $(OBJ_RM) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-rm.xpt
 
 $(SHAREDTARGET)-gmp: $(OBJ_GMP)
-	$(CXX) -shared -o $(SHAREDTARGET)-gmp.so $(OBJ_GMP) $(CFLAGS) $(LDFLAGS)
+	$(CXX) -G -o $(SHAREDTARGET)-gmp.so $(OBJ_GMP) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-gmp.xpt
 
 $(SHAREDTARGET)-dvx: $(OBJ_DVX)
-	$(CXX) -shared -o $(SHAREDTARGET)-dvx.so $(OBJ_DVX) $(CFLAGS) $(LDFLAGS)
+	$(CXX) -G -o $(SHAREDTARGET)-dvx.so $(OBJ_DVX) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-dvx.xpt
 
 np_entry.o: plugingate/np_entry.cpp
