Index: msynctool-trunk/build/linux/osync_build.py
===================================================================
--- msynctool-trunk/build/linux/osync_build.py	(revision 332)
+++ msynctool-trunk/build/linux/osync_build.py	(working copy)
@@ -31,15 +31,14 @@
 	env.ParseConfig('pkg-config --cflags --libs libxml-2.0')
 	env.ParseConfig('pkg-config --cflags --libs opensync-1.0')
 	env.Append(CCFLAGS = r'-I.')
-	env.Append(CCFLAGS = [r'-Wall', r'-Werror', r'-O2'])
 	env.Append(CCFLAGS = r'-DVERSION="\"' + config.version + r'\""')
 	
 	testenv = env.Copy()
 	testenv.Append(CCFLAGS = r'-I' + testenv.GetLaunchDir() + '/tests')
 
 	if env['enable_rpath'] == 1:
-		testenv.Append(LINKFLAGS = [r'-Wl,--rpath', r'-Wl,' + testenv.GetLaunchDir() + r'/opensync/'])
-		env.Append(LINKFLAGS = [r'-Wl,--rpath', r'-Wl,$prefix/$libsuffix'])
+		testenv.Append(LINKFLAGS = [r'-Wl,-R', r'-Wl,' + testenv.GetLaunchDir() + r'/opensync/'])
+		env.Append(LINKFLAGS = [r'-Wl,-R', r'-Wl,$prefix/$libsuffix'])
 
 	
 	return testenv
