--- mugshot-1.1.56/Makefile-libhippofirefox.am.orig	2007-10-14 21:38:33.026337000 -0400
+++ mugshot-1.1.56/Makefile-libhippofirefox.am	2007-10-14 21:39:00.374671000 -0400
@@ -56,30 +56,30 @@
 libhippofirefox_la_CPPFLAGS =			\
 	$(AM_CPPFLAGS)				\
 	$(LIBHIPPOFIREFOX_CFLAGS)		\
-	-I $(FIREFOX_SRCDIR)/include		\
-	-I $(COMMONSRCDIR)/hippoipc		\
-	-I config/hippo
+	-I$(FIREFOX_SRCDIR)/include		\
+	-I$(COMMONSRCDIR)/hippoipc		\
+	-Iconfig/hippo
 
 # The include of $(GECKO_INCLUDEDIR) in the FULL_GECKO case is for 
 # Ubuntu, which has a mix of the gecko-sdk directory arrangement
 # with including the full set of headers
 if HAVE_FULL_GECKO
 libhippofirefox_la_CPPFLAGS +=			\
-	-I $(GECKO_INCLUDEDIR)			\
-	-I $(GECKO_INCLUDEDIR)/caps		\
-	-I $(GECKO_INCLUDEDIR)/docshell		\
-	-I $(GECKO_INCLUDEDIR)/dom		\
-	-I $(GECKO_INCLUDEDIR)/gfx		\
-	-I $(GECKO_INCLUDEDIR)/js		\
-	-I $(GECKO_INCLUDEDIR)/necko		\
-	-I $(GECKO_INCLUDEDIR)/string		\
-	-I $(GECKO_INCLUDEDIR)/widget		\
-	-I $(GECKO_INCLUDEDIR)/xpcom		\
-	-I $(GECKO_INCLUDEDIR)/xpconnect
+	-I$(GECKO_INCLUDEDIR)			\
+	-I$(GECKO_INCLUDEDIR)/caps		\
+	-I$(GECKO_INCLUDEDIR)/docshell		\
+	-I$(GECKO_INCLUDEDIR)/dom		\
+	-I$(GECKO_INCLUDEDIR)/gfx		\
+	-I$(GECKO_INCLUDEDIR)/js		\
+	-I$(GECKO_INCLUDEDIR)/necko		\
+	-I$(GECKO_INCLUDEDIR)/string		\
+	-I$(GECKO_INCLUDEDIR)/widget		\
+	-I$(GECKO_INCLUDEDIR)/xpcom		\
+	-I$(GECKO_INCLUDEDIR)/xpconnect
 else
 libhippofirefox_la_CPPFLAGS +=			\
-	-I $(GECKO_INCLUDEDIR)			\
-	-I $(GECKO_PRIVATE_INCLUDEDIR)
+	-I$(GECKO_INCLUDEDIR)			\
+	-I$(GECKO_PRIVATE_INCLUDEDIR)
 endif
 
 nodist_libhippofirefox_la_SOURCES=			\
@@ -99,12 +99,12 @@
 
 if HAVE_FULL_GECKO
 XPIDL_INCLUDES =				\
-	-I $(GECKO_IDLDIR)			\
-	-I $(FIREFOX_SRCDIR)/public
+	-I$(GECKO_IDLDIR)			\
+	-I$(FIREFOX_SRCDIR)/public
 else
 XPIDL_INCLUDES =				\
-	-I $(GECKO_IDLDIR)			\
-	-I $(FIREFOX_SRCDIR)/public
+	-I$(GECKO_IDLDIR)			\
+	-I$(FIREFOX_SRCDIR)/public
 endif
 
 IDL_FILES =							\
--- mugshot-1.1.56/Makefile-libhippoipc.am.orig	2007-10-14 21:45:28.763275000 -0400
+++ mugshot-1.1.56/Makefile-libhippoipc.am	2007-10-14 21:45:41.432192000 -0400
@@ -3,7 +3,7 @@
 libhippoipc_la_CPPFLAGS =			\
 	$(AM_CPPFLAGS)				\
 	$(LIBHIPPOIPC_CFLAGS)			\
-	-I $(COMMONSRCDIR)/hippoipc
+	-I$(COMMONSRCDIR)/hippoipc
 libhippoipc_la_LIBADD =
 
 nodist_libhippoipc_la_SOURCES =					\
--- mugshot-1.1.56/Makefile.am.orig	2007-10-14 21:46:10.340767000 -0400
+++ mugshot-1.1.56/Makefile.am	2007-10-14 21:46:16.778737000 -0400
@@ -52,8 +52,8 @@
 
 if USE_INCLUDED_CANVAS
 CANVAS_CFLAGS =				\
-	-I $(CANVASSRCDIR)/common	\
-	-I $(CANVASSRCDIR)/linux
+	-I$(CANVASSRCDIR)/common	\
+	-I$(CANVASSRCDIR)/linux
 CANVAS_LIBADD= libhippocanvas-1.la
 
 include Makefile-libhippocanvas.am
--- mugshot-1.1.56/Makefile-mugshot.am.orig	2007-10-14 21:46:28.998151000 -0400
+++ mugshot-1.1.56/Makefile-mugshot.am	2007-10-14 21:46:37.467872000 -0400
@@ -2,7 +2,7 @@
 
 mugshot_CPPFLAGS = 			\
 	$(AM_CPPFLAGS)			\
-	-I $(top_srcdir)		\
+	-I$(top_srcdir)		\
 	$(MUGSHOT_CFLAGS)		\
 	$(CANVAS_CFLAGS)		\
        -DHOST_CPU=\"$(host_cpu)\"	\
--- mugshot-1.1.56/Makefile-libddm.am.orig	2007-10-14 21:47:00.426244000 -0400
+++ mugshot-1.1.56/Makefile-libddm.am	2007-10-14 21:47:05.261093000 -0400
@@ -6,7 +6,7 @@
 
 libddm_1_la_CPPFLAGS =			\
 	$(AM_CPPFLAGS)			\
-	-I $(MARSHAL_DIR)		\
+	-I$(MARSHAL_DIR)		\
 	$(LIBDDM_CFLAGS)		\
 	 -DDDM_COMPILATION=1
 
--- mugshot-1.1.56/Makefile-libhippo.am.orig	2007-10-14 21:47:22.326299000 -0400
+++ mugshot-1.1.56/Makefile-libhippo.am	2007-10-14 21:47:33.719886000 -0400
@@ -2,7 +2,7 @@
 
 libhippo_la_CPPFLAGS =			\
 	$(AM_CPPFLAGS)			\
-	-I $(MARSHAL_DIR)		\
+	-I$(MARSHAL_DIR)		\
 	$(CANVAS_CFLAGS)		\
 	$(LIBHIPPO_CFLAGS)		\
 	-DDDM_I_KNOW_THIS_IS_UNSTABLE=1
--- mugshot-1.1.56/Makefile-libhippocanvas.am.orig	2007-10-14 21:47:59.745090000 -0400
+++ mugshot-1.1.56/Makefile-libhippocanvas.am	2007-10-14 21:48:03.375862000 -0400
@@ -4,7 +4,7 @@
 
 libhippocanvas_1_la_CPPFLAGS =		\
 	$(AM_CPPFLAGS)			\
-	-I $(MARSHAL_DIR)		\
+	-I$(MARSHAL_DIR)		\
 	$(CANVAS_CFLAGS)		\
 	$(LIBHIPPOCANVAS_CFLAGS)
 
--- mugshot-1.1.56/Makefile-test-hippo-ipc.am.orig	2007-10-14 21:48:15.639587000 -0400
+++ mugshot-1.1.56/Makefile-test-hippo-ipc.am	2007-10-14 21:48:23.078698000 -0400
@@ -2,7 +2,7 @@
 
 test_hippo_ipc_CPPFLAGS =		\
 	$(AM_CPPFLAGS)			\
-	-I $(COMMONSRCDIR)/hippoipc	\
+	-I$(COMMONSRCDIR)/hippoipc	\
 	$(TEST_HIPPO_IPC_CFLAGS)
 
 test_hippo_ipc_LDFLAGS=
