--- multitail-5.2.0/makefile.solaris_gcc.orig	2008-01-28 23:59:51.455889322 +0530
+++ multitail-5.2.0/makefile.solaris_gcc	2008-01-29 00:00:35.260524206 +0530
@@ -5,8 +5,8 @@
 
 CC=gcc
 DEBUG=#-g -D_DEBUG #-pg #-fprofile-arcs
-CFLAGS=-O2 -I/usr/local/include/ -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\"
-LDFLAGS=-L/usr/local/lib/ -lsocket -lpanel -lncurses -lnsl -lm $(DEBUG)
+CFLAGS=${EXTRA_CFLAGS} -O2 -I/usr/local/include/ -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\"
+LDFLAGS=${EXTRA_LDFLAGS} -L/usr/local/lib/ -lsocket -lpanel -lncurses -lnsl -lm $(DEBUG)
 
 OBJS=utils.o mt.o error.o my_pty.o term.o scrollback.o help.o mem.o cv.o color.o stripstring.o selbox.o misc.o ui.o exec.o diff.o config.o cmdline.o globals.o history.o
 
@@ -16,17 +16,25 @@
 	$(CC) $(OBJS) $(LDFLAGS) -o multitail
 
 install: multitail
+	/usr/sbin/install -m 0755 -u root -g sys -d $(DESTDIR)/usr
+	/usr/sbin/install -m 0755 -u root -g bin -d $(DESTDIR)/usr/bin
 	cp multitail $(DESTDIR)/usr/bin
-	cp multitail.1 $(DESTDIR)/usr/man/man1/multitail.1
+
+	/usr/sbin/install -m 0755 -u root -g sys -d $(DESTDIR)/usr/share
+	/usr/sbin/install -m 0755 -u root -g bin -d $(DESTDIR)/usr/share/man
+	/usr/sbin/install -m 0755 -u root -g bin -d $(DESTDIR)/usr/share/man/man1
+	cp multitail.1 $(DESTDIR)/usr/share/man/man1/multitail.1
 	#
 	### COPIED multitail.conf.new, YOU NEED TO REPLACE THE multitail.conf
 	### YOURSELF WITH THE NEW FILE
 	#
+	/usr/sbin/install -m 0755 -u root -g sys -d $(DESTDIR)/etc
 	cp multitail.conf $(DESTDIR)/etc/multitail.conf.new
-	mkdir -p $(DESTDIR)/usr/share/doc/multitail-$(VERSION)
+
+	/usr/sbin/install -m 0755 -u root -g other -d $(DESTDIR)/usr/share/doc
+	/usr/sbin/install -m 0755 -u root -g other -d $(DESTDIR)/usr/share/doc/multitail-$(VERSION)
+	#mkdir -p $(DESTDIR)/usr/share/doc/multitail-$(VERSION)
 	cp *.txt INSTALL manual.html $(DESTDIR)/usr/share/doc/multitail-$(VERSION)
-	rm -f $(DESTDIR)/usr/man/man1/multitail.1.gz
-	gzip -9 $(DESTDIR)/usr/man/man1/multitail.1
 	#
 	# +-=-------------------------------------------------------------=-+
 	# | There's a mailinglist!                                          |
--- multitail-5.2.0/makefile.solaris_sunwspro.orig	2008-01-29 00:00:01.294540929 +0530
+++ multitail-5.2.0/makefile.solaris_sunwspro	2008-01-29 00:00:47.116815922 +0530
@@ -4,8 +4,8 @@
 CONFIG_FILE=$(DESTDIR)/etc/multitail.conf
 
 DEBUG=#-g -D_DEBUG #-pg #-fprofile-arcs
-CFLAGS=-O2 -I/usr/local/include/ -D_STDC_C99 -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\"
-LDFLAGS=-L/usr/local/lib/ -lsocket -lpanel -lncurses -lnsl -lm $(DEBUG)
+CFLAGS=${EXTRA_CFLAGS} -O2 -I/usr/local/include/ -D_STDC_C99 -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\"
+LDFLAGS=${EXTRA_LDFLAGS} -L/usr/local/lib/ -lsocket -lpanel -lncurses -lnsl -lm $(DEBUG)
 
 OBJS=utils.o mt.o error.o my_pty.o term.o scrollback.o help.o mem.o cv.o color.o stripstring.o selbox.o misc.o ui.o exec.o diff.o config.o cmdline.o globals.o history.o
 
@@ -15,17 +15,25 @@
 	$(CC) $(OBJS) $(LDFLAGS) -o multitail
 
 install: multitail
+	/usr/sbin/install -m 0755 -u root -g sys -d $(DESTDIR)/usr
+	/usr/sbin/install -m 0755 -u root -g bin -d $(DESTDIR)/usr/bin
 	cp multitail $(DESTDIR)/usr/bin
-	cp multitail.1 $(DESTDIR)/usr/man/man1/multitail.1
+
+	/usr/sbin/install -m 0755 -u root -g sys -d $(DESTDIR)/usr/share
+	/usr/sbin/install -m 0755 -u root -g bin -d $(DESTDIR)/usr/share/man
+	/usr/sbin/install -m 0755 -u root -g bin -d $(DESTDIR)/usr/share/man/man1
+	cp multitail.1 $(DESTDIR)/usr/share/man/man1/multitail.1
 	#
 	### COPIED multitail.conf.new, YOU NEED TO REPLACE THE multitail.conf
 	### YOURSELF WITH THE NEW FILE
 	#
+	/usr/sbin/install -m 0755 -u root -g sys -d $(DESTDIR)/etc
 	cp multitail.conf $(DESTDIR)/etc/multitail.conf.new
+
+	/usr/sbin/install -m 0755 -u root -g other -d $(DESTDIR)/usr/share/doc
+	/usr/sbin/install -m 0755 -u root -g other -d $(DESTDIR)/usr/share/doc/multitail-$(VERSION)
 	mkdir -p $(DESTDIR)/usr/share/doc/multitail-$(VERSION)
 	cp *.txt INSTALL manual.html $(DESTDIR)/usr/share/doc/multitail-$(VERSION)
-	rm -f $(DESTDIR)/usr/man/man1/multitail.1.gz
-	gzip -9 $(DESTDIR)/usr/man/man1/multitail.1
 	#
 	# +-=-------------------------------------------------------------=-+
 	# | There's a mailinglist!                                          |
