--- mutt-1.4.2.2/Makefile.in-p1	2006-07-14 20:38:58.000000000 +0200
+++ mutt-1.4.2.2/Makefile.in	2007-05-21 13:41:40.422427000 +0200
@@ -1000,7 +1000,7 @@
 		$(INSTALL) -m 644 $(srcdir)/Muttrc $(DESTDIR)$(sysconfdir) ; \
 	fi
 	-if [ ! -f $(DESTDIR)$(sysconfdir)/mime.types ]; then \
-		$(INSTALL) -m 644 $(srcdir)/mime.types $(DESTDIR)$(sysconfdir); \
+		$(INSTALL) -m 644 $(srcdir)/mime.types $(DESTDIR)$(sysconfdir)/mutt-mime.types; \
 	fi
 
 # Don't make this one ChangeLog - it's intended to be
