--- rf-2.4.0/adddrv.sh.orig	2008-02-09 23:46:22.963959755 +0530
+++ rf-2.4.0/adddrv.sh	2008-02-09 23:46:44.153664114 +0530
@@ -35,27 +35,5 @@
 #  Corega Inc,
 set DEVLIST = ( $DEVLIST '"pci1259,a11e"' )
 
-#echo $DEVLIST
-
-set DEVLIST2 = ( )
-foreach i ($DEVLIST)
-	set pcidev = `grep $i /etc/driver_aliases`
-	if ("$pcidev" == "") then
-		set DEVLIST2 = ( $DEVLIST2 "$i" )
-	endif
-end
-
-#echo $DEVLIST2
-if ("$DEVLIST2" == "") then
-	echo nothing to do.
-	exit 1
-endif
-
-set existing = `grep "rf " /etc/driver_aliases`
-echo $existing
-if ("$existing" == "") then
-	/usr/sbin/add_drv -n -v -m '* 0600 root sys' -i "$DEVLIST2" rf
-else 
-	/usr/sbin/update_drv -a -v -m '* 0600 root sys' -i "$DEVLIST2" rf
-endif
+/usr/sbin/add_drv -n -v -m '* 0600 root sys' -i "$DEVLIST" rf
 sync
