--- alta-2.6.0/gem.h.orig	Sat Feb  2 21:41:18 2008
+++ alta-2.6.0/gem.h	Sat Mar 22 23:00:26 2008
@@ -672,7 +672,15 @@
 int gem_pci_set_power_state(dev_info_t *, ddi_acc_handle_t, uint_t);
 int gem_pci_regs_map_setup(dev_info_t *, uint32_t, uint32_t,
 	struct ddi_device_acc_attr *, caddr_t *, ddi_acc_handle_t *);
-#ifndef GEM_CONFIG_GLDv3
+#ifdef GEM_CONFIG_GLDv3
+#define	gem_getinfo	nulldev
+#define	gem_open	NULL
+#define	gem_close	NULL
+#define	gem_wput	NULL
+#define	gem_wsrv	NULL
+#define	gem_rsrv	NULL
+#define	gem_power	NULL
+#else
 #define	gem_getinfo	gld_getinfo
 #define	gem_open	gld_open
 #define	gem_close	gld_close
