--- ocaml-3.09.2/otherlibs/systhreads/Makefile.ldflags	2006-07-15 19:05:08.827315000 -0400
+++ ocaml-3.09.2/otherlibs/systhreads/Makefile	2006-07-15 19:05:51.001367000 -0400
@@ -32,7 +32,7 @@
 allopt: libthreadsnat.a threads.cmxa
 
 libthreads.a: $(BYTECODE_C_OBJS)
-	$(MKLIB) -o threads $(BYTECODE_C_OBJS)
+	$(MKLIB) -ldopt "$(BYTECCLIBS)" -o threads $(BYTECODE_C_OBJS)
 
 posix_b.o: posix.c
 	$(BYTECC) -O -I../../byterun $(BYTECCCOMPOPTS) $(SHAREDCCCOMPOPTS) \
--- ocaml-3.09.2/otherlibs/threads/Makefile.ldflags	2004-11-29 09:53:32.000000000 -0500
+++ ocaml-3.09.2/otherlibs/threads/Makefile	2006-07-15 18:43:14.116772000 -0400
@@ -53,7 +53,7 @@
 allopt:
 
 libvmthreads.a: $(C_OBJS)
-	$(MKLIB) -o threads -oc vmthreads $(C_OBJS)
+	$(MKLIB) -ldopt "$(BYTECCLIBS)" -o threads -oc vmthreads $(C_OBJS)
 
 threads.cma: $(CAML_OBJS)
 	$(MKLIB) -ocamlc '$(CAMLC)' -o threads -oc vmthreads $(CAML_OBJS)
--- ocaml-3.09.2/otherlibs/unix/Makefile.ldflags	2004-11-29 09:53:32.000000000 -0500
+++ ocaml-3.09.2/otherlibs/unix/Makefile	2006-07-15 18:43:14.117178000 -0400
@@ -49,7 +49,7 @@
 allopt: libunix.a unix.cmxa
 
 libunix.a: $(OBJS)
-	$(MKLIB) -o unix $(OBJS)
+	$(MKLIB) -ldopt "$(BYTECCLIBS)" -o unix $(OBJS) $(LIBS)
 
 unix.cma: $(MLOBJS)
 	$(MKLIB) -o unix -ocamlc '$(CAMLC)' -linkall $(MLOBJS)
