diff -ur ode-060223-orig/ode/src/Makefile.am ode-060223/ode/src/Makefile.am
--- ode-060223-orig/ode/src/Makefile.am	2005-12-03 05:01:55.000000000 +0700
+++ ode-060223/ode/src/Makefile.am	2007-05-06 20:02:11.381298518 +0700
@@ -2,7 +2,7 @@
 AM_CPPFLAGS = @ARCHFLAGS@ -I$(top_srcdir)/include -I$(top_builddir)/include
 AM_CFLAGS = @ARCHFLAGS@ -I$(topsrcdir)/include -I$(top_builddir)/include
 lib_LIBRARIES = libode.a
-libode_a_CPPFLAGS = -O2
+libode_a_CPPFLAGS = 
 
 # Fake an executable in order to get a shared library
 # Note the elegant and cunning way to trick Autotools to install a program
@@ -11,7 +11,7 @@
 EXEEXT=@so_ext@
 traplib_PROGRAMS=libode
 libode_DEPENDENCIES = libfast.a libode.a
-libode_LDFLAGS=-shared -fPIC
+libode_LDFLAGS=-G -KPIC
 libode_SOURCES=$(libode_a_SOURCES) $(libfast_a_SOURCES)
 if OPCODE
 libode_DEPENDENCIES+= libOPCODE.a
@@ -20,7 +20,7 @@
 
 # convenience library to simulate per object cflags
 noinst_LIBRARIES= libfast.a
-libfast_a_CFLAGS= -O1
+libfast_a_CFLAGS= $(CFLAGS)
 libfast_a_SOURCES= fastldlt.c fastltsolve.c fastdot.c fastlsolve.c
 
 libode_a_DEPENDENCIES = libfast.a
@@ -71,7 +71,7 @@
 
 if OPCODE
 noinst_LIBRARIES+= libOPCODE.a
-libOPCODE_a_CPPFLAGS= -O1
+libOPCODE_a_CPPFLAGS= 
 libOPCODE_a_SOURCES=	@TOPDIR@/OPCODE/OPC_AABBCollider.cpp \
 			@TOPDIR@/OPCODE/OPC_AABBTree.cpp \
 			@TOPDIR@/OPCODE/OPC_BaseModel.cpp \
