diff -ur opencity-0.0.4stable-orig/Makefile.am opencity-0.0.4stable/Makefile.am
--- opencity-0.0.4stable-orig/Makefile.am	2006-10-26 01:51:32.000000000 +0700
+++ opencity-0.0.4stable/Makefile.am	2007-04-22 05:06:06.605921637 +0700
@@ -43,10 +43,10 @@
 			cp -r $(srcdir)/$$data/* $(DESTDIR)$(pkgdatadir)/$$data/; \
 		fi; \
 	done;
-#	$(mkinstalldirs) $(DESTDIR)$(datadir)/pixmaps/;
-#	$(INSTALL_DATA) $(srcdir)/OpenCity.png $(DESTDIR)$(datadir)/pixmaps/;
-#	$(mkinstalldirs) $(DESTDIR)$(datadir)/applications/;
-#	$(INSTALL_DATA) $(srcdir)/OpenCity.desktop $(DESTDIR)$(datadir)/applications/;
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/pixmaps/;
+	$(INSTALL_DATA) $(srcdir)/OpenCity.png $(DESTDIR)$(datadir)/pixmaps/;
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/applications/;
+	$(INSTALL_DATA) $(srcdir)/OpenCity.desktop $(DESTDIR)$(datadir)/applications/;
 
 	@echo -e "\n\r";
 	@echo "========================================================================";
diff -ur opencity-0.0.4stable-orig/Makefile.in opencity-0.0.4stable/Makefile.in
--- opencity-0.0.4stable-orig/Makefile.in	2006-10-26 02:01:07.000000000 +0700
+++ opencity-0.0.4stable/Makefile.in	2007-04-22 05:08:14.616772512 +0700
@@ -691,10 +691,10 @@
 			cp -r $(srcdir)/$$data/* $(DESTDIR)$(pkgdatadir)/$$data/; \
 		fi; \
 	done;
-#	$(mkinstalldirs) $(DESTDIR)$(datadir)/pixmaps/;
-#	$(INSTALL_DATA) $(srcdir)/OpenCity.png $(DESTDIR)$(datadir)/pixmaps/;
-#	$(mkinstalldirs) $(DESTDIR)$(datadir)/applications/;
-#	$(INSTALL_DATA) $(srcdir)/OpenCity.desktop $(DESTDIR)$(datadir)/applications/;
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/pixmaps/;
+	$(INSTALL_DATA) $(srcdir)/OpenCity.png $(DESTDIR)$(datadir)/pixmaps/;
+	$(mkinstalldirs) $(DESTDIR)$(datadir)/applications/;
+	$(INSTALL_DATA) $(srcdir)/OpenCity.desktop $(DESTDIR)$(datadir)/applications/;
 
 	@echo -e "\n\r";
 	@echo "========================================================================";
diff -ur opencity-0.0.4stable-orig/OpenCity.desktop opencity-0.0.4stable/OpenCity.desktop
--- opencity-0.0.4stable-orig/OpenCity.desktop	2006-10-08 17:57:40.000000000 +0700
+++ opencity-0.0.4stable/OpenCity.desktop	2007-04-22 05:09:33.458165319 +0700
@@ -10,4 +10,4 @@
 Name=OpenCity
 Name[fr]=OpenCity
 Terminal=false
-Type=Application
\ No newline at end of file
+Type=Application
