diff -ur opencity-0.0.4stable-orig/src/main.cpp opencity-0.0.4stable/src/main.cpp
--- opencity-0.0.4stable-orig/src/main.cpp	2006-10-26 01:51:30.000000000 +0700
+++ opencity-0.0.4stable/src/main.cpp	2007-04-22 05:37:03.251736368 +0700
@@ -706,22 +706,7 @@
 
 // IF the homedir is not set THEN try to get it from BinReloc routines
 	if (gsHomeDir == "") {
-	// Init the BinReloc routines
-		if (br_init(&brError) != 1) {
-			OPENCITY_INFO(
-				"The initialization of BinReloc routines has failed." << endl
-				 << "The error was: " << brError
-			);
-		}
-		else {
-		// Construct the datadir from the prefix
-			pTemp = br_find_prefix( PREFIX );
-			gsHomeDir = pTemp;
-			gsHomeDir += "/share/";
-			gsHomeDir += PACKAGE;
-			free(pTemp);
-			formatHomeDir();
-		}
+		gsHomeDir = "/usr/share/opencity/";
 	}
 
 // IF the save directory is not set the find it
