diff -Naur pidgin-musictracker-0.4.11.old/src/Makefile.am pidgin-musictracker-0.4.11/src/Makefile.am
--- pidgin-musictracker-0.4.11.old/src/Makefile.am	2008-10-28 19:37:36.736299539 +0200
+++ pidgin-musictracker-0.4.11/src/Makefile.am	2008-10-28 19:39:03.965615653 +0200
@@ -36,8 +36,6 @@
 
 AM_CPPFLAGS = \
         -DDATADIR=\"$(PIDGIN_DATADIR)\" \
-	-Werror -W -Wall -Wno-unused-parameter \
-	-std=gnu99 \
         $(PIDGIN_CFLAGS) \
         $(GLIB_CFLAGS) \
 		$(GTK_CFLAGS) \
diff -Naur pidgin-musictracker-0.4.11.old/src/Makefile.in pidgin-musictracker-0.4.11/src/Makefile.in
--- pidgin-musictracker-0.4.11.old/src/Makefile.in	2008-10-28 19:37:36.741721549 +0200
+++ pidgin-musictracker-0.4.11/src/Makefile.in	2008-10-28 19:39:22.279453020 +0200
@@ -240,8 +240,6 @@
 musictracker_la_LIBADD = $(GLIB_LIBS) $(GTK_LIBS) $(DBUS_LIBS)
 AM_CPPFLAGS = \
         -DDATADIR=\"$(PIDGIN_DATADIR)\" \
-	-Werror -W -Wall -Wno-unused-parameter \
-	-std=gnu99 \
         $(PIDGIN_CFLAGS) \
         $(GLIB_CFLAGS) \
 		$(GTK_CFLAGS) \
diff -Naur pidgin-musictracker-0.4.11.old/src/musictracker.c pidgin-musictracker-0.4.11/src/musictracker.c
--- pidgin-musictracker-0.4.11.old/src/musictracker.c	2008-10-28 19:37:36.745567028 +0200
+++ pidgin-musictracker-0.4.11/src/musictracker.c	2008-10-29 03:03:51.461376140 +0200
@@ -236,7 +236,13 @@
 	status = put_field(status, 'b', buf);
 
         // Music symbol: U+266B 'beamed eighth notes'
-	status = put_field(status, 'm', "\u266b");
+        char symbol[4];
+        symbol[0] = 0xE2;
+        symbol[1] = 0x99;
+        symbol[2] = 0xAB;
+        symbol[3] = 0;
+
+	status = put_field(status, 'm', symbol);
 
 	trace("Formatted status: %s", status);
 
diff -Naur pidgin-musictracker-0.4.11.old/src/utils.h pidgin-musictracker-0.4.11/src/utils.h
--- pidgin-musictracker-0.4.11.old/src/utils.h	2008-10-28 19:37:36.736001356 +0200
+++ pidgin-musictracker-0.4.11/src/utils.h	2008-10-28 19:39:43.253980951 +0200
@@ -2,7 +2,7 @@
 #define _UTILS_H_
 
 #include <stdio.h>
-#include <pcre.h>
+#include <pcre/pcre.h>
 
 #ifndef WIN32
 #include <dbus/dbus-glib.h>
diff -Naur pidgin-musictracker-0.4.11.old/src/xmmsctrl.c pidgin-musictracker-0.4.11/src/xmmsctrl.c
--- pidgin-musictracker-0.4.11.old/src/xmmsctrl.c	2008-10-28 19:37:36.741332515 +0200
+++ pidgin-musictracker-0.4.11/src/xmmsctrl.c	2008-10-28 23:04:08.704020563 +0200
@@ -72,9 +72,9 @@
 			}
 
                         char regexp[100];
-                        sprintf(regexp, "^(.*)\\%s(.*)\\%s(.*)$", sep, sep);
+                        sprintf(regexp, "^(.*)\\%s(.*)$", sep, sep);
                         pcre *re = regex(regexp, 0);
-                        capture(re, title, strlen(title), ti->artist, ti->album, ti->track);
+                        capture(re, title, strlen(title), ti->artist, ti->track);
                         pcre_free(re);
 		}
 
