--- pigment-python-0.3.1/pgm/Makefile.am-orig	2008-01-16 19:38:36.364803000 -0600
+++ pigment-python-0.3.1/pgm/Makefile.am	2008-01-16 19:41:15.333246000 -0600
@@ -33,7 +33,7 @@ pyexec_LTLIBRARIES = _pgmmodule.la
 pgm.c: pgm.defs $(overrides)
 
 _pgmmodule_la_LDFLAGS = \
-	@PYPGM_LDFLAGS@ \
+	@PYPGM_LDFLAGS@ -lpigment-0.3 \
 	-module -avoid-version \
 	-export-symbols-regex "^(init_pgm|_PyGObject_API).*"
 
