--- pigment-0.3.2/libs/pypgm/pgm/Makefile.am-orig	2007-10-05 15:09:06.269633000 -0500
+++ pigment-0.3.2/libs/pypgm/pgm/Makefile.am	2007-10-05 15:09:32.749681000 -0500
@@ -18,7 +18,7 @@ pgm.c: pgm.defs $(overrides)
 _pgmmodule_la_LDFLAGS = \
 	@PY_PGM_LDFLAGS@ \
 	-module -avoid-version \
-	-export-symbols-regex init_pgm
+	-export-symbols-regex init_pgm $(top_builddir)/pgm/libpigment-0.3.la
 
 _pgmmodule_la_CFLAGS = \
 	@PY_PGM_CFLAGS@
