--- planner-0.14/python/Makefile.am.orig	2006-10-13 16:23:46.014780000 -0400
+++ planner-0.14/python/Makefile.am	2006-10-13 16:23:57.319015000 -0400
@@ -10,13 +10,13 @@
 
 pkgpyexec_LTLIBRARIES = planner.la plannerui.la
 
-planner_la_CFLAGS = -Wno-missing-declarations -Wno-missing-prototypes
+planner_la_CFLAGS = 
 planner_la_LDFLAGS = -module -avoid-version -export-symbols-regex initplanner
 planner_la_LIBADD = $(LIBPLANNER_LIBS) $(top_builddir)/libplanner/libplanner-1.la
 planner_la_SOURCES = plannermodule.c
 nodist_planner_la_SOURCES = planner.c
 
-plannerui_la_CFLAGS = -Wno-missing-declarations -Wno-missing-prototypes
+plannerui_la_CFLAGS = 
 plannerui_la_LDFLAGS = -module -avoid-version -export-symbols-regex initplannerui
 plannerui_la_LIBADD = $(PLANNER_LIBS) $(top_builddir)/libplanner/libplanner-1.la
 plannerui_la_SOURCES = planneruimodule.c
