diff -ur portaudio-orig/src/hostapi/oss/pa_unix_oss.c portaudio/src/hostapi/oss/pa_unix_oss.c
--- portaudio-orig/src/hostapi/oss/pa_unix_oss.c	2006-08-30 03:56:04.000000000 +0700
+++ portaudio/src/hostapi/oss/pa_unix_oss.c	2007-05-01 17:45:38.721347869 +0700
@@ -63,7 +63,7 @@
 #include <limits.h>
 #include <semaphore.h>
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__sun)
 # include <sys/soundcard.h>
 # define DEVICE_NAME_BASE            "/dev/dsp"
 #elif defined __linux__
