--- /opt/jdsbld/packages/PKGS/SFEpulseaudio-root/reloc/etc/pulse/default.pa	Tue Sep  4 00:47:33 2007
+++ default.pa	Tue Sep  4 14:57:38 2007
@@ -27,11 +27,11 @@
 #load-module module-pipe-sink
 
 ### Automatically load driver modules depending on the hardware available
-load-module module-hal-detect
+#load-module module-hal-detect
 
 ### Alternatively use the static hardware detection module (for systems that
 ### lack HAL support
-#load-module module-detect
+load-module module-detect
 
 ### Load audio drivers automatically on access
 #add-autoload-sink output module-oss device="/dev/dsp" sink_name=output source_name=input
@@ -42,13 +42,15 @@
 #add-autoload-source input module-alsa-source source_name=input
 
 ### Load several protocols
-load-module module-esound-protocol-unix
+#load-module module-esound-protocol-unix
 load-module module-native-protocol-unix
 
 ### Network access (may be configured with paprefs, so leave this commented
 ### here if you plan to use paprefs)
-#load-module module-esound-protocol-tcp
-#load-module module-native-protocol-tcp
+load-module module-esound-protocol-tcp auth-ip-acl=127.0.0.1
+load-module module-native-protocol-tcp auth-ip-acl=127.0.0.1
+#load-module module-esound-protocol-tcp auth-ip-acl=127.0.0.1;192.168.0.0/16
+#load-module module-native-protocol-tcp auth-ip-acl=127.0.0.1;192.168.0.0/16
 #load-module module-zeroconf-publish
 
 ### Load the RTP reciever module (also configured via paprefs, see above)
