diff -r 84c6816a6e2b src/transform/core.c
--- a/src/transform/core.c	Sun Oct 14 23:51:56 2007 +0530
+++ b/src/transform/core.c	Sun Oct 14 23:52:01 2007 +0530
@@ -78,7 +78,7 @@ PyGSL_transform_helpers_alloc(PyObject *
         ((helpers != NULL) && ((helpers->free_table != 0) &&  (helpers->free_space != 0)) ? \
         PyGSL_transform_helpers_free(helpers) : 0)
 
-static void
+static int
 PyGSL_transform_helpers_free(struct _pygsl_transform_help_rf_s * h)
 {
 	FUNC_MESS_BEGIN();
@@ -101,6 +101,7 @@ PyGSL_transform_helpers_free(struct _pyg
 		h->free_space = 0;
 	}
 	FUNC_MESS_END();
+    return GSL_SUCCESS;
 }
 
 #ifdef _PyGSL_HAS_WAVELET
