diff -ru qt-x11-opensource-src-4.4.3.orig/src/3rdparty/kdebase/runtime/phonon/gstreamer/artssink.cpp qt-x11-opensource-src-4.4.3/src/3rdparty/kdebase/runtime/phonon/gstreamer/artssink.cpp
--- qt-x11-opensource-src-4.4.3.orig/src/3rdparty/kdebase/runtime/phonon/gstreamer/artssink.cpp	2008-09-27 10:58:44.000000000 +0200
+++ qt-x11-opensource-src-4.4.3/src/3rdparty/kdebase/runtime/phonon/gstreamer/artssink.cpp	2008-11-29 02:02:06.175962000 +0100
@@ -26,6 +26,10 @@
 #include <gst/audio/gstaudiosink.h>
 #include "artssink.h"
 
+/* because of Sun C++ , -feature=extensions... */
+
+#define __func__ ((const char*) ("???"))
+
 QT_BEGIN_NAMESPACE
 
 namespace Phonon
