diff -urN ../qtpfsgui-1.9.2.x/project.pro ./project.pro
--- ../qtpfsgui-1.9.2.x/project.pro	2008-04-30 16:23:36.000000000 +0200
+++ ./project.pro	2008-05-25 23:11:47.457289598 +0200
@@ -2,7 +2,7 @@
 CONFIG += release qt thread
 DEFINES += QT_NO_DEBUG_OUTPUT
 
-QMAKE_CXXFLAGS += -funroll-loops -fstrength-reduce -fschedule-insns2 -felide-constructors -frerun-loop-opt -fexceptions -fno-strict-aliasing -fexpensive-optimizations -ffast-math -pipe
+QMAKE_CXXFLAGS += -funroll-loops -fstrength-reduce -fschedule-insns2 -felide-constructors -frerun-loop-opt -fexceptions -fno-strict-aliasing -fexpensive-optimizations -ffast-math -pipe -DSOLARIS
 
 TARGET = qtpfsgui
 
@@ -185,7 +185,7 @@
 message ( "Detecting exiv2:" )
 #I think these are the only paths where we have to search for.
 #If your system is more exotic let me know.
-EXIV2IMAGEHPP = /usr/include/exiv2/image.hpp /usr/local/include/exiv2/image.hpp $$(LOCALSOFT)/include/exiv2/image.hpp 
+EXIV2IMAGEHPP = /usr/include/exiv2/image.hpp /usr/local/include/exiv2/image.hpp $$LOCALSOFT/include/exiv2/image.hpp 
 for(path, EXIV2IMAGEHPP) {
 	exists($$path) {
 		EXIV2PATH = $$dirname(path)
@@ -203,7 +203,7 @@
 	error( "fatal error, bailing out." )
 }
 INCLUDEPATH	*= $$EXIV2PATH
-LIBS		+= -lexiv2
+LIBS		+= -R$$LOCALSOFT/lib -L$$LOCALSOFT/lib -lexiv2
 
 ########################################### OPENEXR ###########################################
 #openEXR dependencies
diff -urN ../qtpfsgui-1.9.2.x/src/MainWindow/hdrviewer.cpp ./src/MainWindow/hdrviewer.cpp
--- ../qtpfsgui-1.9.2.x/src/MainWindow/hdrviewer.cpp	2008-02-16 11:01:25.000000000 +0100
+++ ./src/MainWindow/hdrviewer.cpp	2008-05-25 21:18:53.063130192 +0200
@@ -23,6 +23,7 @@
  */
 
 #include <math.h>
+#include <ieeefp.h>
 #include <QApplication>
 #include <QVBoxLayout>
 #include <QMessageBox>
diff -urN ../qtpfsgui-1.9.1.x/src/Common/commandline.cpp ./src/Common/commandline.cpp
--- ../qtpfsgui-1.9.1.x/src/Common/commandline.cpp	2008-03-02 21:13:07.000000000 +0100
+++ ./src/Common/commandline.cpp	2008-05-26 00:15:01.708058493 +0200
@@ -36,7 +36,7 @@
 void writeRGBEfile (pfs::Frame* inputpfsframe, const char* outfilename);
 void writeEXRfile  (pfs::Frame* inputpfsframe, const char* outfilename);
 
-#if defined(__FreeBSD__) || defined(WIN32)
+#if defined(__FreeBSD__) || defined(WIN32) || defined(SOLARIS)
 #define error(Z) { fprintf(stderr,Z); exit(1); }
 #else
 #include <error.h>

