--- libtorrent-0.12.2.orig/scripts/common.m4	2008-05-07 08:19:12.000000000 -0400
+++ libtorrent-0.12.2/scripts/common.m4	2008-06-23 17:04:24.567475080 -0400
@@ -170,7 +170,7 @@
   AC_COMPILE_IFELSE(
     [[#include <sys/types.h>
           #include <sys/mman.h>
-          void f() { static char test[1024]; madvise((void *)test, sizeof(test), MADV_NORMAL); }
+          void f() { static char test[1024]; madvise((char *)test, sizeof(test), MADV_NORMAL); }
     ]],
     [
       AC_MSG_RESULT(yes)
