diff -ur ruby-1.8.6-orig/configure.in ruby-1.8.6/configure.in
--- ruby-1.8.6-orig/configure.in	2007-02-28 20:23:42.000000000 +0700
+++ ruby-1.8.6/configure.in	2007-04-21 11:14:03.714251484 +0700
@@ -459,6 +459,9 @@
                       ac_cv_func_isinf=yes
               fi
                LIBS="-lm $LIBS"
+	      ;;
+solaris*2.11) ac_cv_func_isinf=yes
+               LIBS="-lm $LIBS"
               ;;
 *) LIBS="-lm $LIBS";;
 esac
