--- sed-4.1.5/configure.ac.patch01	2007-01-06 17:53:05.079145000 -0500
+++ sed-4.1.5/configure.ac	2007-01-06 18:16:26.184937000 -0500
@@ -142,20 +142,16 @@
 
 
 AC_DEFUN([AM_MKINSTALLDIRS], [MKINSTALLDIRS="$mkdir_p" AC_SUBST(MKINSTALLDIRS)])
-AM_GNU_GETTEXT_VERSION(0.14)
-AM_GNU_GETTEXT(no-libtool, need-ngettext, ../intl)
-AC_CONFIG_COMMANDS([gettext-fix], [
-  sed -e '/^mkinstalldirs *=/a\' \
-      -e "install_sh=$install_sh" \
-      -e 's/^mkinstalldirs *=.*/mkinstalldirs=$(MKINSTALLDIRS)/' \
-      intl/Makefile > intl/Makefile.tmp
-  mv intl/Makefile.tmp intl/Makefile
-  sed -e '/^mkinstalldirs *=/a\' \
-      -e "install_sh=$install_sh" \
-      -e 's/^mkinstalldirs *=.*/mkinstalldirs=$(MKINSTALLDIRS)/' \
-      po/Makefile > po/Makefile.tmp
-  mv po/Makefile.tmp po/Makefile], [
-  install_sh="$install_sh"])
+LIBINTL=
+ALL_LINGUAS=`cat po/LINGUAS`
+AM_GLIB_GNU_GETTEXT
+USE_INCLUDED_LIBINTL=no
+AC_SUBST(USE_INCLUDED_LIBINTL)
+BUILD_INCLUDED_LIBINTL=no
+AC_SUBST(BUILD_INCLUDED_LIBINTL)
+AC_SUBST(LIBINTL)
+GETTEXT_PACKAGE=$PACKAGE
+AC_SUBST(GETTEXT_PACKAGE)
 
 AC_CONFIG_FILES([bootstrap.sh], chmod +x bootstrap.sh)
 AC_CONFIG_FILES([Makefile doc/Makefile \
--- sed-4.1.5/Makefile.am.patch01	2007-01-06 17:57:17.244798000 -0500
+++ sed-4.1.5/Makefile.am	2007-01-06 17:57:25.115824000 -0500
@@ -6,7 +6,7 @@
 
 PACKAGE = sed
 
-SUBDIRS = intl lib po sed doc testsuite
+SUBDIRS = lib po sed doc testsuite
 
 noinst_DATA = bootstrap.sh
 noinst_HEADERS = basicdefs.h
