Index: songbird0.6/app/branding/songbird.desktop
===================================================================
--- songbird0.6/app/branding/songbird.desktop	(revision 0)
+++ songbird0.6/app/branding/songbird.desktop	(revision 0)
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Songbird
+Comment=The desktop media player mashed-up with the Web
+Exec=songbird
+Icon=songbird.png
+Terminal=false
+Type=Application
+Categories=Application;AudioVideo;
+StartupNotify=true
