--- sox-14.0.1.old/src/gsm.c	2007-11-04 18:28:29.000000000 +0200
+++ sox-14.0.1/src/gsm.c	2008-06-30 00:44:45.203647244 +0300
@@ -29,7 +29,7 @@
 #include "sox_i.h"
 
 #ifdef EXTERNAL_GSM
-#include <gsm/gsm.h>
+#include <gsm.h>
 #else
 #include "../libgsm/gsm.h"
 #endif
--- sox-14.0.1.old/src/wav.c	2007-11-04 18:28:31.000000000 +0200
+++ sox-14.0.1/src/wav.c	2008-06-30 00:47:05.928523045 +0300
@@ -24,7 +24,7 @@
 #include "ima_rw.h"
 #include "adpcm.h"
 #ifdef EXTERNAL_GSM
-#include <gsm/gsm.h>
+#include <gsm.h>
 #else
 #include "../libgsm/gsm.h"
 #endif
