--- sqlite-3.3.6/src/os_unix.c.orig	2006-07-25 10:02:58.005134000 -0400
+++ sqlite-3.3.6/src/os_unix.c	2006-07-25 10:05:16.861810000 -0400
@@ -462,6 +462,9 @@
 ** threadsOverrideEachOthersLocks variable appropriately.
 */
 static void testThreadLockingBehavior(int fd_orig){
+#if defined (sun) && defined (__SVR4)
+  threadsOverrideEachOthersLocks =  1;
+#else
   int fd;
   struct threadTestData d[2];
   pthread_t t[2];
@@ -482,6 +485,7 @@
   pthread_join(t[1], 0);
   close(fd);
   threadsOverrideEachOthersLocks =  d[0].result==0 && d[1].result==0;
+#endif
 }
 #endif /* SQLITE_UNIX_THREADS */
 
