--- tea-17.1.1/configure.in.orig	2007-07-29 17:30:48.865754727 +0700
+++ tea-17.1.1/configure.in	2007-07-29 17:31:29.207815342 +0700
@@ -53,9 +53,6 @@
 
 AM_CONDITIONAL(FINK, test "x$OSTYPE" = "xDARWIN")
 
-dnl CFLAGS="$CFLAGS -w -s -pipe"
-dnl CFLAGS="$CFLAGS -s -pipe"
-
 
 AC_ARG_ENABLE(hacking_mode,
 [  --enable-hacking
@@ -65,17 +62,6 @@
 fi
 
 
-dnl Add DARWIN flag for Mac OS X
-if test "x$OSTYPE" = "xDARWIN";then
-	CFLAGS="$CFLAGS -DDARWIN -pipe"
-else
-if test -n "$enable_debug"; then
-   CFLAGS="$CFLAGS -w -pipe"
-else
-    CFLAGS="$CFLAGS -w -s -pipe"
-fi
-fi
-
 
 AC_ARG_ENABLE(sounds_mode,
 [  --enable-sounds
