Index: tracker-trunk/src/trackerd/tracker-dbus-methods.c
===================================================================
--- tracker-trunk/src/trackerd/tracker-dbus-methods.c	(revision 1012)
+++ tracker-trunk/src/trackerd/tracker-dbus-methods.c	(working copy)
@@ -490,6 +490,9 @@
 	} else if (strcasecmp (option, "EnableEvolution") == 0) {
 		tracker_log ("evolution support set to %d", value);
 		tracker->index_evolution_emails = value;
+	} else if (strcasecmp (option, "EnableThunderbird") == 0) {
+		tracker_log ("thunderbird support set to %d", value);
+		tracker->index_thunderbird_emails = value;
 	} else if (strcasecmp (option, "FastMerges") == 0) {
 		tracker_log ("Fast merges set to %d", value);
 		tracker->fast_merges = value;
Index: tracker-trunk/src/tracker-preferences/tracker-configuration.c
===================================================================
--- tracker-trunk/src/tracker-preferences/tracker-configuration.c	(revision 1012)
+++ tracker-trunk/src/tracker-preferences/tracker-configuration.c	(working copy)
@@ -362,6 +362,7 @@
 					"# Enables use of language-specific stemmer\n",
 					"EnableStemmer=true\n", "[Emails]\n",
 					"IndexEvolutionEmails=true\n",
+					"IndexThunderbirdEmails=true\n",
 					"[Performance]\n",
 					"# Maximum size of text in bytes to index from a file's text contents\n",
 					"MaxTextToIndex=1048576\n",
Index: tracker-trunk/src/tracker-preferences/tracker-preferences.c
===================================================================
--- tracker-trunk/src/tracker-preferences/tracker-preferences.c	(revision 1012)
+++ tracker-trunk/src/tracker-preferences/tracker-preferences.c	(working copy)
@@ -268,6 +268,15 @@
 					value);
 
 
+	} else 	if (g_str_equal (name, "chkEnableThunderbirdIndexing")) {
+
+		set_bool_option (priv, "EnableThunderbird", value);
+
+		tracker_configuration_set_bool (configuration,
+					"/Emails/IndexThunderbirdEmails",
+					value);
+
+
 	} else 	if (g_str_equal (name, "chkIndexContents")) {
 
 		set_bool_option (priv, "IndexFileContents", value);
@@ -559,6 +568,18 @@
 
 	g_signal_connect (GTK_TOGGLE_BUTTON (widget), "toggled",
 			  G_CALLBACK (check_toggled_cb), priv);
+
+	widget = glade_xml_get_widget (priv->gxml,
+				       "chkEnableThunderbirdIndexing");
+
+	value = tracker_configuration_get_bool (configuration,
+						"/Emails/IndexThunderbirdEmails",
+						NULL);
+
+	gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (widget), value);
+
+	g_signal_connect (GTK_TOGGLE_BUTTON (widget), "toggled",
+			  G_CALLBACK (check_toggled_cb), priv);
 }
 
 static void
@@ -780,7 +801,19 @@
 					"/Emails/IndexEvolutionEmails",
 					value);
 
+	widget = glade_xml_get_widget (priv->gxml,
+				       "chkEnableThunderbirdIndexing");
+	value = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (widget));
+	value_old = tracker_configuration_get_bool (configuration,
+						    "/Emails/IndexThunderbirdEmails",
+						    NULL);
+	if (value != value_old)
+		flag_restart = TRUE;
+	tracker_configuration_set_bool (configuration,
+					"/Emails/IndexThunderbirdEmails",
+					value);
 
+
 	tracker_configuration_write (configuration);
 
 	if (flag_restart && if_trackerd_start (priv)) {
