diff -r -u tracker-0.5.4/src/trackerd/tracker-utils.c tracker-0.5.4-new/src/trackerd/tracker-utils.c
--- tracker-0.5.4/src/trackerd/tracker-utils.c	2007-01-26 08:15:11.000000000 +0800
+++ tracker-0.5.4-new/src/trackerd/tracker-utils.c	2007-02-07 19:32:21.727206000 +0800
@@ -2970,11 +2970,11 @@
 
 
 gboolean
-tracker_spawn (char **argv, int timeout, char **stdout, int *exit_status)
+tracker_spawn (char **argv, int timeout, char **tmp_stdout, int *exit_status)
 {
 	GSpawnFlags flags;
 
-	if (!stdout) {
+	if (!tmp_stdout) {
 		flags = G_SPAWN_SEARCH_PATH | G_SPAWN_STDOUT_TO_DEV_NULL |  G_SPAWN_STDERR_TO_DEV_NULL;
 	} else {
 		flags = G_SPAWN_SEARCH_PATH | G_SPAWN_STDERR_TO_DEV_NULL;
@@ -2986,7 +2986,7 @@
 			  flags,
 			  tracker_child_cb,
 			  GINT_TO_POINTER (timeout),
-			  stdout,
+			  tmp_stdout,
 			  NULL,
 			  exit_status,
 			  NULL);
diff -r -u tracker-0.5.4/src/trackerd/tracker-utils.h tracker-0.5.4-new/src/trackerd/tracker-utils.h
--- tracker-0.5.4/src/trackerd/tracker-utils.h	2007-01-26 09:04:02.000000000 +0800
+++ tracker-0.5.4-new/src/trackerd/tracker-utils.h	2007-02-07 19:32:32.210160000 +0800
@@ -434,7 +434,7 @@
 
 char *		tracker_get_snippet 		(const char *txt, char **terms, int length);
 
-gboolean	tracker_spawn 			(char **argv, int timeout, char **stdout, int *exit_status);
+gboolean	tracker_spawn 			(char **argv, int timeout, char **tmp_stdout, int *exit_status);
 void		tracker_child_cb 		(gpointer user_data);
 
 #endif
diff -r -u tracker-0.5.4/src/tracker-extract/tracker-extract.c tracker-0.5.4-new/src/tracker-extract/tracker-extract.c
--- tracker-0.5.4/src/tracker-extract/tracker-extract.c	2007-01-25 20:47:09.000000000 +0800
+++ tracker-0.5.4-new/src/tracker-extract/tracker-extract.c	2007-02-07 19:33:08.976576000 +0800
@@ -165,7 +165,7 @@
 
 
 gboolean
-tracker_spawn (char **argv, int timeout, char **stdout, int *exit_status)
+tracker_spawn (char **argv, int timeout, char **tmp_stdout, int *exit_status)
 {
 
 	return g_spawn_sync (NULL,
@@ -174,7 +174,7 @@
 			  G_SPAWN_SEARCH_PATH | G_SPAWN_STDERR_TO_DEV_NULL,
 			  tracker_child_cb,
 			  GINT_TO_POINTER (timeout),
-			  stdout,
+			  tmp_stdout,
 			  NULL,
 			  exit_status,
 			  NULL);
diff -r -u tracker-0.5.4/src/tracker-extract/tracker-extract.h tracker-0.5.4-new/src/tracker-extract/tracker-extract.h
--- tracker-0.5.4/src/tracker-extract/tracker-extract.h	2007-01-25 20:47:09.000000000 +0800
+++ tracker-0.5.4-new/src/tracker-extract/tracker-extract.h	2007-02-07 19:32:46.127149000 +0800
@@ -19,4 +19,4 @@
 
 #include <glib.h>
 
-gboolean	tracker_spawn (char **argv, int timeout, char **stdout, int *exit_status);
+gboolean	tracker_spawn (char **argv, int timeout, char **tmp_stdout, int *exit_status);
