Index: tracker-trunk/src/trackerd/trackerd.c
===================================================================
--- tracker-trunk/src/trackerd/trackerd.c	(revision 882)
+++ tracker-trunk/src/trackerd/trackerd.c	(working copy)
@@ -854,9 +854,9 @@
 	gboolean pushed_events, first_run;
 
         /* block all signals in this thread */
-        sigfillset (&signal_set);
+//        sigfillset (&signal_set);
 #ifndef OS_WIN32
-        pthread_sigmask (SIG_BLOCK, &signal_set, NULL);
+//        pthread_sigmask (SIG_BLOCK, &signal_set, NULL);
 #endif
 
 	g_mutex_lock (tracker->files_signal_mutex);
