Index: trunk/src/tracker-preferences/tracker-preferences.c
===================================================================
--- trunk/src/tracker-preferences/tracker-preferences.c	(revision 1099)
+++ trunk/src/tracker-preferences/tracker-preferences.c	(working copy)
@@ -731,6 +731,7 @@
 					"/Indexing/SkipMountPoints", NULL);
 	if (bvalue != bvalue_old) {
 		flag_restart = TRUE;
+		flag_reindex = TRUE;
 		set_bool_option (priv, "SkipMountPoints", bvalue);
 		tracker_configuration_set_bool (configuration,
 					"/Indexing/SkipMountPoints", bvalue);
@@ -749,6 +750,7 @@
 					       G_TYPE_STRING, NULL);
 	if (!str_slist_equal (list, list_old)) {
 		flag_restart = TRUE;
+		flag_reindex = TRUE;
 		tracker_configuration_set_list (configuration,
 					"/Watches/WatchDirectoryRoots", list,
 					G_TYPE_STRING);
@@ -766,6 +768,7 @@
 					       G_TYPE_STRING, NULL);
 	if (!str_slist_equal (list, list_old)) {
 		flag_restart = TRUE;
+		flag_reindex = TRUE;
 		tracker_configuration_set_list (configuration,
 					"/Watches/CrawlDirectory", list,
 					G_TYPE_STRING);
@@ -784,6 +787,7 @@
 					       G_TYPE_STRING, NULL);
 	if (!str_slist_equal (list, list_old)) {
 		flag_restart = TRUE;
+		flag_reindex = TRUE;
 		tracker_configuration_set_list (configuration,
 					"/Watches/NoWatchDirectory", list,
 					G_TYPE_STRING);
@@ -801,6 +805,7 @@
 					       G_TYPE_STRING, NULL);
 	if (!str_slist_equal (list, list_old)) {
 		flag_restart = TRUE;
+		flag_reindex = TRUE;
 		tracker_configuration_set_list (configuration,
 					"/Indexing/NoIndexFileTypes", list,
 					G_TYPE_STRING);
