--- twolame-0.3.10-orig/libtwolame/common.h	2007-03-21 00:01:01.000000000 +0000
+++ twolame-0.3.10/libtwolame/common.h	2007-11-03 03:54:56.021178702 +0000
@@ -35,8 +35,8 @@
 # define NO_DLL_INLINE
 #else
 # include "config.h"
-# ifdef __APPLE__
-/* Same problem under Mac OS X */
+# if defined(__APPLE__) || (defined(sun) && defined(__SVR4))
+/* Same problem under Mac OS X and Solaris */
 #  define NO_DLL_INLINE
 # else
 #  define NO_DLL_INLINE inline
