--- twolame-0.3.10-orig/configure.ac	2007-03-21 00:01:01.000000000 +0000
+++ twolame-0.3.10/configure.ac	2007-11-03 03:48:59.052484589 +0000
@@ -113,15 +113,13 @@
 
 dnl ############## Compiler and Linker Flags
 
-CFLAGS="$CFLAGS -std=c99 -Wunused -Wall"
 LDFLAGS="$LDFLAGS"
 
 # If debugging is enabled then make warnings errors
 if test "$Debugging" = "Enabled"; then
-	CFLAGS="$CFLAGS -g -Werror -pedantic"
+	CFLAGS="$CFLAGS -xc99=all -features=no%extinl -g"
 else
-	# Optimize flag. 3 is about as high as you can sanely go with GCC3.2.
-	CFLAGS="$CFLAGS -O3"
+	CFLAGS="$CFLAGS -xc99=all -features=no%extinl"
 fi
 
 
