Index: trunk/configure.ac
===================================================================
--- trunk/configure.ac	(revision 172)
+++ trunk/configure.ac	(working copy)
@@ -15,8 +15,8 @@
 AM_INIT_AUTOMAKE(1.7)
 AM_CONFIG_HEADER(build/config.h)
 
+GNOME_COMPILE_WARNINGS(yes)
 
-
 dnl ############# Library Version
 dnl
 dnl libtool version: current:revision:age
@@ -113,12 +113,15 @@
 
 dnl ############## Compiler and Linker Flags
 
-CFLAGS="$CFLAGS -std=c99 -Wunused -Wall"
+CFLAGS="$CFLAGS -std=c99"
 LDFLAGS="$LDFLAGS"
 
 # If debugging is enabled then make warnings errors
 if test "$Debugging" = "Enabled"; then
-	CFLAGS="$CFLAGS -g -Werror -pedantic"
+	CFLAGS="$CFLAGS -g"
+        if test "x$GCC" = xyes; then
+		CFLAGS="$CFLAGS -Werror -pedantic"
+	fi
 else
 	# Optimize flag. 3 is about as high as you can sanely go with GCC3.2.
 	CFLAGS="$CFLAGS -O3"
