--- ufraw-0.8/configure.ac.orig	2006-05-08 17:26:46.224736000 +0100
+++ ufraw-0.8/configure.ac	2006-05-08 17:27:22.369585000 +0100
@@ -32,9 +32,9 @@
     AC_MSG_ERROR([size of double must be 8])
 fi
 
-UFRAW_CFLAGS="-ffast-math -fomit-frame-pointer"
+UFRAW_CFLAGS=""
 if test "x$GCC" = "xyes"; then
-    UFRAW_CFLAGS="$UFRAW_CFLAGS -W -Wall -O3"
+    UFRAW_CFLAGS="$UFRAW_CFLAGS -ffast-math -fomit-frame-pointer -W -Wall -O3"
 fi
 
 # windows will require some special attention
