diff -ur vice-1.21-orig/configure.in vice-1.21/configure.in
--- vice-1.21-orig/configure.in	2007-03-05 04:58:50.000000000 +0700
+++ vice-1.21/configure.in	2007-04-27 13:35:25.180289788 +0700
@@ -456,7 +456,7 @@
   	    [CATOBJEXT=.gmo
   	     DATADIRNAME=share],
   	    [CATOBJEXT=.mo
-  	     DATADIRNAME=lib])
+  	     DATADIRNAME=share])
   	  INSTOBJEXT=.mo
   	fi
         USE_NLS=yes
diff -ur vice-1.21-orig/po/Makefile.in.in vice-1.21/po/Makefile.in.in
--- vice-1.21-orig/po/Makefile.in.in	2007-02-26 00:31:39.000000000 +0700
+++ vice-1.21/po/Makefile.in.in	2007-04-27 13:34:24.449939930 +0700
@@ -19,7 +19,7 @@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 datadir = $(DESTDIR)$(prefix)/@DATADIRNAME@
-localedir = $(DESTDIR)$(datadir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(DESTDIR)$(prefix)/share/locale
 gettextsrcdir = $(DESTDIR)$(prefix)/share/gettext/po
 subdir = po
