Index: trunk/configure.ac
===================================================================
--- trunk/configure.ac	(revision 205)
+++ trunk/configure.ac	(working copy)
@@ -37,6 +37,7 @@
 GTK_REQUIRED=2.12.0
 LIBGLADE_REQUIRED=2.6.0
 GCONF_REQUIRED=2.16.0
+GTHREAD_REQUIRED=2.0.0
 GTK_VNC_REQUIRED=0.3.3
 
 PKG_CHECK_MODULES(VINAGRE, \ 
@@ -45,6 +46,7 @@
   gtk+-2.0 >= $GTK_REQUIRED \ 
   libglade-2.0 >= $LIBGLADE_REQUIRED \
   gconf-2.0 >= $GCONF_REQUIRED \
+  gthread-2.0 >= $GTHREAD_REQUIRED \
   gnome-keyring-1 \
   gtk-vnc-1.0 >= $GTK_VNC_REQUIRED)
 AC_SUBST(VINAGRE_CFLAGS)
