--- vlc-0.8.6a/src/stream_output/sap.c.orig	2007-03-23 15:58:05.898730720 +0000
+++ vlc-0.8.6a/src/stream_output/sap.c	2007-03-23 15:59:55.655857471 +0000
@@ -649,7 +649,7 @@
                         ipv, p_addr->psz_machine,
                         psz_name, ipv, psz_uri,
                         /* FIXME: 1 is IPv4 default TTL, not that of IPv6 */
-                        p_session->i_ttl ?: (config_GetInt( p_sap, "ttl" ) ?: 1),
+                        p_session->i_ttl ? p_session->i_ttl : (config_GetInt( p_sap, "ttl" ) ? config_GetInt( p_sap, "ttl" ) : 1),
                         (sfilter != NULL) ? sfilter : "",
                         p_session->i_port,
                         p_session->b_rtp ? "RTP/AVP" : "udp",
--- vlc-0.8.6a/src/network/rootwrap.c.orig	2007-03-23 16:44:41.013175432 +0000
+++ vlc-0.8.6a/src/network/rootwrap.c	2007-03-23 16:44:53.200380013 +0000
@@ -273,7 +273,7 @@
     initgroups (pw->pw_name, pw->pw_gid);
     setgid (pw->pw_gid);
 
-    if (socketpair (AF_LOCAL, SOCK_STREAM, 0, pair))
+    if (socketpair (AF_UNIX, SOCK_STREAM, 0, pair))
     {
         perror ("socketpair");
         goto nofork;
