diff -ur vlc-0.8.6c-orig/configure.ac vlc-0.8.6c/configure.ac
--- vlc-0.8.6c-orig/configure.ac	2007-06-16 21:25:19.000000000 +0700
+++ vlc-0.8.6c/configure.ac	2007-07-14 14:05:58.420120423 +0700
@@ -5491,10 +5491,10 @@
 AC_SUBST(VERSION_MAJOR)
 AC_SUBST(VERSION_MINOR)
 AC_SUBST(VERSION_REVISION)
-AC_DEFINE_UNQUOTED(VLC_COMPILE_BY, "`whoami`", [user who ran configure]) 
+AC_DEFINE_UNQUOTED(VLC_COMPILE_BY, "`/usr/xpg4/bin/id -un`", [user who ran configure]) 
 AC_DEFINE_UNQUOTED(VLC_COMPILE_HOST, "`hostname`", [host which ran configure]) 
 AC_DEFINE_UNQUOTED(VLC_COMPILE_DOMAIN, "`dnsdomainname 2>/dev/null || domainname 2>/dev/null || echo unknown`", [domain of the host which ran configure]) 
-AC_DEFINE_UNQUOTED(VLC_COMPILER, "`$CC -v 2>&1 | tail -n 1`", [compiler]) 
+AC_DEFINE_UNQUOTED(VLC_COMPILER, "`$CC -v 2>&1 | /usr/xpg4/bin/tail -n 1`", [compiler]) 
 
 dnl Old definitions for version-dependant plugins
 dnl VLC_SYMBOL="`echo ${VERSION} | sed -e 'y/.-+/___/'`"
diff -ur vlc-0.8.6c-orig/share/Makefile.am vlc-0.8.6c/share/Makefile.am
--- vlc-0.8.6c-orig/share/Makefile.am	2007-06-16 21:25:05.000000000 +0700
+++ vlc-0.8.6c/share/Makefile.am	2007-07-14 14:04:57.601168308 +0700
@@ -258,7 +258,7 @@
 
 skins2/default.vlt:
 	mkdir -p skins2
-	tar cvvzf skins2/default.vlt --exclude .svn -C $(srcdir)/skins2 default/
+	gtar cvvzf skins2/default.vlt --exclude .svn -C $(srcdir)/skins2 default/
 
 .PHONY: $(nobase_pkgdata_DATA)
 
diff -ur vlc-0.8.6c-orig/share/Makefile.in vlc-0.8.6c/share/Makefile.in
--- vlc-0.8.6c-orig/share/Makefile.in	2007-06-16 21:26:33.000000000 +0700
+++ vlc-0.8.6c/share/Makefile.in	2007-07-14 14:05:15.937804684 +0700
@@ -813,7 +813,7 @@
 
 skins2/default.vlt:
 	mkdir -p skins2
-	tar cvvzf skins2/default.vlt --exclude .svn -C $(srcdir)/skins2 default/
+	gtar cvvzf skins2/default.vlt --exclude .svn -C $(srcdir)/skins2 default/
 
 .PHONY: $(nobase_pkgdata_DATA)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
