diff -ur vlc-0.8.6c-orig/src/network/rootwrap.c vlc-0.8.6c/src/network/rootwrap.c
--- vlc-0.8.6c-orig/src/network/rootwrap.c	2007-06-16 21:25:04.000000000 +0700
+++ vlc-0.8.6c/src/network/rootwrap.c	2007-07-14 11:52:43.414971354 +0700
@@ -38,6 +38,11 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <fcntl.h>
+#ifdef sun
+#define _XPG4_2
+#define __EXTENSIONS__
+#include <inttypes.h>
+#endif
 #include <sys/socket.h>
 #ifdef HAVE_SYS_TIME_H
 #include <sys/time.h>
@@ -273,7 +278,7 @@
     initgroups (pw->pw_name, pw->pw_gid);
     setgid (pw->pw_gid);
 
-    if (socketpair (AF_LOCAL, SOCK_STREAM, 0, pair))
+    if (socketpair (AF_UNIX, SOCK_STREAM, 0, pair))
     {
         perror ("socketpair");
         goto nofork;
