--- vlc-0.8.6/modules/demux/live555.cpp~	2006-12-09 03:12:13.000000000 +0200
+++ vlc-0.8.6/modules/demux/live555.cpp	2006-12-12 23:32:31.281541540 +0200
@@ -37,9 +37,9 @@
 #   include <winsock2.h>
 #endif
 
-#include "BasicUsageEnvironment.hh"
-#include "GroupsockHelper.hh"
-#include "liveMedia.hh"
+#include <liveMedia/BasicUsageEnvironment.hh>
+#include <liveMedia/GroupsockHelper.hh>
+#include <liveMedia/liveMedia.hh>
 
 extern "C" {
 #include "../access/mms/asf.h"  /* Who said ugly ? */
