--- wbxml2-0.9.2/src/Makefile.am.orig	2006-12-04 16:00:23.360323000 +0800
+++ wbxml2-0.9.2/src/Makefile.am	2006-12-05 16:46:03.746976000 +0800
@@ -20,8 +20,7 @@
 	 -DWBXML_SUPPORT_SYNCML\
 	 -DWBXML_SUPPORT_WV\
 	 -DWBXML_SUPPORT_AIRSYNC\
-	 -Wall -Wimplicit -Wreturn-type -Wunused -Wswitch -Wcomment -Wuninitialized -Wparentheses -Wpointer-arith -Wmissing-prototypes\
-	 -O3\
+	 -O\
 	 -g
 
 lib_LTLIBRARIES = libwbxml2.la
--- wbxml2-0.9.2/tools/Makefile.am.orig	2006-12-04 16:00:37.736585000 +0800
+++ wbxml2-0.9.2/tools/Makefile.am	2006-12-05 16:46:22.083443000 +0800
@@ -20,8 +20,7 @@
 	 -DWBXML_SUPPORT_SYNCML\
 	 -DWBXML_SUPPORT_WV\
 	 -DWBXML_SUPPORT_AIRSYNC\
-	 -Wall -Wimplicit -Wreturn-type -Wunused -Wswitch -Wcomment -Wuninitialized -Wparentheses -Wpointer-arith -Wmissing-prototypes\
-	 -O3\
+	 -O\
 	 -g
 
 bin_PROGRAMS = wbxml2xml xml2wbxml
--- wbxml2-0.9.2/Makefile.am.orig	2007-01-11 15:28:26.553389000 +0800
+++ wbxml2-0.9.2/Makefile.am	2007-01-11 15:28:34.015060000 +0800
@@ -26,7 +26,6 @@
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(wbxmldocdir)/manual
-	cp -Rp doc/* $(DESTDIR)$(wbxmldocdir)/manual
 
 dist-bz2: distdir
 	sed -e "s/tar.gz/tar.bz2/g" $(distdir)/wbxml2.spec > $(distdir)/wbxml2.spec.aux
