--- wbxml2-0.9.2/tools/getopt.h.orig	2006-12-05 16:26:00.254564000 +0800
+++ wbxml2-0.9.2/tools/getopt.h	2006-12-05 16:26:09.876052000 +0800
@@ -11,7 +11,7 @@
 #ifndef WBXML_GETOPT_H
 #define WBXML_GETOPT_H
 
-int getopt(int argc, char **argv, char *opts);
+//int getopt(int argc, char **argv, char *opts);
 extern int opterr;
 extern int optind;
 extern int optopt;
--- wbxml2-0.9.2/tools/attgetopt.c.orig	2007-01-11 19:00:03.395313000 +0800
+++ wbxml2-0.9.2/tools/attgetopt.c	2007-01-11 19:00:45.337449000 +0800
@@ -22,6 +22,7 @@
 int     optopt;
 char    *optarg;
 
+/*
 int getopt(int argc, char **argv, char *opts)
 {
     static int sp = 1;
@@ -65,3 +66,4 @@
     }
     return(c);
 }
+*/
