--- wesnoth-1.3.13/src/show_dialog.hpp.fixundef2	2008-01-01 02:05:57.174491264 +0100
+++ wesnoth-1.3.13/src/show_dialog.hpp	2008-01-01 02:03:14.202848752 +0100
@@ -189,8 +189,9 @@
 				const std::vector<preview_pane*>* preview_panes=NULL,
 				const std::string& text_widget_label="",
 				std::string* text_widget_text=NULL,
-				const int text_widget_max_chars = 256,
-				std::vector<check_item>* options=NULL, int xloc=-1, int yloc=-1,
+				int text_widget_max_chars = 256,
+				std::vector<check_item>* options=NULL, 
+                                int xloc=-1, int yloc=-1,
 				const dialog_frame::style* dialog_style=NULL,
 				std::vector<dialog_button_info>* buttons=NULL,
 				const menu::sorter* sorter=NULL,
