--- wesnoth-1.3.14/src/help.cpp.fixconst	2008-01-16 21:24:29.283442706 +0100
+++ wesnoth-1.3.14/src/help.cpp	2008-01-16 21:26:00.141935321 +0100
@@ -2929,7 +2929,7 @@
 //! Open the help browser, show unit with id unit_id.
 //!
 //! If show_topic is the empty string, the default topic will be shown.
-void show_unit_help(display &disp, std::string show_topic, bool hidden, int xloc, int yloc)
+void show_unit_help(display &disp, const std::string show_topic, bool hidden, int xloc, int yloc)
 {
 	show_help(disp, toplevel, (hidden ? ".":"") + unit_prefix + show_topic, xloc, yloc);
 }
