--- wesnoth-1.3.13/src/dialogs.cpp.fixgccism	2007-12-31 15:56:29.991360646 +0100
+++ wesnoth-1.3.13/src/dialogs.cpp	2007-12-31 15:58:20.203663155 +0100
@@ -33,6 +33,7 @@
 #include "unit.hpp"
 #include "wml_separators.hpp"
 #include "widgets/progressbar.hpp"
+#include <locale.h>
 
 
 #define LOG_NG LOG_STREAM(info, engine)
--- wesnoth-1.3.13/src/scoped_resource.hpp.fixgccism	2007-12-31 15:59:58.993353976 +0100
+++ wesnoth-1.3.13/src/scoped_resource.hpp	2007-12-31 16:01:45.621225718 +0100
@@ -182,9 +182,9 @@
  */
 struct close_FILE
 {
-	void operator()(FILE* f) const { if(f != NULL) { fclose(f); } }
+	void operator()(std::FILE* f) const { if(f != NULL) { std::fclose(f); } }
 };
-typedef scoped_resource<FILE*,close_FILE> scoped_FILE;
+typedef scoped_resource<std::FILE*,close_FILE> scoped_FILE;
 
 }
 
--- wesnoth-1.3.13/src/soundsource.cpp.fixgccism	2007-12-31 15:33:39.972515470 +0100
+++ wesnoth-1.3.13/src/soundsource.cpp	2007-12-31 15:52:30.780552231 +0100
@@ -30,7 +30,7 @@
 
 	// An obscure formula to calculate SDL_Mixer's "distance" based on the source's
 	// distance from screen's center
-	return maximum<int>(0, 128 * static_cast<int>(sqrt(dx + dy) / (sqrt(area.w*area.w + area.h * area.h))));
+	return maximum<int>(0, 128 * static_cast<int>(std::sqrt((double) (dx + dy)) / (sqrt((double) (area.w*area.w + area.h * area.h)))));
 }
 
 namespace soundsource {
--- wesnoth-1.3.13/src/mouse_events.cpp.fixgccism	2007-12-31 16:08:38.191679126 +0100
+++ wesnoth-1.3.13/src/mouse_events.cpp	2007-12-31 16:09:33.517500197 +0100
@@ -744,7 +744,7 @@
 	int mx = drag_from_x_; // some default value to prevent unlikely SDL bug
 	int my = drag_from_y_;
 	if (dragging_ && !dragging_started_ && (SDL_GetMouseState(&mx,&my) & SDL_BUTTON_LEFT != 0)) {
-		const double drag_distance = pow(drag_from_x_- mx, 2) + pow(drag_from_y_- my, 2);
+		const double drag_distance = pow((double) (drag_from_x_- mx), 2) + pow((double) (drag_from_y_- my), 2);
 		if (drag_distance > drag_threshold*drag_threshold) {
 			dragging_started_ = true;
 			cursor::set_dragging(true);
--- wesnoth-1.3.13/src/network_worker.hpp.fixgccism	2007-12-31 16:13:48.145973053 +0100
+++ wesnoth-1.3.13/src/network_worker.hpp	2007-12-31 16:17:18.408950889 +0100
@@ -18,14 +18,9 @@
 #include <map>
 #include <vector>
 #include "config.hpp"
-
+#include "network.hpp"
 #include "SDL_net.h"
 
-namespace network
-{
-	struct statistics;
-}
-
 namespace network_worker_pool
 {
 
--- wesnoth-1.3.13/src/game.cpp.fixgccism	2007-12-31 16:02:41.611262373 +0100
+++ wesnoth-1.3.13/src/game.cpp	2007-12-31 16:03:23.512720709 +0100
@@ -71,6 +71,7 @@
 #include <iterator>
 #include <sstream>
 #include <string>
+#include <locale.h>
 
 #include <boost/iostreams/copy.hpp>
 #include <boost/iostreams/filtering_streambuf.hpp>
