--- wine-0.9.59.orig/programs/winetest/Makefile.in	Wed Apr  9 22:46:31 2008
+++ wine-0.9.59/programs/winetest/Makefile.in	Wed Apr  9 22:46:47 2008
@@ -6,7 +6,7 @@
 APPMODE   = -mconsole
 IMPORTS   = comctl32 version user32 gdi32 advapi32 wsock32 kernel32
 
-EXTRARCFLAGS = -DBUILD_SHA1=\"$$(GIT_DIR=$(TOPSRCDIR)/.git git rev-parse HEAD 2>/dev/null)\"
+EXTRARCFLAGS = -DBUILD_SHA1=\"$(shell GIT_DIR=$(TOPSRCDIR)/.git git rev-parse HEAD 2>/dev/null)\"
 
 C_SRCS = \
 	gui.c \
