--- wine-1.1.10/tools/wineprefixcreate.in.orig	2008-12-15 20:17:00.553537000 +0800
+++ wine-1.1.10/tools/wineprefixcreate.in	2008-12-15 20:17:05.542212000 +0800
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Script to create the initial WINEPREFIX directory
 #
--- wine-1.1.10/tools/winewrapper.orig	2008-12-15 20:15:53.280259000 +0800
+++ wine-1.1.10/tools/winewrapper	2008-12-15 20:16:00.293639000 +0800
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # Wrapper script to run Wine and Winelib apps from inside the source tree
 #
