--- wine-0.9.49/tools/widl/widl.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/tools/widl/widl.c	2007-11-12 02:42:58.482046789 +0100
@@ -42,6 +42,7 @@
 #include "parser.h"
 #include "wine/wpp.h"
 #include "header.h"
+#include "wine/list.h"
 
 /* future options to reserve characters for: */
 /* a = alignment of structures */
--- wine-0.9.49/tools/widl/header.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/tools/widl/header.c	2007-11-12 02:43:23.376390419 +0100
@@ -36,6 +36,7 @@
 #include "utils.h"
 #include "parser.h"
 #include "header.h"
+#include "wine/list.h"
 
 static int indentation = 0;
 user_type_list_t user_type_list = LIST_INIT(user_type_list);
--- wine-0.9.49/tools/widl/typelib.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/tools/widl/typelib.c	2007-11-12 02:43:37.537412145 +0100
@@ -37,6 +37,7 @@
 
 #include "windef.h"
 #include "winbase.h"
+#include "wine/list.h"
 
 #include "widl.h"
 #include "utils.h"
--- wine-0.9.49/tools/widl/parser.y.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/tools/widl/parser.y	2007-11-12 02:43:50.156540149 +0100
@@ -32,6 +32,7 @@
 #endif
 
 #include "windef.h"
+#include "wine/list.h"
 
 #include "widl.h"
 #include "utils.h"
--- wine-0.9.49/tools/widl/server.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/tools/widl/server.c	2007-11-12 02:51:51.279294478 +0100
@@ -36,6 +36,7 @@
 #include "parser.h"
 #include "header.h"
 #include "windef.h"
+#include "wine/list.h"
 
 #include "typelib.h"
 #include "typelib_struct.h"
--- wine-0.9.49/tools/widl/client.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/tools/widl/client.c	2007-11-12 02:52:15.062837176 +0100
@@ -35,6 +35,7 @@
 #include "utils.h"
 #include "parser.h"
 #include "header.h"
+#include "wine/list.h"
 
 #include "widltypes.h"
 #include "typelib.h"
--- wine-0.9.49/tools/widl/proxy.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/tools/widl/proxy.c	2007-11-12 02:52:38.103073315 +0100
@@ -37,6 +37,7 @@
 #include "parser.h"
 #include "header.h"
 #include "typegen.h"
+#include "wine/list.h"
 
 #define END_OF_LIST(list)       \
   do {                          \
--- wine-0.9.49/dlls/ddraw/ddraw.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/dlls/ddraw/ddraw.c	2007-11-12 09:21:56.618439648 +0100
@@ -35,6 +35,7 @@
 #include "winerror.h"
 #include "wingdi.h"
 #include "wine/exception.h"
+#include "wine/list.h"
 
 #include "ddraw.h"
 #include "d3d.h"
--- wine-0.9.49/dlls/ddraw/main.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/dlls/ddraw/main.c	2007-11-12 09:22:04.969897457 +0100
@@ -40,6 +40,7 @@
 #include "wingdi.h"
 #include "wine/exception.h"
 #include "winreg.h"
+#include "wine/list.h"
 
 #include "ddraw.h"
 #include "d3d.h"
--- wine-0.9.49/dlls/ddraw/surface.c.orig	2007-11-09 17:56:12.000000000 +0100
+++ wine-0.9.49/dlls/ddraw/surface.c	2007-11-12 09:22:15.888134810 +0100
@@ -39,6 +39,7 @@
 #include "winerror.h"
 #include "wingdi.h"
 #include "wine/exception.h"
+#include "wine/list.h"
 
 #include "ddraw.h"
 #include "d3d.h"
