--- blah/winetricks.orig	2008-06-17 14:59:41.706757206 -0400
+++ blah/winetricks	2008-06-17 15:00:48.104439734 -0400
@@ -48,13 +48,13 @@
 }
 
 which "$WINE" > /dev/null
-if test "$?" -ne "0"
+if [ "$?" -ne "0" ]
 then
   die "Cannot find wine ($WINE)"
 fi
 
 which "$WINEPREFIXCREATE" > /dev/null
-if test "$?" -ne "0"
+if [ "$?" -ne "0" ]
 then
   die "Cannot find wineprefixcreate ($WINEPREFIXCREATE)"
 fi
@@ -173,7 +173,7 @@
     shift
     col2name="$1"
     shift
-    while test $# -gt 0
+    while [ $# -gt 0 ]
     do
         args="$args $1 $1 off"
         shift
@@ -190,12 +190,12 @@
     shift
     col2name="$1"
     shift
-    if test $# -gt 0
+    if [ $# -gt 0 ]
     then
         args="$1"
         shift
     fi
-    while test $# -gt 0
+    while [ $# -gt 0 ]
     do
         args="$args,$1"
         shift
@@ -225,10 +225,11 @@
     export todo="`sh $WINETRICKS_TMP/zenity.sh | tr '|' ' '`"
   else
     packages=`usage | awk '/^ [a-z]/ {print $1}'`
-    export todo="`showmenu "winetricks" "Select a package to install" "Install?" "Package" $packages`"
+    todo="`showmenu "winetricks" "Select a package to install" "Install?" "Package" $packages`"
+    export todo
   fi
 
-  if test "$todo"x = x
+  if [ "$todo"x = x ]
   then
      exit 0
   fi
@@ -252,7 +253,7 @@
 programfilesdir_unix=`$WINE winepath -u "$programfilesdir_win"`
 
 # Did the user rename Fonts to fonts?
-if ! test -d "$WINEPREFIX"/drive_c/windows/Fonts && test -d "$WINEPREFIX"/drive_c/windows/fonts 
+if [ ! -d "$WINEPREFIX"/drive_c/windows/Fonts -a -d "$WINEPREFIX"/drive_c/windows/fonts ]
 then
     winefontsdir="$WINEPREFIX"/drive_c/windows/fonts
 else
@@ -268,7 +269,7 @@
 fi
 
 which cabextract > /dev/null
-if test "$?" -ne "0"
+if [ "$?" -ne "0" ]
 then
   echo "Cannot find cabextract.  Please install it (e.g. 'sudo apt-get install cabextract' or 'sudo yum install cabextract')."
 fi
@@ -284,7 +285,7 @@
     echo Executing "$@"
     "$@"
     status=$?
-    if test $status -ne 0
+    if [ $status -ne 0 ]
     then
         die "Note: command '$@' returned status $status.  Aborting."
     fi
@@ -314,7 +315,7 @@
     fi
     cache="$WINETRICKS_CACHE/$1"
     mkdir -p "$cache"
-    if ! test -f "$cache/$file"
+    if [ ! -f "$cache/$file" ]
     then
         cd "$cache"
         # Mac folks tend to have curl rather than wget
@@ -376,7 +377,7 @@
 
 [HKEY_CURRENT_USER\Software\Wine\DllOverrides]
 _EOF_
-    while test "$1" != ""
+    while [ "$1" != "" ]
     do
         case "$1" in
         comctl32)
@@ -403,7 +404,7 @@
     echo "[HKEY_CURRENT_USER\\Software\\Wine\\AppDefaults\\$app\\DllOverrides]" 
     ) > "$WINETRICKS_TMP"/override-dll.reg 
 
-    while test "$1" != ""
+    while [ "$1" != "" ]
     do
         case "$1" in
         comctl32)
@@ -754,7 +755,7 @@
     # some versions of ukai.ttf and uming.ttf crash .net and picasa
     # See http://bugs.winehq.org/show_bug.cgi?id=7098#c9
     # Could fix globally, but that needs root, so just fix for wine
-    if test -f /usr/share/fonts/truetype/arphic/ukai.ttf 
+    if [ -f /usr/share/fonts/truetype/arphic/ukai.ttf ]
     then
         gotsum=`$SHA1SUM < /usr/share/fonts/truetype/arphic/ukai.ttf | sed 's/ .*//'`
         # FIXME: do all affected versions of the font have same sha1sum as Gutsy?  Seems unlikely.
@@ -768,7 +769,7 @@
         fi
     fi
 
-    if test -f /usr/share/fonts/truetype/arphic/uming.ttf 
+    if [ -f /usr/share/fonts/truetype/arphic/uming.ttf ]
     then
         gotsum=`$SHA1SUM < /usr/share/fonts/truetype/arphic/uming.ttf | sed 's/ .*//'`
         if [ "$gotsum"x = "2a4f4a69e343c21c24d044b2cb19fd4f0decc82c"x ]
@@ -793,7 +794,7 @@
     # so we're adding a maintenance burden here.)
     GECKO_VERSION=0.1.0
 
-    if ! test -f "$WINETRICKS_CACHE"/wine_gecko-$GECKO_VERSION.cab
+    if [ ! -f "$WINETRICKS_CACHE"/wine_gecko-$GECKO_VERSION.cab ]
     then
        # FIXME: busted if using curl!
        download . "http://source.winehq.org/winegecko.php?v=$GECKO_VERSION" c16f1072dc6b0ced20935662138dcf019a38cd56 wine_gecko-$GECKO_VERSION.cab
@@ -973,7 +974,7 @@
 load_msls31() {
     # Install native Microsoft Line Services (needed by e-Sword, possibly only when using native riched20)
     # http://www.microsoft.com/reader/downloads/pc.mspx
-    if ! test -f "$programfilesdir_unix/Microsoft Reader"/msls31.dll
+    if [ ! -f "$programfilesdir_unix/Microsoft Reader"/msls31.dll ]
     then
         echo "Installing Microsoft Reader (which includes a copy of msls31.dll)"
         download . http://download.microsoft.com/download/d/0/9/d099ad62-41d1-4e92-9935-67955a437460/MSReaderSetupUSA.exe 055af1f589e2ae9896f17dbf33d8a3f156d10282
@@ -999,13 +1000,13 @@
 
 load_msxml4() {
     # http://www.microsoft.com/downloads/details.aspx?familyid=24B7D141-6CDF-4FC4-A91B-6F18FE6921D4
-    if ! test -f "$WINETRICKS_CACHE"/msxml4.msi
+    if [ ! -f "$WINETRICKS_CACHE"/msxml4.msi ]
     then
        download . http://download.microsoft.com/download/e/2/e/e2e92e52-210b-4774-8cd9-3a7a0130141d/msxml4-KB927978-enu.exe d364f9fe80c3965e79f6f64609fc253dfeb69c25
        rm -rf "$WINETRICKS_TMP"/*
 
        try $WINE "$WINETRICKS_CACHE"/msxml4-KB927978-enu.exe "/x:`$WINE winepath -w "$WINETRICKS_TMP"`" $WINETRICKS_QUIET
-       if ! test -f "$WINETRICKS_TMP"/msxml.msi
+       if [ ! -f "$WINETRICKS_TMP"/msxml.msi ]
        then
           die msxml.msi not found
        fi
@@ -1039,14 +1040,14 @@
     # http://www.apple.com/support/downloads/quicktime72forwindows.html
     download quicktime72 'http://wsidecar.apple.com/cgi-bin/nph-reg3rdpty2.pl/product=14402&cat=59&platform=osx&method=sa/QuickTimeInstaller.exe' bb89981f10cf21de57b9453e53cf81b9194271a9
     unset QUICKTIME_QUIET
-    if test "$WINETRICKS_QUIET"x != x
+    if [ "$WINETRICKS_QUIET"x != x ]
     then
        QUICKTIME_QUIET="/qn"  # ISSETUPDRIVEN=0
     fi
     # set vista mode to inhibit directdraw overlay use that blacks the screen
     set_winver vista
     try $WINE "$WINETRICKS_CACHE"/quicktime72/QuickTimeInstaller.exe ALLUSERS=1 DESKTOP_SHORTCUTS=0 QTTaskRunFlags=0 QTINFO.BISQTPRO=1 SCHEDULE_ASUW=0 REBOOT_REQUIRED=No $QUICKTIME_QUIET
-    if test "$WINETRICKS_QUIET"x = x
+    if [ "$WINETRICKS_QUIET"x = x ]
     then
         echo "You probably want to select Advanced / Safe Mode in the Quicktime control panel"
         try $WINE control ${programfilesdir_win}'\QuickTime\QTSystem\QuickTime.cpl'
@@ -1073,7 +1074,7 @@
     # You may be able to work around this by using the installer's /x or /extract switch,
     # but renaming drive_c to "harddiskvolume0" lets you just run the installer as normal.
 
-    if ! test -d "$WINEPREFIX"/harddiskvolume0/
+    if [ ! -d "$WINEPREFIX"/harddiskvolume0/ ]
     then
 	ln -s drive_c "$WINEPREFIX"/harddiskvolume0
 	rm "$WINEPREFIX"/dosdevices/c:
@@ -1167,19 +1168,19 @@
 
 load_vcrun6() {
     # Load the Visual C++ 6 runtime libraries, including the elusive mfc42u.dll
-    if test -f "$WINEPREFIX"/drive_c/windows/system32/mfc42u.dll
+    if [ -f "$WINEPREFIX"/drive_c/windows/system32/mfc42u.dll ]
     then
         echo "vcrun6 already installed, skipping"
         return
     fi
 
-    if ! test -f "$WINETRICKS_CACHE"/vcredist.exe
+    if [ ! -f "$WINETRICKS_CACHE"/vcredist.exe ]
     then
        download . http://download.microsoft.com/download/vc60pro/update/1/w9xnt4/en-us/vc6redistsetup_enu.exe 382c8f5a7f41189af8d4165cf441f274b7e2a457
        rm -rf "$WINETRICKS_TMP"/*
        
        try $WINE "$WINETRICKS_CACHE"/vc6redistsetup_enu.exe "/T:`$WINE winepath -w "$WINETRICKS_TMP"`" /c $WINETRICKS_QUIET
-       if ! test -f "$WINETRICKS_TMP"/vcredist.exe
+       if  [ ! -f "$WINETRICKS_TMP"/vcredist.exe ]
        then
           die vcredist.exe not found
        fi
@@ -1347,7 +1348,7 @@
 #----------------------------------------------------------------
 
 
-while test "$1" != ""
+while [ "$1" != "" ]
 do
     PACKAGE=$1
     case $1 in
