diff -ur wine-0.9.35-orig/configure wine-0.9.35/configure
--- wine-0.9.35-orig/configure	2007-04-13 22:47:52.000000000 +0700
+++ wine-0.9.35/configure	2007-04-23 19:01:09.173124977 +0700
@@ -13354,13 +13354,13 @@
 fi
 
 
-      { echo "$as_me:$LINENO: checking whether the compiler supports -fPIC -shared -Wl,-Bsymbolic,-z,defs" >&5
-echo $ECHO_N "checking whether the compiler supports -fPIC -shared -Wl,-Bsymbolic,-z,defs... $ECHO_C" >&6; }
+      { echo "$as_me:$LINENO: checking whether the compiler supports -fPIC -Wl,G,-Bsymbolic,-z,defs" >&5
+echo $ECHO_N "checking whether the compiler supports -fPIC -Wl,G,-Bsymbolic,-z,defs... $ECHO_C" >&6; }
 if test "${ac_cv_cflags__fPIC__shared__Wl__Bsymbolic__z_defs+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_wine_try_cflags_saved=$CFLAGS
-CFLAGS="$CFLAGS -fPIC -shared -Wl,-Bsymbolic,-z,defs"
+CFLAGS="$CFLAGS -fPIC -Wl,G,-Bsymbolic,-z,defs"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
diff -ur wine-0.9.35-orig/configure.ac wine-0.9.35/configure.ac
--- wine-0.9.35-orig/configure.ac	2007-04-13 22:47:52.000000000 +0700
+++ wine-0.9.35/configure.ac	2007-04-23 19:28:02.136339455 +0700
@@ -1010,14 +1010,14 @@
                               ac_cv_c_dll_gnuelf="yes",ac_cv_c_dll_gnuelf="no")])
     if test "$ac_cv_c_dll_gnuelf" = "yes"
     then
-      LDSHARED="\$(CC) -shared"
-      LDDLLFLAGS="-shared -Wl,-Bsymbolic"
-      WINE_TRY_CFLAGS([-fPIC -shared -Wl,-soname,confest.so.1],
-                      [LDSHARED="\$(CC) -shared \$(SONAME:%=-Wl,-soname,%)"],
-                      [WINE_TRY_CFLAGS([-fPIC -shared -Wl,-h,confest.so.1],
-                                       [LDSHARED="\$(CC) -shared \$(SONAME:%=-Wl,-h,%)"])])
+      LDSHARED="\$(CC) -Wl,-G"
+      LDDLLFLAGS="-Wl,-G,-Bsymbolic"
+      WINE_TRY_CFLAGS([-fPIC -Wl,-G,-soname,confest.so.1],
+                      [LDSHARED="\$(CC) -Wl,-G \$(SONAME:%=-Wl,-soname,%)"],
+                      [WINE_TRY_CFLAGS([-fPIC -Wl,-G,-h,confest.so.1],
+                                       [LDSHARED="\$(CC) -Wl,-G \$(SONAME:%=-Wl,-h,%)"])])
 
-      WINE_TRY_CFLAGS([-fPIC -shared -Wl,-Bsymbolic,-z,defs], [LDDLLFLAGS="$LDDLLFLAGS,-z,defs"])
+      WINE_TRY_CFLAGS([-fPIC -Wl,-G,-Bsymbolic,-z,defs], [LDDLLFLAGS="$LDDLLFLAGS,-z,defs"])
 
       WINE_TRY_CFLAGS([-fPIC -shared -Wl,-Bsymbolic,-init,__wine_spec_init,-fini,__wine_spec_fini],
                       [LDDLLFLAGS="$LDDLLFLAGS,-init,__wine_spec_init,-fini,__wine_spec_fini"])
