Index: wv-trunk/wvText.in
===================================================================
--- wv-trunk/wvText.in	(revision 21979)
+++ wv-trunk/wvText.in	(working copy)
@@ -18,20 +18,25 @@
 else
     which links >/dev/null 2>&1
     if [ ${?} -eq "0" ]; then
-	USE_DUMP=2
+		USE_DUMP=2
     else
-	which lynx >/dev/null 2>&1
-	if [ ${?} -eq "0" ]; then
-	    USE_DUMP=1
-	fi
+		which lynx >/dev/null 2>&1
+		if [ ${?} -eq "0" ]; then
+			USE_DUMP=1
+		else
+			which w3m >/dev/null 2>&1
+			if [ ${?} -eq "0" ]; then
+				USE_DUMP=4
+			fi
+		fi
     fi    
 fi
 
 if [ $USE_DUMP -eq "1" ]; then
-	echo "Could not find required program 'elinks' or 'links'"
+	echo "Could not find required program 'elinks' or 'links' or 'w3m'"
 	echo "Using lynx. Ouput will be pretty ugly."
 elif [ $USE_DUMP -eq "0" ]; then
-	echo "Could not find required program 'elinks', 'links', or even 'lynx'"
+	echo "Could not find required program 'elinks', 'links', 'w3m', or even 'lynx'"
 	echo "Using wvWare -x wvText.xml. Ouput will be pretty bad."
 fi
 
@@ -54,7 +59,10 @@
 	exit 1
     fi
 
-    if [ $USE_DUMP -eq "3" ]; then
+    if [ $USE_DUMP -eq "4" ]; then
+	# w3m
+	w3m -T text/html -dump "${tmpdir}/${TMP_FILE}" > "${2}"
+    elif [ $USE_DUMP -eq "3" ]; then
 	# elinks does the best
 	elinks -dump -force-html "${tmpdir}/${TMP_FILE}" > "${2}"
     elif [ $USE_DUMP -eq "2" ]; then
