--- wxsvg-1.0b7/svgview/svgview.cpp.orig	2006-11-29 20:28:31.221883000 -0500
+++ wxsvg-1.0b7/svgview/svgview.cpp	2006-11-29 20:30:03.074920000 -0500
@@ -78,12 +78,15 @@
 {
     // Make a menubar
     wxMenu *fileMenu = new wxMenu;
+    wxMenuItem* it;
     fileMenu->Append(wxID_OPEN, _T("&Open..."));
     fileMenu->Append(wxID_SAVE, _T("&Save as..."));
     fileMenu->AppendSeparator();
     fileMenu->Append(wxID_EXIT, _T("&Close"));
     fileMenu->AppendSeparator();
-    fileMenu->AppendCheckItem(wxID_FIT, _T("&FitToFrame"))->Check();
+    fileMenu->AppendCheckItem(wxID_FIT, _T("&FitToFrame"));
+    it = fileMenu->FindChildItem( wxID_FIT );
+    it->Check();
     fileMenu->AppendCheckItem(wxID_HITTEST, _T("&Hit-Test"));
     
     wxMenuBar *menuBar = new wxMenuBar;
