--- wxWidgets-2.8.8/src/gtk/gnome/gvfs.cpp-orig	2008-07-25 09:04:53.751137000 -0500
+++ wxWidgets-2.8.8/src/gtk/gnome/gvfs.cpp	2008-07-25 09:05:04.529114000 -0500
@@ -91,7 +91,7 @@ wxGnomeVFSLibrary::~wxGnomeVFSLibrary()
         // which we link implicitly (GTK+ itself?) which are not uninstalled
         // when it's unloaded resulting in this crash, so just leave it in
         // memory -- it's a lesser evil
-        m_gnome_vfs_lib->.Detach();
+        m_gnome_vfs_lib->Detach();
         delete m_gnome_vfs_lib;
     }
 }
