--- wxWidgets-2.8.4/src/gtk/gnome/gprint.cpp.orig	2007-08-07 16:57:39.225509000 +0800
+++ wxWidgets-2.8.4/src/gtk/gnome/gprint.cpp	2007-08-07 16:58:37.014728000 +0800
@@ -1304,7 +1304,7 @@
 void wxGnomePrintDC::makeEllipticalPath(wxCoord x, wxCoord y,
                                         wxCoord width, wxCoord height)
 {
-    double r = 4 * (sqrt (2) - 1) / 3;
+    double r = 4 * (sqrt ((double)2) - 1) / 3;
     double  halfW = 0.5 * width,
             halfH = 0.5 * height,
             halfWR = r * halfW,
