--- xchat-2.8.0/src/common/dbus/Makefile.am.patch03	2007-01-07 00:18:46.729862000 -0500
+++ xchat-2.8.0/src/common/dbus/Makefile.am	2007-01-07 00:19:26.933990000 -0500
@@ -23,7 +23,7 @@
 
 noinst_PROGRAMS = example
 example_SOURCES = example.c 
-example_LDADD = $(DBUS_LIBS) $(GLIB_LIBS)
+example_LDADD = $(DBUS_LIBS) $(GLIB_LIBS) $(GTK_LIBS)
 
 remote-object-glue.h: remote-object.xml
 	$(LIBTOOL) --mode=execute $(DBUS_BINDING_TOOL) --prefix=remote_object --mode=glib-server --output=$@ $<
