--- xchat-2.6.4/src/common/msproxy.h.orig	2006-06-24 00:06:48.459416000 -0400
+++ xchat-2.6.4/src/common/msproxy.h	2006-06-24 00:07:33.187344000 -0400
@@ -120,7 +120,7 @@
 			char			pad50[6];			/* 211-216						*/
 			guint16			magic55;				/* 217-218: 0x30, 0x00					*/
 			char			pad55[2];			/* 219-220						*/
-			char			data[0];			/* Dummy end marker, no real data required		*/
+			char			data[1];			/* Dummy end marker, no real data required		*/
 		} auth;
 
 		struct {
@@ -180,7 +180,7 @@
 			guint16			boundport;			/* 77-78: bind request; port used on client behalf.	*/
 			guint32			boundaddr;			/* 79-82: addr used on client behalf			*/
 			char			pad30[90];			/* 83-172						*/
-			char			data[0];			/* End marker						*/
+			char			data[1];			/* End marker						*/
 		} connack;
 
 	} packet;
