--- xvidcore-1.1.3/build/generic/configure.in.orig	2007-07-31 14:21:53.877885306 +0700
+++ xvidcore-1.1.3/build/generic/configure.in	2007-07-31 14:36:22.539055836 +0700
@@ -265,13 +265,24 @@
 SPECIFIC_CFLAGS=""
 PRE_SHARED_LIB=""
 case "$target_os" in
-     linux*|solaris*)
+     linux*)
 	AC_MSG_RESULT([ok])
 	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
 	SHARED_LIB="libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR).\$(API_MINOR)"
 	SPECIFIC_LDFLAGS="-Wl,-soname,libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR) -shared -Wl,--version-script=libxvidcore.ld -lc -lm"
 	SPECIFIC_CFLAGS="-fPIC"
 	;;
+     solaris*)
+	AC_MSG_RESULT([ok])
+	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
+	SHARED_LIB="libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR).\$(API_MINOR)"
+	SPECIFIC_LDFLAGS="-Wl,-h,libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR) -Wl,-G -Wl,-M,libxvidcore.ld -lc -lm"
+	if test "`basename $CC`" = "gcc" ; then
+	    SPECIFIC_CFLAGS="-fPIC"
+	else
+	    SPECIFIC_CFLAGS="-KPIC -D__inline=inline"
+	fi
+	;;
      *bsd*|irix*)
 	AC_MSG_RESULT([ok])
 	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
