--- yasm-0.7.1.orig/configure	2008-05-14 21:40:34.000000000 -0400
+++ yasm-0.7.1/configure	2008-06-27 14:25:12.095991000 -0400
@@ -10448,12 +10448,6 @@
 # Add some more CFLAGS for various options.
 #
 
-if test "$debugging" = "no" ; then
-
-  CFLAGS="`echo $CFLAGS' ' | sed -e 's/-g[0-9] //g' | sed -e 's/-g//g'`"
-
-fi
-
 # Turn warnings into errors
 if test "$warnerror" = "yes"; then
   if test "$GCC" = "yes"; then
